<!DOCTYPE html>
<html lang="en">

<!-- layout.ejs-->
<head><meta name="generator" content="Hexo 3.8.0">
    <!-- hexo-inject:begin --><!-- hexo-inject:end --><meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="Adelaide">
    <meta name="author" content="xt">
    <meta name="keyword" content="">
    <link rel="canonical" href="https://taoshengxu.github.io/hexo/hexo/2019/03/13/2019-03-13_TP53_BRCA/">
    <link rel="shortcut icon" href="/hexo/img/favicon.png">
    <link rel="alternate" type="application/atom+xml" title="IIM" href="/atom.xml">
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/animate.css/3.5.2/animate.min.css">
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
    <link rel="stylesheet" href="https://ajax.googleapis.com/ajax/libs/jqueryui/1.12.1/themes/smoothness/jquery-ui.css">
    <script src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.12.1/jquery-ui.min.js"></script>
    <script src="/hexo/js/search.js"></script>

    <title>
        
        TP53突变型和TP53野生型BRCA病人的差异分析结果（jimmy）｜undefined
        
    </title>

    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">

    <link rel="stylesheet" href="/hexo/css/main.css">

    
      <link href="//fonts.googleapis.com/css?family=Source+Code+Pro" rel="stylesheet" type="text/css">
      <link rel="stylesheet" href="/hexo/css/highlight.css">
    

    

    <script async src="//dn-lbstatics.qbox.me/busuanzi/2.3/busuanzi.pure.mini.js"></script>


    
      <meta name="google-site-verification" content="xxx">
    

    

    


    
  


    
<script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
                (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
            m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
    ga('create', 'UA-xxxxxx-xx', 'auto');
    ga('send', 'pageview');
</script>



<script>
    var _baId = 'xxx';
    // Originial
    var _hmt = _hmt || [];
    (function() {
        var hm = document.createElement("script");
        hm.src = "//hm.baidu.com/hm.js?" + _baId;
        var s = document.getElementsByTagName("script")[0];
        s.parentNode.insertBefore(hm, s);
    })();
</script>



    <script async defer src="https://buttons.github.io/buttons.js"></script>

<link rel="stylesheet" href="/hexo/css/prism-solarizedlight.css" type="text/css">
<link rel="stylesheet" href="/hexo/css/prism-line-numbers.css" type="text/css"><!-- hexo-inject:begin --><!-- hexo-inject:end --></head>

<style>
    header.intro-header {
        background-image: url('/hexo/img/northernlights-sisimiut-lake.jpg')
    }
</style>
<!-- hack iOS CSS :active style -->
<body ontouchstart="" class="animated fadeIn">
<!-- hexo-inject:begin --><!-- hexo-inject:end --><header>
  <nav class="navbar navbar-default header-navbar" id="nav-top" data-ispost="true" data-istags="false" data-ishome="false">
    <div class="container-fluid">
      <div class="navbar-header page-scroll">
        <button type="button" class="navbar-toggle" data-toggle="collapse" aria-expanded="false" data-target="#website_navbar">
          <span class="sr-only">Toggle navigation</span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <span class="navbar-brand animated pulse">
          <a class="brand-logo" href="/hexo/">
                <img src="/hexo/img/banner.jpg?h=350&amp;auto=compress&amp;cs=tinysrgb">
          </a>
        </span>
      </div>

      <div class="collapse navbar-collapse" id="website_navbar">
          <ul class="nav navbar-nav navbar-right">
              
                <li>
                  <a href="/hexo/">home</a>
                </li>
              
                <li>
                  <a href="/hexo/archives/">archives</a>
                </li>
              
                <li>
                  <a href="/hexo/categories/">categories</a>
                </li>
              
                <li>
                  <a href="/hexo/tags/">tags</a>
                </li>
              
                <li>
                  <a href="/hexo/columns/">columns</a>
                </li>
              
          </ul>
      </div>
  </div></nav>


  
    <style>
       .intro-header {
          background-image: url('/hexo/post_cover_images/moon.jpg');
      }
    </style>

    <div class="intro-header">
        <div class="container">
            <div class="row">
                <div class="col-lg-8 col-lg-offset-2 col-md-10 col-md-offset-1 text-center">
                    <div class="site-heading">
                        <h1>TP53突变型和TP53野生型BRCA病人的差异分析结果（jimmy）</h1>
                        
                        

                        
                          <span class="meta">
                               <span class="meta-item">Author: xt</span>
                               <span class="meta-item">Date: Mar 13, 2019</span>
                               
                          </span>
                          <div class="tags text-center">
                              Categories: 
                              <a class="tag" href="/hexo/categories/#Bioinformatics" title="Bioinformatics">Bioinformatics</a>
                              
                          </div>
                          <div class="tags text-center">
                              Tags: 
                          </div>
                        
                    </div>
                </div>
            </div>
        </div>
    </div>
  
</header>


<!-- Main Content -->
<!-- post.ejs -->
<article>
    <div class="container">
      <div class="col-lg-8 col-lg-offset-1 col-sm-9">
          
          <span class="post-count">2,578 words in total, 15 minutes required.</span>
          <hr>
          
          <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/hint.css/2.4.1/hint.min.css"><p>这个教程有很多学习练习之处，首先解决了国内安装Bioconductor的问题。</p>
<hr>
<p>第一步 安装必要的R包</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># Step0 Before starting your project --------------------------------------</span></span><br><span class="line"></span><br><span class="line"><span class="comment">## Remove everything in the working environment, not including loaded libraries.</span></span><br><span class="line">rm(list = objects( all = <span class="literal">TRUE</span> ))</span><br><span class="line"></span><br><span class="line"><span class="keyword">if</span> (!is.null( dev.list() )) dev.off() </span><br><span class="line"></span><br><span class="line">clearhistory &lt;- <span class="keyword">function</span>() &#123;</span><br><span class="line">  write( <span class="string">""</span>, file = <span class="string">".blank"</span> )</span><br><span class="line">  loadhistory( <span class="string">".blank"</span> )</span><br><span class="line">  unlink( <span class="string">".blank"</span> )</span><br><span class="line">&#125;</span><br><span class="line">clearhistory()</span><br><span class="line"></span><br><span class="line"><span class="comment">## basecal packages</span></span><br><span class="line">sysPackages &lt;- (.packages())</span><br><span class="line"></span><br><span class="line"><span class="comment">## data.frame(..., row.names = NULL, check.rows = FALSE,</span></span><br><span class="line"><span class="comment">##            check.names = TRUE, fix.empty.names = TRUE,</span></span><br><span class="line"><span class="comment">##            stringsAsFactors = default.stringsAsFactors())</span></span><br><span class="line">options( stringsAsFactors = <span class="literal">FALSE</span> )</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">## Now winInet not supported for use in service, but the default setting of </span></span><br><span class="line"><span class="comment">## download.file.method is "wininet". </span></span><br><span class="line"><span class="comment">## If your system support "libcurl", set the downloaded method to libcurl.</span></span><br><span class="line"><span class="keyword">if</span> ( capabilities( <span class="string">"libcurl"</span> ) == <span class="literal">T</span> ) &#123;</span><br><span class="line">  options( download.file.method = <span class="string">"libcurl"</span> )</span><br><span class="line">&#125;</span><br><span class="line">options()$download.file.method</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">## Change the library location of the packages</span></span><br><span class="line"><span class="comment">## Even your updated your R, you can still use your packages.</span></span><br><span class="line">.libPaths( c( <span class="string">"G:/R-packages"</span>,</span><br><span class="line">              <span class="string">"C:/Program Files/R/R-3.5.2/library"</span>) )</span><br><span class="line">.libPaths()</span><br><span class="line"></span><br><span class="line"><span class="comment">## 上面的步骤是为了创建一个方便处理数据的环境，之后的每步运行之前我都会先运行一下这部分代码，由于篇幅有限，就不重复出现了</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># Step1 Setting CRAN mirror -----------------------------------------------</span></span><br><span class="line"></span><br><span class="line">local(&#123;</span><br><span class="line">  options( repos  = <span class="string">"https://mirrors.ustc.edu.cn/CRAN/"</span> )</span><br><span class="line">  options( BioC_mirror = <span class="string">"https://mirrors.ustc.edu.cn/bioc/"</span> )</span><br><span class="line">&#125;)</span><br><span class="line"><span class="comment"># Step2 List of the used packages  ----------------------------------------</span></span><br><span class="line"></span><br><span class="line">bioPackages &lt;- </span><br><span class="line">c( </span><br><span class="line">  <span class="string">"dplyr"</span>, <span class="string">"stringi"</span>, <span class="string">"purrr"</span>, <span class="comment">## ERROR</span></span><br><span class="line">  <span class="string">"R.utils"</span>, <span class="string">"data.table"</span>, <span class="comment">## unzip and read table</span></span><br><span class="line">  <span class="string">"GEOquery"</span>, <span class="comment">## download</span></span><br><span class="line">  <span class="string">"FactoMineR"</span>, <span class="string">"factoextra"</span>, <span class="string">"ggfortify"</span>, <span class="comment">## PCA</span></span><br><span class="line">  <span class="string">"pheatmap"</span>, <span class="comment">## heatmap</span></span><br><span class="line">  <span class="string">"ggplot2"</span>, <span class="comment">## Volcano plot</span></span><br><span class="line">  <span class="string">"limma"</span>, <span class="string">"DESeq2"</span>, <span class="string">"edgeR"</span>, <span class="comment">## DEG</span></span><br><span class="line">  <span class="string">"clusterProfiler"</span>, <span class="string">"org.Hs.eg.db"</span>, <span class="comment">## annotation</span></span><br><span class="line">  <span class="string">"pathview"</span> <span class="comment">## kegg</span></span><br><span class="line">)</span><br><span class="line"><span class="comment"># Step3 Install the packages ----------------------------------------------</span></span><br><span class="line"></span><br><span class="line">lapply( bioPackages, </span><br><span class="line">  <span class="keyword">function</span>(bioPackage) &#123;</span><br><span class="line">    <span class="keyword">if</span> ( !<span class="keyword">require</span>( bioPackage, character.only = <span class="literal">T</span> ) ) &#123;</span><br><span class="line">      CRANpackages &lt;- available.packages()</span><br><span class="line"></span><br><span class="line">      <span class="comment">## install packages by CRAN</span></span><br><span class="line">      <span class="keyword">if</span> ( bioPackage %<span class="keyword">in</span>% rownames( CRANpackages ) ) &#123;</span><br><span class="line">        install.packages( bioPackage )</span><br><span class="line"></span><br><span class="line">      &#125;<span class="keyword">else</span>&#123;</span><br><span class="line">        <span class="comment">## install packages by bioconductor</span></span><br><span class="line">          <span class="comment">## R version &gt;= 3.5 ===&gt; BiocManager</span></span><br><span class="line">        <span class="keyword">if</span> ( as.character( sessionInfo()$R.version$minor ) &gt;= <span class="number">3.5</span> ) &#123;</span><br><span class="line">          <span class="keyword">if</span> (!requireNamespace(<span class="string">"BiocManager"</span>, quietly = <span class="literal">TRUE</span>))</span><br><span class="line">            install.packages(<span class="string">"BiocManager"</span>)</span><br><span class="line">          BiocManager::install(bioPackage, update = <span class="literal">TRUE</span>, ask = <span class="literal">FALSE</span>)</span><br><span class="line"></span><br><span class="line">        &#125;<span class="keyword">else</span>&#123;</span><br><span class="line">          <span class="comment">## R version &lt; 3.5 ===&gt; BiocInstaller</span></span><br><span class="line">          <span class="keyword">if</span> (!requireNamespace(<span class="string">"BiocInstaller"</span>, quietly = <span class="literal">TRUE</span>))</span><br><span class="line">            <span class="keyword">source</span>( <span class="string">"https://bioconductor.org/biocLite.R"</span> )</span><br><span class="line">          BiocInstaller::biocLite( bioPackage, ask = <span class="literal">FALSE</span>)</span><br><span class="line">        &#125;</span><br><span class="line">      &#125;</span><br><span class="line">    &#125;</span><br><span class="line">  &#125;</span><br><span class="line">)</span><br><span class="line"><span class="comment"># Step4 Remove new loaded packages ----------------------------------------</span></span><br><span class="line"></span><br><span class="line">allPackages &lt;- (.packages())</span><br><span class="line">newPackages &lt;- setdiff( allPackages, sysPackages )</span><br><span class="line">lapply( newPackages,</span><br><span class="line">        <span class="keyword">function</span>(package) &#123;</span><br><span class="line">          package &lt;- paste(<span class="string">'package:'</span>, package, sep = <span class="string">""</span>, collapse = <span class="literal">NULL</span>)</span><br><span class="line">          <span class="keyword">detach</span>( package, character.only = <span class="literal">TRUE</span> )</span><br><span class="line">        &#125;</span><br><span class="line">)</span><br><span class="line"><span class="comment">## 这一步是卸载已经加载的包，篇幅有限，同Step0步骤一样，之后就不重复书写了</span></span><br></pre></td></tr></table></figure>
<p>上面的代码组织的非常棒，值得初学者花十几个小时自行理解和模范，尤其是R包存放，已经批量下载，这些小技巧会极大的增加你使用R语言完成自己项目的效率。而且里面隐含着一些R不常用的概念，是你在大多数书籍上面都学不到的，更别说是视频教程了。</p>
<p>第二步 下载和处理TCGA数据</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># Step1 download TCGA dateset ---------------------------------------------</span></span><br><span class="line"></span><br><span class="line"><span class="keyword">if</span> (!file.exists( <span class="string">'./data/TCGA-BRCA.htseq_counts.Rdata'</span> )) &#123;</span><br><span class="line">  gzfile &lt;- <span class="string">"./raw_data/TCGA-BRCA.htseq_counts.tsv.gz"</span></span><br><span class="line">  download.file(<span class="string">"https://gdc.xenahubs.net/download/TCGA-BRCA/Xena_Matrices/TCGA-BRCA.htseq_counts.tsv.gz"</span>, </span><br><span class="line">                destfile = gzfile)</span><br><span class="line">  <span class="keyword">library</span>(R.utils)</span><br><span class="line">  gunzip(gzfile, remove = <span class="literal">F</span>)</span><br><span class="line">  <span class="keyword">library</span>(data.table)</span><br><span class="line">  raw_data &lt;- fread( <span class="string">"./raw_data/TCGA-BRCA.htseq_counts.tsv"</span>,</span><br><span class="line">                     sep = <span class="string">'	'</span>, header = <span class="literal">T</span>)</span><br><span class="line">  raw_data &lt;- as.data.frame( raw_data )</span><br><span class="line">  raw_data[<span class="number">1</span>:<span class="number">5</span>, <span class="number">1</span>:<span class="number">6</span>] </span><br><span class="line">  rownames( raw_data ) &lt;- raw_data[, <span class="number">1</span>]</span><br><span class="line">  raw_data &lt;- raw_data[, -<span class="number">1</span>]</span><br><span class="line">  raw_data[<span class="number">1</span>:<span class="number">5</span>, <span class="number">1</span>:<span class="number">6</span>]</span><br><span class="line">  raw_data &lt;- <span class="number">2</span>^raw_data - <span class="number">1</span></span><br><span class="line">  raw_data &lt;- ceiling( raw_data )</span><br><span class="line">  raw_data[<span class="number">1</span>:<span class="number">5</span>, <span class="number">1</span>:<span class="number">6</span>]</span><br><span class="line">  pick_row &lt;- apply( raw_data, <span class="number">1</span>, <span class="keyword">function</span>(x)&#123;</span><br><span class="line">    sum(x == <span class="number">0</span>) &lt; <span class="number">10</span></span><br><span class="line">  &#125;)</span><br><span class="line">  raw_data &lt;- raw_data[pick_row, ]</span><br><span class="line">  dim(raw_data  )</span><br><span class="line">  save( raw_data, file = <span class="string">'./data/TCGA-BRCA.htseq_counts.Rdata'</span> )</span><br><span class="line">&#125;<span class="keyword">else</span>&#123;</span><br><span class="line">  load(<span class="string">'./data/TCGA-BRCA.htseq_counts.Rdata'</span>)</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment"># Step2 Grouping by special clinical information --------------------------</span></span><br><span class="line"></span><br><span class="line"><span class="keyword">if</span> (!file.exists( <span class="string">'./raw_data/TCGA-BRCA.GDC_phenotype.tsv.gz'</span> )) &#123;</span><br><span class="line">  gzfile &lt;- <span class="string">"./raw_data/TCGA-BRCA.GDC_phenotype.tsv.gz"</span></span><br><span class="line">  download.file(<span class="string">"https://gdc.xenahubs.net/download/TCGA-BRCA/Xena_Matrices/TCGA-BRCA.GDC_phenotype.tsv.gz"</span>, </span><br><span class="line">                destfile = gzfile)</span><br><span class="line">  phenoData &lt;- read.table( gzfile,</span><br><span class="line">                           header = <span class="literal">T</span>,</span><br><span class="line">                           sep = <span class="string">'	'</span>,</span><br><span class="line">                           quote = <span class="string">''</span> )</span><br><span class="line">  save( phenoData, file = <span class="string">'./data/TCGA-BRCA.GDC_phenotype.Rdata'</span> )</span><br><span class="line">&#125;<span class="keyword">else</span>&#123;</span><br><span class="line">  load(<span class="string">'./data/TCGA-BRCA.GDC_phenotype.Rdata'</span>)</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">pheno_num &lt;- c()</span><br><span class="line">invisible(</span><br><span class="line">  lapply(<span class="number">1</span>:ncol(phenoData), </span><br><span class="line">         <span class="keyword">function</span>(col_num)&#123;</span><br><span class="line">           <span class="comment">## Assume that the classification project is between 2 and 4</span></span><br><span class="line">           <span class="keyword">if</span> (<span class="number">1</span> &lt; dim(table(phenoData[,col_num])) &amp; </span><br><span class="line">               dim(table(phenoData[,col_num])) &lt; <span class="number">5</span>) &#123;</span><br><span class="line">             pheno_num &lt;&lt;- append(pheno_num, col_num, after = length(pheno_num))</span><br><span class="line">           &#125;</span><br><span class="line">         &#125;</span><br><span class="line">  )</span><br><span class="line">)</span><br><span class="line">View(phenoData[, pheno_num])</span><br><span class="line">names(phenoData[, pheno_num])</span><br><span class="line"></span><br><span class="line"><span class="comment">## Category 3: TP53</span></span><br><span class="line"><span class="keyword">if</span> (!file.exists( <span class="string">'./raw_data/TCGA-BRCA.mutect2_snv.tsv.gz'</span> )) &#123;</span><br><span class="line">  gzfile &lt;- <span class="string">"./raw_data/TCGA-BRCA.mutect2_snv.tsv.gz"</span></span><br><span class="line">  download.file(<span class="string">"https://gdc.xenahubs.net/download/TCGA-BRCA/Xena_Matrices/TCGA-BRCA.mutect2_snv.tsv.gz"</span>, </span><br><span class="line">                destfile = gzfile)</span><br><span class="line">  mutype_file &lt;- read.table( gzfile,</span><br><span class="line">                             header = <span class="literal">T</span>,</span><br><span class="line">                             sep = <span class="string">'	'</span>,</span><br><span class="line">                             quote = <span class="string">''</span> )</span><br><span class="line">  save( mutype_file, file = <span class="string">'./data/TCGA-BRCA.mutect2_snv.Rdata'</span> )</span><br><span class="line">&#125;<span class="keyword">else</span>&#123;</span><br><span class="line">  load(<span class="string">'./data/TCGA-BRCA.mutect2_snv.Rdata'</span>)</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line"><span class="comment">## Pick columns that contains 'tp53'</span></span><br><span class="line">TP53 &lt;- mutype_file[mutype_file$gene == <span class="string">'tp53'</span> | mutype_file$gene == <span class="string">'TP53'</span>,]</span><br><span class="line">TP53_sample &lt;- unique( sort( TP53$Sample_ID ) )</span><br><span class="line">tumor_sample &lt;- colnames(raw_data)[substr( colnames(raw_data),<span class="number">14</span>,<span class="number">15</span>) &lt; <span class="number">10</span>]</span><br><span class="line">TP53_sample &lt;- intersect(tumor_sample, TP53_sample)</span><br><span class="line">noTP53_sample &lt;- setdiff(tumor_sample, TP53_sample)</span><br><span class="line">save(TP53_sample, noTP53_sample, file = <span class="string">'./data/sample_by_TP53.Rdata'</span>)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment"># Step3 Filt sample ------------------------------------------------</span></span><br><span class="line"></span><br><span class="line">load(<span class="string">'./data/TCGA-BRCA.htseq_counts.Rdata'</span>)</span><br><span class="line"></span><br><span class="line">tp53_sample &lt;- c(TP53_sample, noTP53_sample)</span><br><span class="line">AssayData &lt;- raw_data[, tp53_sample]</span><br><span class="line">dim(AssayData)</span><br><span class="line">group_list &lt;- c(rep(<span class="string">'TP53'</span>, length(TP53_sample)),</span><br><span class="line">                rep(<span class="string">'NO_TP53'</span>, length(noTP53_sample)))</span><br><span class="line">save(AssayData, group_list, file = <span class="string">'./data/tnbc_tumor_TP53_AssayData.Rdata'</span>)</span><br></pre></td></tr></table></figure>
<p><strong>XENA数据中心</strong>一直是我们大力推荐的TCGA下载方式，我们也有视频指导：</p>
<p><a href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247486746&amp;idx=1&amp;sn=b7c5ad7eff8cffb3620756f5feaff587&amp;chksm=9b484da1ac3fc4b741a6e3b59ba1bf668a11e21eb610f1a1d4582e33d429c67c14e6659c0771&amp;scene=21#wechat_redirect" target="_blank" rel="noopener">生信小技巧第8课，加上  TCGA的28篇教程-  批量下载TCGA所有数据</a></p>
<p>第三步 差异分析并绘图</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br><span class="line">127</span><br><span class="line">128</span><br><span class="line">129</span><br><span class="line">130</span><br><span class="line">131</span><br><span class="line">132</span><br><span class="line">133</span><br><span class="line">134</span><br><span class="line">135</span><br><span class="line">136</span><br><span class="line">137</span><br><span class="line">138</span><br><span class="line">139</span><br><span class="line">140</span><br><span class="line">141</span><br><span class="line">142</span><br><span class="line">143</span><br><span class="line">144</span><br><span class="line">145</span><br><span class="line">146</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">## 绘制热图和火山图的函数</span></span><br><span class="line">draw_heatmap &lt;- <span class="keyword">function</span>(nrDEG, type)&#123;</span><br><span class="line">  <span class="keyword">library</span>( <span class="string">"pheatmap"</span> )</span><br><span class="line">  nrDEG_Z = nrDEG[ order( nrDEG$logFC ), ]</span><br><span class="line">  nrDEG_F = nrDEG[ order( -nrDEG$logFC ), ]</span><br><span class="line">  choose_gene = c( rownames( nrDEG_Z )[<span class="number">1</span>:<span class="number">50</span>], rownames( nrDEG_F )[<span class="number">1</span>:<span class="number">50</span>] )</span><br><span class="line">  choose_matrix = AssayData[ choose_gene, ]</span><br><span class="line">  choose_matrix = t( scale( t( choose_matrix ) ) )</span><br><span class="line"></span><br><span class="line">  choose_matrix[choose_matrix &gt; <span class="number">2</span>] = <span class="number">2</span></span><br><span class="line">  choose_matrix[choose_matrix &lt; -<span class="number">2</span>] = -<span class="number">2</span></span><br><span class="line"></span><br><span class="line">  annotation_col = data.frame( CellType = factor( group_list ) )</span><br><span class="line">  rownames( annotation_col ) = colnames( AssayData )</span><br><span class="line">  filename &lt;- paste(<span class="string">'./fig/'</span>, type, <span class="string">'_heatmap_top100_logFC.png'</span>,</span><br><span class="line">                    sep = <span class="string">""</span>, collapse = <span class="literal">NULL</span>)</span><br><span class="line">  pheatmap( fontsize = <span class="number">6</span>, choose_matrix, annotation_col = annotation_col, </span><br><span class="line">            show_rownames = <span class="literal">T</span>, show_colnames = <span class="literal">F</span>,</span><br><span class="line">            annotation_legend = <span class="literal">T</span>, cluster_cols = <span class="literal">T</span>, </span><br><span class="line">            filename = filename)</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">draw_volcano &lt;- <span class="keyword">function</span>(nrDEG, type)&#123;</span><br><span class="line">  <span class="keyword">library</span>( <span class="string">"ggplot2"</span> )</span><br><span class="line">  logFC_cutoff &lt;- with( nrDEG, mean( abs( logFC ) ) + <span class="number">2</span> * sd( abs( logFC ) ) )</span><br><span class="line"></span><br><span class="line">  nrDEG$change = as.factor( ifelse( </span><br><span class="line">    nrDEG$P.Value &lt; <span class="number">0.01</span> &amp; abs(nrDEG$logFC) &gt; logFC_cutoff,</span><br><span class="line">    ifelse( nrDEG$logFC &gt; logFC_cutoff, <span class="string">'UP'</span>, <span class="string">'DOWN'</span> ), <span class="string">'NOT'</span> ) )</span><br><span class="line">  nrDEGfile &lt;- paste(<span class="string">'./data/'</span>, type, <span class="string">'_nrDEG_by_logFC.Rdata'</span>,</span><br><span class="line">                    sep = <span class="string">""</span>, collapse = <span class="literal">NULL</span>)</span><br><span class="line">  save( nrDEG, file = nrDEGfile )</span><br><span class="line"></span><br><span class="line">  this_tile &lt;- paste0( </span><br><span class="line">    <span class="string">'Cutoff for logFC is '</span>, round( logFC_cutoff, <span class="number">3</span> ),</span><br><span class="line">    <span class="string">'</span></span><br><span class="line"><span class="string">The number of up gene is '</span>, nrow(nrDEG[ nrDEG$change == <span class="string">'UP'</span>, ] ),</span><br><span class="line">    <span class="string">'</span></span><br><span class="line"><span class="string">The number of down gene is '</span>, nrow(nrDEG[ nrDEG$change == <span class="string">'DOWN'</span>, ] ) )</span><br><span class="line"></span><br><span class="line">  volcano = ggplot(data = nrDEG, </span><br><span class="line">                   aes( x = logFC, y = -log10(P.Value), color = change)) +</span><br><span class="line">    geom_point( alpha = <span class="number">0.4</span>, size = <span class="number">1.75</span> ) +</span><br><span class="line">    theme_set( theme_set( theme_bw( base_size = <span class="number">15</span> ) ) ) +</span><br><span class="line">    xlab( <span class="string">"log2 fold change"</span> ) + ylab( <span class="string">"-log10 p-value"</span> ) +</span><br><span class="line">    ggtitle( this_tile ) + </span><br><span class="line">    theme( plot.title = element_text( size = <span class="number">15</span>, hjust = <span class="number">0.5</span> )) +</span><br><span class="line">    scale_colour_manual( values = c(<span class="string">'blue'</span>,<span class="string">'black'</span>,<span class="string">'red'</span>) )</span><br><span class="line">  print( volcano )</span><br><span class="line">  filename &lt;- paste(<span class="string">'./fig/'</span>, type, <span class="string">'_volcano_logFC.png'</span>,</span><br><span class="line">                    sep = <span class="string">""</span>, collapse = <span class="literal">NULL</span>)</span><br><span class="line">  ggsave( volcano, filename = filename )</span><br><span class="line">  dev.off()</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment"># Step2 Then run edgeR ----------------------------------------------------</span></span><br><span class="line"></span><br><span class="line"><span class="keyword">library</span>(edgeR)</span><br><span class="line"><span class="comment">## A list-based S4 class for storing read counts and associated information </span></span><br><span class="line"><span class="comment">## from digital gene expression or sequencing technologies.</span></span><br><span class="line">DGElist &lt;- DGEList( counts = AssayData, group = factor(group_list) )</span><br><span class="line"><span class="comment">## Counts per Million or Reads per Kilobase per Million</span></span><br><span class="line">keep_gene &lt;- rowSums( cpm(DGElist) &gt; <span class="number">1</span> ) &gt;= <span class="number">2</span></span><br><span class="line">table(keep_gene)</span><br><span class="line">DGElist &lt;- DGElist[ keep_gene, , keep.lib.sizes = <span class="literal">FALSE</span> ]</span><br><span class="line"><span class="comment">## Calculate Normalization Factors to Align Columns of a Count Matrix</span></span><br><span class="line">DGElist &lt;- calcNormFactors( DGElist )</span><br><span class="line">DGElist$samples</span><br><span class="line"></span><br><span class="line">design &lt;- model.matrix( ~<span class="number">0</span> + factor(group_list) )</span><br><span class="line">rownames(design) &lt;- colnames(DGElist)</span><br><span class="line">colnames(design) &lt;- levels(factor(group_list))</span><br><span class="line"></span><br><span class="line"><span class="comment">## Estimate Common Dispersion for Negative Binomial GLMs</span></span><br><span class="line">DGElist &lt;- estimateGLMCommonDisp(DGElist, design)</span><br><span class="line"><span class="comment">## Estimate Trended Dispersion for Negative Binomial GLMs</span></span><br><span class="line">DGElist &lt;- estimateGLMTrendedDisp(DGElist, design)</span><br><span class="line"><span class="comment">## Empirical Bayes Tagwise Dispersions for Negative Binomial GLMs</span></span><br><span class="line">DGElist &lt;- estimateGLMTagwiseDisp(DGElist, design)</span><br><span class="line"></span><br><span class="line"><span class="comment">## glmFit fits genewise negative binomial glms, all with the same design matrix </span></span><br><span class="line"><span class="comment">## but possibly different dispersions, offsets and weights</span></span><br><span class="line">fit &lt;- glmFit(DGElist, design)</span><br><span class="line"><span class="comment">## https://www.biostars.org/p/110861/</span></span><br><span class="line"><span class="comment">## glmLRT conducts likelihood ratio tests for one or more coefficients in the </span></span><br><span class="line"><span class="comment">## linear model.</span></span><br><span class="line">results &lt;- glmLRT(fit, contrast = c(-<span class="number">1</span>, <span class="number">1</span>)) </span><br><span class="line">nrDEG_edgeR &lt;- topTags(results, n = nrow(DGElist))</span><br><span class="line">nrDEG_edgeR &lt;- as.data.frame(nrDEG_edgeR)</span><br><span class="line">head(nrDEG_edgeR)</span><br><span class="line">colnames(nrDEG_edgeR)[<span class="number">4</span>] &lt;- c(<span class="string">"P.Value"</span>) </span><br><span class="line">draw_heatmap(nrDEG = nrDEG_edgeR, type = <span class="string">'edgeR'</span>)</span><br><span class="line">draw_volcano(nrDEG = nrDEG_edgeR, type = <span class="string">'edgeR'</span>)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment"># Step3 Lastly run voom from limma ----------------------------------------</span></span><br><span class="line"></span><br><span class="line"><span class="keyword">library</span>(limma)</span><br><span class="line"><span class="comment">## A list-based S4 class for storing read counts and associated information </span></span><br><span class="line"><span class="comment">## from digital gene expression or sequencing technologies.</span></span><br><span class="line">DGElist &lt;- DGEList( counts = AssayData, group = factor(group_list) )</span><br><span class="line"><span class="comment">## Counts per Million or Reads per Kilobase per Million</span></span><br><span class="line">keep_gene &lt;- rowSums( cpm(DGElist) &gt; <span class="number">1</span> ) &gt;= <span class="number">2</span></span><br><span class="line">table(keep_gene)</span><br><span class="line">DGElist &lt;- DGElist[ keep_gene, , keep.lib.sizes = <span class="literal">FALSE</span> ]</span><br><span class="line"><span class="comment">## Calculate Normalization Factors to Align Columns of a Count Matrix</span></span><br><span class="line">DGElist &lt;- calcNormFactors( DGElist )</span><br><span class="line">DGElist$samples</span><br><span class="line"></span><br><span class="line">design &lt;- model.matrix( ~<span class="number">0</span> + factor(group_list) )</span><br><span class="line">rownames(design) &lt;- colnames(DGElist)</span><br><span class="line">colnames(design) &lt;- levels(factor(group_list))</span><br><span class="line"></span><br><span class="line"><span class="comment">## Transform RNA-Seq Data Ready for Linear Modelling</span></span><br><span class="line">v &lt;- voom(DGElist, design, plot = <span class="literal">TRUE</span>, normalize = <span class="string">"quantile"</span>)</span><br><span class="line"><span class="comment">## Fit linear model for each gene given a series of arrays</span></span><br><span class="line">fit &lt;- lmFit(v, design)</span><br><span class="line"></span><br><span class="line"><span class="comment">## Construct the contrast matrix corresponding to specified contrasts of a set </span></span><br><span class="line"><span class="comment">## of parameters.</span></span><br><span class="line">cont.matrix &lt;- makeContrasts(contrasts = c(<span class="string">'TP53-NO_TP53'</span>), levels = design)</span><br><span class="line"><span class="comment">## Given a linear model fit to microarray data, compute estimated coefficients </span></span><br><span class="line"><span class="comment">## and standard errors for a given set of contrasts.</span></span><br><span class="line">fit2 &lt;- contrasts.fit(fit, cont.matrix)</span><br><span class="line"><span class="comment">## Empirical Bayes Statistics for Differential Expression</span></span><br><span class="line">fit2 &lt;- eBayes(fit2)</span><br><span class="line"></span><br><span class="line">nrDEG_limma_voom = topTable(fit2, coef = <span class="string">'TP53-NO_TP53'</span>, n = <span class="literal">Inf</span>)</span><br><span class="line">nrDEG_limma_voom = na.omit(nrDEG_limma_voom)</span><br><span class="line">head(nrDEG_limma_voom)</span><br><span class="line">draw_heatmap(nrDEG = nrDEG_limma_voom, type = <span class="string">'limma_voom'</span>)</span><br><span class="line">draw_volcano(nrDEG = nrDEG_limma_voom, type = <span class="string">'limma_voom'</span>)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment"># Step4 Compare three methods ---------------------------------------------</span></span><br><span class="line"></span><br><span class="line">cor(na.omit(lf))</span><br><span class="line">mi &lt;- unique(c(rownames(nrDEG_edgeR),</span><br><span class="line">               rownames(nrDEG_limma_voom)))</span><br><span class="line">lf &lt;- data.frame(edgeR = nrDEG_edgeR[mi, <span class="number">1</span>],</span><br><span class="line">                 limma_voom = nrDEG_limma_voom[mi, <span class="number">1</span>])</span><br><span class="line">cor(na.omit(lf))</span><br><span class="line"><span class="comment">##                edgeR     limma_voom</span></span><br><span class="line"><span class="comment">##    edgeR      1.0000000  0.9118619</span></span><br><span class="line"><span class="comment">##    limma_voom 0.9118619  1.0000000</span></span><br></pre></td></tr></table></figure>
<p><img src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wzlMng3v2EzaoJlhf8OW0br7p5FcC9ZIjmDtFKIFgfIEV16948IQwK9JA3S98CKsS05J0ibuz7g8vQ/640?wx_fmt=png&amp;tp=webp&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" alt="img"><span class="image-caption-center">img</span></p>
<p>左图是edgeR的火山图结果，右图是limma的火山图结果</p>
<p><img src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wzlMng3v2EzaoJlhf8OW0brRtMabnvxhIHFhvfd8Y7ibbEDTURDtND6dA9I3oGiargIG5AdQOuZHmSw/640?wx_fmt=png&amp;tp=webp&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" alt="img"><span class="image-caption-center">img</span></p>
<p>左图是edgeR的火山图结果，右图是limma的火山图结果</p>
<p>第四步 通路注释</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">## function of KEGG pathway</span></span><br><span class="line">kegg_plot &lt;- <span class="keyword">function</span>(type) &#123;</span><br><span class="line">  kk.up &lt;- enrichKEGG(   gene          =  gene_up    ,</span><br><span class="line">                         organism      =  <span class="string">'hsa'</span>      ,</span><br><span class="line">                         universe      =  gene_all   ,</span><br><span class="line">                         pvalueCutoff  =  <span class="number">0.8</span>        ,</span><br><span class="line">                         qvalueCutoff  =  <span class="number">0.8</span>        )</span><br><span class="line">  kk.down &lt;- enrichKEGG( gene          =  gene_down  ,</span><br><span class="line">                         organism      =  <span class="string">'hsa'</span>      ,</span><br><span class="line">                         universe      =  gene_all   ,</span><br><span class="line">                         pvalueCutoff  =  <span class="number">0.8</span>        ,</span><br><span class="line">                         qvalueCutoff  =  <span class="number">0.8</span>        )</span><br><span class="line">  <span class="keyword">library</span>( <span class="string">"ggplot2"</span> )</span><br><span class="line">  kegg_down_dt &lt;- as.data.frame( kk.down )</span><br><span class="line">  kegg_up_dt &lt;- as.data.frame( kk.up )</span><br><span class="line">  down_kegg &lt;- kegg_down_dt[ kegg_down_dt$pvalue &lt; <span class="number">0.05</span>, ]</span><br><span class="line">  down_kegg$group &lt;- -<span class="number">1</span></span><br><span class="line">  up_kegg &lt;- kegg_up_dt[ kegg_up_dt$pvalue &lt; <span class="number">0.05</span>, ]</span><br><span class="line">  up_kegg$group &lt;- <span class="number">1</span></span><br><span class="line">  dat = rbind( up_kegg, down_kegg )</span><br><span class="line">  dat$pvalue = -log10( dat$pvalue )</span><br><span class="line">  dat$pvalue = dat$pvalue * dat$group</span><br><span class="line">  dat = dat[ order( dat$pvalue, decreasing = <span class="literal">F</span> ), ]</span><br><span class="line">  g_kegg &lt;- ggplot( dat, </span><br><span class="line">                    aes(x = reorder( Description, order( pvalue, decreasing = <span class="literal">F</span> )), </span><br><span class="line">                        y = pvalue, fill = group)) + </span><br><span class="line">    geom_bar( stat = <span class="string">"identity"</span> ) + </span><br><span class="line">    scale_fill_gradient( low = <span class="string">"blue"</span>, high = <span class="string">"red"</span>, guide = <span class="literal">FALSE</span> ) + </span><br><span class="line">    scale_x_discrete( name = <span class="string">"Pathway names"</span> ) +</span><br><span class="line">    scale_y_continuous( name = <span class="string">"log10P-value"</span> ) +</span><br><span class="line">    coord_flip() + theme_bw() + </span><br><span class="line">    theme( plot.title = element_text( hjust = <span class="number">0.5</span> ), </span><br><span class="line">           axis.text.x = element_text(size = <span class="number">10</span>),</span><br><span class="line">           axis.text.y = element_text(size = <span class="number">7</span>)) +</span><br><span class="line">    ggtitle( <span class="string">"Pathway Enrichment"</span> ) </span><br><span class="line">  print( g_kegg )</span><br><span class="line">  filename &lt;- paste(<span class="string">'./fig/kegg_up_down_'</span>, type, <span class="string">'.png'</span>, sep = <span class="string">""</span>, collapse = <span class="literal">NULL</span>)</span><br><span class="line">  ggsave( g_kegg, filename = filename )</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line"><span class="comment">## function of GO pathway</span></span><br><span class="line">go_plot &lt;- <span class="keyword">function</span>(type) &#123;</span><br><span class="line">  go_enrich_results &lt;- lapply( g_list, <span class="keyword">function</span>(gene) &#123;</span><br><span class="line">    lapply( c( <span class="string">'BP'</span>, <span class="string">'MF'</span>, <span class="string">'CC'</span> ) , <span class="keyword">function</span>(ont) &#123;</span><br><span class="line">      cat( paste( <span class="string">'Now process'</span>, ont ) )</span><br><span class="line">      ego &lt;- enrichGO( gene          =  gene,</span><br><span class="line">                       universe      =  gene_all,</span><br><span class="line">                       OrgDb         =  org.Hs.eg.db,</span><br><span class="line">                       ont           =  ont ,</span><br><span class="line">                       pAdjustMethod =  <span class="string">"BH"</span>,</span><br><span class="line">                       pvalueCutoff  =  <span class="number">0.99</span>,</span><br><span class="line">                       qvalueCutoff  =  <span class="number">0.99</span>,</span><br><span class="line">                       readable      =  <span class="literal">TRUE</span>)</span><br><span class="line">      print( head( ego ) )</span><br><span class="line">      <span class="keyword">return</span>( ego )</span><br><span class="line">    &#125;)</span><br><span class="line">  &#125;)</span><br><span class="line">  gofilename &lt;- paste(<span class="string">'./data/go_enrich_result'</span>, type, <span class="string">'.Rdata'</span>, </span><br><span class="line">                      sep = <span class="string">""</span>, collapse = <span class="literal">NULL</span>)</span><br><span class="line">  save( go_enrich_results, file = gofilename )</span><br><span class="line"></span><br><span class="line">  n1 = c( <span class="string">'gene_up'</span>, <span class="string">'gene_down'</span>, <span class="string">'gene_diff'</span> )</span><br><span class="line">  n2 = c( <span class="string">'BP'</span>, <span class="string">'MF'</span>, <span class="string">'CC'</span> ) </span><br><span class="line">  <span class="keyword">for</span> (i <span class="keyword">in</span> <span class="number">1</span>:<span class="number">3</span>) &#123;</span><br><span class="line">    <span class="keyword">for</span> (j <span class="keyword">in</span> <span class="number">1</span>:<span class="number">3</span>) &#123;</span><br><span class="line">      fn = paste0( <span class="string">'./fig/dotplot_'</span>, n1[i], <span class="string">'_'</span>, n2[j], <span class="string">'_'</span>, type, <span class="string">'.png'</span> )</span><br><span class="line">      cat( paste0( fn, <span class="string">'</span></span><br><span class="line"><span class="string">'</span> ) )</span><br><span class="line">      png( fn, res = <span class="number">150</span>, width = <span class="number">1080</span> )</span><br><span class="line">      print( dotplot( go_enrich_results[[i]][[j]] ) )</span><br><span class="line">      dev.off()</span><br><span class="line">    &#125;</span><br><span class="line">  &#125;</span><br><span class="line">&#125;</span><br><span class="line"><span class="comment"># Step1 annotation --------------------------------------------------------</span></span><br><span class="line"></span><br><span class="line"><span class="keyword">library</span>( <span class="string">"clusterProfiler"</span> )</span><br><span class="line"><span class="keyword">library</span>( <span class="string">"org.Hs.eg.db"</span> )</span><br><span class="line">keytypes(org.Hs.eg.db)</span><br><span class="line"><span class="keyword">library</span>(<span class="string">"stringr"</span>)</span><br><span class="line"></span><br><span class="line">load( <span class="string">"./data/edgeR_nrDEG_by_logFC.Rdata"</span> )</span><br><span class="line">load( <span class="string">"./data/limma_voom_nrDEG_by_logFC.Rdata"</span> )</span><br><span class="line"></span><br><span class="line"><span class="comment">## tans1: ENSEMBL2ENTREZID</span></span><br><span class="line">table( nrDEG$change )</span><br><span class="line">rownames( nrDEG ) &lt;- str_sub(rownames( nrDEG ), start = <span class="number">1</span>, end = <span class="number">15</span>)</span><br><span class="line">nrDEG$ENSEMBL &lt;- rownames( nrDEG )</span><br><span class="line">df &lt;- bitr( rownames( nrDEG ), fromType = <span class="string">"ENSEMBL"</span>, toType = c( <span class="string">"ENTREZID"</span> ), </span><br><span class="line">            OrgDb = org.Hs.eg.db )</span><br><span class="line">head( df )</span><br><span class="line">nrDEG = merge( nrDEG, df, by = <span class="string">'ENSEMBL'</span> )</span><br><span class="line">head( nrDEG )</span><br><span class="line"></span><br><span class="line">gene_up = nrDEG[ nrDEG$change == <span class="string">'UP'</span>, <span class="string">'ENTREZID'</span> ] </span><br><span class="line">gene_down = nrDEG[ nrDEG$change == <span class="string">'DOWN'</span>, <span class="string">'ENTREZID'</span> ]</span><br><span class="line">gene_diff = c( gene_up, gene_down )</span><br><span class="line">gene_all = as.character(nrDEG[ ,<span class="string">'ENTREZID'</span>] )</span><br><span class="line">g_list = list( gene_up = gene_up, gene_down = gene_down, gene_diff = gene_diff)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment"># Step2 pathway analysis ------------------------------------------</span></span><br><span class="line"></span><br><span class="line">kegg_plot(<span class="string">"edgeR"</span>)</span><br><span class="line">go_plot(<span class="string">"edgeR"</span>)</span><br><span class="line"></span><br><span class="line">kegg_plot(<span class="string">"limma_voom"</span>)</span><br><span class="line">go_plot(<span class="string">"limma_voom"</span>)</span><br><span class="line"></span><br><span class="line"><span class="keyword">library</span>(pathview)</span><br><span class="line">geneList &lt;- nrDEG$logFC</span><br><span class="line">names( geneList ) &lt;- nrDEG$ENTREZID</span><br><span class="line">geneList &lt;- sort( geneList, decreasing = <span class="literal">T</span> )</span><br><span class="line">pathview( gene.data = geneList, </span><br><span class="line">          pathway.id = dat$ID, </span><br><span class="line">          species = <span class="string">"hsa"</span>, </span><br><span class="line">          limit = list(gene = <span class="number">5</span>, cpd = <span class="number">1</span>))</span><br></pre></td></tr></table></figure>
<p><img src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wzlMng3v2EzaoJlhf8OW0brye8M0AforAmlWS17ILLqiae1NvJ3hTWUTy4GCgl2ibAAicKph7qrK1sibA/640?wx_fmt=png&amp;tp=webp&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" alt="img"><span class="image-caption-center">img</span></p>
<p>edgeR的kegg结果</p>
<p><img src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wzlMng3v2EzaoJlhf8OW0brNMYdexLRp9U6a1A0cYEVnibFiaydcG8KweTaJwLwqibhVd2YpxEyHyk6w/640?wx_fmt=png&amp;tp=webp&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" alt="img"><span class="image-caption-center">img</span></p>

          
          <hr>
          <ul class="pager no-print">
              
              <li class="previous">
                  <a href="/hexo/2019/03/15/2019-03-16_Circos_plot/" data-toggle="tooltip" data-placement="left" title="Circos plot">&larr; Previous Post</a>
              </li>
              
              
              <li class="next">
                  <a href="/hexo/2019/03/13/2019-03-13_breastcancer/" data-toggle="tooltip" data-placement="top" title="乳腺癌专题">Next Post&rarr;</a>
              </li>
              
          </ul>
          

      </div>
      
  <div class="hidden-xs col-sm-3 toc-col">
    <div class="toc-wrap">
        Table of Contents
        
          
        
    </div>
  </div>


    </div>
</article>

<!-- Footer -->
<!-- footer.ejs -->
<footer class="no-print">

    <div class="text-center">

        <ul class="list-inline">
            
            
            

            

            

            

            

            

        </ul>
        <div class="text-muted copyright">
            &copy;
            
            2017 - 2019
            
            -
            xt
            <br>
            
            
            
            | 
            <p>
                <span id="busuanzi_container_site_pv"><span id="busuanzi_value_site_pv"></span> <b>PV</b></span> -
                <span id="busuanzi_container_site_uv"><span id="busuanzi_value_site_uv"></span> <b>UV</b></span> -
                115.0k <b>Words</b>
        </p></div>

        <div class="search-container">
            <form class="site-search-form">
                <span class="glyphicon glyphicon-search"></span>
                <input type="text" id="local-search-input" class="st-search-input" placeholder="Search...">
            </form>
            <div id="local-search-result" class="local-search-result-cls"></div>
        </div>
        <script type="text/javascript" id="local.search.active">
            var inputArea = document.querySelector("#local-search-input");
            inputArea.onclick = function () {
                var root = "/hexo/";
                getSearchFile(root);
                this.onclick = null
            }
            inputArea.onkeydown = function () {
                if (event.keyCode == 13) return false
            }
        </script>

    </div>
</footer>

<!-- Custom Theme JavaScript -->
<script src="/hexo/js/main.js"></script>

<!-- async load function -->
<script>
    function async(u, c) {
        var d = document, t = 'script',
            o = d.createElement(t),
            s = d.getElementsByTagName(t)[0];
        o.src = u;
        if (c) {
            o.addEventListener('load', function (e) {
                c(null, e);
            }, false);
        }
        s.parentNode.insertBefore(o, s);
    }
</script>

<!--fastClick.js -->
<script>
    async("//cdn.bootcss.com/fastclick/1.0.6/fastclick.min.js", function () {
        var $nav = document.querySelector("nav");
        if ($nav) FastClick.attach($nav);
    })
</script><!-- hexo-inject:begin --><!-- hexo-inject:end -->



</body>

</html>
