<!DOCTYPE html>
<html lang="en">

<!-- layout.ejs-->
<head><meta name="generator" content="Hexo 3.8.0">
    <!-- hexo-inject:begin --><!-- hexo-inject:end --><meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="Adelaide">
    <meta name="author" content="xt">
    <meta name="keyword" content="">
    <link rel="canonical" href="https://taoshengxu.github.io/hexo/hexo/2020/01/06/2020-01-06_lincs_database/">
    <link rel="shortcut icon" href="/hexo/img/favicon.png">
    <link rel="alternate" type="application/atom+xml" title="IIM" href="/atom.xml">
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/animate.css/3.5.2/animate.min.css">
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
    <link rel="stylesheet" href="https://ajax.googleapis.com/ajax/libs/jqueryui/1.12.1/themes/smoothness/jquery-ui.css">
    <script src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.12.1/jquery-ui.min.js"></script>
    <script src="/hexo/js/search.js"></script>

    <title>
        
        lincs数据库资源｜undefined
        
    </title>

    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">

    <link rel="stylesheet" href="/hexo/css/main.css">

    
      <link href="//fonts.googleapis.com/css?family=Source+Code+Pro" rel="stylesheet" type="text/css">
      <link rel="stylesheet" href="/hexo/css/highlight.css">
    

    

    <script async src="//dn-lbstatics.qbox.me/busuanzi/2.3/busuanzi.pure.mini.js"></script>


    

    

    


    
  


    




    <script async defer src="https://buttons.github.io/buttons.js"></script>

<link rel="stylesheet" href="/hexo/css/prism-solarizedlight.css" type="text/css">
<link rel="stylesheet" href="/hexo/css/prism-line-numbers.css" type="text/css"><!-- hexo-inject:begin --><!-- hexo-inject:end --></head>

<style>
    header.intro-header {
        background-image: url('/hexo/img/northernlights-sisimiut-lake.jpg')
    }
</style>
<!-- hack iOS CSS :active style -->
<body ontouchstart="" class="animated fadeIn">
<!-- hexo-inject:begin --><!-- hexo-inject:end --><header>
  <nav class="navbar navbar-default header-navbar" id="nav-top" data-ispost="true" data-istags="false" data-ishome="false">
    <div class="container-fluid">
      <div class="navbar-header page-scroll">
        <button type="button" class="navbar-toggle" data-toggle="collapse" aria-expanded="false" data-target="#website_navbar">
          <span class="sr-only">Toggle navigation</span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <span class="navbar-brand animated pulse">
          <a class="brand-logo" href="/hexo/">
                <img src="/hexo/img/banner.jpg?h=350&amp;auto=compress&amp;cs=tinysrgb">
          </a>
        </span>
      </div>

      <div class="collapse navbar-collapse" id="website_navbar">
          <ul class="nav navbar-nav navbar-right">
              
                <li>
                  <a href="/hexo/">home</a>
                </li>
              
                <li>
                  <a href="/hexo/archives/">archives</a>
                </li>
              
                <li>
                  <a href="/hexo/categories/">categories</a>
                </li>
              
                <li>
                  <a href="/hexo/tags/">tags</a>
                </li>
              
                <li>
                  <a href="/hexo/columns/">columns</a>
                </li>
              
          </ul>
      </div>
  </div></nav>


  
    <style>
       .intro-header {
          background-image: url('/hexo/post_cover_images/moon.jpg');
      }
    </style>

    <div class="intro-header">
        <div class="container">
            <div class="row">
                <div class="col-lg-8 col-lg-offset-2 col-md-10 col-md-offset-1 text-center">
                    <div class="site-heading">
                        <h1>lincs数据库资源</h1>
                        
                        

                        
                          <span class="meta">
                               <span class="meta-item">Author: XT</span>
                               <span class="meta-item">Date: Jan 6, 2020</span>
                               
                          </span>
                          <div class="tags text-center">
                              Categories: 
                              <a class="tag" href="/hexo/categories/#Bioinformatics" title="Bioinformatics">Bioinformatics</a>
                              
                          </div>
                          <div class="tags text-center">
                              Tags: 
                          </div>
                        
                    </div>
                </div>
            </div>
        </div>
    </div>
  
</header>


<!-- Main Content -->
<!-- post.ejs -->
<article>
    <div class="container">
      <div class="col-lg-8 col-lg-offset-1 col-sm-9">
          
          <span class="post-count">1,667 words in total, 10 minutes required.</span>
          <hr>
          
          <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/hint.css/2.4.1/hint.min.css"><div class="table-container">
<table>
<thead>
<tr>
<th>GSE92742_Broad_LINCS_Level1_LXB_n1403502.tar.gz</th>
<th>1.2 Tb</th>
<th><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE92nnn/GSE92742/suppl/GSE92742_Broad_LINCS_Level1_LXB_n1403502.tar.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE92742&amp;format=file&amp;file=GSE92742_Broad_LINCS_Level1_LXB_n1403502.tar.gz" target="_blank" rel="noopener">(http)</a></th>
<th>TAR</th>
</tr>
</thead>
<tbody>
<tr>
<td>GSE92742_Broad_LINCS_Level2_GEX_delta_n49216x978.gctx.gz</td>
<td>98.9 Mb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE92nnn/GSE92742/suppl/GSE92742_Broad_LINCS_Level2_GEX_delta_n49216x978.gctx.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE92742&amp;format=file&amp;file=GSE92742_Broad_LINCS_Level2_GEX_delta_n49216x978.gctx.gz" target="_blank" rel="noopener">(http)</a></td>
<td>GCTX</td>
</tr>
<tr>
<td>GSE92742_Broad_LINCS_Level2_GEX_epsilon_n1269922x978.gctx.gz</td>
<td>2.3 Gb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE92nnn/GSE92742/suppl/GSE92742_Broad_LINCS_Level2_GEX_epsilon_n1269922x978.gctx.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE92742&amp;format=file&amp;file=GSE92742_Broad_LINCS_Level2_GEX_epsilon_n1269922x978.gctx.gz" target="_blank" rel="noopener">(http)</a></td>
<td>GCTX</td>
</tr>
<tr>
<td>GSE92742_Broad_LINCS_Level3_INF_mlr12k_n1319138x12328.gctx.gz</td>
<td>48.8 Gb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE92nnn/GSE92742/suppl/GSE92742_Broad_LINCS_Level3_INF_mlr12k_n1319138x12328.gctx.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE92742&amp;format=file&amp;file=GSE92742_Broad_LINCS_Level3_INF_mlr12k_n1319138x12328.gctx.gz" target="_blank" rel="noopener">(http)</a></td>
<td>GCTX</td>
</tr>
<tr>
<td>GSE92742_Broad_LINCS_Level4_ZSPCINF_mlr12k_n1319138x12328.gctx.gz</td>
<td>49.6 Gb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE92nnn/GSE92742/suppl/GSE92742_Broad_LINCS_Level4_ZSPCINF_mlr12k_n1319138x12328.gctx.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE92742&amp;format=file&amp;file=GSE92742_Broad_LINCS_Level4_ZSPCINF_mlr12k_n1319138x12328.gctx.gz" target="_blank" rel="noopener">(http)</a></td>
<td>GCTX</td>
</tr>
<tr>
<td>GSE92742_Broad_LINCS_Level5_COMPZ.MODZ_n473647x12328.gctx.gz</td>
<td>19.9 Gb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE92nnn/GSE92742/suppl/GSE92742_Broad_LINCS_Level5_COMPZ.MODZ_n473647x12328.gctx.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE92742&amp;format=file&amp;file=GSE92742_Broad_LINCS_Level5_COMPZ.MODZ_n473647x12328.gctx.gz" target="_blank" rel="noopener">(http)</a></td>
<td>GCTX</td>
</tr>
<tr>
<td>GSE92742_Broad_LINCS_README.pdf</td>
<td>25.8 Kb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE92nnn/GSE92742/suppl/GSE92742_Broad_LINCS_README.pdf" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE92742&amp;format=file&amp;file=GSE92742_Broad_LINCS_README.pdf" target="_blank" rel="noopener">(http)</a></td>
<td>PDF</td>
</tr>
<tr>
<td>GSE92742_Broad_LINCS_auxiliary_datasets.tar.gz</td>
<td>1.6 Gb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE92nnn/GSE92742/suppl/GSE92742_Broad_LINCS_auxiliary_datasets.tar.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE92742&amp;format=file&amp;file=GSE92742_Broad_LINCS_auxiliary_datasets.tar.gz" target="_blank" rel="noopener">(http)</a></td>
<td>TAR</td>
</tr>
<tr>
<td>GSE92742_Broad_LINCS_cell_info.txt.gz</td>
<td>2.5 Kb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE92nnn/GSE92742/suppl/GSE92742_Broad_LINCS_cell_info.txt.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE92742&amp;format=file&amp;file=GSE92742_Broad_LINCS_cell_info.txt.gz" target="_blank" rel="noopener">(http)</a></td>
<td>TXT</td>
</tr>
<tr>
<td>GSE92742_Broad_LINCS_gene_info.txt.gz</td>
<td>211.6 Kb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE92nnn/GSE92742/suppl/GSE92742_Broad_LINCS_gene_info.txt.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE92742&amp;format=file&amp;file=GSE92742_Broad_LINCS_gene_info.txt.gz" target="_blank" rel="noopener">(http)</a></td>
<td>TXT</td>
</tr>
<tr>
<td>GSE92742_Broad_LINCS_gene_info_delta_landmark.txt.gz</td>
<td>18.3 Kb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE92nnn/GSE92742/suppl/GSE92742_Broad_LINCS_gene_info_delta_landmark.txt.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE92742&amp;format=file&amp;file=GSE92742_Broad_LINCS_gene_info_delta_landmark.txt.gz" target="_blank" rel="noopener">(http)</a></td>
<td>TXT</td>
</tr>
<tr>
<td>GSE92742_Broad_LINCS_inst_info.txt.gz</td>
<td>11.5 Mb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE92nnn/GSE92742/suppl/GSE92742_Broad_LINCS_inst_info.txt.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE92742&amp;format=file&amp;file=GSE92742_Broad_LINCS_inst_info.txt.gz" target="_blank" rel="noopener">(http)</a></td>
<td>TXT</td>
</tr>
<tr>
<td>GSE92742_Broad_LINCS_pert_info.txt.gz</td>
<td>1.1 Mb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE92nnn/GSE92742/suppl/GSE92742_Broad_LINCS_pert_info.txt.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE92742&amp;format=file&amp;file=GSE92742_Broad_LINCS_pert_info.txt.gz" target="_blank" rel="noopener">(http)</a></td>
<td>TXT</td>
</tr>
<tr>
<td>GSE92742_Broad_LINCS_pert_metrics.txt.gz</td>
<td>908.9 Kb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE92nnn/GSE92742/suppl/GSE92742_Broad_LINCS_pert_metrics.txt.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE92742&amp;format=file&amp;file=GSE92742_Broad_LINCS_pert_metrics.txt.gz" target="_blank" rel="noopener">(http)</a></td>
<td>TXT</td>
</tr>
<tr>
<td>GSE92742_Broad_LINCS_sig_info.txt.gz</td>
<td>10.6 Mb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE92nnn/GSE92742/suppl/GSE92742_Broad_LINCS_sig_info.txt.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE92742&amp;format=file&amp;file=GSE92742_Broad_LINCS_sig_info.txt.gz" target="_blank" rel="noopener">(http)</a></td>
<td>TXT</td>
</tr>
<tr>
<td>GSE92742_Broad_LINCS_sig_metrics.txt.gz</td>
<td>11.9 Mb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE92nnn/GSE92742/suppl/GSE92742_Broad_LINCS_sig_metrics.txt.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE92742&amp;format=file&amp;file=GSE92742_Broad_LINCS_sig_metrics.txt.gz" target="_blank" rel="noopener">(http)</a></td>
<td>TXT</td>
</tr>
<tr>
<td>GSE92742_SHA512SUMS.txt.gz</td>
<td>1.4 Kb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE92nnn/GSE92742/suppl/GSE92742_SHA512SUMS.txt.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE92742&amp;format=file&amp;file=GSE92742_SHA512SUMS.txt.gz" target="_blank" rel="noopener">(http)</a></td>
<td>TXT</td>
</tr>
</tbody>
</table>
</div>
<p>分隔线</p>
<div class="table-container">
<table>
<thead>
<tr>
<th>GSE70138_Broad_LINCS_Level1_LXB_n345976_2017-03-06.tar.gz</th>
<th>223.0 Gb</th>
<th><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE70nnn/GSE70138/suppl/GSE70138_Broad_LINCS_Level1_LXB_n345976_2017-03-06.tar.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE70138&amp;format=file&amp;file=GSE70138_Broad_LINCS_Level1_LXB_n345976_2017-03-06.tar.gz" target="_blank" rel="noopener">(http)</a></th>
<th>TAR</th>
</tr>
</thead>
<tbody>
<tr>
<td>GSE70138_Broad_LINCS_Level2_GEX_n113012x978_2015-12-31.gct.gz</td>
<td>189.8 Mb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE70nnn/GSE70138/suppl/GSE70138_Broad_LINCS_Level2_GEX_n113012x978_2015-12-31.gct.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE70138&amp;format=file&amp;file=GSE70138_Broad_LINCS_Level2_GEX_n113012x978_2015-12-31.gct.gz" target="_blank" rel="noopener">(http)</a></td>
<td>GCT</td>
</tr>
<tr>
<td>GSE70138_Broad_LINCS_Level2_GEX_n345976x978_2017-03-06.gctx.gz</td>
<td>637.9 Mb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE70nnn/GSE70138/suppl/GSE70138_Broad_LINCS_Level2_GEX_n345976x978_2017-03-06.gctx.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE70138&amp;format=file&amp;file=GSE70138_Broad_LINCS_Level2_GEX_n345976x978_2017-03-06.gctx.gz" target="_blank" rel="noopener">(http)</a></td>
<td>GCTX</td>
</tr>
<tr>
<td>GSE70138_Broad_LINCS_Level2_GEX_n78980x978_2015-06-30.gct.gz</td>
<td>144.4 Mb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE70nnn/GSE70138/suppl/GSE70138_Broad_LINCS_Level2_GEX_n78980x978_2015-06-30.gct.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE70138&amp;format=file&amp;file=GSE70138_Broad_LINCS_Level2_GEX_n78980x978_2015-06-30.gct.gz" target="_blank" rel="noopener">(http)</a></td>
<td>GCT</td>
</tr>
<tr>
<td>GSE70138_Broad_LINCS_Level3_INF_mlr12k_n113012x22268_2015-12-31.gct.gz</td>
<td>11.1 Gb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE70nnn/GSE70138/suppl/GSE70138_Broad_LINCS_Level3_INF_mlr12k_n113012x22268_2015-12-31.gct.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE70138&amp;format=file&amp;file=GSE70138_Broad_LINCS_Level3_INF_mlr12k_n113012x22268_2015-12-31.gct.gz" target="_blank" rel="noopener">(http)</a></td>
<td>GCT</td>
</tr>
<tr>
<td>GSE70138_Broad_LINCS_Level3_INF_mlr12k_n345976x12328_2017-03-06.gctx.gz</td>
<td>12.6 Gb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE70nnn/GSE70138/suppl/GSE70138_Broad_LINCS_Level3_INF_mlr12k_n345976x12328_2017-03-06.gctx.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE70138&amp;format=file&amp;file=GSE70138_Broad_LINCS_Level3_INF_mlr12k_n345976x12328_2017-03-06.gctx.gz" target="_blank" rel="noopener">(http)</a></td>
<td>GCTX</td>
</tr>
<tr>
<td>GSE70138_Broad_LINCS_Level3_INF_mlr12k_n78980x22268_2015-06-30.gct.gz</td>
<td>4.3 Gb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE70nnn/GSE70138/suppl/GSE70138_Broad_LINCS_Level3_INF_mlr12k_n78980x22268_2015-06-30.gct.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE70138&amp;format=file&amp;file=GSE70138_Broad_LINCS_Level3_INF_mlr12k_n78980x22268_2015-06-30.gct.gz" target="_blank" rel="noopener">(http)</a></td>
<td>GCT</td>
</tr>
<tr>
<td>GSE70138_Broad_LINCS_Level4_ZSPCINF_mlr12k_n113012x22268_2015-12-31.gct.gz</td>
<td>14.5 Gb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE70nnn/GSE70138/suppl/GSE70138_Broad_LINCS_Level4_ZSPCINF_mlr12k_n113012x22268_2015-12-31.gct.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE70138&amp;format=file&amp;file=GSE70138_Broad_LINCS_Level4_ZSPCINF_mlr12k_n113012x22268_2015-12-31.gct.gz" target="_blank" rel="noopener">(http)</a></td>
<td>GCT</td>
</tr>
<tr>
<td>GSE70138_Broad_LINCS_Level4_ZSPCINF_mlr12k_n345976x12328_2017-03-06.gctx.gz</td>
<td>13.1 Gb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE70nnn/GSE70138/suppl/GSE70138_Broad_LINCS_Level4_ZSPCINF_mlr12k_n345976x12328_2017-03-06.gctx.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE70138&amp;format=file&amp;file=GSE70138_Broad_LINCS_Level4_ZSPCINF_mlr12k_n345976x12328_2017-03-06.gctx.gz" target="_blank" rel="noopener">(http)</a></td>
<td>GCTX</td>
</tr>
<tr>
<td>GSE70138_Broad_LINCS_Level4_ZSPCINF_mlr12k_n78980x22268_2015-06-30.gct.gz</td>
<td>4.5 Gb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE70nnn/GSE70138/suppl/GSE70138_Broad_LINCS_Level4_ZSPCINF_mlr12k_n78980x22268_2015-06-30.gct.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE70138&amp;format=file&amp;file=GSE70138_Broad_LINCS_Level4_ZSPCINF_mlr12k_n78980x22268_2015-06-30.gct.gz" target="_blank" rel="noopener">(http)</a></td>
<td>GCT</td>
</tr>
<tr>
<td>GSE70138_Broad_LINCS_Level4_ZSVCINF_mlr12k_n113012x22268_2015-12-31.gct.gz</td>
<td>14.7 Gb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE70nnn/GSE70138/suppl/GSE70138_Broad_LINCS_Level4_ZSVCINF_mlr12k_n113012x22268_2015-12-31.gct.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE70138&amp;format=file&amp;file=GSE70138_Broad_LINCS_Level4_ZSVCINF_mlr12k_n113012x22268_2015-12-31.gct.gz" target="_blank" rel="noopener">(http)</a></td>
<td>GCT</td>
</tr>
<tr>
<td>GSE70138_Broad_LINCS_Level4_ZSVCINF_mlr12k_n78980x22268_2015-06-30.gct.gz</td>
<td>4.6 Gb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE70nnn/GSE70138/suppl/GSE70138_Broad_LINCS_Level4_ZSVCINF_mlr12k_n78980x22268_2015-06-30.gct.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE70138&amp;format=file&amp;file=GSE70138_Broad_LINCS_Level4_ZSVCINF_mlr12k_n78980x22268_2015-06-30.gct.gz" target="_blank" rel="noopener">(http)</a></td>
<td>GCT</td>
</tr>
<tr>
<td>GSE70138_Broad_LINCS_Level5_COMPZ_n118050x12328_2017-03-06.gctx.gz</td>
<td>5.0 Gb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE70nnn/GSE70138/suppl/GSE70138_Broad_LINCS_Level5_COMPZ_n118050x12328_2017-03-06.gctx.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE70138&amp;format=file&amp;file=GSE70138_Broad_LINCS_Level5_COMPZ_n118050x12328_2017-03-06.gctx.gz" target="_blank" rel="noopener">(http)</a></td>
<td>GCTX</td>
</tr>
<tr>
<td>GSE70138_Broad_LINCS_README.pdf</td>
<td>25.8 Kb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE70nnn/GSE70138/suppl/GSE70138_Broad_LINCS_README.pdf" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE70138&amp;format=file&amp;file=GSE70138_Broad_LINCS_README.pdf" target="_blank" rel="noopener">(http)</a></td>
<td>PDF</td>
</tr>
<tr>
<td>GSE70138_Broad_LINCS_cell_info_2017-04-28.txt.gz</td>
<td>2.5 Kb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE70nnn/GSE70138/suppl/GSE70138_Broad_LINCS_cell_info_2017-04-28.txt.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE70138&amp;format=file&amp;file=GSE70138_Broad_LINCS_cell_info_2017-04-28.txt.gz" target="_blank" rel="noopener">(http)</a></td>
<td>TXT</td>
</tr>
<tr>
<td>GSE70138_Broad_LINCS_gene_info_2017-03-06.txt.gz</td>
<td>211.6 Kb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE70nnn/GSE70138/suppl/GSE70138_Broad_LINCS_gene_info_2017-03-06.txt.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE70138&amp;format=file&amp;file=GSE70138_Broad_LINCS_gene_info_2017-03-06.txt.gz" target="_blank" rel="noopener">(http)</a></td>
<td>TXT</td>
</tr>
<tr>
<td>GSE70138_Broad_LINCS_inst_info_2017-03-06.txt.gz</td>
<td>4.6 Mb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE70nnn/GSE70138/suppl/GSE70138_Broad_LINCS_inst_info_2017-03-06.txt.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE70138&amp;format=file&amp;file=GSE70138_Broad_LINCS_inst_info_2017-03-06.txt.gz" target="_blank" rel="noopener">(http)</a></td>
<td>TXT</td>
</tr>
<tr>
<td>GSE70138_Broad_LINCS_pert_info.txt.gz</td>
<td>81.7 Kb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE70nnn/GSE70138/suppl/GSE70138_Broad_LINCS_pert_info.txt.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE70138&amp;format=file&amp;file=GSE70138_Broad_LINCS_pert_info.txt.gz" target="_blank" rel="noopener">(http)</a></td>
<td>TXT</td>
</tr>
<tr>
<td>GSE70138_Broad_LINCS_pert_info_2017-03-06.txt.gz</td>
<td>80.4 Kb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE70nnn/GSE70138/suppl/GSE70138_Broad_LINCS_pert_info_2017-03-06.txt.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE70138&amp;format=file&amp;file=GSE70138_Broad_LINCS_pert_info_2017-03-06.txt.gz" target="_blank" rel="noopener">(http)</a></td>
<td>TXT</td>
</tr>
<tr>
<td>GSE70138_Broad_LINCS_sig_info_2017-03-06.txt.gz</td>
<td>1.9 Mb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE70nnn/GSE70138/suppl/GSE70138_Broad_LINCS_sig_info_2017-03-06.txt.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE70138&amp;format=file&amp;file=GSE70138_Broad_LINCS_sig_info_2017-03-06.txt.gz" target="_blank" rel="noopener">(http)</a></td>
<td>TXT</td>
</tr>
<tr>
<td>GSE70138_Broad_LINCS_sig_metrics_2017-03-06.txt.gz</td>
<td>3.0 Mb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE70nnn/GSE70138/suppl/GSE70138_Broad_LINCS_sig_metrics_2017-03-06.txt.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE70138&amp;format=file&amp;file=GSE70138_Broad_LINCS_sig_metrics_2017-03-06.txt.gz" target="_blank" rel="noopener">(http)</a></td>
<td>TXT</td>
</tr>
<tr>
<td>GSE70138_SHA512SUMS.txt.gz</td>
<td>1.0 Kb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE70nnn/GSE70138/suppl/GSE70138_SHA512SUMS.txt.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE70138&amp;format=file&amp;file=GSE70138_SHA512SUMS.txt.gz" target="_blank" rel="noopener">(http)</a></td>
<td>TXT</td>
</tr>
</tbody>
</table>
</div>
<p>分隔线</p>
<div class="table-container">
<table>
<thead>
<tr>
<th><strong>Supplementary file</strong></th>
<th><strong>Size</strong></th>
<th><strong>Download</strong></th>
<th><strong>File type/resource</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>GSE92743_Broad_Affymetrix_training_Level3_Q2NORM_n100000x12320.gctx.gz</td>
<td>2.2 Gb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE92nnn/GSE92743/suppl/GSE92743_Broad_Affymetrix_training_Level3_Q2NORM_n100000x12320.gctx.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE92743&amp;format=file&amp;file=GSE92743_Broad_Affymetrix_training_Level3_Q2NORM_n100000x12320.gctx.gz" target="_blank" rel="noopener">(http)</a></td>
<td>GCTX</td>
</tr>
<tr>
<td>GSE92743_Broad_GTEx_L1000_Holdout_Level3_Q2NORM_n1000x12320.gctx.gz</td>
<td>37.7 Mb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE92nnn/GSE92743/suppl/GSE92743_Broad_GTEx_L1000_Holdout_Level3_Q2NORM_n1000x12320.gctx.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE92743&amp;format=file&amp;file=GSE92743_Broad_GTEx_L1000_Holdout_Level3_Q2NORM_n1000x12320.gctx.gz" target="_blank" rel="noopener">(http)</a></td>
<td>GCTX</td>
</tr>
<tr>
<td>GSE92743_Broad_GTEx_L1000_Level3_Q2NORM_n3176x12320.gctx.gz</td>
<td>120.0 Mb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE92nnn/GSE92743/suppl/GSE92743_Broad_GTEx_L1000_Level3_Q2NORM_n3176x12320.gctx.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE92743&amp;format=file&amp;file=GSE92743_Broad_GTEx_L1000_Level3_Q2NORM_n3176x12320.gctx.gz" target="_blank" rel="noopener">(http)</a></td>
<td>GCTX</td>
</tr>
<tr>
<td>GSE92743_Broad_GTEx_L1000_NotUsed_Level3_Q2NORM_n1526x12320.gctx.gz</td>
<td>57.6 Mb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE92nnn/GSE92743/suppl/GSE92743_Broad_GTEx_L1000_NotUsed_Level3_Q2NORM_n1526x12320.gctx.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE92743&amp;format=file&amp;file=GSE92743_Broad_GTEx_L1000_NotUsed_Level3_Q2NORM_n1526x12320.gctx.gz" target="_blank" rel="noopener">(http)</a></td>
<td>GCTX</td>
</tr>
<tr>
<td>GSE92743_Broad_GTEx_L1000_Test_Level3_Q2NORM_n650x12320.gctx.gz</td>
<td>24.5 Mb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE92nnn/GSE92743/suppl/GSE92743_Broad_GTEx_L1000_Test_Level3_Q2NORM_n650x12320.gctx.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE92743&amp;format=file&amp;file=GSE92743_Broad_GTEx_L1000_Test_Level3_Q2NORM_n650x12320.gctx.gz" target="_blank" rel="noopener">(http)</a></td>
<td>GCTX</td>
</tr>
<tr>
<td>GSE92743_Broad_GTEx_RNAseq_Log2RPKM_q2norm_n3176x12320.gctx.gz</td>
<td>110.3 Mb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE92nnn/GSE92743/suppl/GSE92743_Broad_GTEx_RNAseq_Log2RPKM_q2norm_n3176x12320.gctx.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE92743&amp;format=file&amp;file=GSE92743_Broad_GTEx_RNAseq_Log2RPKM_q2norm_n3176x12320.gctx.gz" target="_blank" rel="noopener">(http)</a></td>
<td>GCTX</td>
</tr>
<tr>
<td>GSE92743_Broad_GTEx_gene_info.txt.gz</td>
<td>212.6 Kb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE92nnn/GSE92743/suppl/GSE92743_Broad_GTEx_gene_info.txt.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE92743&amp;format=file&amp;file=GSE92743_Broad_GTEx_gene_info.txt.gz" target="_blank" rel="noopener">(http)</a></td>
<td>TXT</td>
</tr>
<tr>
<td>GSE92743_Broad_GTEx_inst_info.txt.gz</td>
<td>42.0 Kb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE92nnn/GSE92743/suppl/GSE92743_Broad_GTEx_inst_info.txt.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE92743&amp;format=file&amp;file=GSE92743_Broad_GTEx_inst_info.txt.gz" target="_blank" rel="noopener">(http)</a></td>
<td>TXT</td>
</tr>
<tr>
<td>GSE92743_Broad_OLS_WEIGHTS_n979x11350.gctx.gz</td>
<td>20.3 Mb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE92nnn/GSE92743/suppl/GSE92743_Broad_OLS_WEIGHTS_n979x11350.gctx.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE92743&amp;format=file&amp;file=GSE92743_Broad_OLS_WEIGHTS_n979x11350.gctx.gz" target="_blank" rel="noopener">(http)</a></td>
<td>GCTX</td>
</tr>
<tr>
<td>GSE92743_Broad_README.pdf</td>
<td>25.8 Kb</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE92nnn/GSE92743/suppl/GSE92743_Broad_README.pdf" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE92743&amp;format=file&amp;file=GSE92743_Broad_README.pdf" target="_blank" rel="noopener">(http)</a></td>
<td>PDF</td>
</tr>
<tr>
<td>GSE92743_SHA512SUMS.txt.gz</td>
<td>939 b</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE92nnn/GSE92743/suppl/GSE92743_SHA512SUMS.txt.gz" target="_blank" rel="noopener">(ftp)</a><a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE92743&amp;format=file&amp;file=GSE92743_SHA512SUMS.txt.gz" target="_blank" rel="noopener">(http)</a></td>
<td>TXT</td>
</tr>
</tbody>
</table>
</div>
<hr>
<h2 id="1-原始数据下载"><a href="#1-原始数据下载" class="headerlink" title="1 .原始数据下载"></a>1 .原始数据下载</h2><ul>
<li><ol>
<li>L1000 LINCS Pilot PHASE I (n=1,319,138; updated March 03, 2017) <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE70138" target="_blank" rel="noopener">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE70138</a></li>
</ol>
</li>
<li><ol>
<li>L1000  LINCS PHASE <em>II</em> (n=354,123; updated March 30, 2017) <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE92742" target="_blank" rel="noopener">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE92742</a></li>
</ol>
</li>
<li><ol>
<li>Broad Institute L1000 Connectivity Map and RNAseq gene expression profiles on tissue samples from the GTEx project<br><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE92743" target="_blank" rel="noopener">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE92743</a></li>
</ol>
</li>
</ul>
<p>R语言</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br></pre></td><td class="code"><pre><span class="line">L1000_phase1 &lt;- read.csv(<span class="string">"./download/L1000_phase1.txt"</span>, header=<span class="literal">FALSE</span>, row.names=<span class="literal">NULL</span>,as.is = <span class="literal">TRUE</span>)</span><br><span class="line"><span class="keyword">library</span>(stringr)</span><br><span class="line"><span class="keyword">for</span>(i <span class="keyword">in</span> <span class="number">1</span>:nrow(L1000_phase1))</span><br><span class="line">&#123;</span><br><span class="line">  a=L1000_phase1[i,]</span><br><span class="line">  index1=str_locate(a,<span class="string">"ftp://"</span>)</span><br><span class="line">  index2=str_locate(a,<span class="string">"\\)\\[\\(http\\)\\]"</span>)</span><br><span class="line">  L1000_phase1[i,]=str_sub(a,index1[<span class="number">1</span>],index2[<span class="number">1</span>]-<span class="number">1</span>)</span><br><span class="line">&#125;</span><br><span class="line">write.table(L1000_phase1,file=<span class="string">"./download/L1000_phase1_ftp.txt"</span>,row.names = <span class="literal">FALSE</span>,col.names = <span class="literal">FALSE</span>,quote=<span class="literal">F</span>)</span><br><span class="line"></span><br><span class="line">L1000_phase2 &lt;- read.csv(<span class="string">"./download/L1000_phase2.txt"</span>, header=<span class="literal">FALSE</span>, row.names=<span class="literal">NULL</span>,as.is = <span class="literal">TRUE</span>)</span><br><span class="line"><span class="keyword">for</span>(i <span class="keyword">in</span> <span class="number">1</span>:nrow(L1000_phase2))</span><br><span class="line">&#123;</span><br><span class="line">  a=L1000_phase2[i,]</span><br><span class="line">  index1=str_locate(a,<span class="string">"ftp://"</span>)</span><br><span class="line">  index2=str_locate(a,<span class="string">"\\)\\[\\(http\\)\\]"</span>)</span><br><span class="line">  L1000_phase2[i,]=str_sub(a,index1[<span class="number">1</span>],index2[<span class="number">1</span>]-<span class="number">1</span>)</span><br><span class="line">&#125;</span><br><span class="line">write.table(L1000_phase2,file=<span class="string">"./download/L1000_phase2_ftp.txt"</span>,row.names = <span class="literal">FALSE</span>,col.names = <span class="literal">FALSE</span>,quote=<span class="literal">F</span>)</span><br><span class="line"></span><br><span class="line">CMap &lt;- read.csv(<span class="string">"./download/CMap.txt"</span>, header=<span class="literal">FALSE</span>, row.names=<span class="literal">NULL</span>,as.is = <span class="literal">TRUE</span>)</span><br><span class="line"><span class="keyword">for</span>(i <span class="keyword">in</span> <span class="number">1</span>:nrow(CMap))</span><br><span class="line">&#123;</span><br><span class="line">  a=CMap[i,]</span><br><span class="line">  index1=str_locate(a,<span class="string">"ftp://"</span>)</span><br><span class="line">  index2=str_locate(a,<span class="string">"\\)\\[\\(http\\)\\]"</span>)</span><br><span class="line">  CMap[i,]=str_sub(a,index1[<span class="number">1</span>],index2[<span class="number">1</span>]-<span class="number">1</span>)</span><br><span class="line">&#125;</span><br><span class="line">write.table(CMap,file=<span class="string">"./download/CMap_ftp.txt"</span>,row.names = <span class="literal">FALSE</span>,col.names = <span class="literal">FALSE</span>,quote=<span class="literal">F</span>)</span><br></pre></td></tr></table></figure>
<p>bash</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#echo $(basename $&#123;id##*/&#125;)</span></span><br><span class="line"><span class="comment">#下载</span></span><br><span class="line">cat L1000_phase1_ftp.txt | <span class="keyword">while</span> <span class="built_in">read</span> id</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">  nohup axel <span class="variable">$id</span> 1&gt; $(basename <span class="variable">$id</span> <span class="string">".gz"</span>).<span class="built_in">log</span> 2&gt;&amp;1 &amp; </span><br><span class="line"><span class="keyword">done</span></span><br><span class="line"></span><br><span class="line">cat L1000_phase2_ftp.txt | <span class="keyword">while</span> <span class="built_in">read</span> id</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">  <span class="built_in">echo</span> $(basename <span class="variable">$id</span> <span class="string">".gz"</span>) </span><br><span class="line">  nohup axel <span class="variable">$id</span> 1&gt; $(basename <span class="variable">$id</span> <span class="string">".gz"</span>).<span class="built_in">log</span> 2&gt;&amp;1 &amp;</span><br><span class="line"><span class="keyword">done</span></span><br><span class="line"></span><br><span class="line">cat CMap_ftp.txt | <span class="keyword">while</span> <span class="built_in">read</span> id</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">  <span class="built_in">echo</span> $(basename <span class="variable">$id</span> <span class="string">".gz"</span>) </span><br><span class="line">  nohup axel <span class="variable">$id</span> 1&gt; $(basename <span class="variable">$id</span> <span class="string">".gz"</span>).<span class="built_in">log</span> 2&gt;&amp;1 &amp;</span><br><span class="line"><span class="keyword">done</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line">ls *.<span class="built_in">log</span> | <span class="keyword">while</span> <span class="built_in">read</span> id; </span><br><span class="line"><span class="keyword">do</span> </span><br><span class="line">  <span class="built_in">echo</span> <span class="variable">$id</span>  </span><br><span class="line">  tail <span class="variable">$id</span> </span><br><span class="line">  <span class="built_in">echo</span> <span class="string">"#####################"</span></span><br><span class="line"><span class="keyword">done</span></span><br><span class="line">  </span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line">nohup axel ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE92nnn/GSE92742/suppl/GSE92742_Broad_LINCS_Level5_COMPZ.MODZ_n473647x12328.gctx.gz 1&gt; GSE92742_Broad_LINCS_Level5_COMPZ.MODZ_n473647x12328.gctx.log 2&gt;&amp;1 &amp;</span><br><span class="line"></span><br><span class="line">nohup axel ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE70nnn/GSE70138/suppl/GSE70138_Broad_LINCS_Level1_LXB_n345976_2017-03-06.tar.gz 1&gt; GSE70138_Broad_LINCS_Level1_LXB_n345976_2017-03-06.tar.log 2&gt;&amp;1 &amp;</span><br><span class="line"></span><br><span class="line">nohup axel ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE70nnn/GSE70138/suppl/GSE70138_Broad_LINCS_Level4_ZSPCINF_mlr12k_n345976x12328_2017-03-06.gctx.gz 1&gt; GSE70138_Broad_LINCS_Level4_ZSPCINF_mlr12k_n345976x12328_2017-03-06.gctx.log 2&gt;&amp;1 &amp;</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">nohup axel ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE92nnn/GSE92742/suppl/GSE92742_Broad_LINCS_Level1_LXB_n1403502.tar.gz 1&gt; GSE92742_Broad_LINCS_Level1_LXB_n1403502.tar.log 2&gt;&amp;1 &amp;</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line">nohup wget -c ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE92nnn/GSE92742/suppl/GSE92742_Broad_LINCS_Level1_LXB_n1403502.tar.gz 1&gt; GSE92742_Broad_LINCS_Level1_LXB_n1403502.tar.log 2&gt;&amp;1 &amp;</span><br></pre></td></tr></table></figure>
<p>NIH The Library of Integrated Network-Based Cellular Signatures</p>
<p><a href="http://www.lincsproject.org/LINCS/" target="_blank" rel="noopener">http://www.lincsproject.org/LINCS/</a></p>
<p> The LINCS L1000 project</p>
<p><a href="http://www.lincsproject.org/LINCS/tools/workflows/find-the-best-place-to-obtain-the-lincs-l1000-data" target="_blank" rel="noopener">http://www.lincsproject.org/LINCS/tools/workflows/find-the-best-place-to-obtain-the-lincs-l1000-data</a></p>
<p>The Harvard Medical School (HMS) LINCS Center is funded by NIH grant U54 HL127365 and is part of the NIH Library of Integrated Network-based Cellular Signatures (LINCS) Program. </p>
<p>Homepage： <a href="http://lincs.hms.harvard.edu/" target="_blank" rel="noopener">http://lincs.hms.harvard.edu/</a></p>
<p><a href="http://lincs.hms.harvard.edu/db/" target="_blank" rel="noopener">HMS LINCS Database</a></p>
<p><a href="http://lincs.hms.harvard.edu/resources/methods-software/software/" target="_blank" rel="noopener">software page</a></p>
<p> <a href="https://docs.google.com/document/d/1R_d_1UWO0C9y1TceXpKIUkhjk08DfvP1D19txi4Tbas/edit" target="_blank" rel="noopener">HMS LINCS Database URL Scheme</a></p>
<p><a href="https://docs.google.com/document/d/1IE9AqrF49sGgJwNvGxr0-f6gB8eG1AGguWbCdMgx2Qg/pub" target="_blank" rel="noopener">HMS LINCS Database Tutorial</a></p>
<p><a href="http://lincsportal.ccs.miami.edu/dcic-portal/" target="_blank" rel="noopener">LINCS Data Portal</a>.</p>
<p><em>reagent</em>   n. [试剂] 试剂; 反应物</p>
<p>perturbing agents 扰动剂</p>
<ul>
<li><p>1377 Proteins</p>
</li>
<li><p>165 Cell Lines</p>
</li>
<li><p>89 Antibodies</p>
</li>
<li><p>346 Datasets</p>
</li>
</ul>
<p><a href="https://clue.io" target="_blank" rel="noopener">https://clue.io</a></p>
<p><a href="https://clue.io/GEO-guide" target="_blank" rel="noopener">https://clue.io/GEO-guide</a></p>
<p>GEO CMap LINCS User Guide</p>
<p> User Guide for CMap L1000 datasets deposited in GEO</p>
<p>User guide for LINCS Phase I (GSE92742) which includes the GEO Series containing RNAi and CRISPR datasets(GSE106127), Phase II (GSE70138), and the LINCS-CMap contest dataset (GSE92743) is available<a href="https://docs.google.com/document/d/1q2gciWRhVCAAnlvF2iRLuJ7whrGP6QjpsCMq1yWz7dU/edit#heading=h.usef9o7fuux3" target="_blank" rel="noopener"> here.</a>**</p>
<p>GSE70138 that corresponds to genetic pertubational signatures from shRNA and CRISPR reagents</p>
<p>genetic perturbagens profiled 遗传扰动分析</p>
<p>LINCS-CMap （Connectivity Map）project </p>
<blockquote>
<h4 id="L1000-mRNA-profiling-assay"><a href="#L1000-mRNA-profiling-assay" class="headerlink" title="L1000 mRNA profiling assay"></a>L1000 mRNA profiling assay</h4><p>L1000 is a gene-expression profiling assay based on the direct measurement of a reduced representation of the transcriptome and computational inference of the portion of the transcriptome not explicitly measured. The number of landmark transcripts whose abundance is measured directly is approximately one thousand. Eighty additional invariant transcripts are also explicitly measured to enable quality control, scaling and normalization. Measurements of transcript abundance are made with a combination of a coupled ligase detection and polymerase chain reaction, optically-addressed microspheres, and a flow-cytometric detection system. The dataset consists of a collection of transcriptional expression data from cultured human cells treated with bioactive small molecules and genetic perturbagens profiled on the Broad Institute L1000 assay. Each well contains a biological sample and fluorescent Luminex beads. When a plate is scanned, the Luminex scanner reads off two values from each bead in each well. The color identifies the two possible genes that the bead may be measuring, and the intensity identifies the expression of the gene being measured. The two genes for each bead color are chosen such that one is a high expresser, and one is a low expresser. The scanning process yields intensity data for each bead in each well. Consider all beads of a particular color in a particular well. If there are 100 such beads, then roughly 70 of them detect the presence of some gene a, while roughly 30 detect the presence of gene b. Because we only have color and intensity information on each bead, we cannot measure directly which of the two genes a particular bead detects. However, we can indirectly determine the expression levels of genes a and b. To deconvolute the expression levels, a histogram where the x axis is the fluorescence intensity, and the y axis is the number of beads measured to fluoresce at that intensity is constructed. As most of the beads detect the presence of gene a, the larger peak indicates the expression of this gene. Similarly the smaller peak indicates the expression of gene b. The peaks are distinct because the genes were chosen to express at different levels. The peaks are separated using the K-means clustering algorithm. These values are further normalized to account for differences in overall fluorescence measurements between different plates, beads, and scanners. There are two sequential normalization steps: Luminex Invariant Set Normalization (LISS), and Quantile Normalization (QNORM). Normalization yields the expression values of the 978 Landmark L1000 genes. To obtain expression values for all genes in the transcriptome, it is assumed that an unmeasured gene can be predicted from the measured landmark genes via linear regression. To obtain a measure of relative gene expression, a Z-scoring procedure is used to generate signatures from normalized profiles. In the case of L1000 data, all signatures are represented by vectors measuring differential expression. For gene knockdown data, a single treatment represents a hairpin targeting a single gene. Because these hairpins are known to have off-target effects, the effects of targeting the actual gene are summarized over the different hairpins into concensus gene score (CGS). The figure below illustrates the flow of data through the signature generation pipeline.</p>
</blockquote>
<p> Normalization yields the expression values of the 978 Landmark L1000 genes</p>
<p>facilitates the discovery of connections  between genes, drugs, and diseases.</p>

          
          <hr>
          <ul class="pager no-print">
              
              <li class="previous">
                  <a href="/hexo/2020/01/08/2020-01-06_nextflow/" data-toggle="tooltip" data-placement="left" title="nextflow-rna_seq">&larr; Previous Post</a>
              </li>
              
              
              <li class="next">
                  <a href="/hexo/2020/01/04/2020-01-04_ncbi_database/" data-toggle="tooltip" data-placement="top" title="NCBI数据库资源">Next Post&rarr;</a>
              </li>
              
          </ul>
          

      </div>
      
  <div class="hidden-xs col-sm-3 toc-col">
    <div class="toc-wrap">
        Table of Contents
        
          <ol class="toc"><li class="toc-item toc-level-2"><a class="toc-link" href="#1-原始数据下载"><span class="toc-number">1.</span> <span class="toc-text">1 .原始数据下载</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#L1000-mRNA-profiling-assay"><span class="toc-number">1.0.1.</span> <span class="toc-text">L1000 mRNA profiling assay</span></a></li></ol></li></ol>
        
    </div>
  </div>


    </div>
</article>

<!-- Footer -->
<!-- footer.ejs -->
<footer class="no-print">

    <div class="text-center">

        <ul class="list-inline">
            
            
            

            

            

            

            

            

        </ul>
        <div class="text-muted copyright">
            &copy;
            
            2017 - 2020
            
            -
            xt
            <br>
            
            
            
            | 
            <p>
                <span id="busuanzi_container_site_pv"><span id="busuanzi_value_site_pv"></span> <b>PV</b></span> -
                <span id="busuanzi_container_site_uv"><span id="busuanzi_value_site_uv"></span> <b>UV</b></span> -
                170.1k <b>Words</b>
        </p></div>

        <div class="search-container">
            <form class="site-search-form">
                <span class="glyphicon glyphicon-search"></span>
                <input type="text" id="local-search-input" class="st-search-input" placeholder="Search...">
            </form>
            <div id="local-search-result" class="local-search-result-cls"></div>
        </div>
        <script type="text/javascript" id="local.search.active">
            var inputArea = document.querySelector("#local-search-input");
            inputArea.onclick = function () {
                var root = "/hexo/";
                getSearchFile(root);
                this.onclick = null
            }
            inputArea.onkeydown = function () {
                if (event.keyCode == 13) return false
            }
        </script>

    </div>
</footer>

<!-- Custom Theme JavaScript -->
<script src="/hexo/js/main.js"></script>

<!-- async load function -->
<script>
    function async(u, c) {
        var d = document, t = 'script',
            o = d.createElement(t),
            s = d.getElementsByTagName(t)[0];
        o.src = u;
        if (c) {
            o.addEventListener('load', function (e) {
                c(null, e);
            }, false);
        }
        s.parentNode.insertBefore(o, s);
    }
</script>

<!--fastClick.js -->
<script>
    async("//cdn.bootcss.com/fastclick/1.0.6/fastclick.min.js", function () {
        var $nav = document.querySelector("nav");
        if ($nav) FastClick.attach($nav);
    })
</script><!-- hexo-inject:begin --><!-- hexo-inject:end -->



</body>

</html>
