<!DOCTYPE html>
<html lang="en">

<!-- layout.ejs-->
<head><meta name="generator" content="Hexo 3.8.0">
    <!-- hexo-inject:begin --><!-- hexo-inject:end --><meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="Adelaide">
    <meta name="author" content="xt">
    <meta name="keyword" content="">
    <link rel="canonical" href="https://taoshengxu.github.io/hexo/hexo/2019/12/05/2019-12-05_TCGA_Datadownload/">
    <link rel="shortcut icon" href="/hexo/img/favicon.png">
    <link rel="alternate" type="application/atom+xml" title="IIM" href="/atom.xml">
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/animate.css/3.5.2/animate.min.css">
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
    <link rel="stylesheet" href="https://ajax.googleapis.com/ajax/libs/jqueryui/1.12.1/themes/smoothness/jquery-ui.css">
    <script src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.12.1/jquery-ui.min.js"></script>
    <script src="/hexo/js/search.js"></script>

    <title>
        
        TCGA数据类型下载总结｜undefined
        
    </title>

    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">

    <link rel="stylesheet" href="/hexo/css/main.css">

    
      <link href="//fonts.googleapis.com/css?family=Source+Code+Pro" rel="stylesheet" type="text/css">
      <link rel="stylesheet" href="/hexo/css/highlight.css">
    

    

    <script async src="//dn-lbstatics.qbox.me/busuanzi/2.3/busuanzi.pure.mini.js"></script>


    

    

    


    
  


    




    <script async defer src="https://buttons.github.io/buttons.js"></script>

<link rel="stylesheet" href="/hexo/css/prism-solarizedlight.css" type="text/css">
<link rel="stylesheet" href="/hexo/css/prism-line-numbers.css" type="text/css"><!-- hexo-inject:begin --><!-- hexo-inject:end --></head>

<style>
    header.intro-header {
        background-image: url('/hexo/img/northernlights-sisimiut-lake.jpg')
    }
</style>
<!-- hack iOS CSS :active style -->
<body ontouchstart="" class="animated fadeIn">
<!-- hexo-inject:begin --><!-- hexo-inject:end --><header>
  <nav class="navbar navbar-default header-navbar" id="nav-top" data-ispost="true" data-istags="false" data-ishome="false">
    <div class="container-fluid">
      <div class="navbar-header page-scroll">
        <button type="button" class="navbar-toggle" data-toggle="collapse" aria-expanded="false" data-target="#website_navbar">
          <span class="sr-only">Toggle navigation</span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <span class="navbar-brand animated pulse">
          <a class="brand-logo" href="/hexo/">
                <img src="/hexo/img/banner.jpg?h=350&amp;auto=compress&amp;cs=tinysrgb">
          </a>
        </span>
      </div>

      <div class="collapse navbar-collapse" id="website_navbar">
          <ul class="nav navbar-nav navbar-right">
              
                <li>
                  <a href="/hexo/">home</a>
                </li>
              
                <li>
                  <a href="/hexo/archives/">archives</a>
                </li>
              
                <li>
                  <a href="/hexo/categories/">categories</a>
                </li>
              
                <li>
                  <a href="/hexo/tags/">tags</a>
                </li>
              
                <li>
                  <a href="/hexo/columns/">columns</a>
                </li>
              
          </ul>
      </div>
  </div></nav>


  
    <style>
       .intro-header {
          background-image: url('/hexo/post_cover_images/moon.jpg');
      }
    </style>

    <div class="intro-header">
        <div class="container">
            <div class="row">
                <div class="col-lg-8 col-lg-offset-2 col-md-10 col-md-offset-1 text-center">
                    <div class="site-heading">
                        <h1>TCGA数据类型下载总结</h1>
                        
                        

                        
                          <span class="meta">
                               <span class="meta-item">Author: XT</span>
                               <span class="meta-item">Date: Dec 5, 2019</span>
                               
                          </span>
                          <div class="tags text-center">
                              Categories: 
                              <a class="tag" href="/hexo/categories/#Bioinformatics" title="Bioinformatics">Bioinformatics</a>
                              
                          </div>
                          <div class="tags text-center">
                              Tags: 
                          </div>
                        
                    </div>
                </div>
            </div>
        </div>
    </div>
  
</header>


<!-- Main Content -->
<!-- post.ejs -->
<article>
    <div class="container">
      <div class="col-lg-8 col-lg-offset-1 col-sm-9">
          
          <span class="post-count">1,887 words in total, 8 minutes required.</span>
          <hr>
          
          <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/hint.css/2.4.1/hint.min.css"><h2 id="1-Clinical-data"><a href="#1-Clinical-data" class="headerlink" title="1. Clinical data"></a>1. Clinical data</h2><p>用 <a href="https://bioconductor.org/packages/release/bioc/vignettes/TCGAbiolinks/inst/doc/clinical.html" target="_blank" rel="noopener">TCGAbiolinks</a>下载临床数据，</p>
<p>优点： 临床数据量不大，容易下载。 Parse XML clinical data 会获取非常高质量的follow-up数据，对生存分析作用非常大。</p>
<h1 id="2-甲基化数据"><a href="#2-甲基化数据" class="headerlink" title="2.甲基化数据"></a>2.甲基化数据</h1><p>甲基化数据量非常大，从firehose获取神经连接用迅雷下载，其他工具下载慢，容易中断。然后用TCGA Assembler 2进行数据处理。</p>
<h2 id="3-CNV数据"><a href="#3-CNV数据" class="headerlink" title="3.CNV数据"></a>3.CNV数据</h2><h5 id="1-Germline和Somatic-的区别"><a href="#1-Germline和Somatic-的区别" class="headerlink" title="1. Germline和Somatic 的区别"></a>1. Germline<code>和</code>Somatic 的区别</h5><p>对于DNA测序而言，主要识别SNP和CNV 两大类型的变异，每种变异类型又有<code>Germline</code>和<code>Somatic</code>的区别。</p>
<ul>
<li><code>Germline</code>指的是在胚胎发育早起出现的变异，这种变异会在所有细胞中广泛存在，是可以遗传给后代的变异；</li>
<li><code>Somatic</code>指的是体细胞变异，身体特定区域或者组织中出现的变异，通常不会遗传给后代。（<a href="https://www.jianshu.com/p/825e7d618838）" target="_blank" rel="noopener">https://www.jianshu.com/p/825e7d618838）</a></li>
</ul>
<h5 id="2-明白什么是CNV"><a href="#2-明白什么是CNV" class="headerlink" title="2. 明白什么是CNV"></a>2. 明白什么是CNV</h5><p>CNV（copy-number variant）是指拷贝数目变异，也称拷贝数目多态性（copy-number polymorphism，CNP），是一个大小介于1kb至3MB的DNA片段的变异，在人类及动植物基因组中广泛分布，其覆盖的核苷酸总数大大超过单核苷酸多态性（SNP）的总数，极大地丰富了基因组遗传变异的多样性。按照CNV是否致病可分为<strong>致病性CNV</strong>、<strong>非致病性CNV</strong>和<strong>不明临床意义CNV</strong>。</p>
<p><strong>CNV是相对参考基因组而言的概念</strong>。对正常人来说，基因组应该是二倍体的，所以凡是测到非2倍体的地方都是CNV。但是CNV本身就是人群遗传物质多样性的体现，所以对癌症样本来说，是需要过滤掉正常人体内的germline的CNV，得到somatic的CNV。</p>
<h5 id="3-TCGA的CNV测量及计算"><a href="#3-TCGA的CNV测量及计算" class="headerlink" title="3. TCGA的CNV测量及计算"></a>3. TCGA的CNV测量及计算</h5><p>TCGA里面主要是通过Affymetrix SNP6.0 array这款芯片来测拷贝数变异。值得注意的是，并不是只有TCGA利用了SNP6.这个芯片数据，著名的CCLE计划也对一千多细胞系处理了SNP6.0芯片，数据也是可以下载的。对SNP6.0的拷贝数芯片来说，通常是用PICNIC等软件处理原始数据，就可以得到的segment记录文件，每个样本一个结果，表明了某条染色体的某个区域内，SNP6.0芯片设计了多少个探针，芯片结果的拷贝数值是多少(这个区域的拷贝数用Segment_Mean)。</p>
<blockquote>
<p>Chromosome Start   End    Num_Probes   Segment_Mean<br>1    6173515  108012 26               -0.0397<br>11  627918    1672603 17               -0.92<br>11  687587    161534978176        0.0077<br>11  6153536  161539255              -2.7441<br>11  6154201  161550104              -0.8711<br>11  616566    17276849834630    0.0048<br>17  276891    67281114846          -1.7394<br>17  281190    49567471014901    0.0026<br>19  567651    1956765182            -1.6636</p>
</blockquote>
<p><code>通常二倍体的Segment_Mean值为0，可以用-0.2和0.2来作为该区域是否缺失或者扩增。</code></p>
<p>具体数据处理流程见NIH的TCGA官网：</p>
<p><a href="https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines/CNV_Pipeline/" target="_blank" rel="noopener">https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines/CNV_Pipeline/</a> </p>
<p>参考文献：<a href="http://mcr.aacrjournals.org/content/12/4/485.long" target="_blank" rel="noopener">http://mcr.aacrjournals.org/content/12/4/485.long</a> </p>
<h5 id="4-TCGA的CNV数据下载"><a href="#4-TCGA的CNV数据下载" class="headerlink" title="4. TCGA的CNV数据下载"></a>4. TCGA的CNV数据下载</h5><p>众所周知，TCGA的数据的开放程度分成了4个等级，一般人都是下载level 3 的数据，对CNV数据也是如此。我比较喜欢去broad institute下载TCGA的数据，所有的文件都以目录的形式存放着：</p>
<p><a href="https://gdac.broadinstitute.org/runs/stddata__latest/" target="_blank" rel="noopener">https://gdac.broadinstitute.org/runs/stddata__latest/</a></p>
<p>如果要下载level3的数据，就用<code>stddata__latest</code> 这个url即可，打开可以看到里面列出了所有的癌症种类，假如我们感兴趣的是BRCA，就直接点击进入，用下面的url即可。</p>
<p><a href="https://gdac.broadinstitute.org/runs/analyses__latest/data/BRCA/20160128/" target="_blank" rel="noopener">https://gdac.broadinstitute.org/runs/analyses__latest/data/BRCA/20160128/</a> </p>
<p>打开url可以看到非常多的文件，这里我们感兴趣的是snp6芯片的拷贝数结果，而且一般是基于hg19版本的。</p>
<figure class="highlight awk"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">wget -c -r -np -nH -k -L --cut-dirs <span class="number">6</span> -p  -A <span class="string">"*snp_6*hg19*Level_3*"</span>  http:<span class="regexp">//g</span>dac.broadinstitute.org<span class="regexp">/runs/</span>stddata__2016_01_28<span class="regexp">/data/</span>BRCA<span class="regexp">/20160128/</span></span><br></pre></td></tr></table></figure>
<figure class="highlight gcode"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">Merge_s<span class="symbol">np__genome_wide_snp_6</span>__broad_mit_edu__Level_<span class="number">3</span>__segme<span class="symbol">nted_scna_hg19</span>__seg </span><br><span class="line">Merge_s<span class="symbol">np__genome_wide_snp_6</span>__broad_mit_edu__Level_<span class="number">3</span>__segme<span class="symbol">nted_scna_minus_germline_cnv_hg19</span>__seg</span><br></pre></td></tr></table></figure>
<blockquote>
<p><code>其中minus了germline的CNV的就是我们想要的癌症相关的somatic CNV!</code></p>
</blockquote>
<p>而在TCGA Assembler 2中分别用 na_cnv.hg18, cna_cnv.hg19, cna_nocnv.hg18, and cna_nocnv.hg19 来区别表示</p>
<h2 id="4-RNASeq数据"><a href="#4-RNASeq数据" class="headerlink" title="4.  RNASeq数据"></a>4.  RNASeq数据</h2><h3 id="TCGA-RNAseq数据中FPKM与TPM转换介绍"><a href="#TCGA-RNAseq数据中FPKM与TPM转换介绍" class="headerlink" title="TCGA RNAseq数据中FPKM与TPM转换介绍"></a>TCGA RNAseq数据中FPKM与TPM转换介绍</h3><p><a href="https://shengxin.ren/topic/2" target="_blank" rel="noopener">TCGA</a>     <a href="https://shengxin.ren/topic/91" target="_blank" rel="noopener">TCGA下载</a>  <a href="https://shengxin.ren/topic/243" target="_blank" rel="noopener">TPM</a>  <a href="https://shengxin.ren/topic/217" target="_blank" rel="noopener">FPKM</a></p>
<p>在新版TCGA（GDC Portral）的RNAseq数据主要提供了三种数据下载，<strong>HTSeq-Counts，HTSeq-FPKM，HTSeq-FPKM-UQ</strong>.</p>
<p><a href="https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/" target="_blank" rel="noopener">TCGA官方文档</a></p>
<p><img src="https://taoshengxu.github.io/DocumentGit/img/GDC_RNASeq.png" alt=""></p>
<p>Counts是数据后台没有处理的原始表达量，而FPKM和FPKM-UQ是两种数据处理方法，也就是说，如果下载Counts数据，是表达量数据，如果下载FPKM数据，那么要注意这些数据是经过处理的。正常情况下，我们下载Counts数据就可以了，特殊情况选择FPKM数据也是可以的。<br><img src="https://taoshengxu.github.io/DocumentGit/img/FPKM.png" alt=""></p>
<p>下载数据后，在数据分析时，用的方法也是不同的，Counts数据一般使用edgeR包或DESeq包，对数据做分析；如果下载FPKM数据，就不能使用edgeR包，只能只用DESeq包进行处理。<br>在使用edgeR包做Counts数据处理时，是需要对数据进行normalize的，所以我们在下载数据时，下载counts是比较常用的。 </p>
<p>如果要用edgR等筛选差异的话会下载使用Counts数据，但是笔者在过去的数据分析中发现TCGA数据使用edgR等软件筛选差…</p>
<p>但是笔者在过去的数据分析中发现TCGA数据使用edgR等软件筛选差异基因并不理想，细思主要有两方面原因：</p>
<ul>
<li>一、肿瘤数据本身异质性很高</li>
<li>二、正常样本严重偏少</li>
</ul>
<p>基于此笔者几乎很少使用edgR等软件来筛选差异了，那么就很少下载Counts数据了，所以大多数情况下都是用RPKM，但是RPKM数据本身也是饱受诟病，像<a href="http://www.cbioportal.org/" target="_blank" rel="noopener">cbioportal</a>使用的是RSEM软件做的定量即TPM，在老版TCGA中也可以直接下载到这些数据，那么新版如何得到这样的数据呢，简单的办法是从FPKM转TPM.</p>
<p>FPKM: Fragments Per Kilobase of exon model per Million mapped fragments 即每千个碱基的转录每百万映射读取的fragments<br>TPM：TranscriptsPerKilobase of exonmodel per Million mapped reads 即每千个碱基的转录每百万映射读取的Transcripts<br>他们的计算公式如下：</p>
<figure class="highlight stata"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">FPKM= <span class="keyword">total</span> exon reads/ (mapped reads (Millions) * exon <span class="built_in">length</span>(KB))</span><br><span class="line">                    <span class="keyword">total</span> exon reads：某个样本mapping到特定基因的外显子上的所有的reads</span><br><span class="line">                    mapped reads (Millions) :某个样本的所有reads总和</span><br><span class="line">                    exon <span class="built_in">length</span>(KB)：某个基因的长度（外显子的长度的总和，以KB为单位）</span><br><span class="line">TPMi=(Ni/<span class="keyword">Li</span>)*1000000/<span class="built_in">sum</span>(N0/L0+……..+ Nm/Lm)</span><br><span class="line">                     Ni：mapping到基因i上的<span class="keyword">read</span>数； <span class="keyword">Li</span>：基因i的外显子长度的总和;<span class="keyword">m</span>：为所有基因的总数</span><br></pre></td></tr></table></figure>
<p>从以上公式我们可以进一步推导如下：</p>
<p>从以上公式我们可以进一步推导如下：</p>
<figure class="highlight routeros"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">1、FPKMi*(mapped reads (Millions)= total exon reads/ (mapped reads (Millions) * exon length(KB))</span><br><span class="line">2、total exon reads/ (mapped reads (Millions) * exon length(KB))=(Ni/Li)</span><br><span class="line">3、FPKMi*(mapped reads (Millions)=(Ni/Li)</span><br><span class="line">4、<span class="attribute">TPMi</span>=FPKMi*(mapped reads (Millions)<span class="number">*1000000</span>/(N0/L0+……<span class="built_in">..</span>+ Nm/Lm)</span><br><span class="line">5、<span class="attribute">TPMi</span>=FPKMi*(mapped reads (Millions)<span class="number">*1000000</span>/(FPKM0*(mapped reads (Millions)+……<span class="built_in">..</span>+ FPKMm*(mapped reads (Millions))</span><br><span class="line">6、<span class="attribute">TPMi</span>=FPKMi*1000000/(FPKM0+……..+ FPKMm)</span><br></pre></td></tr></table></figure>
<p>最终我们得到了TPM和FPKM的转换公公式，从公式里可以看到TPM就是等于该基因的FPKM占所有基因的FPKM的总和的比例乘以一百万，那么值得注意的是每个样本所有基因的TPM加和就等于一百万了，这很类似样本间标准化<br>最后你下载的TCGA FPKM数据转换TPM就不会懵逼了吧<br>参考文献：<a href="https://academic.oup.com/bioinformatics/article/26/4/493/243395/RNA-Seq-gene-expression-estimation-with-read" target="_blank" rel="noopener">https://academic.oup.com/bioinformatics/article/26/4/493/243395/RNA-Seq-gene-expression-estimation-with-read</a></p>
<h2 id="5-其他数据"><a href="#5-其他数据" class="headerlink" title="5. 其他数据"></a>5. 其他数据</h2><p>可以直接用TCGA assembler 2 下载。</p>

          
          <hr>
          <ul class="pager no-print">
              
              <li class="previous">
                  <a href="/hexo/2019/12/05/2019-12-05_TCGA_UCSC_xena/" data-toggle="tooltip" data-placement="left" title=" TCGA数据下载 UCSC Xena">&larr; Previous Post</a>
              </li>
              
              
              <li class="next">
                  <a href="/hexo/2019/12/05/2019-12-05_TCGAsampleName/" data-toggle="tooltip" data-placement="top" title="TCGA样本命名详解">Next Post&rarr;</a>
              </li>
              
          </ul>
          

      </div>
      
  <div class="hidden-xs col-sm-3 toc-col">
    <div class="toc-wrap">
        Table of Contents
        
          <ol class="toc"><li class="toc-item toc-level-2"><a class="toc-link" href="#1-Clinical-data"><span class="toc-number">1.</span> <span class="toc-text">1. Clinical data</span></a></li></ol><li class="toc-item toc-level-1"><a class="toc-link" href="#2-甲基化数据"><span class="toc-number"></span> <span class="toc-text">2.甲基化数据</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#3-CNV数据"><span class="toc-number">1.</span> <span class="toc-text">3.CNV数据</span></a><ol class="toc-child"><li class="toc-item toc-level-5"><a class="toc-link" href="#1-Germline和Somatic-的区别"><span class="toc-number">1.0.0.1.</span> <span class="toc-text">1. Germline和Somatic 的区别</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#2-明白什么是CNV"><span class="toc-number">1.0.0.2.</span> <span class="toc-text">2. 明白什么是CNV</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#3-TCGA的CNV测量及计算"><span class="toc-number">1.0.0.3.</span> <span class="toc-text">3. TCGA的CNV测量及计算</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#4-TCGA的CNV数据下载"><span class="toc-number">1.0.0.4.</span> <span class="toc-text">4. TCGA的CNV数据下载</span></a></li></ol></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#4-RNASeq数据"><span class="toc-number">2.</span> <span class="toc-text">4.  RNASeq数据</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#TCGA-RNAseq数据中FPKM与TPM转换介绍"><span class="toc-number">2.1.</span> <span class="toc-text">TCGA RNAseq数据中FPKM与TPM转换介绍</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#5-其他数据"><span class="toc-number">3.</span> <span class="toc-text">5. 其他数据</span></a></li>
        
    </div>
  </div>


    </div>
</article>

<!-- Footer -->
<!-- footer.ejs -->
<footer class="no-print">

    <div class="text-center">

        <ul class="list-inline">
            
            
            

            

            

            

            

            

        </ul>
        <div class="text-muted copyright">
            &copy;
            
            2017 - 2019
            
            -
            xt
            <br>
            
            
            
            | 
            <p>
                <span id="busuanzi_container_site_pv"><span id="busuanzi_value_site_pv"></span> <b>PV</b></span> -
                <span id="busuanzi_container_site_uv"><span id="busuanzi_value_site_uv"></span> <b>UV</b></span> -
                118.8k <b>Words</b>
        </p></div>

        <div class="search-container">
            <form class="site-search-form">
                <span class="glyphicon glyphicon-search"></span>
                <input type="text" id="local-search-input" class="st-search-input" placeholder="Search...">
            </form>
            <div id="local-search-result" class="local-search-result-cls"></div>
        </div>
        <script type="text/javascript" id="local.search.active">
            var inputArea = document.querySelector("#local-search-input");
            inputArea.onclick = function () {
                var root = "/hexo/";
                getSearchFile(root);
                this.onclick = null
            }
            inputArea.onkeydown = function () {
                if (event.keyCode == 13) return false
            }
        </script>

    </div>
</footer>

<!-- Custom Theme JavaScript -->
<script src="/hexo/js/main.js"></script>

<!-- async load function -->
<script>
    function async(u, c) {
        var d = document, t = 'script',
            o = d.createElement(t),
            s = d.getElementsByTagName(t)[0];
        o.src = u;
        if (c) {
            o.addEventListener('load', function (e) {
                c(null, e);
            }, false);
        }
        s.parentNode.insertBefore(o, s);
    }
</script>

<!--fastClick.js -->
<script>
    async("//cdn.bootcss.com/fastclick/1.0.6/fastclick.min.js", function () {
        var $nav = document.querySelector("nav");
        if ($nav) FastClick.attach($nav);
    })
</script><!-- hexo-inject:begin --><!-- hexo-inject:end -->



</body>

</html>
