<!DOCTYPE html>
<html lang="en">

<!-- layout.ejs-->
<head><meta name="generator" content="Hexo 3.8.0">
    <!-- hexo-inject:begin --><!-- hexo-inject:end --><meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="Adelaide">
    <meta name="author" content="xt">
    <meta name="keyword" content="">
    <link rel="canonical" href="https://taoshengxu.github.io/hexo/hexo/2017/11/07/2017-11-07_WES_Seq/">
    <link rel="shortcut icon" href="/hexo/img/favicon.png">
    <link rel="alternate" type="application/atom+xml" title="IIM" href="/atom.xml">
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/animate.css/3.5.2/animate.min.css">
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
    <link rel="stylesheet" href="https://ajax.googleapis.com/ajax/libs/jqueryui/1.12.1/themes/smoothness/jquery-ui.css">
    <script src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.12.1/jquery-ui.min.js"></script>
    <script src="/hexo/js/search.js"></script>

    <title>
        
        WES分析｜undefined
        
    </title>

    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">

    <link rel="stylesheet" href="/hexo/css/main.css">

    
      <link href="//fonts.googleapis.com/css?family=Source+Code+Pro" rel="stylesheet" type="text/css">
      <link rel="stylesheet" href="/hexo/css/highlight.css">
    

    

    <script async src="//dn-lbstatics.qbox.me/busuanzi/2.3/busuanzi.pure.mini.js"></script>


    

    

    


    
  
    <script type="text/x-mathjax-config">
      MathJax.Hub.Config({
        tex2jax: {
          inlineMath: [ ['$','$'], ["\\(","\\)"]  ],
          processEscapes: true,
          skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
        }
      });
    </script>

    <script type="text/x-mathjax-config">
      MathJax.Hub.Queue(function() {
        var all = MathJax.Hub.getAllJax(), i;
        for (i=0; i < all.length; i += 1) {
          all[i].SourceElement().parentNode.className += ' has-jax';
        }
      });
    </script>
    <script type="text/javascript" src="//cdn.bootcss.com/mathjax/2.7.1/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
  


    




    <script async defer src="https://buttons.github.io/buttons.js"></script>

<link rel="stylesheet" href="/hexo/css/prism-solarizedlight.css" type="text/css">
<link rel="stylesheet" href="/hexo/css/prism-line-numbers.css" type="text/css"><!-- hexo-inject:begin --><!-- hexo-inject:end --></head>

<style>
    header.intro-header {
        background-image: url('/hexo/img/northernlights-sisimiut-lake.jpg')
    }
</style>
<!-- hack iOS CSS :active style -->
<body ontouchstart="" class="animated fadeIn">
<!-- hexo-inject:begin --><!-- hexo-inject:end --><header>
  <nav class="navbar navbar-default header-navbar" id="nav-top" data-ispost="true" data-istags="false" data-ishome="false">
    <div class="container-fluid">
      <div class="navbar-header page-scroll">
        <button type="button" class="navbar-toggle" data-toggle="collapse" aria-expanded="false" data-target="#website_navbar">
          <span class="sr-only">Toggle navigation</span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <span class="navbar-brand animated pulse">
          <a class="brand-logo" href="/hexo/">
                <img src="/hexo/img/banner.jpg?h=350&amp;auto=compress&amp;cs=tinysrgb">
          </a>
        </span>
      </div>

      <div class="collapse navbar-collapse" id="website_navbar">
          <ul class="nav navbar-nav navbar-right">
              
                <li>
                  <a href="/hexo/">home</a>
                </li>
              
                <li>
                  <a href="/hexo/archives/">archives</a>
                </li>
              
                <li>
                  <a href="/hexo/categories/">categories</a>
                </li>
              
                <li>
                  <a href="/hexo/tags/">tags</a>
                </li>
              
                <li>
                  <a href="/hexo/columns/">columns</a>
                </li>
              
          </ul>
      </div>
  </div></nav>


  
    <style>
       .intro-header {
          background-image: url('/hexo/post_cover_images/moon.jpg');
      }
    </style>

    <div class="intro-header">
        <div class="container">
            <div class="row">
                <div class="col-lg-8 col-lg-offset-2 col-md-10 col-md-offset-1 text-center">
                    <div class="site-heading">
                        <h1>WES分析</h1>
                        
                        

                        
                          <span class="meta">
                               <span class="meta-item">Author: xt</span>
                               <span class="meta-item">Date: Nov 7, 2017</span>
                               
                          </span>
                          <div class="tags text-center">
                              Categories: 
                              <a class="tag" href="/hexo/categories/#Bioinformatics" title="Bioinformatics">Bioinformatics</a>
                              
                          </div>
                          <div class="tags text-center">
                              Tags: 
                              <a class="tag" href="/hexo/tags/#English" title="English">English</a>
                              
                          </div>
                        
                    </div>
                </div>
            </div>
        </div>
    </div>
  
</header>


<!-- Main Content -->
<!-- post.ejs -->
<article>
    <div class="container">
      <div class="col-lg-8 col-lg-offset-1 col-sm-9">
          
          <span class="post-count">4,362 words in total, 23 minutes required.</span>
          <hr>
          
          <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/hint.css/2.4.1/hint.min.css"><p><a href="https://taoshengxu.github.io/DocumentGit/html/给学徒的WES数据分析流程.html">给学徒的WES数据分析流程(无数据)</a>   <a href="https://www.bilibili.com/video/av28030610/?p=6" target="_blank" rel="noopener">对应B站视频</a></p>
<p><a href="https://taoshengxu.github.io/DocumentGit/html/ 给学徒的WES数据分析流程_配套.html">给学徒的WES数据分析流程—练习</a>  </p>
<p><a href="https://taoshengxu.github.io/DocumentGit/html/肿瘤全外显子测序数据分析流程大放送.html">肿瘤全外显子测序数据分析流程大放送</a>  本文采用这个数据</p>
<p><img src="https://taoshengxu.github.io/DocumentGit/img/WES_flow.PNG" alt=""></p>
<ol>
<li><p>测序数据的输入文件为raw fastq，质控软件为fastqc 和trim_galore，通过参数设置，得到结果文件clean fastq。</p>
</li>
<li><p>比对。输入文件为clean/HQ fastq，比对软件使用BWA，结果文件为raw sam，然后转成bam文件。</p>
<ol>
<li><p>使用BWA构建参考基因组的index数据库</p>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">bwa index genome<span class="selector-class">.fasta</span> -<span class="selector-tag">p</span> genome</span><br><span class="line"></span><br><span class="line">bwa index hg38.fa</span><br></pre></td></tr></table></figure>
</li>
<li><p>BWA-MEM比对（BWA-SW 和 BWA-backtrack用的少）</p>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">bwa mem -t <span class="number">4</span> hg38 F2-<span class="number">19.1</span><span class="selector-class">.fastq</span> F2-<span class="number">19.2</span><span class="selector-class">.fastq</span> &gt; F2-<span class="number">19</span><span class="selector-class">.mem</span><span class="selector-class">.sam</span></span><br></pre></td></tr></table></figure>
</li>
</ol>
</li>
<li><p>找变异。IGV可视化bam，GATK过滤bam，GATK找变异得到vcf</p>
</li>
<li><p>注释。输入文件为HQ vcf，比对软件Annovar，使用参考数据库，得到结果文件即与功能相关的变异信息.(<a href="https://www.jianshu.com/p/9d4ba8cf8d2a" target="_blank" rel="noopener">https://www.jianshu.com/p/9d4ba8cf8d2a</a>)</p>
</li>
</ol>
<p>2019.12.25</p>
<h3 id="软件安装"><a href="#软件安装" class="headerlink" title="软件安装"></a>软件安装</h3><figure class="highlight mipsasm"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">conda create -n wes python=<span class="number">2</span></span><br><span class="line">conda info --envs</span><br><span class="line">source activate wes</span><br><span class="line"><span class="comment">#需要软件ascp,samtools,vcftools,bcftools,fastqc,multiqc,trim_galore,bwa,sra-tools,cutadapt,gatk4</span></span><br><span class="line">conda <span class="keyword">install </span><span class="keyword">sra-tools</span></span><br><span class="line"><span class="keyword">conda </span><span class="keyword">install </span>samtools gatk4</span><br><span class="line">conda <span class="keyword">install </span>-y <span class="keyword">bcftools </span>vcftools  snpeff</span><br><span class="line">conda <span class="keyword">install </span>-y <span class="keyword">multiqc </span>qualimap</span><br><span class="line">conda <span class="keyword">install </span>fastqc</span><br><span class="line">conda <span class="keyword">install </span><span class="keyword">bedtools</span></span><br><span class="line"><span class="keyword">conda </span><span class="keyword">install </span><span class="keyword">Subread </span><span class="comment">## 包含featureCounts</span></span><br></pre></td></tr></table></figure>
<h3 id="GATK依赖数据库下载"><a href="#GATK依赖数据库下载" class="headerlink" title="GATK依赖数据库下载"></a>GATK依赖数据库下载</h3><p><a href="https://software.broadinstitute.org/gatk/download/bundle" target="_blank" rel="noopener">https://software.broadinstitute.org/gatk/download/bundle</a><br>ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash">下载GATK Bundle 依赖文件</span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">#/data/reference/GATK/bundle/hg19</span></span></span><br><span class="line">cd GATK </span><br><span class="line">nohup wget -r -c -nH ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg38/*&amp;</span><br><span class="line"></span><br><span class="line">ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/1000G_phase1.indels.hg19.sites.vcf.gz</span><br><span class="line">ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/1000G_phase1.indels.hg19.sites.vcf.idx.gz</span><br><span class="line">ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/dbsnp_138.hg19.vcf.gz</span><br><span class="line">ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/ucsc.hg19.dict.gz</span><br><span class="line">ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/ucsc.hg19.fasta.gz</span><br><span class="line"><span class="meta">#</span><span class="bash">https://github.com/snewhouse/ngs_nextflow/wiki/GATK-Bundle</span></span><br><span class="line">231M Jul 2 05:14 1000G_phase1.indels.hg19.sites.vcf</span><br><span class="line">1.2M Jul 2 10:45 1000G_phase1.indels.hg19.sites.vcf.idx</span><br><span class="line">11G Jul 2 08:05 dbsnp_138.hg19.vcf</span><br><span class="line">2.5K Jul 1 04:31 hg19.dict</span><br><span class="line">3.0G Jun 30 21:29 hg19.fasta</span><br><span class="line">6.6K Jun 30 22:54 hg19.fasta.amb</span><br><span class="line">944 Jun 30 22:54 hg19.fasta.ann</span><br><span class="line">2.9G Jun 30 22:54 hg19.fasta.bwt</span><br><span class="line">788 Jul 2 01:53 hg19.fasta.fai</span><br><span class="line">739M Jun 30 22:54 hg19.fasta.pac</span><br><span class="line">1.5G Jun 30 23:23 hg19.fasta.sa</span><br><span class="line">87M Jul 2 05:37 Mills_and_1000G_gold_standard.indels.hg19.sites.vcf</span><br><span class="line">2.3M Jul 2 10:45 Mills_and_1000G_gold_standard.indels.hg19.sites.vcf.idx</span><br><span class="line"></span><br><span class="line">/GATK/resources/bundle/hg38/bwa_index/</span><br><span class="line">|-- [ 20K]  gatk_hg38.amb</span><br><span class="line">|-- [445K]  gatk_hg38.ann</span><br><span class="line">|-- [3.0G]  gatk_hg38.bwt</span><br><span class="line">|-- [767M]  gatk_hg38.pac</span><br><span class="line">|-- [1.5G]  gatk_hg38.sa</span><br><span class="line">|-- [6.2K]  hg38.bwa_index.log </span><br><span class="line">`-- [ 566]  run.sh</span><br><span class="line"></span><br><span class="line">/GATK/resources/bundle/hg38/</span><br><span class="line">|-- [1.8G]  1000G_phase1.snps.high_confidence.hg38.vcf.gz</span><br><span class="line">|-- [2.0M]  1000G_phase1.snps.high_confidence.hg38.vcf.gz.tbi</span><br><span class="line">|-- [3.2G]  dbsnp_146.hg38.vcf.gz</span><br><span class="line">|-- [3.0M]  dbsnp_146.hg38.vcf.gz.tbi</span><br><span class="line">|-- [ 59M]  hapmap_3.3.hg38.vcf.gz</span><br><span class="line">|-- [1.5M]  hapmap_3.3.hg38.vcf.gz.tbi</span><br><span class="line">|-- [568K]  Homo_sapiens_assembly38.dict</span><br><span class="line">|-- [3.0G]  Homo_sapiens_assembly38.fasta</span><br><span class="line">|-- [157K]  Homo_sapiens_assembly38.fasta.fai</span><br><span class="line">|-- [ 20M]  Mills_and_1000G_gold_standard.indels.hg38.vcf.gz</span><br><span class="line">|-- [1.4M]  Mills_and_1000G_gold_standard.indels.hg38.vcf.gz.tbi</span><br></pre></td></tr></table></figure>
<h3 id="文章"><a href="#文章" class="headerlink" title="文章"></a>文章</h3><blockquote>
<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4812767/" target="_blank" rel="noopener">Whole-exome sequencing identifies MST1R as a genetic susceptibility gene in nasopharyngeal carcinoma</a></p>
</blockquote>
<h3 id="1-数据下载"><a href="#1-数据下载" class="headerlink" title="1.数据下载"></a>1.数据下载</h3><blockquote>
<p>We sequenced the blood samples from 161 NPC cases, including 39 EAO cases, 63 FH+ cases from 52 independent families, and 59 sporadic cases by WES and achieved an average coverage of 49-fold on target (range of 32- to 76-fold). An additional 2,160 NPC cases and 2,433 healthy controls from Hong Kong were further examined for the selected candidate variants. </p>
<p>Data deposition: Whole-exome sequencing data for the early-age onset cases have been deposited in the Sequence Read Archive (SRA) database (accession ID <a href="https://www.ncbi.nlm.nih.gov/sra?term=SRA291701" target="_blank" rel="noopener">SRA291701</a>).</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">#挑选5个样本进行分析 npc.sra.txt (现在GEO数据从新组织了，这种byExp下载不到了)</span></span></span><br><span class="line">SRX445405 MALE NPC15 SRR1139956 NPC15F NO SRS540548 NPC15F-T</span><br><span class="line">SRX445406 MALE NPC15 SRR1139958 NPC15F NO SRS540549 NPC15F-N</span><br><span class="line">SRX445407 MALE NPC29 SRR1139966 NPC29F YES SRS540550 NPC29F-T</span><br><span class="line">SRX445408 MALE NPC29 SRR1139973 NPC29F YES SRS540551 NPC29F-N</span><br><span class="line">SRX445409 FEMALE NPC10 SRR1139999 NPC10F NO SRS540552 NPC10F-T</span><br><span class="line">SRX445410 FEMALE NPC10 SRR1140007 NPC10F NO SRS540553 NPC10F-N</span><br><span class="line">SRX445411 FEMALE NPC34 SRR1140015 NPC34F NO SRS540554 NPC34F-T</span><br><span class="line">SRX445412 FEMALE NPC34 SRR1140023 NPC34F NO SRS540555 NPC34F-N</span><br><span class="line">SRX445413 MALE NPC37 SRR1140044 NPC37F YES SRS540556 NPC37F-T</span><br><span class="line">SRX445414 MALE NPC37 SRR1140045 NPC37F YES SRS540557 NPC37F-N</span><br><span class="line"></span><br><span class="line">cat npc.sra.txt | cut -d " " -f 4| while read id</span><br><span class="line">do echo $id</span><br><span class="line">done</span><br><span class="line">cat npc.sra.txt | while read id</span><br><span class="line">do</span><br><span class="line">array=($id)</span><br><span class="line">echo  $&#123;array[3]&#125;.sra  $&#123;array[7]&#125; </span><br><span class="line">fastq-dump --gzip --split-3 -A $&#123;array[7]&#125;  $&#123;array[3]&#125;.sra </span><br><span class="line">done </span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># SRR_Acc_List.txt挑选前4个样本数据下载一下。这种常规方法产生的地址不能用。用prefetch 安装SRR下载没有问题，但是国内网络会断。</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">##cat SRR_Acc_List.txt | head -n 4 | while read id</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">#do </span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">#echo ftp://ftp.ncbi.nlm.nih.gov/sra/sra-##instant/reads/ByRun/sra/SRR/$&#123;id:0:6&#125;/$id/$id.sra &gt;&gt; ftp.txt</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">#done</span></span></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash">正确的地址如下（挑4个样本分析一下，用迅雷下载）：</span></span><br><span class="line">https://sra-downloadb.be-md.ncbi.nlm.nih.gov/sos1/sra-pub-run-5/SRR2188163/SRR2188163.1</span><br><span class="line">https://sra-downloadb.be-md.ncbi.nlm.nih.gov/sos1/sra-pub-run-5/SRR2188164/SRR2188164.1</span><br><span class="line">https://sra-downloadb.be-md.ncbi.nlm.nih.gov/sos1/sra-pub-run-5/SRR2188165/SRR2188165.1</span><br><span class="line">https://sra-downloadb.be-md.ncbi.nlm.nih.gov/sos1/sra-pub-run-5/SRR2188166/SRR2188166.1</span><br></pre></td></tr></table></figure>
</blockquote>
<h3 id="2-抽取Fastq文件-fastq-dump"><a href="#2-抽取Fastq文件-fastq-dump" class="headerlink" title="2. 抽取Fastq文件(fastq-dump)"></a>2. 抽取Fastq文件(<a href="https://taoshengxu.github.io/DocumentGit/html/Fastq_dump一个神奇的软件.html">fastq-dump</a>)</h3><blockquote>
</blockquote>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br></pre></td><td class="code"><pre><span class="line">mkdir &#123;raw,clean,qc,align,mutation&#125;</span><br><span class="line"></span><br><span class="line"><span class="comment">##直接用excel制作config文件</span></span><br><span class="line"><span class="comment">##fastq-dump从SRA文件中提取fastq文件,fastq-dump是sra-tools软件中的一个功能，</span></span><br><span class="line"><span class="comment">##fastq-dump -h  （查看是否添加完成）</span></span><br><span class="line"></span><br><span class="line"><span class="comment">#arr=($id)用于将字符串$id分割到数组$arr $&#123;arr[0]&#125; $&#123;arr[1]&#125; ... 分别存储分割后的数组第1 2 ... 项 ，$&#123;arr[@]&#125;存储整个数组。变量$IFS存储着分隔符，将其设为逗号 "," OLD_IFS用于备份默认的分隔符，使用完后将之恢复默认。</span></span><br><span class="line"><span class="built_in">cd</span> WES</span><br><span class="line">cat config.txt|<span class="keyword">while</span> <span class="built_in">read</span> id;</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">    OLD_IFS=<span class="string">"<span class="variable">$IFS</span>"</span> </span><br><span class="line">    IFS=<span class="string">","</span> </span><br><span class="line">    arr=(<span class="variable">$id</span>)</span><br><span class="line">    IFS=<span class="string">"<span class="variable">$OLD_IFS</span>"</span> </span><br><span class="line">    srr=<span class="variable">$&#123;arr[0]&#125;</span></span><br><span class="line">    sample=<span class="variable">$&#123;arr[1]&#125;</span></span><br><span class="line">    <span class="built_in">echo</span> <span class="variable">$sample</span></span><br><span class="line">    nohup fastq-dump -A  <span class="variable">$sample</span> -O ./raw  --gzip --split-3  ./sra/<span class="variable">$srr</span>.sra &amp; </span><br><span class="line"><span class="keyword">done</span></span><br><span class="line"><span class="comment">#.gz就可以进行比对分析，不需要解压。</span></span><br><span class="line"><span class="comment">#cd raw</span></span><br><span class="line"><span class="comment">#ls *.gz | while read id</span></span><br><span class="line"><span class="comment">#do</span></span><br><span class="line"><span class="comment">#echo $id</span></span><br><span class="line"><span class="comment">#gunzip $id</span></span><br><span class="line"><span class="comment"># #gunzip -c $id #加-c可以保留原来的.gz文件</span></span><br><span class="line"><span class="comment">#done</span></span><br></pre></td></tr></table></figure>
<h3 id="3-QC质控fasqc和multiqc"><a href="#3-QC质控fasqc和multiqc" class="headerlink" title="3.QC质控fasqc和multiqc"></a>3.QC质控fasqc和multiqc</h3><p>使用fasqc和multiqc两个软件查看测序质量，以及使用trim_galore软件进行过滤低质量reads和去除接头。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">cd</span> qc</span><br><span class="line"><span class="comment">## 单独运行fastqc会报错，但是不影响使用。</span></span><br><span class="line"><span class="comment">## 速度比较快</span></span><br><span class="line">find ../raw  -name *gz |xargs fastqc -t 16 -o ./ </span><br><span class="line"><span class="comment"># multiqc 对所有qc结果综合一下</span></span><br><span class="line">multiqc ./</span><br></pre></td></tr></table></figure>
<h3 id="假设质量很差，trim-galore过滤"><a href="#假设质量很差，trim-galore过滤" class="headerlink" title="假设质量很差，trim_galore过滤"></a>假设质量很差，trim_galore过滤</h3><blockquote>
<p><strong>本文章中的测序reads的编码格式是phred+64 而不是传统的33</strong></p>
</blockquote>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">### filter the bad quality reads and remove adaptors. </span></span><br><span class="line"></span><br><span class="line"><span class="built_in">cd</span> clean</span><br><span class="line">find ../raw  -name *gz |grep 1.fastq.gz &gt; 1</span><br><span class="line">find ../raw  -name *gz |grep 2.fastq.gz &gt; 2</span><br><span class="line">paste 1 2  &gt; config </span><br><span class="line">cat config  |<span class="keyword">while</span> <span class="built_in">read</span> id</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">  arr=(<span class="variable">$&#123;id&#125;</span>)</span><br><span class="line">  fq1=<span class="variable">$&#123;arr[0]&#125;</span></span><br><span class="line">  fq2=<span class="variable">$&#123;arr[1]&#125;</span> </span><br><span class="line">  <span class="built_in">echo</span> <span class="variable">$fq1</span> <span class="variable">$fq2</span> </span><br><span class="line">  <span class="comment">#nohup trim_galore -q 25 --phred33 --length 36 -e 0.1 --stringency 3 --paired -o ./  $fq1 $fq2 &amp; </span></span><br><span class="line">  nohup trim_galore -q 25 --phred33 --length 36 -e 0.1 --stringency 3 --paired -o ./  <span class="variable">$fq1</span> <span class="variable">$fq2</span> &amp;</span><br><span class="line"><span class="keyword">done</span></span><br></pre></td></tr></table></figure>
<h3 id="4-比对到基因组-bwa"><a href="#4-比对到基因组-bwa" class="headerlink" title="4.比对到基因组 bwa"></a>4.比对到基因组 bwa</h3><blockquote>
<p>bwa mem 推荐使用的算法，支持较长的read长度，同时支持剪接性比对（split alignments)，但是BWA-MEM是更新的算法，也更快，更准确，且 BWA-MEM 对于 70bp-100bp 的 Illumina 数据来说，效果也更好些。</p>
<p>bwa -t 线程数</p>
<p>bwa index  建立索引</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#根据reference genome data(e.g. ref.fa) 建立索引，索引可以直接使用，这样就不需要参考基因组fastq文件了</span></span><br><span class="line">$ bwa index ref.fa -p genome <span class="comment">###可以不加-p genome，这样建立索引都是以ref为前缀</span></span><br><span class="line"></span><br><span class="line"><span class="comment">## 构建bwa索引</span></span><br><span class="line"><span class="built_in">cd</span> /data/reference/GATK/bundle/hg38</span><br><span class="line">mkdir bwa_index </span><br><span class="line"><span class="built_in">cd</span> bwa_index</span><br><span class="line">nohup bwa index -a bwtsw -p gatk_hg38 ../Homo_sapiens_assembly38.fasta 1&gt;hg38.bwa_index.log 2&gt;&amp;1 &amp;</span><br></pre></td></tr></table></figure>
<p>samtools sort -@ 线程数</p>
<p>索引文件都要取到路径下的文件名（去掉文件类型，点前面的部分）<br><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">cd</span> align</span><br><span class="line">ls ../clean/*1.fq.gz &gt;1</span><br><span class="line">ls ../clean/*2.fq.gz &gt;2</span><br><span class="line">cut -d<span class="string">"/"</span> -f 3 1 |cut -d<span class="string">"_"</span> -f 1,2  &gt; 0 <span class="comment">#取样本名</span></span><br><span class="line">paste 0 1 2 &gt; config </span><br><span class="line">conda activate wes</span><br><span class="line"><span class="comment">#bwa_index=/data/reference/hg38/UCSC/bwa_index_UCSC_hg38/hg38</span></span><br><span class="line">bwa_index=/data/reference/GATK/bundle/hg38/bwa_index/hg38</span><br><span class="line">cat config |<span class="keyword">while</span> <span class="built_in">read</span> id</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">  arr=(<span class="variable">$id</span>)</span><br><span class="line">  fq1=<span class="variable">$&#123;arr[1]&#125;</span></span><br><span class="line">  fq2=<span class="variable">$&#123;arr[2]&#125;</span></span><br><span class="line">  sample=<span class="variable">$&#123;arr[0]&#125;</span></span><br><span class="line">  <span class="built_in">echo</span> <span class="variable">$sample</span> <span class="variable">$fq1</span> <span class="variable">$fq2</span> </span><br><span class="line">  bwa mem -t 18 -R <span class="string">"@RG\tID:<span class="variable">$sample</span>\tSM:<span class="variable">$sample</span>\tLB:WGS\tPL:Illumina"</span> <span class="variable">$bwa_index</span> <span class="variable">$fq1</span> <span class="variable">$fq2</span>  | samtools sort -@ 18 -o <span class="variable">$sample</span>.bam -    </span><br><span class="line"><span class="keyword">done</span></span><br><span class="line"><span class="comment">## 管道输出的是一个内容，需要用 “-” 来占位， samtools sort 需要输入文件</span></span><br><span class="line"><span class="comment">## bwa -R "@RG\tID:$sample\tSM:$sample\tLB:WGS\tPL:Illumina" 参数是为了后续GATK分析，否者GATK会报错</span></span><br></pre></td></tr></table></figure></p>
</blockquote>
<h3 id="5-去除PCR重复"><a href="#5-去除PCR重复" class="headerlink" title="5. 去除PCR重复"></a>5. 去除PCR重复</h3><blockquote>
<p>NGS上机测序前需要进行PCR一步，由一个模板链扩增出一簇，从而在上机测序的时候表现出为1个增强的良点，即一个reads。若一个模板扩增出了多簇，结果得到了多个reads，这些reads的坐标(coordinates)是相近的。在进行了reads比对后需要将这些由PCR duplicates获得的reads去掉，并只保留最高比对质量的read。使用rmdup命令即可完成。</p>
<p><strong>‘samtoolsmarkdup操作的正确顺序</strong></p>
</blockquote>
<figure class="highlight css"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">//https://www.jianshu.com/p/7752a6633d1a</span><br><span class="line">//1.The first sort can be omitted if the file is already name ordered</span><br><span class="line"><span class="selector-tag">samtools</span> <span class="selector-tag">sort</span> <span class="selector-tag">-n</span> <span class="selector-tag">-o</span> <span class="selector-tag">namesort</span><span class="selector-class">.bam</span> <span class="selector-tag">example</span><span class="selector-class">.bam</span></span><br><span class="line">//2.Add ms and MC tags for markdup to use later</span><br><span class="line"><span class="selector-tag">samtools</span> <span class="selector-tag">fixmate</span> <span class="selector-tag">-m</span> <span class="selector-tag">namesort</span><span class="selector-class">.bam</span> <span class="selector-tag">fixmate</span><span class="selector-class">.bam</span></span><br><span class="line">//3. Markdup needs position order</span><br><span class="line"><span class="selector-tag">samtools</span> <span class="selector-tag">sort</span> <span class="selector-tag">-o</span> <span class="selector-tag">positionsort</span><span class="selector-class">.bam</span> <span class="selector-tag">fixmate</span><span class="selector-class">.bam</span></span><br><span class="line">//4. Finally mark duplicates</span><br><span class="line"><span class="selector-tag">samtools</span> <span class="selector-tag">markdup</span> <span class="selector-tag">positionsort</span><span class="selector-class">.bam</span> <span class="selector-tag">markdup</span><span class="selector-class">.bam</span></span><br></pre></td></tr></table></figure>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">cd</span> align</span><br><span class="line">ls *.bam | <span class="keyword">while</span> <span class="built_in">read</span> id</span><br><span class="line">  <span class="keyword">do</span> (nohup samtools markdup -r <span class="variable">$id</span> $(basename <span class="variable">$id</span> <span class="string">".bam"</span>).rmdup.bam &amp; );</span><br><span class="line"><span class="keyword">done</span></span><br><span class="line"><span class="comment">#如果不报错忽略以下操作</span></span><br><span class="line"><span class="comment">#如果报错</span></span><br><span class="line"><span class="comment">#[markdup] error: no ms score tag. Please run samtools fixmate on file first.</span></span><br><span class="line"><span class="comment">#[markdup] error: no ms score tag. Please run samtools fixmate on file first.</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 所以进行以下操作,开20个线程，每个线程分配2G内存 </span></span><br><span class="line"><span class="comment"># ‐n 设定排序方式按short reads的ID排序,不带-n 表示position order</span></span><br><span class="line"></span><br><span class="line">ls *.bam | <span class="keyword">while</span> <span class="built_in">read</span> id</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line"><span class="built_in">echo</span> $(basename <span class="variable">$id</span> <span class="string">".bam"</span>).sort.bam</span><br><span class="line">samtools sort -@ 20 -m 2000000000 -n -o $(basename <span class="variable">$id</span> <span class="string">".bam"</span>).sort.bam <span class="variable">$id</span></span><br><span class="line"><span class="keyword">done</span></span><br><span class="line"></span><br><span class="line"><span class="comment">#Add ms and MC tags for markdup to use later.samtools fixmate ，为以名称排序的定位alignment填入配对坐标</span></span><br><span class="line">ls *.sort.bam | <span class="keyword">while</span> <span class="built_in">read</span> id</span><br><span class="line"><span class="keyword">do</span> (nohup samtools fixmate -m <span class="variable">$id</span> $(basename <span class="variable">$id</span> <span class="string">".sort.bam"</span>).fixmate.bam &amp;);</span><br><span class="line"><span class="keyword">done</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">#Markdup needs position order</span></span><br><span class="line">ls *.fixmate.bam | <span class="keyword">while</span> <span class="built_in">read</span> id</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">  samtools sort -@ 20 -m 1000000000 -o $(basename <span class="variable">$id</span> <span class="string">".fixmate.bam"</span>).fixmat.sort.bam <span class="variable">$id</span></span><br><span class="line"><span class="keyword">done</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># Markdup去除PCR重复</span></span><br><span class="line">ls *.fixmat.sort.bam | <span class="keyword">while</span> <span class="built_in">read</span> id</span><br><span class="line">  <span class="keyword">do</span> (nohup samtools markdup -r <span class="variable">$id</span> $(basename <span class="variable">$id</span> <span class="string">".fixmat.sort.bam"</span>).rmdup.bam &amp; );</span><br><span class="line"><span class="keyword">done</span></span><br><span class="line"></span><br><span class="line"><span class="comment">#查看下rmdup.bam</span></span><br><span class="line">samtools view ST_1.rmdup.bam |less -S</span><br><span class="line"><span class="comment"># 看rm后的是否有差异 ,这是一个输出reads报告的方法</span></span><br><span class="line">samtools flagstat ST_1.bam</span><br><span class="line">samtools flagstat ST_1.rmdup.bam</span><br><span class="line"></span><br><span class="line"><span class="comment">## 删除中间文件，节省空间</span></span><br><span class="line">rm *.fixmate*</span><br><span class="line"></span><br><span class="line"><span class="comment">## 构建bam文件索引</span></span><br><span class="line">ls  *.rmdup.bam  |xargs -i samtools index &#123;&#125; </span><br><span class="line"><span class="comment">## 输出bam比对结果的统计</span></span><br><span class="line">ls  *.rmdup.bam  | <span class="keyword">while</span> <span class="built_in">read</span> id ;</span><br><span class="line">   <span class="keyword">do</span> (nohup samtools flagstat <span class="variable">$id</span> &gt; $(basename <span class="variable">$id</span> <span class="string">".bam"</span>).<span class="built_in">stat</span> &amp; );</span><br><span class="line"><span class="keyword">done</span></span><br></pre></td></tr></table></figure>
<h3 id="5-bcftools找变异call-variation"><a href="#5-bcftools找变异call-variation" class="headerlink" title="5. bcftools找变异call variation"></a><a href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247483812&amp;idx=1&amp;sn=c3f7ce44a4f164f0d5549e11a22e702f&amp;chksm=9b48411fac3fc8093925fbad417636544ea9af0c58ee9f4f31d1fa1fc2c80af610a835d09311&amp;scene=21#wechat_redirect" target="_blank" rel="noopener">5. bcftools找变异call variation</a></h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">conda activate wes</span><br><span class="line">ref=/data/reference/GATK/bundle/hg38/Homo_sapiens_assembly38.fasta</span><br><span class="line"><span class="comment">#这一步非常耗时 14个小时跑完</span></span><br><span class="line">time samtools mpileup -ugf <span class="variable">$ref</span> *.rmdup.bam | bcftools call -vmO z -o out.rmdup.vcf.gz</span><br><span class="line">less -S out.rmdup.vcf.gz</span><br><span class="line"></span><br><span class="line">------------------------------</span><br><span class="line"><span class="comment">#因为时间太长，只跑一个数据看一下</span></span><br><span class="line">conda activate wes</span><br><span class="line">ref=/data/reference/GATK/bundle/hg38/Homo_sapiens_assembly38.fasta</span><br><span class="line">time samtools mpileup -ugf <span class="variable">$ref</span> ST_1.rmdup.bam | bcftools call -vmO z -o ST_1.rmdup.vcf.gz</span><br><span class="line">less -S ST_1.rmdup.vcf.gz</span><br></pre></td></tr></table></figure>
<h4 id="6-GATK4流程"><a href="#6-GATK4流程" class="headerlink" title="6. GATK4流程"></a>6. GATK4流程</h4><blockquote>
<p>为什么做GATK, 因为做人类的染色体变异会跟严谨</p>
</blockquote>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><span class="line">ref=/data/reference/GATK/bundle/hg38/Homo_sapiens_assembly38.fasta</span><br><span class="line">snp=/data/reference/GATK/bundle/hg38/dbsnp_146.hg38.vcf.gz</span><br><span class="line">indel=/data/reference/GATK/bundle/hg38/Mills_and_1000G_gold_standard.indels.hg38.vcf.gz  </span><br><span class="line"></span><br><span class="line"><span class="built_in">cd</span> align</span><br><span class="line"><span class="keyword">for</span> sample <span class="keyword">in</span> &#123;ST_1.rmdup,ST_2.rmdup,ST_3.rmdup,ST_4.rmdup&#125;</span><br><span class="line"><span class="keyword">do</span> </span><br><span class="line"><span class="built_in">echo</span> <span class="variable">$sample</span>  </span><br><span class="line"><span class="comment"># Elapsed time: 7.91 minutes Xmx20G表示20G内存</span></span><br><span class="line">gatk --java-options <span class="string">"-Xmx20G -Djava.io.tmpdir=./"</span> MarkDuplicates -I <span class="variable">$sample</span>.bam -O <span class="variable">$&#123;sample&#125;</span>_marked.bam -M <span class="variable">$sample</span>.metrics 1&gt;<span class="variable">$&#123;sample&#125;</span>_log.mark 2&gt;&amp;1 </span><br><span class="line"></span><br><span class="line"><span class="comment">## Elapsed time: 13.61 minutes</span></span><br><span class="line">gatk --java-options <span class="string">"-Xmx20G -Djava.io.tmpdir=./"</span> FixMateInformation -I <span class="variable">$&#123;sample&#125;</span>_marked.bam -O <span class="variable">$&#123;sample&#125;</span>_marked_fixed.bam -SO coordinate 1&gt;<span class="variable">$&#123;sample&#125;</span>_log.fix 2&gt;&amp;1 </span><br><span class="line"></span><br><span class="line">samtools index <span class="variable">$&#123;sample&#125;</span>_marked_fixed.bam</span><br><span class="line"></span><br><span class="line"><span class="comment">##  17.2 minutes</span></span><br><span class="line">gatk --java-options <span class="string">"-Xmx20G -Djava.io.tmpdir=./"</span>  BaseRecalibrator -R <span class="variable">$ref</span> -I <span class="variable">$&#123;sample&#125;</span>_marked_fixed.bam --known-sites <span class="variable">$snp</span> --known-sites <span class="variable">$indel</span> -O <span class="variable">$&#123;sample&#125;</span>_recal.table 1&gt;<span class="variable">$&#123;sample&#125;</span>_log.recal 2&gt;&amp;1 </span><br><span class="line">    </span><br><span class="line">gatk --java-options <span class="string">"-Xmx20G -Djava.io.tmpdir=./"</span>   ApplyBQSR -R <span class="variable">$ref</span> -I <span class="variable">$&#123;sample&#125;</span>_marked_fixed.bam -bqsr <span class="variable">$&#123;sample&#125;</span>_recal.table -O <span class="variable">$&#123;sample&#125;</span>_bqsr.bam 1&gt;<span class="variable">$&#123;sample&#125;</span>_log.ApplyBQSR  2&gt;&amp;1 </span><br><span class="line">    </span><br><span class="line"><span class="comment">## 使用GATK的HaplotypeCaller命令</span></span><br><span class="line">gatk --java-options <span class="string">"-Xmx20G -Djava.io.tmpdir=./"</span> HaplotypeCaller -R <span class="variable">$ref</span> -I <span class="variable">$&#123;sample&#125;</span>_bqsr.bam --dbsnp <span class="variable">$snp</span> -O <span class="variable">$&#123;sample&#125;</span>_raw.vcf 1&gt;<span class="variable">$&#123;sample&#125;</span>_log.HC 2&gt;&amp;1  </span><br><span class="line"><span class="keyword">done</span></span><br></pre></td></tr></table></figure>
<h4 id="7-检查感兴趣基因区域内比对和找变异情况"><a href="#7-检查感兴趣基因区域内比对和找变异情况" class="headerlink" title="7. 检查感兴趣基因区域内比对和找变异情况"></a>7. 检查感兴趣基因区域内比对和找变异情况</h4><blockquote>
<p> 把这些bam里面的BRCA1基因的reads拿出来：</p>
</blockquote>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">cd</span> /data/reference/gencode</span><br><span class="line">cat gencode.v32.annotation.gtf | grep -w BRCA1|head|less -SN</span><br><span class="line"><span class="comment">##  1 chr17   HAVANA  gene    43044295        43170245 </span></span><br><span class="line"></span><br><span class="line"><span class="built_in">cd</span> align</span><br><span class="line">mkdir BRCA1</span><br><span class="line">samtools view ST_1.rmdup.bam chr17:43044295-43170245 |wc</span><br><span class="line"></span><br><span class="line">samtools  view -h  ST_1.rmdup.bam chr17:43044295-43170245 |samtools sort -o  ./BRCA1/ST_1.rmdup.brca1.bam -</span><br><span class="line">samtools  view -h  ST_1.rmdup_bqsr.bam chr17:43044295-43170245 |samtools sort -o  ./BRCA1/ST_1.rmdup_bqsr.brca1.bam -</span><br><span class="line">samtools  view -h  ST_1.rmdup_marked.bam chr17:43044295-43170245 |samtools sort -o  ./BRCA1/ST_1.rmdup_marked.brca1.bam -</span><br><span class="line">samtools  view -h  ST_1.rmdup_marked_fixed.bam chr17:43044295-43170245 |samtools sort -o  ./BRCA1/ST_1.rmdup_marked_fixed.brca1.bam -</span><br><span class="line"><span class="built_in">cd</span> BRCA1</span><br><span class="line">ls  *brca1.bam|xargs -i samtools index &#123;&#125;</span><br><span class="line"></span><br><span class="line"><span class="comment">#有了这些特定基因区域的bam，就可以针对特定基因找变异</span></span><br><span class="line">ref=/data/reference/GATK/bundle/hg38/Homo_sapiens_assembly38.fasta</span><br><span class="line">samtools mpileup -ugf <span class="variable">$ref</span> ST_1.rmdup.brca1.bam | bcftools call -vmO z -o ST_1.rmdup_bqsr_1.vcf.gz</span><br></pre></td></tr></table></figure>
<blockquote>
<p>所有得到的BRCA1的bam及 vcf 载入IGV看结果</p>
</blockquote>
<p><img src="https://taoshengxu.github.io/DocumentGit/img/wes_brca1.jpg" alt=""></p>
<h4 id="8-所有样本走samtools-mpileup-和bcftools-call-流程"><a href="#8-所有样本走samtools-mpileup-和bcftools-call-流程" class="headerlink" title="8.所有样本走samtools mpileup 和bcftools call 流程"></a>8.所有样本走samtools mpileup 和bcftools call 流程</h4><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">cd</span> align </span><br><span class="line">ls  *_bqsr.bam  |xargs -i samtools index &#123;&#125;</span><br><span class="line">ref=/data/reference/GATK/bundle/hg38/Homo_sapiens_assembly38.fasta</span><br><span class="line">samtools mpileup -ugf <span class="variable">$ref</span> *_bqsr.bam | bcftools call -vmO z -o all_bqsr.vcf.gz</span><br></pre></td></tr></table></figure>
<h4 id="9-比对及找变异结果的质控"><a href="#9-比对及找变异结果的质控" class="headerlink" title="9.比对及找变异结果的质控"></a>9.比对及找变异结果的质控</h4><blockquote>
<p>raw和clean的fastq文件都需要使用fastqc质控。<br>比对的各个阶段的bam文件都可以质控。<br>使用qualimap对wes的比对bam文件总结<strong>测序深度及覆盖度</strong>。</p>
<p><strong>Conserved CDS database(CCDS)</strong></p>
<p>不同的科研小组使用他们各自的方法研究同一物种基因组时，对于基因组中的基因定位可能会得到相似但不完全相同的结论。这样，就会对其它的科研工作者造成困扰。在所有的模式生物中，目前对人类和小鼠的基因组序列研究得最多也最透彻，因此它们最适合用来作为“标准的(consensus)”基因注释的“实验材料”。</p>
<p>CCDS数据库计划(www.ncbi.nlm.nih.gov/CCDS/)就是由NCBI、欧洲生物信息学研究院(European Bioinformatics Institute)、韦尔科姆基金会桑格研究院(Wellcome Trust Sanger Institute)和加州大学圣克鲁兹分校(UCSC)共同合作建立的标准的有关人类和小鼠基因蛋白质编码区的数据库，该数据库会不断更新来保持其高水准。到目前为止，CCDS数据库共收录了超过20,000条人类基因CDS注释数据和17,500条小鼠基因CDS注释数据。用户可以在CCDS的网页上使用基因名或序列ID进行搜索，还可以链接到Entrez Gene数据库、历史记录信息、转录体和蛋白质序列、Map Viewer、Ensemble Genome Browser、UCSC Genome Browser和桑格研究院的Vega Browser。用户可以到 (<a href="ftp.ncbi.nlm.nih.gov/pub/CCDS/current_human/">ftp.ncbi.nlm.nih.gov/pub/CCDS/</a>)下载CCDS序列数据。</p>
</blockquote>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">cd</span> clean</span><br><span class="line">ls *.gz |xargs fastqc -t 10 -o ./</span><br><span class="line"><span class="built_in">cd</span> align</span><br><span class="line">mkdir marked_bam</span><br><span class="line">mv *mark* ./marked_bam/</span><br><span class="line">mv *sort* ./marked_bam/</span><br><span class="line"></span><br><span class="line"><span class="comment">## 质控之前要建立index</span></span><br><span class="line">ls  *.bam  |xargs -i samtools index &#123;&#125; </span><br><span class="line">ls  *.bam  | <span class="keyword">while</span> <span class="built_in">read</span> id ;<span class="keyword">do</span> (samtools flagstat -@ 20 <span class="variable">$id</span> &gt; $(basename <span class="variable">$id</span> <span class="string">".bam"</span>).<span class="built_in">stat</span>);<span class="keyword">done</span></span><br><span class="line">mkdir stats</span><br><span class="line">mv *<span class="built_in">stat</span> ./stats</span><br><span class="line"><span class="built_in">cd</span> stats</span><br><span class="line">multiqc ./</span><br><span class="line"></span><br><span class="line">cat /data/reference/annotation/CCDS/human/CCDS.20180614.txt |perl -alne <span class="string">'&#123;/\[(.*?)\]/;next unless $1;$gene=$F[2];$exons=$1;$exons=~s/\s//g;$exons=~s/-/\t/g;print "$F[0]\t$_\t$gene" foreach split/,/,$exons;&#125;'</span>|sort -u |bedtools sort -i |awk <span class="string">'&#123;print "chr"$0"\t0\t+"&#125;'</span>  &gt; ./hg38.exon.bed </span><br><span class="line">ls  *_bqsr.bam | <span class="keyword">while</span> <span class="built_in">read</span> id;</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">sample=<span class="variable">$&#123;id%%.*&#125;</span></span><br><span class="line"><span class="built_in">echo</span> <span class="variable">$sample</span></span><br><span class="line">qualimap bamqc --java-mem-size=20G -gff hg38.exon.bed -bam <span class="variable">$id</span>  &amp; </span><br><span class="line"><span class="keyword">done</span> </span><br><span class="line"></span><br><span class="line"><span class="built_in">cd</span> align</span><br><span class="line">mkdir qualimap</span><br><span class="line">mv *_stats qualimap/</span><br><span class="line"><span class="built_in">cd</span> qualimap</span><br><span class="line">multiqc ./</span><br><span class="line"><span class="comment">## 最后看multiqc的网页报告</span></span><br></pre></td></tr></table></figure>
<blockquote>
<p> <strong><a href="http://bioinformatics.oxfordjournals.org/cgi/reprint/btt656?ijkey=ZzPz96t2lqzAH6F&amp;keytype=ref" target="_blank" rel="noopener">featureCounts</a></strong>: a software program developed for counting reads to genomic features such as genes, exons, promoters and genomic bins.</p>
</blockquote>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">### featureCounts  (Assigned to exon, 比对到外显子的数目)</span></span><br><span class="line">gtf=/data/reference/gencode/gencode.v32.annotation.gtf.gz</span><br><span class="line">nohup featureCounts -T 5 -p -t exon -g gene_id -a  <span class="variable">$gtf</span> *_bqsr.bam -o all.id.txt  1&gt;counts.id.log 2&gt;&amp;1 &amp;</span><br></pre></td></tr></table></figure>
<h4 id="比较两个找变异工具的区别"><a href="#比较两个找变异工具的区别" class="headerlink" title="比较两个找变异工具的区别"></a>比较两个找变异工具的区别</h4><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">chr1    139213  .       A       G       388     .   </span><br><span class="line">DP=984;VDB=0.831898;SGB=-223.781;RPB=0.599582;MQB=0.0001971;MQSB=0.00457372;BQB=2.59396e-09;MQ0F=0.281504;ICB=0.333333;HOB=0.5;AC=3;AN=6;DP4=371,169,160,84;MQ=21</span><br><span class="line">GT:PL   0/1:198,0,255   0/1:176,0,255   0/1:50,0,158</span><br><span class="line"></span><br><span class="line">chr1    139213  rs370723703 A   G   3945.77 .   AC=1;AF=0.500;AN=2;BaseQRankSum=-2.999;ClippingRankSum=0.000;DB;DP=244;ExcessHet=3.0103;FS=2.256;MLEAC=1;MLEAF=0.500;MQ=29.33;MQRankSum=-0.929;QD=16.17;ReadPosRankSum=1.462;SOR=0.863  GT:AD:DP:GQ:PL  0/1:136,108:244:99:3974,0,6459</span><br><span class="line">chr1    139213  rs370723703 A   G   2261.77 .   AC=1;AF=0.500;AN=2;BaseQRankSum=-1.191;ClippingRankSum=0.000;DB;DP=192;ExcessHet=3.0103;FS=9.094;MLEAC=1;MLEAF=0.500;MQ=32.03;MQRankSum=-0.533;QD=11.78;ReadPosRankSum=0.916;SOR=0.321  GT:AD:DP:GQ:PL  0/1:126,66:192:99:2290,0,7128</span><br><span class="line">chr1    139213  rs370723703 A   G   2445.77 .   AC=1;AF=0.500;AN=2;BaseQRankSum=-2.495;ClippingRankSum=0.000;DB;DP=223;ExcessHet=3.0103;FS=10.346;MLEAC=1;MLEAF=0.500;MQ=30.18;MQRankSum=0.486;QD=10.97;ReadPosRankSum=-0.808;SOR=0.300 GT:AD:DP:GQ:PL  0/1:152,71:223:99:2474,0,7901</span><br></pre></td></tr></table></figure>
<h4 id="VCF下游分析"><a href="#VCF下游分析" class="headerlink" title="VCF下游分析"></a>VCF下游分析</h4><p>主要是：注释和过滤</p>
<ul>
<li>注释( 可以用三个软件 VEP,snpEFF,ANNOVAR）</li>
</ul>
<p><a href="https://www.jianshu.com/p/49d035b121b8" target="_blank" rel="noopener">1.Annovar使用记录</a> (<a href="https://www.jianshu.com/p/49d035b121b8" target="_blank" rel="noopener">http://www.bio-info-trainee.com/641.html</a>)</p>
<p><a href="https://www.jianshu.com/p/49d035b121b8" target="_blank" rel="noopener">2.用annovar对snp进行注释</a> (<a href="https://www.jianshu.com/p/49d035b121b8" target="_blank" rel="noopener">http://www.bio-info-trainee.com/441.html</a>)</p>
<p><a href="https://www.jianshu.com/p/49d035b121b8" target="_blank" rel="noopener">3.对感兴趣的基因call variation</a>(<a href="https://www.jianshu.com/p/49d035b121b8" target="_blank" rel="noopener">http://www.bio-info-trainee.com/2013.html</a>)</p>
<p><a href="https://www.jianshu.com/p/49d035b121b8" target="_blank" rel="noopener">4.WES（六）用annovar注释</a>(<a href="https://www.jianshu.com/p/49d035b121b8" target="_blank" rel="noopener">http://www.bio-info-trainee.com/1158.html</a>)</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<p>===========================================================================</p>
<p><strong>学习提纲</strong>：</p>
<p>(1) <a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247484873&amp;idx=1&amp;sn=e455578dfac51afece7adb21cb315a17&amp;chksm=9b484572ac3fcc6460b6b9bbb88306f73c399d8e7a96ad8bb472a179a8de518bc70e21a72b0b&amp;mpshare=1&amp;scene=1&amp;srcid=0819fvKioudGMT1TYHXwnXoG#rd" target="_blank" rel="noopener">WES分析七步走</a></p>
<p>(2) <a href="http://www.bio-info-trainee.com/1108.html" target="_blank" rel="noopener">WES （一）测序质量控制</a></p>
<p>(3) <a href="http://www.bio-info-trainee.com/1114.html" target="_blank" rel="noopener">WES（二）snp-calling</a></p>
<p>(4) <a href="http://www.bio-info-trainee.com/1137.html" target="_blank" rel="noopener">WES（三）snp-filter</a></p>
<p>(5) <a href="http://www.bio-info-trainee.com/1138.html" target="_blank" rel="noopener">WES（四）不同个体的比较</a></p>
<p>(6) <a href="http://www.bio-info-trainee.com/1150.html" target="_blank" rel="noopener">WES（五）不同软件比较</a></p>
<p>(7) <a href="http://www.bio-info-trainee.com/1158.html" target="_blank" rel="noopener">WES（六）用annovar注释</a></p>
<p>(8) <a href="http://www.bio-info-trainee.com/1159.html" target="_blank" rel="noopener">WES（七）看de novo变异情况</a></p>
<p>(9) <a href="http://blog.csdn.net/zhu_si_tao/article/details/53321374" target="_blank" rel="noopener">GATK流程</a></p>
<p>(10) <a href="http://www.bio-info-trainee.com/838.html" target="_blank" rel="noopener">GATK使用注意事项</a></p>
<p>WES数据分析步骤：A Survey of Computational Tools to Analyze and Interpret Whole Exome Sequencing Data(2016)</p>

          
          <hr>
          <ul class="pager no-print">
              
              <li class="previous">
                  <a href="/hexo/2017/11/07/2017-11-07_DNA_seq/" data-toggle="tooltip" data-placement="left" title="DNA-SEQ 直播我的基因组">&larr; Previous Post</a>
              </li>
              
              
              <li class="next">
                  <a href="/hexo/2017/11/07/2017-11-07_WES_RNASEQ_CHIPSEQ/" data-toggle="tooltip" data-placement="top" title="WGS,WES,RNA-seq组与ChIP-seq之间的异同">Next Post&rarr;</a>
              </li>
              
          </ul>
          

      </div>
      
  <div class="hidden-xs col-sm-3 toc-col">
    <div class="toc-wrap">
        Table of Contents
        
          <ol class="toc"><li class="toc-item toc-level-3"><a class="toc-link" href="#软件安装"><span class="toc-number">1.</span> <span class="toc-text">软件安装</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#GATK依赖数据库下载"><span class="toc-number">2.</span> <span class="toc-text">GATK依赖数据库下载</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#文章"><span class="toc-number">3.</span> <span class="toc-text">文章</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#1-数据下载"><span class="toc-number">4.</span> <span class="toc-text">1.数据下载</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#2-抽取Fastq文件-fastq-dump"><span class="toc-number">5.</span> <span class="toc-text">2. 抽取Fastq文件(fastq-dump)</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#3-QC质控fasqc和multiqc"><span class="toc-number">6.</span> <span class="toc-text">3.QC质控fasqc和multiqc</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#假设质量很差，trim-galore过滤"><span class="toc-number">7.</span> <span class="toc-text">假设质量很差，trim_galore过滤</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#4-比对到基因组-bwa"><span class="toc-number">8.</span> <span class="toc-text">4.比对到基因组 bwa</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#5-去除PCR重复"><span class="toc-number">9.</span> <span class="toc-text">5. 去除PCR重复</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#5-bcftools找变异call-variation"><span class="toc-number">10.</span> <span class="toc-text">5. bcftools找变异call variation</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#6-GATK4流程"><span class="toc-number">10.1.</span> <span class="toc-text">6. GATK4流程</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#7-检查感兴趣基因区域内比对和找变异情况"><span class="toc-number">10.2.</span> <span class="toc-text">7. 检查感兴趣基因区域内比对和找变异情况</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#8-所有样本走samtools-mpileup-和bcftools-call-流程"><span class="toc-number">10.3.</span> <span class="toc-text">8.所有样本走samtools mpileup 和bcftools call 流程</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#9-比对及找变异结果的质控"><span class="toc-number">10.4.</span> <span class="toc-text">9.比对及找变异结果的质控</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#比较两个找变异工具的区别"><span class="toc-number">10.5.</span> <span class="toc-text">比较两个找变异工具的区别</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#VCF下游分析"><span class="toc-number">10.6.</span> <span class="toc-text">VCF下游分析</span></a></li></ol></li></ol>
        
    </div>
  </div>


    </div>
</article>

<!-- Footer -->
<!-- footer.ejs -->
<footer class="no-print">

    <div class="text-center">

        <ul class="list-inline">
            
            
            

            

            

            

            

            

        </ul>
        <div class="text-muted copyright">
            &copy;
            
            2017 - 2020
            
            -
            xt
            <br>
            
            
            
            | 
            <p>
                <span id="busuanzi_container_site_pv"><span id="busuanzi_value_site_pv"></span> <b>PV</b></span> -
                <span id="busuanzi_container_site_uv"><span id="busuanzi_value_site_uv"></span> <b>UV</b></span> -
                141.2k <b>Words</b>
        </p></div>

        <div class="search-container">
            <form class="site-search-form">
                <span class="glyphicon glyphicon-search"></span>
                <input type="text" id="local-search-input" class="st-search-input" placeholder="Search...">
            </form>
            <div id="local-search-result" class="local-search-result-cls"></div>
        </div>
        <script type="text/javascript" id="local.search.active">
            var inputArea = document.querySelector("#local-search-input");
            inputArea.onclick = function () {
                var root = "/hexo/";
                getSearchFile(root);
                this.onclick = null
            }
            inputArea.onkeydown = function () {
                if (event.keyCode == 13) return false
            }
        </script>

    </div>
</footer>

<!-- Custom Theme JavaScript -->
<script src="/hexo/js/main.js"></script>

<!-- async load function -->
<script>
    function async(u, c) {
        var d = document, t = 'script',
            o = d.createElement(t),
            s = d.getElementsByTagName(t)[0];
        o.src = u;
        if (c) {
            o.addEventListener('load', function (e) {
                c(null, e);
            }, false);
        }
        s.parentNode.insertBefore(o, s);
    }
</script>

<!--fastClick.js -->
<script>
    async("//cdn.bootcss.com/fastclick/1.0.6/fastclick.min.js", function () {
        var $nav = document.querySelector("nav");
        if ($nav) FastClick.attach($nav);
    })
</script><!-- hexo-inject:begin --><!-- hexo-inject:end -->



</body>

</html>
