<!DOCTYPE html>
<html lang="en">

<!-- layout.ejs-->
<head><meta name="generator" content="Hexo 3.8.0">
    <!-- hexo-inject:begin --><!-- hexo-inject:end --><meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="Adelaide">
    <meta name="author" content="xt">
    <meta name="keyword" content="">
    <link rel="canonical" href="https://taoshengxu.github.io/hexo/hexo/2017/11/07/2017-11-07_DNA_seq/">
    <link rel="shortcut icon" href="/hexo/img/favicon.png">
    <link rel="alternate" type="application/atom+xml" title="IIM" href="/atom.xml">
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/animate.css/3.5.2/animate.min.css">
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
    <link rel="stylesheet" href="https://ajax.googleapis.com/ajax/libs/jqueryui/1.12.1/themes/smoothness/jquery-ui.css">
    <script src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.12.1/jquery-ui.min.js"></script>
    <script src="/hexo/js/search.js"></script>

    <title>
        
        DNA-SEQ 直播我的基因组｜undefined
        
    </title>

    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">

    <link rel="stylesheet" href="/hexo/css/main.css">

    
      <link href="//fonts.googleapis.com/css?family=Source+Code+Pro" rel="stylesheet" type="text/css">
      <link rel="stylesheet" href="/hexo/css/highlight.css">
    

    

    <script async src="//dn-lbstatics.qbox.me/busuanzi/2.3/busuanzi.pure.mini.js"></script>


    

    

    


    
  
    <script type="text/x-mathjax-config">
      MathJax.Hub.Config({
        tex2jax: {
          inlineMath: [ ['$','$'], ["\\(","\\)"]  ],
          processEscapes: true,
          skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
        }
      });
    </script>

    <script type="text/x-mathjax-config">
      MathJax.Hub.Queue(function() {
        var all = MathJax.Hub.getAllJax(), i;
        for (i=0; i < all.length; i += 1) {
          all[i].SourceElement().parentNode.className += ' has-jax';
        }
      });
    </script>
    <script type="text/javascript" src="//cdn.bootcss.com/mathjax/2.7.1/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
  


    




    <script async defer src="https://buttons.github.io/buttons.js"></script>

<link rel="stylesheet" href="/hexo/css/prism-solarizedlight.css" type="text/css">
<link rel="stylesheet" href="/hexo/css/prism-line-numbers.css" type="text/css"><!-- hexo-inject:begin --><!-- hexo-inject:end --></head>

<style>
    header.intro-header {
        background-image: url('/hexo/img/northernlights-sisimiut-lake.jpg')
    }
</style>
<!-- hack iOS CSS :active style -->
<body ontouchstart="" class="animated fadeIn">
<!-- hexo-inject:begin --><!-- hexo-inject:end --><header>
  <nav class="navbar navbar-default header-navbar" id="nav-top" data-ispost="true" data-istags="false" data-ishome="false">
    <div class="container-fluid">
      <div class="navbar-header page-scroll">
        <button type="button" class="navbar-toggle" data-toggle="collapse" aria-expanded="false" data-target="#website_navbar">
          <span class="sr-only">Toggle navigation</span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <span class="navbar-brand animated pulse">
          <a class="brand-logo" href="/hexo/">
                <img src="/hexo/img/banner.jpg?h=350&amp;auto=compress&amp;cs=tinysrgb">
          </a>
        </span>
      </div>

      <div class="collapse navbar-collapse" id="website_navbar">
          <ul class="nav navbar-nav navbar-right">
              
                <li>
                  <a href="/hexo/">home</a>
                </li>
              
                <li>
                  <a href="/hexo/archives/">archives</a>
                </li>
              
                <li>
                  <a href="/hexo/categories/">categories</a>
                </li>
              
                <li>
                  <a href="/hexo/tags/">tags</a>
                </li>
              
                <li>
                  <a href="/hexo/columns/">columns</a>
                </li>
              
          </ul>
      </div>
  </div></nav>


  
    <style>
       .intro-header {
          background-image: url('/hexo/post_cover_images/moon.jpg');
      }
    </style>

    <div class="intro-header">
        <div class="container">
            <div class="row">
                <div class="col-lg-8 col-lg-offset-2 col-md-10 col-md-offset-1 text-center">
                    <div class="site-heading">
                        <h1>DNA-SEQ 直播我的基因组</h1>
                        
                        

                        
                          <span class="meta">
                               <span class="meta-item">Author: xt</span>
                               <span class="meta-item">Date: Nov 7, 2017</span>
                               
                          </span>
                          <div class="tags text-center">
                              Categories: 
                              <a class="tag" href="/hexo/categories/#Bioinformatics" title="Bioinformatics">Bioinformatics</a>
                              
                          </div>
                          <div class="tags text-center">
                              Tags: 
                              <a class="tag" href="/hexo/tags/#English" title="English">English</a>
                              
                          </div>
                        
                    </div>
                </div>
            </div>
        </div>
    </div>
  
</header>


<!-- Main Content -->
<!-- post.ejs -->
<article>
    <div class="container">
      <div class="col-lg-8 col-lg-offset-1 col-sm-9">
          
          <span class="post-count">2,512 words in total, 13 minutes required.</span>
          <hr>
          
          <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/hint.css/2.4.1/hint.min.css"><p><strong>教程</strong> <a href="http://www.biotrainee.com/thread-1376-1-1.html" target="_blank" rel="noopener">Jimmy直播我的基因组分析 - 目录</a></p>
<p><img src="https://taoshengxu.github.io/DocumentGit/img/RNASeq_WES.jpeg" alt=""></p>
<h1 id="下载hg19基因组"><a href="#下载hg19基因组" class="headerlink" title="下载hg19基因组"></a>下载hg19基因组</h1><figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">cd /data/reference</span><br><span class="line">mkdir -<span class="selector-tag">p</span> genome/hg19  &amp;&amp; cd genome/hg19</span><br><span class="line">nohup wget http:<span class="comment">//hgdownload.cse.ucsc.edu/goldenPath/hg19/bigZips/chromFa.tar.gz &amp;</span></span><br><span class="line">tar zvfx chromFa<span class="selector-class">.tar</span><span class="selector-class">.gz</span></span><br><span class="line">cat *<span class="selector-class">.fa</span> &gt; hg19.fa</span><br><span class="line">rm chr*.fa</span><br></pre></td></tr></table></figure>
<h1 id="fastq-sam-bam"><a href="#fastq-sam-bam" class="headerlink" title="fastq-sam-bam"></a>fastq-sam-bam</h1><figure class="highlight mel"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">ls</span> *gz |xargs ~/biosoft/fastqc/FastQC/fastqc -t <span class="number">10</span></span><br><span class="line"></span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> $(seq <span class="number">1</span> <span class="number">6</span>) ;<span class="keyword">do</span> (nohup bwa  mem -t <span class="number">5</span> -M /data/<span class="keyword">reference</span>/index/bwa/hg19  KPGP<span class="number">-00001</span>_L$&#123;i&#125;_R1.fq.gz KPGP<span class="number">-00001</span>_L$&#123;i&#125;_R2.fq.gz <span class="number">1</span>&gt;KPGP<span class="number">-00001</span>_L$&#123;i&#125;.sam <span class="number">2</span>&gt;KPGP<span class="number">-00001</span>_L$&#123;i&#125;.bwa.<span class="keyword">align</span>.<span class="keyword">log</span> &amp;);done</span><br><span class="line"></span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> $(seq <span class="number">1</span> <span class="number">6</span>) ;<span class="keyword">do</span> (nohup samtools <span class="keyword">sort</span> -@ <span class="number">5</span> -o KPGP<span class="number">-00001</span>_L$&#123;i&#125;.sorted.bam  KPGP<span class="number">-00001</span>_L$&#123;i&#125;.sam &amp;);done</span><br><span class="line"></span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> $(seq <span class="number">1</span> <span class="number">6</span>) ;<span class="keyword">do</span> (nohup samtools index KPGP<span class="number">-00001</span>_L$&#123;i&#125;.sorted.bam &amp;);done</span><br><span class="line">samtools merge KPGP<span class="number">-00001.</span>merge.bam *.sorted.bam</span><br><span class="line">samtools <span class="keyword">sort</span> -@ <span class="number">10</span> -O bam -o KPGP<span class="number">-00001.</span>sorted.merge.bam  KPGP<span class="number">-00001.</span>merge.bam</span><br><span class="line">samtools index  KPGP<span class="number">-00001.</span>sorted.merge.bam</span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> $(seq <span class="number">1</span> <span class="number">6</span>) ;<span class="keyword">do</span> ( samtools flagstat KPGP<span class="number">-00001</span>_L$&#123;i&#125;.sorted.bam &gt;KPGP<span class="number">-00001</span>_L$&#123;i&#125;.flagstat.txt );done</span><br></pre></td></tr></table></figure>
<h1 id="根据gtf格式的基因注释文件得到人所有基因的染色体坐标"><a href="#根据gtf格式的基因注释文件得到人所有基因的染色体坐标" class="headerlink" title="根据gtf格式的基因注释文件得到人所有基因的染色体坐标"></a>根据gtf格式的基因注释文件得到人所有基因的染色体坐标</h1><p><a href="http://www.biotrainee.com/thread-472-1-1.html" target="_blank" rel="noopener">原文</a></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br></pre></td><td class="code"><pre><span class="line">mkdir -p ~/reference/gtf/gencode</span><br><span class="line"><span class="built_in">cd</span>  ~/reference/gtf/gencode</span><br><span class="line"><span class="comment">## https://www.gencodegenes.org/releases/current.html</span></span><br><span class="line">wget ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_25/gencode.v25.2wayconspseudos.gtf.gz</span><br><span class="line">wget ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_25/gencode.v25.long_noncoding_RNAs.gtf.gz </span><br><span class="line">wget ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_25/gencode.v25.polyAs.gtf.gz </span><br><span class="line">wget ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_25/gencode.v25.annotation.gtf.gz </span><br><span class="line"></span><br><span class="line"><span class="comment">## https://www.gencodegenes.org/releases/25lift37.html </span></span><br><span class="line">wget ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_25/GRCh37_mapping/gencode.v25lift37.annotation.gtf.gz </span><br><span class="line">wget ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_25/GRCh37_mapping/gencode.v25lift37.metadata.HGNC.gz </span><br><span class="line">wget ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_25/GRCh37_mapping/gencode.v25lift37.metadata.EntrezGene.gz </span><br><span class="line">wget ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_25/GRCh37_mapping/gencode.v25lift37.metadata.RefSeq.gz </span><br><span class="line"> </span><br><span class="line">zcat  gencode.v25.long_noncoding_RNAs.gtf.gz |perl -alne <span class="string">'&#123;next unless $F[2] eq "gene" ;/gene_name \"(.*?)\";/; print "$F[0]\t$F[3]\t$F[4]\t$1" &#125;'</span> &gt;lncRNA.hg38.position</span><br><span class="line">zcat  gencode.v25.2wayconspseudos.gtf.gz     |perl -alne <span class="string">'&#123;next unless $F[2] eq "transcript" ;/gene_name \"(.*?)\";/; print "$F[0]\t$F[3]\t$F[4]\t$1" &#125;'</span> &gt;pseudos.hg38.position</span><br><span class="line">zcat  gencode.v25.annotation.gtf.gz| grep   protein_coding |perl -alne <span class="string">'&#123;next unless $F[2] eq "gene" ;/gene_name \"(.*?)\";/; print "$F[0]\t$F[3]\t$F[4]\t$1" &#125;'</span> &gt;protein_coding.hg38.position</span><br><span class="line">zcat  gencode.v25.annotation.gtf.gz|perl -alne <span class="string">'&#123;next unless $F[2] eq "gene" ;/gene_name \"(.*?)\";/; print "$F[0]\t$F[3]\t$F[4]\t$1" &#125;'</span> &gt;allGene.hg38.position</span><br><span class="line"> </span><br><span class="line">zcat  gencode.v25lift37.annotation.gtf.gz | grep   protein_coding |perl -alne <span class="string">'&#123;next unless $F[2] eq "gene" ;/gene_name \"(.*?)\";/; print "$F[0]\t$F[3]\t$F[4]\t$1" &#125;'</span> &gt;protein_coding.hg19.position</span><br><span class="line">zcat  gencode.v25lift37.annotation.gtf.gz | perl -alne <span class="string">'&#123;next unless $F[2] eq "gene" ;/gene_name \"(.*?)\";/; print "$F[0]\t$F[3]\t$F[4]\t$1" &#125;'</span> &gt;allGene.hg19.position</span><br></pre></td></tr></table></figure>
<p>有个很严重的问题，gencode里面的数据有着HAVANA和ENSEMBL的区别，尤其是在hg38里面，需要区别对待！具体见：<a href="http://www.bio-info-trainee.com/1991.html" target="_blank" rel="noopener">http://www.bio-info-trainee.com/1991.html</a> 的解释.</p>
<h1 id="Call-感兴趣的基因-variation"><a href="#Call-感兴趣的基因-variation" class="headerlink" title="Call 感兴趣的基因 variation"></a>Call 感兴趣的基因 variation</h1><p><a href="http://www.bio-info-trainee.com/2013.html" target="_blank" rel="noopener">原文</a></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">grep H3F3A /data/reference/gtf/gencode/protein_coding.hg19.position</span><br><span class="line">samtools mpileup -r chr1:226249552-226259702  -ugf /data/reference/genome/hg19/hg19.fa *sorted.bam | bcftools call -vmO z -o H3F3A.vcf.gz</span><br><span class="line">gunzip H3F3A.vcf.gz</span><br><span class="line"></span><br><span class="line">/data/biosoft/annovar/convert2annovar.pl -format vcf4old H3F3A.vcf &gt;H3F3A.annovar</span><br><span class="line">/data/biosoft/annovar/annotate_variation.pl -buildver hg19 --geneanno --outfile H3F3A.anno H3F3A.annovar /data/biosoft/annovar/humandb/</span><br><span class="line"></span><br><span class="line"><span class="comment">#/data/biosoft/annovar/annotate_variation.pl --downdb refGene /data/biosoft/annovar/humandb</span></span><br><span class="line"></span><br><span class="line">/data/biosoft/annovar/annotate_variation.pl -buildver hg19 --dbtype refGene --geneanno --outfile H3F3A.anno H3F3A.annovar /data/biosoft/annovar/humandb/</span><br></pre></td></tr></table></figure>
<h1 id="获取感兴趣的基因坐标"><a href="#获取感兴趣的基因坐标" class="headerlink" title="获取感兴趣的基因坐标"></a>获取感兴趣的基因坐标</h1><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">grep H3F3A /data/reference/gtf/gencode/protein_coding.hg19.position &gt;&gt; key.list.txt</span><br><span class="line">grep HLA-DQ /data/reference/gtf/gencode/protein_coding.hg19.position &gt;&gt; key.list.txt</span><br><span class="line">grep AVPR1 /data/reference/gtf/gencode/protein_coding.hg19.position &gt;&gt; key.list.txt</span><br><span class="line">grep IRX3 /data/reference/gtf/gencode/protein_coding.hg19.position &gt;&gt; key.list.txt</span><br><span class="line">grep IRX5 /data/reference/gtf/gencode/protein_coding.hg19.position &gt;&gt; key.list.txt</span><br><span class="line">grep GLI3 /data/reference/gtf/gencode/protein_coding.hg19.position &gt;&gt; key.list.txt</span><br><span class="line">grep PAX1 /data/reference/gtf/gencode/protein_coding.hg19.position &gt;&gt; key.list.txt</span><br><span class="line">grep RUNX2 /data/reference/gtf/gencode/protein_coding.hg19.position &gt;&gt; key.list.txt</span><br></pre></td></tr></table></figure>
<h1 id="批量提取我们感兴趣的基因的变异情况"><a href="#批量提取我们感兴趣的基因的变异情况" class="headerlink" title="批量提取我们感兴趣的基因的变异情况"></a>批量提取我们感兴趣的基因的变异情况</h1><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">cat key.list.txt |<span class="keyword">while</span> <span class="built_in">read</span> id;</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">chr=$(<span class="built_in">echo</span> <span class="variable">$id</span> |cut -d<span class="string">" "</span> -f 1|sed <span class="string">'s/chr//'</span> )</span><br><span class="line">start=$(<span class="built_in">echo</span> <span class="variable">$id</span> |cut -d<span class="string">" "</span> -f 2 )</span><br><span class="line">end=$(<span class="built_in">echo</span> <span class="variable">$id</span> |cut -d<span class="string">" "</span> -f 3 )</span><br><span class="line">gene=$(<span class="built_in">echo</span> <span class="variable">$id</span> |cut -d<span class="string">" "</span> -f 4 )</span><br><span class="line"><span class="built_in">echo</span> <span class="variable">$chr</span>:<span class="variable">$start</span>-<span class="variable">$end</span>  <span class="variable">$gene</span></span><br><span class="line">samtools mpileup -r  <span class="variable">$chr</span>:<span class="variable">$start</span>-<span class="variable">$end</span>   -ugf /data/reference/genome/hg19/hg19.fa KPGP-00001.sorted.merge.bam | bcftools call -vmO z -o <span class="variable">$gene</span>.vcf.gz</span><br><span class="line"><span class="keyword">done</span></span><br></pre></td></tr></table></figure>
<p>前面我们说到有研究表明STAT4上的rs7574865和HLA-DQ的rs9275319是国人群中乙型肝炎病毒（HBV）相关肝细胞癌（HCC）遗传易感基因，那么我们很容易去dbSNP数据库或者我最近强烈推荐 的<a href="https://www.snpedia.com/index.php/Rs7574865" target="_blank" rel="noopener">Snpedia</a>数据库<a href="http://www.bio-info-trainee.com/2100.html" target="_blank" rel="noopener">吐血推荐snpedia数据库，非常丰富的snp信息记录</a>里面找到它的坐标。<br><figure class="highlight elixir"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="number">6</span>	<span class="number">32666295</span> <span class="symbol">:Rs9275319--HLA-DQ</span></span><br><span class="line"><span class="number">2</span>	<span class="number">191964633</span> <span class="symbol">:Rs7574865--STAT4</span></span><br></pre></td></tr></table></figure></p>
<p>然后我检查了我刚才call到的variation文件，<br><figure class="highlight 1c"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">zcat STAT4.vcf.gz <span class="string">|grep -w 191964633 显示为空。</span></span><br><span class="line">zcat HLA-DQ* <span class="string">|grep 32666295  也是空。</span></span><br></pre></td></tr></table></figure></p>
<p>哈哈，我完美的错过了这两个易感位点！！！！谢天谢地！！！</p>
<h1 id="bam文件给按照染色体给分割"><a href="#bam文件给按照染色体给分割" class="headerlink" title="bam文件给按照染色体给分割"></a>bam文件给按照染色体给分割</h1><p>按照染色体（chr1-chr22,chrX,chrY,chrMT）</p>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">bamtools split -<span class="keyword">in</span> KPGP-<span class="number">00001</span><span class="selector-class">.sorted</span><span class="selector-class">.merge</span><span class="selector-class">.bam</span> -reference</span><br></pre></td></tr></table></figure>
<h1 id="提取未比对的测序数据"><a href="#提取未比对的测序数据" class="headerlink" title="提取未比对的测序数据"></a>提取未比对的测序数据</h1><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line">samtools view -f4 KPGP-00001.sorted.merge.bam &gt; KPGP-00001.sorted.merge.unmapped.sam <span class="comment">#more fast,  or</span></span><br><span class="line"><span class="comment"># samtools view sample.bam |perl -alne '&#123;print if $F[2] eq "*" or $F[5] eq "*" &#125;' &gt; sample.unmapped.sam</span></span><br><span class="line"></span><br><span class="line"><span class="comment">#提前未比对成功的测序数据(可以分成3类，仅reads1，仅reads2，和两端reads都没有比对成功)</span></span><br><span class="line"><span class="comment">#小写的f是提取，大写的F是过滤</span></span><br><span class="line">samtools view -u -f 4 -F 264 alignments.bam  &gt; tmps1.bam <span class="comment">#仅reads1</span></span><br><span class="line">samtools view -u -f 8 -F 260 alignments.bam  &gt; tmps2.bam <span class="comment">#仅reads2</span></span><br><span class="line">samtools view -u -f 12 -F 256 alignments.bam &gt; tmps3.bam <span class="comment">#两端reads都没有比对成功</span></span><br><span class="line">samtools merge -u - tmps[123].bam | samtools sort -n - unmapped</span><br><span class="line">bamToFastq -bam unmapped.bam -fq1 unmapped_reads1.fastq -fq2 unmapped_reads2.fastq</span><br><span class="line"></span><br><span class="line"><span class="comment">#统计一下未比对成功的reads有多少</span></span><br><span class="line">cut -f 3,6 KPGP-00001.sorted.merge.unmapped.sam |sort |uniq -c &gt;KPGP-00001.sorted.merge.unmapped.counts</span><br></pre></td></tr></table></figure>
<h1 id="提取左右端测序数据比对到不同染色体的PE-reads"><a href="#提取左右端测序数据比对到不同染色体的PE-reads" class="headerlink" title="提取左右端测序数据比对到不同染色体的PE reads"></a>提取左右端测序数据比对到不同染色体的PE reads</h1><p>左右端测序数据比对到不同染色体的情况，比较有意义，可能是融合基因，也可能是基因之间本来就相似性很大。<br>三种具有代表性的肿瘤融合基因BCR-ABL、SLC45A3-ELK4 和PAX3-FOXO1</p>
<p><strong>融合基因</strong>（英语：Fusion gene）是指两个基因的全部或一部分的序列相互融合为一个新的基因的过程。其有可能是染色体易位、中间缺失或染色体倒置所致的结果.</p>
<figure class="highlight stata"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">samtools <span class="keyword">view</span> KPGP-00001.sorted.<span class="keyword">merge</span>.bam | perl -alne '&#123;<span class="keyword">print</span> <span class="keyword">if</span> <span class="variable">$F</span>[6] ne <span class="string">"="</span>&#125;'  &gt;unpaired.sam </span><br><span class="line">cut -f 3,7 unpaired.sam |<span class="keyword">sort</span> |uniq -c</span><br></pre></td></tr></table></figure>
<h1 id="PCR-duplication"><a href="#PCR-duplication" class="headerlink" title="PCR duplication"></a>PCR duplication</h1><p><a href="http://blog.sina.com.cn/s/blog_69e75efd0102wu57.html" target="_blank" rel="noopener">http://blog.sina.com.cn/s/blog_69e75efd0102wu57.html</a></p>
<p>设一个基因组有A、B两个片段，PCR后得到无论多少条reads，比如nA+mB条，在数据分析的时候，都只保留1条A和1条B（unique reads）用于组装，而去掉(n-1)条A和(m-1)条B。共有(n-1)条A和(m-1)条B被当成duplicatedreads看待，尽管它们是正常PCR的正常产物。</p>
<p>那么为什么要去除这个duplication呢？主要是因为在call snp的时候，如果某个变异位点的变异碱基都是来自于PCR重复，而我们却认为它深度足够判断是真的变异位点，这个结论其实有很大可能是假阳性。</p>
<h1 id="覆盖度详细探究"><a href="#覆盖度详细探究" class="headerlink" title="覆盖度详细探究"></a>覆盖度详细探究</h1><figure class="highlight stata"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">samtools flagstat KPGP-00001.sorted.<span class="keyword">merge</span>.bam</span><br><span class="line"></span><br><span class="line">#全基因组</span><br><span class="line">samtools mpileup KPGP-00001.sorted.<span class="keyword">merge</span>.bam |perl -alne '&#123;<span class="keyword">if</span>(<span class="variable">$F</span>[3]&gt;100)&#123;<span class="variable">$depth</span>&#123;<span class="string">"over100"</span>&#125;++&#125;<span class="keyword">else</span>&#123;<span class="variable">$depth</span>&#123;<span class="variable">$F</span>[3]&#125;++&#125;&#125;END&#123;<span class="keyword">print</span> <span class="string">"$_\t$depth&#123;$_&#125;"</span> <span class="keyword">foreach</span> <span class="keyword">sort</span>&#123;<span class="variable">$a</span> &lt;=&gt; <span class="variable">$b&#125;</span>keys %depth&#125;' </span><br><span class="line"></span><br><span class="line">#每一条染色体</span><br><span class="line"><span class="keyword">ls</span> KPGP-00001.sorted.<span class="keyword">merge</span>.REF*.bam |<span class="keyword">while</span> <span class="keyword">read</span> id </span><br><span class="line"><span class="keyword">do</span> </span><br><span class="line">echo <span class="variable">$id</span></span><br><span class="line">samtools mpileup <span class="variable">$id</span> |perl -alne '&#123;<span class="keyword">if</span>(<span class="variable">$F</span>[3]&gt;100)&#123;<span class="variable">$depth</span>&#123;<span class="string">"over100"</span>&#125;++&#125;<span class="keyword">else</span>&#123;<span class="variable">$depth</span>&#123;<span class="variable">$F</span>[3]&#125;++&#125;&#125; END &#123;<span class="keyword">print</span> <span class="string">"$_\t$depth&#123;$_&#125;"</span> <span class="keyword">foreach</span> <span class="keyword">sort</span>&#123;<span class="variable">$a</span> &lt;=&gt; <span class="variable">$b&#125;</span>keys %depth&#125;' &gt;<span class="variable">$id</span>.depth.txt</span><br><span class="line">done</span><br></pre></td></tr></table></figure>
<h1 id="对比对结果文件进行过滤"><a href="#对比对结果文件进行过滤" class="headerlink" title="对比对结果文件进行过滤"></a>对比对结果文件进行过滤</h1><p>这个地方有问题</p>
<figure class="highlight armasm"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"><span class="symbol">samtools</span> view -h -<span class="built_in">F4</span>  -q <span class="number">5</span> KPGP-<span class="number">00001</span>.sorted.merge.<span class="keyword">bam </span><span class="title">|samtools view -bS |</span>samtools rmdup - KPGP-<span class="number">00001</span>.filter.rmdup.<span class="keyword">bam</span></span><br><span class="line"><span class="keyword">samtools </span>index KPGP-<span class="number">00001</span>.filter.rmdup.<span class="keyword">bam</span></span><br><span class="line"><span class="keyword"></span></span><br><span class="line"><span class="keyword">bam文件是二进制文件，我们需要samtools </span>view的命令进行格式转换。</span><br><span class="line">这个管道的意思分开来说就是运行第一步时过滤的时候已将<span class="keyword">bam文件转成我们能看的sam格式。</span></span><br><span class="line"><span class="keyword">其中h是在输出的结果中包含头header，F和q是过滤掉没有mapped上的reads（也就是multiple </span>mapping的情况）和低质量的reads。</span><br><span class="line">第二步是将上一步得出的sam文件再转成<span class="keyword">bam，</span></span><br><span class="line"><span class="keyword">第三步就是用samtools </span>rmdup过滤掉PCR duplication的情况了，最后得到了过滤了<span class="keyword">multiple </span>mapping、PCR duplication及低质量比对的<span class="keyword">bam文件。</span></span><br><span class="line"><span class="keyword">最后利用samtools </span>index对过滤后的<span class="keyword">bam文件建立索引。</span></span><br></pre></td></tr></table></figure>
<h1 id="用GATK对SAM格式的文件进行重排"><a href="#用GATK对SAM格式的文件进行重排" class="headerlink" title="用GATK对SAM格式的文件进行重排"></a>用GATK对SAM格式的文件进行重排</h1><p><a href="http://www.bio-info-trainee.com/838.html" target="_blank" rel="noopener">GATK使用注意事项</a></p>
<p><a href="https://github.com/snewhouse/ngs_nextflow/wiki/GATK-Bundle" target="_blank" rel="noopener">https://github.com/snewhouse/ngs_nextflow/wiki/GATK-Bundle</a></p>
<p><a href="https://github.com/ddcap/halvade/wiki" target="_blank" rel="noopener">https://github.com/ddcap/halvade/wiki</a></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># ftp://ftp.broadinstitute.org/bundle/hg19</span></span><br><span class="line"></span><br><span class="line">axel ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/ucsc.hg19.fasta.gz</span><br><span class="line">wget ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/ucsc.hg19.fasta.fai.gz</span><br><span class="line">gunzip ucsc.hg19.fasta.gz</span><br><span class="line">bwa index ucsc.hg19.fasta</span><br><span class="line"><span class="comment"># ucsc.hg19.fasta.amb</span></span><br><span class="line"><span class="comment"># ucsc.hg19.fasta.ann</span></span><br><span class="line"><span class="comment"># ucsc.hg19.fasta.bwt</span></span><br><span class="line"><span class="comment"># ucsc.hg19.fasta.pac</span></span><br><span class="line"><span class="comment"># ucsc.hg19.fasta.sa</span></span><br><span class="line">axel ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/1000G_phase1.indels.hg19.sites.vcf.gz</span><br><span class="line">axel ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/1000G_phase1.indels.hg19.sites.vcf.idx.gz</span><br><span class="line">axel ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf.gz</span><br><span class="line">axel ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf.idx.gz</span><br><span class="line">axel ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/ucsc.hg19.dict.gz</span><br><span class="line">axel ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/dbsnp_138.hg19.vcf.gz</span><br><span class="line"></span><br><span class="line">ls *.gz |<span class="keyword">while</span> <span class="built_in">read</span> id;</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">gunzip <span class="variable">$id</span></span><br><span class="line"><span class="keyword">done</span></span><br></pre></td></tr></table></figure>
<p><strong>按染色体下载hg19基因组</strong><br><figure class="highlight ruby"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> $(seq <span class="number">1</span> <span class="number">22</span>) X Y M;</span><br><span class="line"><span class="keyword">do</span> echo $i;</span><br><span class="line">wget <span class="symbol">http:</span>/<span class="regexp">/hgdownload.cse.ucsc.edu/golden</span>Path/hg19/chromosomes/chr$&#123;i&#125;.fa.gz;</span><br><span class="line">done</span><br><span class="line">gunzip *.gz</span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> $(seq <span class="number">1</span> <span class="number">22</span>) X Y M;</span><br><span class="line"><span class="keyword">do</span> cat chr$&#123;i&#125;.fa <span class="meta">&gt;&gt; </span>hg19.fasta;</span><br><span class="line">done</span><br><span class="line">rm -fr chr*.fasta</span><br><span class="line"></span><br><span class="line">bwa index ucsc.hg19.fasta</span><br><span class="line"></span><br><span class="line">bwa index -a bwtsw hg19.fasta</span><br><span class="line">samtools faidx hg19.fasta</span><br></pre></td></tr></table></figure></p>
<p>首先用RealignerTargetCreator找到需要重新比对的区域，输出文件intervals，然后用输出的 tmp.intervals 做输入文件来进行重新比对，也就是用IndelRealigner在这些区域内进行重新比对.</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">nohup java -Xmx60g -jar /data/biosoft/GATK/GenomeAnalysisTK.jar -R \</span><br><span class="line">/data/reference/annotation/GATK/ucsc.hg19.fasta -T RealignerTargetCreator \</span><br><span class="line">-I KPGP-00001.filter.rmdup.bam -o KPGP-00001.filter.rmdup.intervals \</span><br><span class="line">-known /data/reference/annotation/GATK/1000G_phase1.indels.hg19.sites.vcf \</span><br><span class="line">1&gt; KPGP-00001.filter.rmdup.RealignerTargetCreator.log</span><br><span class="line"></span><br><span class="line">nohup java -Xmx60g -jar /data/biosoft/GATK/GenomeAnalysisTK.jar  -R \</span><br><span class="line">/data/reference/annotation/GATK/ucsc.hg19.fasta  -T IndelRealigner  \</span><br><span class="line">-I KPGP-00001.filter.rmdup.bam  -targetIntervals KPGP-00001.filter.rmdup.intervals \</span><br><span class="line">-o KPGP-00001.filter.rmdup.realgn.bam  1&gt;KPGP-00001.filter.rmdup.IndelRealigner.log</span><br></pre></td></tr></table></figure>

          
          <hr>
          <ul class="pager no-print">
              
              <li class="previous">
                  <a href="/hexo/2017/11/07/2017-11-07_Data_format/" data-toggle="tooltip" data-placement="left" title="常用生物信息学数据格式">&larr; Previous Post</a>
              </li>
              
              
              <li class="next">
                  <a href="/hexo/2017/11/07/2017-11-07_WES_Seq/" data-toggle="tooltip" data-placement="top" title="WES分析">Next Post&rarr;</a>
              </li>
              
          </ul>
          

      </div>
      
  <div class="hidden-xs col-sm-3 toc-col">
    <div class="toc-wrap">
        Table of Contents
        
          <ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#下载hg19基因组"><span class="toc-number">1.</span> <span class="toc-text">下载hg19基因组</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#fastq-sam-bam"><span class="toc-number">2.</span> <span class="toc-text">fastq-sam-bam</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#根据gtf格式的基因注释文件得到人所有基因的染色体坐标"><span class="toc-number">3.</span> <span class="toc-text">根据gtf格式的基因注释文件得到人所有基因的染色体坐标</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#Call-感兴趣的基因-variation"><span class="toc-number">4.</span> <span class="toc-text">Call 感兴趣的基因 variation</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#获取感兴趣的基因坐标"><span class="toc-number">5.</span> <span class="toc-text">获取感兴趣的基因坐标</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#批量提取我们感兴趣的基因的变异情况"><span class="toc-number">6.</span> <span class="toc-text">批量提取我们感兴趣的基因的变异情况</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#bam文件给按照染色体给分割"><span class="toc-number">7.</span> <span class="toc-text">bam文件给按照染色体给分割</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#提取未比对的测序数据"><span class="toc-number">8.</span> <span class="toc-text">提取未比对的测序数据</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#提取左右端测序数据比对到不同染色体的PE-reads"><span class="toc-number">9.</span> <span class="toc-text">提取左右端测序数据比对到不同染色体的PE reads</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#PCR-duplication"><span class="toc-number">10.</span> <span class="toc-text">PCR duplication</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#覆盖度详细探究"><span class="toc-number">11.</span> <span class="toc-text">覆盖度详细探究</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#对比对结果文件进行过滤"><span class="toc-number">12.</span> <span class="toc-text">对比对结果文件进行过滤</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#用GATK对SAM格式的文件进行重排"><span class="toc-number">13.</span> <span class="toc-text">用GATK对SAM格式的文件进行重排</span></a></li></ol>
        
    </div>
  </div>


    </div>
</article>

<!-- Footer -->
<!-- footer.ejs -->
<footer class="no-print">

    <div class="text-center">

        <ul class="list-inline">
            
            
            

            

            

            

            

            

        </ul>
        <div class="text-muted copyright">
            &copy;
            
            2017 - 2020
            
            -
            xt
            <br>
            
            
            
            | 
            <p>
                <span id="busuanzi_container_site_pv"><span id="busuanzi_value_site_pv"></span> <b>PV</b></span> -
                <span id="busuanzi_container_site_uv"><span id="busuanzi_value_site_uv"></span> <b>UV</b></span> -
                128.9k <b>Words</b>
        </p></div>

        <div class="search-container">
            <form class="site-search-form">
                <span class="glyphicon glyphicon-search"></span>
                <input type="text" id="local-search-input" class="st-search-input" placeholder="Search...">
            </form>
            <div id="local-search-result" class="local-search-result-cls"></div>
        </div>
        <script type="text/javascript" id="local.search.active">
            var inputArea = document.querySelector("#local-search-input");
            inputArea.onclick = function () {
                var root = "/hexo/";
                getSearchFile(root);
                this.onclick = null
            }
            inputArea.onkeydown = function () {
                if (event.keyCode == 13) return false
            }
        </script>

    </div>
</footer>

<!-- Custom Theme JavaScript -->
<script src="/hexo/js/main.js"></script>

<!-- async load function -->
<script>
    function async(u, c) {
        var d = document, t = 'script',
            o = d.createElement(t),
            s = d.getElementsByTagName(t)[0];
        o.src = u;
        if (c) {
            o.addEventListener('load', function (e) {
                c(null, e);
            }, false);
        }
        s.parentNode.insertBefore(o, s);
    }
</script>

<!--fastClick.js -->
<script>
    async("//cdn.bootcss.com/fastclick/1.0.6/fastclick.min.js", function () {
        var $nav = document.querySelector("nav");
        if ($nav) FastClick.attach($nav);
    })
</script><!-- hexo-inject:begin --><!-- hexo-inject:end -->



</body>

</html>
