<!DOCTYPE html>
<html lang="en">

<!-- layout.ejs-->
<head><meta name="generator" content="Hexo 3.8.0">
    <!-- hexo-inject:begin --><!-- hexo-inject:end --><meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="Adelaide">
    <meta name="author" content="xt">
    <meta name="keyword" content="">
    <link rel="canonical" href="https://taoshengxu.github.io/hexo/hexo/2017/11/07/2017-11-07_RNA_Seq/">
    <link rel="shortcut icon" href="/hexo/img/favicon.png">
    <link rel="alternate" type="application/atom+xml" title="IIM" href="/atom.xml">
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/animate.css/3.5.2/animate.min.css">
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
    <link rel="stylesheet" href="https://ajax.googleapis.com/ajax/libs/jqueryui/1.12.1/themes/smoothness/jquery-ui.css">
    <script src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.12.1/jquery-ui.min.js"></script>
    <script src="/hexo/js/search.js"></script>

    <title>
        
        RNA-Seq分析｜undefined
        
    </title>

    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">

    <link rel="stylesheet" href="/hexo/css/main.css">

    
      <link href="//fonts.googleapis.com/css?family=Source+Code+Pro" rel="stylesheet" type="text/css">
      <link rel="stylesheet" href="/hexo/css/highlight.css">
    

    

    <script async src="//dn-lbstatics.qbox.me/busuanzi/2.3/busuanzi.pure.mini.js"></script>


    

    

    


    
  


    




    <script async defer src="https://buttons.github.io/buttons.js"></script>

<link rel="stylesheet" href="/hexo/css/prism-solarizedlight.css" type="text/css">
<link rel="stylesheet" href="/hexo/css/prism-line-numbers.css" type="text/css"><!-- hexo-inject:begin --><!-- hexo-inject:end --></head>

<style>
    header.intro-header {
        background-image: url('/hexo/img/northernlights-sisimiut-lake.jpg')
    }
</style>
<!-- hack iOS CSS :active style -->
<body ontouchstart="" class="animated fadeIn">
<!-- hexo-inject:begin --><!-- hexo-inject:end --><header>
  <nav class="navbar navbar-default header-navbar" id="nav-top" data-ispost="true" data-istags="false" data-ishome="false">
    <div class="container-fluid">
      <div class="navbar-header page-scroll">
        <button type="button" class="navbar-toggle" data-toggle="collapse" aria-expanded="false" data-target="#website_navbar">
          <span class="sr-only">Toggle navigation</span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <span class="navbar-brand animated pulse">
          <a class="brand-logo" href="/hexo/">
                <img src="/hexo/img/banner.jpg?h=350&amp;auto=compress&amp;cs=tinysrgb">
          </a>
        </span>
      </div>

      <div class="collapse navbar-collapse" id="website_navbar">
          <ul class="nav navbar-nav navbar-right">
              
                <li>
                  <a href="/hexo/">home</a>
                </li>
              
                <li>
                  <a href="/hexo/archives/">archives</a>
                </li>
              
                <li>
                  <a href="/hexo/categories/">categories</a>
                </li>
              
                <li>
                  <a href="/hexo/tags/">tags</a>
                </li>
              
                <li>
                  <a href="/hexo/columns/">columns</a>
                </li>
              
          </ul>
      </div>
  </div></nav>


  
    <style>
       .intro-header {
          background-image: url('/hexo/img/northernlights-sisimiut-lake.jpg');
      }
    </style>

    <div class="intro-header">
        <div class="container">
            <div class="row">
                <div class="col-lg-8 col-lg-offset-2 col-md-10 col-md-offset-1 text-center">
                    <div class="site-heading">
                        <h1>RNA-Seq分析</h1>
                        
                        

                        
                          <span class="meta">
                               <span class="meta-item">Author: xt</span>
                               <span class="meta-item">Date: Nov 7, 2017</span>
                               
                          </span>
                          <div class="tags text-center">
                              Categories: 
                          </div>
                          <div class="tags text-center">
                              Tags: 
                          </div>
                        
                    </div>
                </div>
            </div>
        </div>
    </div>
  
</header>


<!-- Main Content -->
<!-- post.ejs -->
<article>
    <div class="container">
      <div class="col-lg-8 col-lg-offset-1 col-sm-9">
          
          <span class="post-count">4,180 words in total, 21 minutes required.</span>
          <hr>
          
          <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/hint.css/2.4.1/hint.min.css"><p>这是一个学习笔记，跟随生信技能树的学习笔记重复,把几个优秀笔记的内容重复摘录在此。</p>
<p><strong>学习提纲</strong>：<a href="http://www.biotrainee.com/thread-1750-1-1.html" target="_blank" rel="noopener">RNA-seq基础入门传送门</a></p>
<p><strong>文章链接</strong>：<a href="https://www.nature.com/articles/ncomms13347" target="_blank" rel="noopener">https://www.nature.com/articles/ncomms13347</a> </p>
<p><strong>非常棒的学习笔记1</strong>：<a href="https://taoshengxu.github.io/DocumentGit/pdf/沈梦圆2017年转录组入门合辑0-6.pdf">PANDA姐的转录组入门（0-6）合辑</a></p>
<p><strong>非常棒的学习笔记2</strong>:<a href="https://taoshengxu.github.io/DocumentGit/pdf/浙大植物学小白的转录组笔记.pdf">浙大植物学小白的转录组笔记</a> <a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247484895&amp;idx=1&amp;sn=678da702fa929789b177d214070dd39a&amp;chksm=9b484564ac3fcc72914b0ae2c1b71adb63fb359cf7e73221be1ddfd3040efa2944c91bee8e3b&amp;mpshare=1&amp;scene=1&amp;srcid=0824NWlPEoAgwVKtIWnkEDd9#rd" target="_blank" rel="noopener">Link2</a></p>
<p><strong>非常棒的学习笔记3</strong>:<a href="./post/transcriptome_analysis1.html">下一篇</a></p>
<h1 id="分析软件安装"><a href="#分析软件安装" class="headerlink" title="分析软件安装"></a>分析软件安装</h1><p>最方便的安装方式就是 Anaconda</p>
<figure class="highlight llvm"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">wget https://repo.continuum.io/archive/Anaconda<span class="number">3</span><span class="number">-4.4</span>.<span class="number">0</span>-Linux-<span class="keyword">x</span><span class="number">86</span>_<span class="number">64</span>.sh</span><br><span class="line"></span><br><span class="line"> conda install -<span class="keyword">c</span> bioconda samtools=<span class="number">1.5</span></span><br><span class="line"> conda install -<span class="keyword">c</span> bioconda htseq=<span class="number">0.7</span>.<span class="number">2</span></span><br><span class="line"> conda install -<span class="keyword">c</span> bioconda hisat<span class="number">2</span>=<span class="number">2.1</span>.<span class="number">0</span></span><br><span class="line"> conda install -<span class="keyword">c</span> bioconda fastqc=<span class="number">0.11</span>.<span class="number">5</span></span><br><span class="line"> conda install -<span class="keyword">c</span> jfear sratoolkit=<span class="number">2.8</span>.<span class="number">1</span></span><br></pre></td></tr></table></figure>
<h1 id="数据下载"><a href="#数据下载" class="headerlink" title="数据下载"></a>数据下载</h1><p>From NCBI GEO ftp</p>
<p>The RIP-seq an RNA-seq data have been deposited in the Gene Expression Omnibus database, with accession code <strong>GSE81916</strong></p>
<h1 id="数据分析"><a href="#数据分析" class="headerlink" title="数据分析"></a>数据分析</h1><h2 id="质控"><a href="#质控" class="headerlink" title="质控"></a>质控</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">for</span> id <span class="keyword">in</span> `seq 56 62`</span><br><span class="line"><span class="keyword">do</span> fastq-dump --gzip --split-3 -O /data/RNASeq -A SRR35899<span class="variable">$&#123;id&#125;</span></span><br><span class="line"><span class="keyword">done</span> <span class="comment">##很慢，建议后台多线程</span></span><br><span class="line"></span><br><span class="line"><span class="comment">##查看fastq文件</span></span><br><span class="line">zcat SRR3589956_1.fastq.gz | head -n 4</span><br><span class="line"><span class="comment">##安装集成分析工具</span></span><br><span class="line">conda install -c bioconda multiqc</span><br><span class="line"></span><br><span class="line"><span class="comment"># 先获取QC结果</span></span><br><span class="line">ls *gz | <span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> fastqc -t 4 <span class="variable">$id</span>; <span class="keyword">done</span></span><br><span class="line"><span class="comment"># multiqc</span></span><br><span class="line">multiqc *fastqc.zip --pdf</span><br></pre></td></tr></table></figure>
<p><strong>Python质控脚本</strong><br><figure class="highlight vim"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br></pre></td><td class="code"><pre><span class="line">import re</span><br><span class="line">import zipfile</span><br><span class="line"># <span class="keyword">read</span> the zip <span class="keyword">file</span></span><br><span class="line">def zipReader(<span class="keyword">file</span>):</span><br><span class="line">    qcfile =  zipfile.ZipFile(<span class="keyword">file</span>)</span><br><span class="line">    data_txt = [<span class="keyword">file</span> <span class="keyword">for</span> <span class="keyword">file</span> in qcfile.namelist() <span class="keyword">if</span> re.<span class="keyword">match</span>(<span class="string">".*?_data\.txt"</span>, <span class="keyword">file</span>)][<span class="number">0</span>]</span><br><span class="line">    data = [bytes.decode(<span class="built_in">line</span>) <span class="keyword">for</span> <span class="built_in">line</span> in qcfile.<span class="keyword">open</span>(data_txt)]</span><br><span class="line">    <span class="keyword">return</span> data</span><br><span class="line"> </span><br><span class="line">def fastqc_summary(data):</span><br><span class="line">    module_num = <span class="number">0</span></span><br><span class="line">    bases = <span class="number">0</span></span><br><span class="line">    Q20 = <span class="number">0</span></span><br><span class="line">    Q30 = <span class="number">0</span></span><br><span class="line">    <span class="keyword">for</span> <span class="built_in">line</span> in dat<span class="variable">a:</span></span><br><span class="line">        <span class="keyword">if</span> re.<span class="keyword">match</span>(<span class="string">'Filename'</span>, <span class="built_in">line</span>):</span><br><span class="line">            filename = <span class="built_in">line</span>.<span class="keyword">split</span>(sep=<span class="string">"\t"</span>)[<span class="number">1</span>].strip()</span><br><span class="line">        <span class="keyword">if</span> re.<span class="keyword">match</span>(<span class="string">'Total Sequence'</span>, <span class="built_in">line</span>):</span><br><span class="line">            <span class="keyword">read</span> = <span class="built_in">line</span>.<span class="keyword">split</span>(sep=<span class="string">"\t"</span>)[<span class="number">1</span>].strip()</span><br><span class="line">        <span class="keyword">if</span> re.<span class="keyword">match</span>(<span class="string">'%GC'</span>, <span class="built_in">line</span>):</span><br><span class="line">            GC = <span class="built_in">line</span>.<span class="keyword">split</span>(sep=<span class="string">"\t"</span>)[<span class="number">1</span>].strip()</span><br><span class="line">        <span class="keyword">if</span> re.<span class="keyword">match</span>(<span class="string">"[^#](.*?\t)&#123;6&#125;"</span>,<span class="built_in">line</span>):</span><br><span class="line">            bases = bases + <span class="number">1</span></span><br><span class="line">            <span class="keyword">if</span> float(<span class="built_in">line</span>.<span class="keyword">split</span>(<span class="string">"\t"</span>)[<span class="number">1</span>]) &gt; <span class="number">30</span>:</span><br><span class="line">                Q20 = Q20 + <span class="number">1</span></span><br><span class="line">                Q30 = Q30 + <span class="number">1</span></span><br><span class="line">            elif float(<span class="built_in">line</span>.<span class="keyword">split</span>(<span class="string">"\t"</span>)[<span class="number">1</span>]) &gt; <span class="number">20</span>:</span><br><span class="line">                Q20 = Q20 + <span class="number">1</span></span><br><span class="line"> </span><br><span class="line">        <span class="keyword">if</span> re.<span class="keyword">match</span>(<span class="string">"&gt;&gt;END"</span>, <span class="built_in">line</span>) :</span><br><span class="line">            module_num = module_num + <span class="number">1</span></span><br><span class="line">            <span class="keyword">if</span> module_num &gt;= <span class="number">2</span>:</span><br><span class="line">                <span class="keyword">break</span></span><br><span class="line">    Q20 = Q20 / bases</span><br><span class="line">    Q30 = Q30 / bases</span><br><span class="line">    summary = [filename, <span class="keyword">read</span>, GC, str(Q20), str(Q30)]</span><br><span class="line">    <span class="keyword">return</span> summary</span><br><span class="line"> </span><br><span class="line"><span class="keyword">if</span> __name__ == <span class="string">'__main__'</span>:</span><br><span class="line">    import sys</span><br><span class="line">    <span class="keyword">for</span> arg in <span class="built_in">range</span>(<span class="number">1</span>, <span class="built_in">len</span>(sys.<span class="built_in">argv</span>)):</span><br><span class="line">        data = zipReader(sys.<span class="built_in">argv</span>[arg])</span><br><span class="line">        summary = fastqc_summary(data)</span><br><span class="line">        with <span class="keyword">open</span>(<span class="string">'summary.txt'</span>, <span class="string">'a'</span>) <span class="keyword">as</span> <span class="keyword">f</span>:</span><br><span class="line">            <span class="keyword">f</span>.<span class="keyword">write</span>(<span class="string">'\t'</span>.<span class="keyword">join</span>(summary) + <span class="string">'\n'</span>)</span><br></pre></td></tr></table></figure></p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">grep -w 'gene' gencode.v26lift37.annotation.gtf |grep -w 'TP53'|cut -f 1,4,5 &gt;&gt;gene.bed</span><br><span class="line">grep -w 'gene' gencode.v26lift37.annotation.gtf |grep -w 'KRAS'|cut -f 1,4,5 &gt;&gt;gene.bed</span><br><span class="line">grep -w 'gene' gencode.v26lift37.annotation.gtf |grep -w 'EGFR'|cut -f 1,4,5 &gt;&gt;gene.bed</span><br><span class="line">bedtools igv -i gene.bed &gt;Bach_sanpshot.txt</span><br><span class="line">perl -alne '&#123;print "goto $F[0]:$F[1]-$F[2]\nsnapshot $F[3].png"&#125; '</span><br></pre></td></tr></table></figure>
<h2 id="Hisat2比对"><a href="#Hisat2比对" class="headerlink" title="Hisat2比对"></a>Hisat2比对</h2><p>HISAT2是TopHat2/Bowti2的继任者，使用改进的BWT算法，实现了更快的速度和更少的资源占用，作者推荐TopHat2/Bowti2和HISAT的用户转换到HISAT2。<br>官网：<a href="https://ccb.jhu.edu/software/hisat2/index.shtml" target="_blank" rel="noopener">https://ccb.jhu.edu/software/hisat2/index.shtml</a></p>
<p>1.建立基因组索引or index 下载<br><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash">建立基因组索引</span></span><br><span class="line"><span class="meta">#</span><span class="bash">hisat2-build –p 4 genome.fa genome</span></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash">下载索引</span></span><br><span class="line">cd ~/reference</span><br><span class="line">mkdir -p index/hisat &amp;&amp; cd index/hisat</span><br><span class="line">wget -c ftp://ftp.ccb.jhu.edu/pub/infphilo/hisat2/data/hg19.tar.gz</span><br><span class="line">wget -c ftp://ftp.ccb.jhu.edu/pub/infphilo/hisat2/data/mm10.tar.gz</span><br><span class="line">tar zxvf hg19.tar.gz</span><br><span class="line">tar xvzf mm10.tar.gz</span><br></pre></td></tr></table></figure></p>
<p>2.比对<br><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br></pre></td><td class="code"><pre><span class="line">for i in `seq 56 58`</span><br><span class="line">do</span><br><span class="line">    hisat2 -t -p 24 -x /data/Reference/index/hisat2/hg19/genome \</span><br><span class="line">    -1 /data/RNASeq/fastq/SRR35899$&#123;i&#125;_1.fastq.gz \</span><br><span class="line">    -2 SRR35899$&#123;i&#125;_2.fastq.gz \</span><br><span class="line">    -S /data/RNASeq/fastq/SRR35899$&#123;i&#125;.sam &gt; SRR35899$&#123;i&#125;.log &amp;</span><br><span class="line">done</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">#比对结果</span></span></span><br><span class="line">xts@R710:/data/RNASeq/fastq$ for i in `seq 56 58`</span><br><span class="line"><span class="meta">&gt;</span><span class="bash"> <span class="keyword">do</span></span></span><br><span class="line"><span class="meta">&gt;</span><span class="bash">     hisat2 -t -p 24 -x /data/Reference/index/hisat2/hg19/genome \</span></span><br><span class="line"><span class="meta">&gt;</span><span class="bash">     -1 /data/RNASeq/fastq/SRR35899<span class="variable">$&#123;i&#125;</span>_1.fastq.gz \</span></span><br><span class="line"><span class="meta">&gt;</span><span class="bash">     -2 SRR35899<span class="variable">$&#123;i&#125;</span>_2.fastq.gz \</span></span><br><span class="line"><span class="meta">&gt;</span><span class="bash">     -S /data/RNASeq/fastq/SRR35899<span class="variable">$&#123;i&#125;</span>.sam &gt; SRR35899<span class="variable">$&#123;i&#125;</span>.<span class="built_in">log</span> &amp;</span></span><br><span class="line"><span class="meta">&gt;</span><span class="bash"> <span class="keyword">done</span></span></span><br><span class="line">[1] 11177</span><br><span class="line">[2] 11178</span><br><span class="line">[3] 11179</span><br><span class="line">xts@R710:/data/RNASeq/fastq$ tipTime loading forward index: 00:00:24</span><br><span class="line">Time loading forward index: 00:00:24</span><br><span class="line">Time loading forward index: 00:00:24</span><br><span class="line">Time loading reference: 00:00:04</span><br><span class="line">Time loading reference: 00:00:04</span><br><span class="line">Time loading reference: 00:00:04</span><br><span class="line">Multiseed full-index search: 00:13:22</span><br><span class="line">28856780 reads; of these:</span><br><span class="line">  28856780 (100.00%) were paired; of these:</span><br><span class="line">    1838981 (6.37%) aligned concordantly 0 times</span><br><span class="line">    24732654 (85.71%) aligned concordantly exactly 1 time</span><br><span class="line">    2285145 (7.92%) aligned concordantly &gt;1 times</span><br><span class="line">    ----</span><br><span class="line">    1838981 pairs aligned concordantly 0 times; of these:</span><br><span class="line">      90927 (4.94%) aligned discordantly 1 time</span><br><span class="line">    ----</span><br><span class="line">    1748054 pairs aligned 0 times concordantly or discordantly; of these:</span><br><span class="line">      3496108 mates make up the pairs; of these:</span><br><span class="line">        2034939 (58.21%) aligned 0 times</span><br><span class="line">        1221462 (34.94%) aligned exactly 1 time</span><br><span class="line">        239707 (6.86%) aligned &gt;1 times</span><br><span class="line">96.47% overall alignment rate</span><br><span class="line">Time searching: 00:13:26</span><br><span class="line">Overall time: 00:13:50</span><br><span class="line"></span><br><span class="line">Multiseed full-index search: 00:14:42</span><br><span class="line">25914821 reads; of these:</span><br><span class="line">  25914821 (100.00%) were paired; of these:</span><br><span class="line">    1785160 (6.89%) aligned concordantly 0 times</span><br><span class="line">    21786672 (84.07%) aligned concordantly exactly 1 time</span><br><span class="line">    2342989 (9.04%) aligned concordantly &gt;1 times</span><br><span class="line">    ----</span><br><span class="line">    1785160 pairs aligned concordantly 0 times; of these:</span><br><span class="line">      53455 (2.99%) aligned discordantly 1 time</span><br><span class="line">    ----</span><br><span class="line">    1731705 pairs aligned 0 times concordantly or discordantly; of these:</span><br><span class="line">      3463410 mates make up the pairs; of these:</span><br><span class="line">        2187330 (63.16%) aligned 0 times</span><br><span class="line">        1050929 (30.34%) aligned exactly 1 time</span><br><span class="line">        225151 (6.50%) aligned &gt;1 times</span><br><span class="line">95.78% overall alignment rate</span><br><span class="line">Time searching: 00:14:46</span><br><span class="line">Overall time: 00:15:10</span><br><span class="line">[1]   已完成               hisat2 -t -p 24 -x /data/Reference/index/hisat2/hg19/genome -1 /data/RNASeq/fastq/SRR35899$&#123;i&#125;_1.fastq.gz -2 SRR35899$&#123;i&#125;_2.fastq.gz -S /data/RNASeq/fastq/SRR35899$&#123;i&#125;.sam &gt; SRR35899$&#123;i&#125;.log</span><br><span class="line">[3]+  已完成               hisat2 -t -p 24 -x /data/Reference/index/hisat2/hg19/genome -1 /data/RNASeq/fastq/SRR35899$&#123;i&#125;_1.fastq.gz -2 SRR35899$&#123;i&#125;_2.fastq.gz -S /data/RNASeq/fastq/SRR35899$&#123;i&#125;.sam &gt; SRR35899$&#123;i&#125;.log</span><br><span class="line">xts@R710:/data/RNASeq/fastq$ Multiseed full-index search: 00:16:08</span><br><span class="line">29720636 reads; of these:</span><br><span class="line">  29720636 (100.00%) were paired; of these:</span><br><span class="line">    1920019 (6.46%) aligned concordantly 0 times</span><br><span class="line">    25503958 (85.81%) aligned concordantly exactly 1 time</span><br><span class="line">    2296659 (7.73%) aligned concordantly &gt;1 times</span><br><span class="line">    ----</span><br><span class="line">    1920019 pairs aligned concordantly 0 times; of these:</span><br><span class="line">      61683 (3.21%) aligned discordantly 1 time</span><br><span class="line">    ----</span><br><span class="line">    1858336 pairs aligned 0 times concordantly or discordantly; of these:</span><br><span class="line">      3716672 mates make up the pairs; of these:</span><br><span class="line">        2292272 (61.68%) aligned 0 times</span><br><span class="line">        1196099 (32.18%) aligned exactly 1 time</span><br><span class="line">        228301 (6.14%) aligned &gt;1 times</span><br><span class="line">96.14% overall alignment rate</span><br><span class="line">Time searching: 00:16:12</span><br><span class="line">Overall time: 00:16:36</span><br></pre></td></tr></table></figure></p>
<p><strong>数据转换sam-bam-sorted bam</strong><br><figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">for</span> <span class="selector-tag">i</span> <span class="keyword">in</span> `seq <span class="number">56</span> <span class="number">58</span>`</span><br><span class="line">do</span><br><span class="line">    samtools view -S SRR35899$&#123;i&#125;<span class="selector-class">.sam</span> -<span class="selector-tag">b</span> &gt; SRR35899$&#123;i&#125;.bam</span><br><span class="line">    samtools sort SRR35899$&#123;i&#125;<span class="selector-class">.bam</span> -o SRR35899$&#123;i&#125;_sorted.bam</span><br><span class="line">    samtools index SRR35899$&#123;i&#125;_sorted.bam</span><br><span class="line">done</span><br></pre></td></tr></table></figure></p>
<p>SAMtools其他操作</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br></pre></td><td class="code"><pre><span class="line">head -1000 SRR3589957.sam &gt; test.sam</span><br><span class="line">samtools view -b  test.sam &gt; test.bam</span><br><span class="line">samtools view test.bam | head</span><br><span class="line"></span><br><span class="line">samtools sort test.bam -o default.bam</span><br><span class="line">samtools view default.bam | head</span><br><span class="line"> </span><br><span class="line"><span class="meta">#</span><span class="bash"> Sort alignments by leftmost coordinates, or by <span class="built_in">read</span> name when -n is used</span></span><br><span class="line">samtools sort test.bam default</span><br><span class="line">samtools view default.bam | head</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash">提取1号染色体1234-123456区域的比对<span class="built_in">read</span></span></span><br><span class="line">samtools view SRR3589957_sorted.bam chr1:1234-123456 | head</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash">在比如搭配flag(0.1.19版本没有）和flagstat，使用-f或-F参数提取不同匹配情况的<span class="built_in">read</span>。</span></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> 可以先用flagstat看下总体情况</span></span><br><span class="line">samtools flagstat SRR3589957_sorted.bam</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash">筛选恰好配对的<span class="built_in">read</span>,就需要用0x10</span></span><br><span class="line">samtools view -b -f 0x10 SRR3589957_sorted.bam chr1:1234-123456  &gt; flag.bam</span><br><span class="line">samtools flagstat flag.bam</span><br></pre></td></tr></table></figure>
<h2 id="比对质控"><a href="#比对质控" class="headerlink" title="比对质控"></a>比对质控</h2><ul>
<li>RSeQC——<a href="http://rseqc.sourceforge.net/" target="_blank" rel="noopener">http://rseqc.sourceforge.net/</a></li>
<li>Qualimap——<a href="http://qualimap.bioinfo.cipf.es/" target="_blank" rel="noopener">http://qualimap.bioinfo.cipf.es/</a></li>
<li>Picard——<a href="http://broadinstitute.github.io/picard/" target="_blank" rel="noopener">http://broadinstitute.github.io/picard/</a></li>
</ul>
<p>使用RSeQC来对我们的比对结果进行质控,RSeQC包括了十多个Python脚本，实现很多功能，具体每个脚本的参数用法，都可以在官网学习.</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> RSeQC的安装，需要先安装gcc；numpy；R；Python2.7</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> pip install RSeQC</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 对bam文件进行质控，其余都同样的进行</span></span><br><span class="line"><span class="meta">$</span><span class="bash"> bam_stat.py  -i SRR3589956_sorted.bam</span></span><br><span class="line"></span><br><span class="line">基因组覆盖率的QC需要提供bed文件，可以直接RSeQC的网站下载，或者可以用gtf转换</span><br><span class="line">read_distribution.py -i RNA-Seq/aligned/SRR3589956_sorted.bam -r reference/hg19_RefSeq.bed</span><br></pre></td></tr></table></figure>
<h1 id="Reads-计数-From-hoptop"><a href="#Reads-计数-From-hoptop" class="headerlink" title="Reads 计数 (From hoptop)"></a>Reads 计数 (From hoptop)</h1><p>定量分为三个水平</p>
<ul>
<li>基因水平(gene-level)</li>
<li>转录本水平(transcript-level)</li>
<li>外显子使用水平(exon-usage-level)。</li>
</ul>
<p>1.在<strong>基因水平</strong>上，常用的软件为HTSeq-count，featureCounts，BEDTools, Qualimap, Rsubread, GenomicRanges等。以常用的HTSeq-count为例，这些工具要解决的问题就是<strong>根据read和基因位置的overlap判断这个read到底是谁家的孩子</strong>。值得注意的是不同工具对multimapping reads处理方式也是不同的，例如HTSeq-count就直接当它们不存在。而Qualimpa则是一人一份，平均分配。</p>
<p>对每个基因计数之后得到的count matrix再后续的分析中，要注意标准化的问题。如果你要比较同一个样本(within-sample)不同基因之间的表达情况，你就需要考虑到<strong>转录本长度</strong>，因为转录本越长，那么检测的片段也会更多，直接比较等于让小孩和大人进行赛跑。如果你是比较不同样本（across sample）同一个基因的表达情况，虽然不必在意转录本长度，但是你要考虑到<strong>测序深度</strong>（sequence depth)，毕竟测序深度越高，检测到的概率越大。除了这两个因素外，你还需要考虑GC%所导致的偏差，以及测序仪器的系统偏差。目前对read count标准化的算法有RPKM（SE）, FPKM（PE），TPM, TMM等，不同算法之间的差异与换算方法已经有文章进行整理和吐槽了。但是，有一些下游分析的软件会要求是输入的count matrix是原始数据，未经标准化，比如说DESeq2，这个时候你需要注意你上一步所用软件会不会进行标准化。</p>
<p>2.在<strong>转录本水平</strong>上，一般常用工具为Cufflinks和它的继任者StringTie， eXpress。这些软件要处理的难题就时转录本亚型（isoforms）之间通常是有重叠的，当二代测序读长低于转录本长度时，如何进行区分？这些工具大多采用的都是expectation maximization（EM）。好在我们有三代测序。<br>上述软件都是alignment-based，目前许多alignment-free软件，如kallisto, silfish, salmon，能够省去比对这一步，直接得到read count，在运行效率上更高。不过最近一篇文献[1]指出这类方法在估计丰度时存在样本特异性和读长偏差。</p>
<p>3.在<strong>外显子使用水平</strong>上，其实和基因水平的统计类似。但是值得注意的是为了更好的计数，我们需要提供无重叠的外显子区域的gtf文件。用于分析差异外显子使用的DEXSeq提供了一个Python脚本（dexseq_prepare_annotation.py）执行这个任务。</p>
<figure class="highlight stata"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">mkdir</span> -p /data/RNASeq/fastq/<span class="keyword">matrix</span>/</span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> $(seq 56 58); </span><br><span class="line"><span class="keyword">do</span> htseq-<span class="keyword">count</span> -f bam -r pos -s <span class="keyword">no</span> \</span><br><span class="line">/data/RNASeq/fastq/SRR35899<span class="variable">$&#123;i&#125;</span>_sorted.bam /data/Reference/gtf/gencode/gencode.v26lift37.annotation.sorted.gtf \</span><br><span class="line">1&gt;<span class="keyword">matrix</span>/SRR35899<span class="variable">$&#123;i&#125;</span>.<span class="keyword">count</span> 2&gt;<span class="keyword">matrix</span>/SRR35899<span class="variable">$&#123;i&#125;</span>_htseq.<span class="keyword">log</span> &amp;; done</span><br></pre></td></tr></table></figure>
<h2 id="表达矩阵"><a href="#表达矩阵" class="headerlink" title="表达矩阵"></a>表达矩阵</h2><p>在RNA-Seq分析中，每一个基因就是一个feature（特征？），而基因被认为是它的所有外显子的和集。在可变剪切分析中，可以单独把每个外显子当作一个feature。而在ChIP-Seq分析中，feature则是预先定义的结合域。但是确定一个read到底属于哪一个feature有时会非常棘手。</p>
<h4 id="这段描述很有意思，信息量也很多-From-hoptop"><a href="#这段描述很有意思，信息量也很多-From-hoptop" class="headerlink" title="这段描述很有意思，信息量也很多(From hoptop)"></a>这段描述很有意思，信息量也很多(From hoptop)</h4><p>我们这次分析是人类mRNA-Seq测序的结果，但是我们其实只下载了3个sra文件。一般而言RNA-Seq数据分析都至少要有2个重复，所以必须要有4个sra文件才行。我在仔细读完文章的方法这一段以后，发现他们有一批数据用的是其他课题组的： For 293 cells, the mRNA-seq results of the control samples include (1) those from the doxycycline-treated parental Flp-In T-REx 293 cells by us and (2) those from the doxycycline-treated control Flp-In T-REx 293 cells performed by another group unrelated to us (sample GSM1095127 in GSE44976)。 然后和Jimmy交流之后，他也承认自己只分析了小鼠的数据，而没有分析人类的数据。所以我们需要根据文章提供的线索下载另外一份数据，才能进行下一步的分析。<br>这个时候就有一个经常被问到的问题：不同来源的RNA-Seq数据能够直接比较吗？甚至说如果不同来源的RNA-seq数据的构建文库都不一样该如何比较?不同来源的RNA-Seq结果之间比较需要考虑 批次效应（batch effect) 的影响。<br>处理批次效应，根据我搜索的结果，是不能使用FPKM/RPKM，关于这个标准化的吐槽，我在biostars上找到了如下观点：<br>FPKM/RPKM 不是标准化的方法，它会引入文库特异的协变量<br>FPKM/RPKM has never been peer-reviewed, it has been introduced as an ad-hoc measure in a supplementary 没有同行评审<br>One of the authors of this paper states, that it should not be used because of faulty arithmetic 作者说算法有问题<br>All reviews so far have shown it to be an inferior scale for DE analysis of genes Length normalization is mostly dispensable imo in DE analysis because gene length is constant<br>有人建议使用一个Bioconductor包<a href="http://www.bioconductor.org/packages/devel/bioc/html/sva.html" target="_blank" rel="noopener">http://www.bioconductor.org/packages/devel/bioc/html/sva.html</a> 我没有具体了解，有生之年去了解补充。<br>还有人引用了一篇文献 IVT-seq reveals extreme bias in RNA-sequencing 证明不同文库的RNA-Seq结果会存在很大差异。<br>结论： 可以问下原作者他们是如何处理数据的，居然有一个居然没有重复的分析也能过审。改用小鼠数据进行分析。或者使用无重复的分析方法，或者模拟一份数据出来，先把流程走完。</p>
<h1 id="PANDA实践完整代码"><a href="#PANDA实践完整代码" class="headerlink" title="PANDA实践完整代码"></a>PANDA实践完整代码</h1><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">##1.SRA to fastq</span></span><br><span class="line"></span><br><span class="line">perl -F<span class="string">'\t'</span> -alne <span class="string">'if($F[7]=~/SRR/)&#123;$F[6]=~s/\s/_/g;$F[13]=~ s/\s|#/_/g;$F[13]=~s/\(|\)//g;print "$F[7]\t$F[6]_$F[13]"&#125;'</span> SraRunTable.txt &gt; Rename.txt</span><br><span class="line"></span><br><span class="line">perl -F<span class="string">'\t'</span> -alne <span class="string">'print "fastq-dump --split-3 --gzip -A $F[1] $F[0].sra &amp;" '</span> Rename.txt &gt;sratofq.sh</span><br><span class="line"></span><br><span class="line">sh sratofq.sh </span><br><span class="line"></span><br><span class="line">md5sum *.fastq.gz &gt;md5sum.txt</span><br><span class="line"></span><br><span class="line"><span class="comment">##2.QC</span></span><br><span class="line"><span class="built_in">cd</span> MYShen</span><br><span class="line">mkdir 1_FastQC_Raw_Data</span><br><span class="line">ls *.gz|<span class="keyword">while</span> <span class="built_in">read</span> id;<span class="keyword">do</span>(fastqc <span class="variable">$id</span> -o 1_FastQC_Raw_Data -t 8);<span class="keyword">done</span></span><br><span class="line"><span class="comment"># 质控统计</span></span><br><span class="line"><span class="built_in">cd</span> 1_FastQC_Raw_Data</span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> *.zip; <span class="keyword">do</span> unzip <span class="variable">$i</span>; <span class="keyword">done</span></span><br><span class="line">perl /data/RNASeq/MYShen/fastqc_table.pl</span><br><span class="line">csvtk tab2csv fastqc_table.txt | csvtk csv2md</span><br><span class="line">------------------------------------------------------------------------------------------------</span><br><span class="line">                                                  |total_reads|GC |Q20              |Q30</span><br><span class="line">:-------------------------------------------------|:----------|:--|:----------------|:----------------</span><br><span class="line">Homo_sapiens_AKAP95_KD_miR_12_293_cell_1_fastqc   |25914821   |50 |0.999810031487387|0.972201583024633</span><br><span class="line">Homo_sapiens_AKAP95_KD_miR_12_293_cell_2_fastqc   |25914821   |50 |0.977900484051192|0.934933990090072</span><br><span class="line">Homo_sapiens_AKAP95_KD_miR_8_293_cell_1_fastqc    |29720636   |50 |0.992602101445318|0.966831930304341</span><br><span class="line">Homo_sapiens_AKAP95_KD_miR_8_293_cell_2_fastqc    |29720636   |50 |0.978714654693123|0.93782410982053</span><br><span class="line">Homo_sapiens_Control_293_cell_1_fastqc            |28856780   |50 |0.989872946867992|0.966726176148248</span><br><span class="line">Homo_sapiens_Control_293_cell_2_fastqc            |28856780   |50 |0.977797280223227|0.940547004897982</span><br><span class="line">Mus_musculus_E14_cells_Akap95_shRNA_rep1_1_fastqc |52972617   |50 |0.99966391028307 |0.987561698545491</span><br><span class="line">Mus_musculus_E14_cells_Akap95_shRNA_rep1_2_fastqc |52972617   |50 |0.991854774481748|0.968082091276275</span><br><span class="line">Mus_musculus_E14_cells_Akap95_shRNA_rep2_1_fastqc |43802631   |49 |0.999679725428068|0.972184686419785</span><br><span class="line">Mus_musculus_E14_cells_Akap95_shRNA_rep2_2_fastqc |43802631   |49 |0.986887120253892|0.937616464383233</span><br><span class="line">Mus_musculus_E14_cells_control_shRNA_rep1_1_fastqc|30468155   |50 |0.99960200376177 |0.987110860654561</span><br><span class="line">Mus_musculus_E14_cells_control_shRNA_rep1_2_fastqc|30468155   |50 |0.991083449869222|0.966506018600785</span><br><span class="line">Mus_musculus_E14_cells_control_shRNA_rep2_1_fastqc|36763726   |50 |0.999694316126568|0.979622471169646</span><br><span class="line">Mus_musculus_E14_cells_control_shRNA_rep2_2_fastqc|36763726   |50 |0.990428680495556|0.951578715079022</span><br><span class="line">-------------------------------------------------------------------------------------------------------</span><br><span class="line"><span class="comment">#截图几个基因的 IGV 可视化结构</span></span><br><span class="line">grep -w <span class="string">'gene'</span> gencode.v26lift37.annotation.gtf |grep -w <span class="string">'TP53'</span>|cut -f 1,4,5 &gt;&gt;gene.bed</span><br><span class="line">grep -w <span class="string">'gene'</span> gencode.v26lift37.annotation.gtf |grep -w <span class="string">'KRAS'</span>|cut -f 1,4,5 &gt;&gt;gene.bed</span><br><span class="line">grep -w <span class="string">'gene'</span> gencode.v26lift37.annotation.gtf |grep -w <span class="string">'EGFR'</span>|cut -f 1,4,5 &gt;&gt;gene.bed</span><br><span class="line">bedtools igv -i gene.bed &gt;Bach_sanpshot.txt </span><br><span class="line"><span class="comment">#perl -alne '&#123;print "goto $F[0]:$F[1]-$F[2]\nsnapshot $F[3].png"&#125; '</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">##3. HISAT2比对</span></span><br><span class="line">---------------------------------------------------------------</span><br><span class="line"><span class="comment">#map.sh</span></span><br><span class="line">--------------------------------------------------------------------------------------------</span><br><span class="line"><span class="meta">#! usr/bin/bash</span></span><br><span class="line"><span class="built_in">set</span> -u</span><br><span class="line"><span class="built_in">set</span> -e</span><br><span class="line"><span class="built_in">set</span> -o pipefail</span><br><span class="line"></span><br><span class="line">hg19_ref=/data/Reference/index/hisat2/hg19/genome</span><br><span class="line">mm10_ref=/data/Reference/index/hisat2/mm10/genome</span><br><span class="line">data_path=/data/RNASeq/MYShen</span><br><span class="line">NUM_THREADS=5</span><br><span class="line">ls Homo*1.fastq.gz|<span class="keyword">while</span> <span class="built_in">read</span> id; \</span><br><span class="line"><span class="keyword">do</span>((hisat2 -t -p <span class="variable">$NUM_THREADS</span> -x <span class="variable">$hg19_ref</span> -1 <span class="variable">$data_path</span>/<span class="variable">$&#123;id%_*&#125;</span>_1.fastq.gz -2 \</span><br><span class="line"><span class="variable">$data_path</span>/<span class="variable">$&#123;id%_*&#125;</span>_2.fastq.gz 2&gt;<span class="variable">$&#123;id%_*&#125;</span>_map.log | samtools view -b - &gt;<span class="variable">$&#123;id%_*&#125;</span>.bam) &amp;);<span class="keyword">done</span> </span><br><span class="line">                               </span><br><span class="line">ls  Mus*1.fastq.gz|<span class="keyword">while</span> <span class="built_in">read</span> id; \</span><br><span class="line"><span class="keyword">do</span>((hisat2 -t -p <span class="variable">$NUM_THREADS</span> -x <span class="variable">$mm10_ref</span> -1 <span class="variable">$data_path</span>/<span class="variable">$&#123;id%_*&#125;</span>_1.fastq.gz -2 \</span><br><span class="line"><span class="variable">$data_path</span>/<span class="variable">$&#123;id%_*&#125;</span>_2.fastq.gz 2&gt;<span class="variable">$&#123;id%_*&#125;</span>_map.log | samtools view -b - &gt;<span class="variable">$&#123;id%_*&#125;</span>.bam) &amp;);<span class="keyword">done</span></span><br><span class="line">---------------------------------------------------------------------------------------------        </span><br><span class="line"></span><br><span class="line">bash map.sh</span><br><span class="line"><span class="comment">##因为前面开了并行，等前面执行完成，后面再单独执行</span></span><br><span class="line">ls *.bam | <span class="keyword">while</span> <span class="built_in">read</span> id;<span class="keyword">do</span> samtools sort --threads 25 <span class="variable">$id</span> -o <span class="variable">$&#123;id%.*&#125;</span>_sorted.bam; <span class="keyword">done</span> </span><br><span class="line">ls *_sorted.bam | <span class="keyword">while</span> <span class="built_in">read</span> id;<span class="keyword">do</span> samtools index <span class="variable">$id</span>; <span class="keyword">done</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">## 4.Count reads</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 人类</span></span><br><span class="line">mkdir matrix</span><br><span class="line">Homo_GTF=/data/Reference/gtf/gencode/gencode.v26lift37.annotation.gtf</span><br><span class="line"><span class="comment">###count 结果显示基因名称，如果用基因的id号，将 -i gene_name 参数删除即可</span></span><br><span class="line">ls Homo_sapiens*sorted.bam | <span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> (htseq-count -f bam -r pos -i gene_name -s no \</span><br><span class="line"><span class="variable">$id</span> <span class="variable">$Homo_GTF</span> &gt; matrix/<span class="variable">$&#123;id%_*&#125;</span>.count 2&gt; matrix/<span class="variable">$&#123;id%_*&#125;</span>.<span class="built_in">log</span> &amp;); <span class="keyword">done</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 老鼠</span></span><br><span class="line"><span class="comment"># 下载 gtf：http://www.gencodegenes.org/mouse_stats/archive.html</span></span><br><span class="line">axel ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_mouse/release_M10/gencode.vM10.annotation.gtf.gz</span><br><span class="line">gzip -d gencode.vM10.annotation.gtf.gz</span><br><span class="line">Mus_GTF=/data/Reference/gtf/gencode/gencode.vM10.annotation.gtf</span><br><span class="line">ls Mus_musculus*sorted.bam|<span class="keyword">while</span> <span class="built_in">read</span> id;<span class="keyword">do</span> (htseq-count -f bam -r pos -i gene_name -s no \</span><br><span class="line"><span class="variable">$id</span> <span class="variable">$Mus_GTF</span> &gt; matrix/<span class="variable">$&#123;id%_*&#125;</span>.count 2&gt; matrix/<span class="variable">$&#123;id%_*&#125;</span>.<span class="built_in">log</span> &amp;);<span class="keyword">done</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">##另外一些相关的代码</span></span><br><span class="line"><span class="comment">## ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_mouse/release_M1/</span></span><br><span class="line"><span class="comment">## http://hgdownload-test.cse.ucsc.edu/goldenPath/mm10/liftOver/</span></span><br><span class="line"><span class="comment">## GRCm38/mm10 (Dec, 2011) </span></span><br><span class="line"><span class="comment">## ls *bam |while read id;do ( ~/.local/bin/htseq-count  -f bam  $id   genecode/mm9/gencode.vM1.annotation.gtf.gz  1&gt;$&#123;id%%.*&#125;.gene.counts ) ;done </span></span><br><span class="line"><span class="comment">## ls *bam |while read id;do ( ~/.local/bin/htseq-count  -f bam -i exon_id  $id   genecode/mm9/gencode.vM1.annotation.gtf.gz  1&gt;$&#123;id%%.*&#125;.exon.counts ) ;done</span></span><br><span class="line"></span><br><span class="line"><span class="built_in">cd</span> matrix</span><br><span class="line"><span class="comment">#wc命令的功能为统计指定文件中的字节数-c,字符数-m,字数-w,行数-l</span></span><br><span class="line">wc -l Homo_sapiens*.count</span><br><span class="line">head -n 4 Homo_sapiens*.count</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">perl -lne <span class="string">'if($ARGV=~/Homo_sapiens_(.*)count/)&#123;print "$1\t$_"&#125;'</span> *|grep -v Homo_sapiens&gt;hg1.count</span><br><span class="line"><span class="comment"># 先把所有文件进行合并</span></span><br><span class="line">setwd(<span class="string">"~/rna_seq/work/matrix"</span>)</span><br><span class="line">hg &lt;- read.csv(file = <span class="string">"hg.count"</span>,header = F,sep = <span class="string">"\t"</span>)</span><br><span class="line">colnames(hg) &lt;- c(<span class="string">'sample'</span>,<span class="string">'gene'</span>,<span class="string">'count'</span>)</span><br><span class="line">library(reshape2)</span><br><span class="line">reads &lt;- dcast(hg,formula = gene ~ sample)</span><br><span class="line">write.table(reads,file = <span class="string">"hg_join.count"</span>,sep = <span class="string">"\t"</span>,quote = FALSE,row.names = FALSE)</span><br></pre></td></tr></table></figure>

          
          <hr>
          <ul class="pager no-print">
              
              <li class="previous">
                  <a href="/hexo/2017/11/07/2017-11-07_RNA_Seq_1/" data-toggle="tooltip" data-placement="left" title="RNA-Seq分析 笔记">&larr; Previous Post</a>
              </li>
              
              
              <li class="next">
                  <a href="/hexo/2017/11/06/2017-11-06_Markdown_Rmarkdown_Shiny/" data-toggle="tooltip" data-placement="top" title="Markdown+Rmarkdown+Shiny手册">Next Post&rarr;</a>
              </li>
              
          </ul>
          
  <br>

  


      </div>
      
  <div class="hidden-xs col-sm-3 toc-col">
    <div class="toc-wrap">
        Table of Contents
        
          <ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#分析软件安装"><span class="toc-number">1.</span> <span class="toc-text">分析软件安装</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#数据下载"><span class="toc-number">2.</span> <span class="toc-text">数据下载</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#数据分析"><span class="toc-number">3.</span> <span class="toc-text">数据分析</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#质控"><span class="toc-number">3.1.</span> <span class="toc-text">质控</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Hisat2比对"><span class="toc-number">3.2.</span> <span class="toc-text">Hisat2比对</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#比对质控"><span class="toc-number">3.3.</span> <span class="toc-text">比对质控</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#Reads-计数-From-hoptop"><span class="toc-number">4.</span> <span class="toc-text">Reads 计数 (From hoptop)</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#表达矩阵"><span class="toc-number">4.1.</span> <span class="toc-text">表达矩阵</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#这段描述很有意思，信息量也很多-From-hoptop"><span class="toc-number">4.1.0.1.</span> <span class="toc-text">这段描述很有意思，信息量也很多(From hoptop)</span></a></li></ol></li></ol></li></ol><li class="toc-item toc-level-1"><a class="toc-link" href="#PANDA实践完整代码"><span class="toc-number">5.</span> <span class="toc-text">PANDA实践完整代码</span></a></li>
        
    </div>
  </div>


    </div>
</article>

<!-- Footer -->
<!-- footer.ejs -->
<footer class="no-print">

    <div class="text-center">

        <ul class="list-inline">
            
            
            

            

            

            

            

            

        </ul>
        <div class="text-muted copyright">
            &copy;
            
            2017 - 2019
            
            -
            xt
            <br>
            
            
            
            | 
            <p>
                <span id="busuanzi_container_site_pv"><span id="busuanzi_value_site_pv"></span> <b>PV</b></span> -
                <span id="busuanzi_container_site_uv"><span id="busuanzi_value_site_uv"></span> <b>UV</b></span> -
                118.4k <b>Words</b>
        </p></div>

        <div class="search-container">
            <form class="site-search-form">
                <span class="glyphicon glyphicon-search"></span>
                <input type="text" id="local-search-input" class="st-search-input" placeholder="Search...">
            </form>
            <div id="local-search-result" class="local-search-result-cls"></div>
        </div>
        <script type="text/javascript" id="local.search.active">
            var inputArea = document.querySelector("#local-search-input");
            inputArea.onclick = function () {
                var root = "/hexo/";
                getSearchFile(root);
                this.onclick = null
            }
            inputArea.onkeydown = function () {
                if (event.keyCode == 13) return false
            }
        </script>

    </div>
</footer>

<!-- Custom Theme JavaScript -->
<script src="/hexo/js/main.js"></script>

<!-- async load function -->
<script>
    function async(u, c) {
        var d = document, t = 'script',
            o = d.createElement(t),
            s = d.getElementsByTagName(t)[0];
        o.src = u;
        if (c) {
            o.addEventListener('load', function (e) {
                c(null, e);
            }, false);
        }
        s.parentNode.insertBefore(o, s);
    }
</script>

<!--fastClick.js -->
<script>
    async("//cdn.bootcss.com/fastclick/1.0.6/fastclick.min.js", function () {
        var $nav = document.querySelector("nav");
        if ($nav) FastClick.attach($nav);
    })
</script><!-- hexo-inject:begin --><!-- hexo-inject:end -->



</body>

</html>
