<!DOCTYPE html>
<html lang="en">

<!-- layout.ejs-->
<head><meta name="generator" content="Hexo 3.8.0">
    <!-- hexo-inject:begin --><!-- hexo-inject:end --><meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="Adelaide">
    <meta name="author" content="xt">
    <meta name="keyword" content="">
    <link rel="canonical" href="https://taoshengxu.github.io/hexo/hexo/2020/08/05/2020-08-05_SCENIC/">
    <link rel="shortcut icon" href="/hexo/img/favicon.png">
    <link rel="alternate" type="application/atom+xml" title="IIM" href="/atom.xml">
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/animate.css/3.5.2/animate.min.css">
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
    <link rel="stylesheet" href="https://ajax.googleapis.com/ajax/libs/jqueryui/1.12.1/themes/smoothness/jquery-ui.css">
    <script src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.12.1/jquery-ui.min.js"></script>
    <script src="/hexo/js/search.js"></script>

    <title>
        
        SCENIC｜undefined
        
    </title>

    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">

    <link rel="stylesheet" href="/hexo/css/main.css">

    
      <link href="//fonts.googleapis.com/css?family=Source+Code+Pro" rel="stylesheet" type="text/css">
      <link rel="stylesheet" href="/hexo/css/highlight.css">
    

    

    <script async src="//dn-lbstatics.qbox.me/busuanzi/2.3/busuanzi.pure.mini.js"></script>


    

    

    


    
  


    




    <script async defer src="https://buttons.github.io/buttons.js"></script>

<link rel="stylesheet" href="/hexo/css/prism-solarizedlight.css" type="text/css">
<link rel="stylesheet" href="/hexo/css/prism-line-numbers.css" type="text/css"><!-- hexo-inject:begin --><!-- hexo-inject:end --></head>

<style>
    header.intro-header {
        background-image: url('/hexo/img/northernlights-sisimiut-lake.jpg')
    }
</style>
<!-- hack iOS CSS :active style -->
<body ontouchstart="" class="animated fadeIn">
<!-- hexo-inject:begin --><!-- hexo-inject:end --><header>
  <nav class="navbar navbar-default header-navbar" id="nav-top" data-ispost="true" data-istags="false" data-ishome="false">
    <div class="container-fluid">
      <div class="navbar-header page-scroll">
        <button type="button" class="navbar-toggle" data-toggle="collapse" aria-expanded="false" data-target="#website_navbar">
          <span class="sr-only">Toggle navigation</span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <span class="navbar-brand animated pulse">
          <a class="brand-logo" href="/hexo/">
                <img src="/hexo/img/banner.jpg?h=350&amp;auto=compress&amp;cs=tinysrgb">
          </a>
        </span>
      </div>

      <div class="collapse navbar-collapse" id="website_navbar">
          <ul class="nav navbar-nav navbar-right">
              
                <li>
                  <a href="/hexo/">home</a>
                </li>
              
                <li>
                  <a href="/hexo/archives/">archives</a>
                </li>
              
                <li>
                  <a href="/hexo/categories/">categories</a>
                </li>
              
                <li>
                  <a href="/hexo/tags/">tags</a>
                </li>
              
                <li>
                  <a href="/hexo/columns/">columns</a>
                </li>
              
          </ul>
      </div>
  </div></nav>


  
    <style>
       .intro-header {
          background-image: url('/hexo/post_cover_images/moon.jpg');
      }
    </style>

    <div class="intro-header">
        <div class="container">
            <div class="row">
                <div class="col-lg-8 col-lg-offset-2 col-md-10 col-md-offset-1 text-center">
                    <div class="site-heading">
                        <h1>SCENIC</h1>
                        
                        

                        
                          <span class="meta">
                               <span class="meta-item">Author: XT</span>
                               <span class="meta-item">Date: Aug 5, 2020</span>
                               
                          </span>
                          <div class="tags text-center">
                              Categories: 
                              <a class="tag" href="/hexo/categories/#Bioinformatics" title="Bioinformatics">Bioinformatics</a>
                              
                          </div>
                          <div class="tags text-center">
                              Tags: 
                          </div>
                        
                    </div>
                </div>
            </div>
        </div>
    </div>
  
</header>


<!-- Main Content -->
<!-- post.ejs -->
<article>
    <div class="container">
      <div class="col-lg-8 col-lg-offset-1 col-sm-9">
          
          <span class="post-count">3,217 words in total, 15 minutes required.</span>
          <hr>
          
          <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/hint.css/2.4.1/hint.min.css"><h2 id="SCENIC-single-cell-RNA-seq数据推断基因调控网络和细胞功能聚类"><a href="#SCENIC-single-cell-RNA-seq数据推断基因调控网络和细胞功能聚类" class="headerlink" title="SCENIC | single-cell RNA-seq数据推断基因调控网络和细胞功能聚类"></a>SCENIC | single-cell RNA-seq数据推断基因调控网络和细胞功能聚类</h2><p><a href="https://vsn-pipelines.readthedocs.io/en/latest/getting-started.html#quick-start" target="_blank" rel="noopener">https://vsn-pipelines.readthedocs.io/en/latest/getting-started.html#quick-start</a>  nextflow</p>
<p><a href="https://pyscenic.readthedocs.io/en/latest/installation.html" target="_blank" rel="noopener">https://pyscenic.readthedocs.io/en/latest/installation.html</a></p>
<p><a href="https://github.com/aertslab/SCENIC" target="_blank" rel="noopener">https://github.com/aertslab/SCENIC</a></p>
<p><a href="https://cloud.tencent.com/developer/article/1606747" target="_blank" rel="noopener">https://cloud.tencent.com/developer/article/1606747</a> 这篇是论文大致翻译，写入了理解，good.</p>
<p><a href="https://github.com/aertslab/SCENICprotocol/" target="_blank" rel="noopener">https://github.com/aertslab/SCENICprotocol/</a>  教程</p>
<p><strong>SENIC</strong>是一种同时重建基因调控网络并从单细胞RNA-seq数据中鉴定stable cell states的工具。基于共表达和DNA模基序 （motif）分析推断基因调控网络 ，然后在每个细胞中分析网络活性以鉴定细胞状态。</p>
<p><strong>输入</strong>：SCENIC需要输入的是单细胞RNA-seq<code>表达矩阵</code>。每列对应于样品（细胞）。每行对应一个基因。</p>
<blockquote>
<p>基因ID应该是gene-symbol并存储为rownames （尤其是基因名字部分是为了与<code>RcisTarget</code>数据库兼容）；表达数据是Gene的reads count。根据作者的测试，提供原始的或Normalized UMI count，无论是否log转换，或使用TPM值，结果相差不大。</p>
</blockquote>
<hr>
<h2 id="SENIC-工作流程"><a href="#SENIC-工作流程" class="headerlink" title="SENIC 工作流程"></a><strong>SENIC</strong> 工作流程</h2><p><strong>建立基因调控网络</strong>（<code>Gene Regulation Network，GRN</code>）：</p>
<ol>
<li><p>基于共表达识别每个转录因子TF的潜在靶标。过滤表达矩阵并运行<strong>GENIE3或者GRNBoost</strong>，它们是利用表达矩阵推断基因调控网络的一种算法，能得到转录因子和潜在靶标的相关性网络；将目标从GENIE3或者GRNBoost格式转为共表达模块。</p>
<p>用GENIE3（随机森林) 或GRNBoost (Gradient Boosting) 推断转录因子与候选靶基因之间的共表达模块。每个模块包含一个转录因子及其靶基因，纯粹基于共表达。</p>
<blockquote>
<p>选择这两个工具的依据为：原则上，推测共表达网络的目前大多方法都可以使用这个过程的输出文件，只要他的输出文件是一个包含每个转录因子的潜在靶点的列表即可。我们选择GENIE3是因为它可以确定非线性的关系，即使他们只存在于一部分样品中。同时它在Network inference  DREAM5挑战中具有很好的表现。同时这个方法可以很好的运行在R中。</p>
<p>然而，GENIE3是非常耗时的以及对计算力有严重的消耗，比如在处理3k到5k个细胞的时候可能会花费几个小时。因此为了处理大数据集，我们开发了GRNBoost算法，这个算法提供与GENIE3相似的结果，但是时间上大大节省了，因此我们强烈推荐大家在处理大数据集的时候使用它</p>
</blockquote>
</li>
<li><p>根据DNA模序分析（<code>motif</code>）选择潜在的直接结合靶标（调节因子）（利用<code>RcisTarget</code>包：TF基序分析）</p>
<p>使用RcisTarget分析每个共表达模块中的基因，以鉴定enriched motifs；仅保留TF motif富集的模块和targets，<strong>每个TF及其潜在的直接targets gene被称作一个<code>调节子（regulon）</code></strong></p>
</li>
</ol>
<p><strong>确定细胞状态及其调节因子</strong>：</p>
<ol>
<li><p>分析每个细胞中的网络活性（AUCell） 在细胞中评分调节子（计算AUC）</p>
<p>使用AUCell评估每个细胞中每个regulon的活性，AUCell分数用于生成Regulon活性矩阵，通过为每个regulon设置AUC阈值，可以将该矩阵进行二值化（0|1，on|off），这将确定Regulon在哪些细胞中处于“打开”状态。</p>
</li>
</ol>
<blockquote>
<p><strong>使用RcisTarget是SCENIC不同于大多共表达算法的重要区别</strong>。</p>
</blockquote>
<p>由于GENIE3模块仅基于共表达，因此结果可能包含许多误报和间接target，为了鉴定推断的直接结合的靶标基因，使用RcisTarget对每个共表达模块进行顺式调控基序（motif）分析。仅保留具有正确基因上游调节子且显着富集TF motif的模块，并对它们进行修剪以除去缺乏基序支持的间接靶标，这些处理后的模块才称为regulon。</p>
<blockquote>
<p><strong>AUCell算法来对每个细胞中的每个regulon的活性进行评分。</strong></p>
</blockquote>
<p>对于一个给定的regulon，通过比较所有细胞间的AUCell打分值，我们可以识别哪些细胞具有更显著高的regulon活性。通过卡阈值得到的二元活性矩阵使矩阵维数减少（可理解为只有 0|1，on|off），对于下游分析很有用。例如，<strong>基于regulon二元活性矩阵的聚类，可以根据某个调控子网络（regulon）的活性来识别细胞群类型和细胞状态。</strong>由于regulon是整体评分的，而不是使用单个基因的表达，因此这种方法对于个别基因的dropouts很有效。</p>
<blockquote>
<p>SCENIC使用建议</p>
</blockquote>
<p>随着单细胞数据集大小的增加，作者建议两种补充方法来扩展网络推断：</p>
<p>1.取采样数据集的子集来推断出GRN，并在AUCell评分步骤中包括所有细胞</p>
<p>2.使用更高效的机器学习和大数据处理解决方案，作者应用GRNBoost 梯度增强来代替random-forest regression，这种实现方式大大减少了推断GRN所需的时间，并将为在非常大的数据集上进行网络推断铺平道路。</p>
<p>SCENIC是一种普遍适用的分析scRNA-seq数据的方法，利用TF和顺式调控序列来指导细胞状态的发现。文章的结果表明，GRNs是确定细胞状态的可靠方法，并且scRNA-seq数据非常适合跟踪基因调控过程，在基因调控过程中特定组合的TFs驱动细胞特异性的转录组。</p>
<hr>
<h3 id="初始化SCENIC设置"><a href="#初始化SCENIC设置" class="headerlink" title="初始化SCENIC设置"></a>初始化SCENIC设置</h3><p>为了在<code>SCENIC</code>的多个步骤中保持设置一致，SCENIC包中的大多数函数使用一个公共对象，该对象存储当前运行的选项并代替大多数函数的“参数”。比如下面的<code>org</code>，<code>dbDir</code>等，可以在开始就将物种rog（<code>mgi</code>—— mouse， <code>hgnc</code> —— human， <code>dmel</code> —— fly）和<code>RcisTarge</code>数据库位置分别读给对象<code>org</code>，<code>dbDir</code>，之后统一用函数<code>initializeScenic</code>得到对象<code>scenicOptions</code>。具体参数设置可以用<code>?initializeScenic</code>help一下。</p>
<h3 id="不同数据格式的读入"><a href="#不同数据格式的读入" class="headerlink" title="不同数据格式的读入"></a>不同数据格式的读入</h3><p>loom文件<br><figure class="highlight applescript"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">download.<span class="built_in">file</span>(<span class="string">"http://loom.linnarssonlab.org/clone/Previously%20Published/Cortex.loom"</span>, <span class="string">"Cortex.loom"</span>)</span><br><span class="line">loomPath &lt;- <span class="string">"Cortex.loom"</span></span><br></pre></td></tr></table></figure></p>
<p>10x的输出文件<br><figure class="highlight haskell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="title">singleCellMatrix</span> &lt;- <span class="type">Seurat</span>::<span class="type">Read10X</span>(<span class="class"><span class="keyword">data</span>.dir="<span class="keyword">data</span>/pbmc3k/filtered_gene_bc_matrices/hg19/")</span></span><br><span class="line"><span class="title">cellInfo</span> &lt;- <span class="class"><span class="keyword">data</span>.frame(<span class="title">seuratCluster</span>=<span class="type">Idents(seuratObject)</span>)</span></span><br></pre></td></tr></table></figure></p>
<p>R objects (e.g. Seurat, SingleCellExperiment)<br><figure class="highlight lisp"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">sce &lt;- load_as_sce(<span class="name">dataPath</span>) # any SingleCellExperiment object</span><br><span class="line">exprMat &lt;- counts(<span class="name">sce</span>)</span><br><span class="line">cellInfo &lt;- colData(<span class="name">sce</span>)</span><br></pre></td></tr></table></figure></p>
<h2 id="1-评分数据库cisTarget-databases"><a href="#1-评分数据库cisTarget-databases" class="headerlink" title="1. 评分数据库cisTarget databases"></a>1. 评分数据库cisTarget databases</h2><p><a href="https://resources.aertslab.org/cistarget/" target="_blank" rel="noopener">https://resources.aertslab.org/cistarget/</a></p>
<blockquote>
<p>最快的下载办法： 复制下载地址用sougou浏览器下载。</p>
</blockquote>
<figure class="highlight makefile"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">mm_url=<span class="string">"https://resources.aertslab.org/cistarget/databases/mus_musculus/mm9/refseq_r45/mc9nr/gene_based/mm9-500bp-upstream-7species.mc9nr.feather"</span></span><br><span class="line">wget -c $mm_url</span><br></pre></td></tr></table></figure>
<h2 id="2-SCENIC-docker运行指令"><a href="#2-SCENIC-docker运行指令" class="headerlink" title="2. SCENIC docker运行指令"></a>2. SCENIC docker运行指令</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br></pre></td><td class="code"><pre><span class="line">docker pull aertslab/pyscenic </span><br><span class="line">或者</span><br><span class="line">docker pull aertslab/pyscenic:latest </span><br><span class="line"></span><br><span class="line">docker run -it --rm -v /data:/data aertslab/pyscenic:0.9.18 pyscenic -h 测试</span><br><span class="line">docker run -it --rm -v /data:/data aertslab/pyscenic:latest pyscenic -h 测试</span><br><span class="line"><span class="comment">#docker run -it --rm -p 8888:8888 -v /data:/data aertslab/pyscenic:0.9.18 指定端口</span></span><br><span class="line">   </span><br><span class="line"><span class="comment">## 执行的bash脚本如下</span></span><br><span class="line"><span class="meta">#!/bin/bash</span></span><br><span class="line">data=<span class="string">"/data/CUHKSZ/jupyter/pbmc10k_filtered_scenic.loom"</span></span><br><span class="line">tf= <span class="string">"/data/CUHKSZ/jupyter/allTFs_hg38.txt"</span></span><br><span class="line">docker run -it --rm -v /data/CUHKSZ/jupyter:/data/CUHKSZ/jupyter aertslab/pyscenic:latest pyscenic grn &#123;data&#125; &#123;tf&#125; -o adj.csv --num_workers 18</span><br><span class="line"></span><br><span class="line">----------------------------------------------</span><br><span class="line"><span class="comment"># 下面网址有详细使用方法</span></span><br><span class="line"><span class="comment"># https://hub.docker.com/r/aertslab/pyscenic</span></span><br><span class="line">docker run -it --rm \</span><br><span class="line">    -v /path/to/data:/scenicdata \</span><br><span class="line">    aertslab/pyscenic:[version] pyscenic grn \</span><br><span class="line">        --num_workers 6 \</span><br><span class="line">        -o /scenicdata/expr_mat.adjacencies.tsv \</span><br><span class="line">        /scenicdata/expr_mat.tsv \</span><br><span class="line">        /scenicdata/allTFs_hg38.txt</span><br><span class="line"></span><br><span class="line">docker run -it --rm \</span><br><span class="line">    -v /path/to/data:/scenicdata \</span><br><span class="line">    aertslab/pyscenic:[version] pyscenic ctx \</span><br><span class="line">        /scenicdata/expr_mat.adjacencies.tsv \</span><br><span class="line">        /scenicdata/hg19-tss-centered-5kb-7species.mc9nr.feather \</span><br><span class="line">        /scenicdata/hg19-tss-centered-10kb-7species.mc9nr.feather \</span><br><span class="line">        --annotations_fname /scenicdata/motifs-v9-nr.hgnc-m0.001-o0.0.tbl \</span><br><span class="line">        --expression_mtx_fname /scenicdata/expr_mat.tsv \</span><br><span class="line">        --mode <span class="string">"dask_multiprocessing"</span> \</span><br><span class="line">        --output /scenicdata/regulons.csv \</span><br><span class="line">        --num_workers 6</span><br><span class="line"></span><br><span class="line">docker run -it --rm \</span><br><span class="line">    -v /path/to/data:/scenicdata \</span><br><span class="line">    aertslab/pyscenic:[version] pyscenic aucell \</span><br><span class="line">        /scenicdata/expr_mat.tsv \</span><br><span class="line">        /scenicdata/regulons.csv \</span><br><span class="line">        -o /scenicdata/auc_mtx.csv \</span><br><span class="line">        --num_workers 6</span><br></pre></td></tr></table></figure>
<h2 id="docker-关联-jupyter-notebook（成功过一次，后来不成功）"><a href="#docker-关联-jupyter-notebook（成功过一次，后来不成功）" class="headerlink" title="docker 关联 jupyter notebook（成功过一次，后来不成功）"></a>docker 关联 jupyter notebook（成功过一次，后来不成功）</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line">pip3 install remote_ikernel</span><br><span class="line">/soft/miniconda3/envs/scenic_protocol/bin/remote_ikernel manage</span><br><span class="line"></span><br><span class="line">remote_ikernel manage --add \</span><br><span class="line">    --kernel_cmd=<span class="string">"docker run -it --rm aertslab/pyscenic:0.9.18 ipython kernel -f &#123;connection_file&#125;"</span> \</span><br><span class="line">    --name=<span class="string">"pyscenic-0918-docker"</span> \</span><br><span class="line">    --interface=ssh \</span><br><span class="line">    --host=hostname \</span><br><span class="line">    --workdir=<span class="string">"/data"</span> \</span><br><span class="line">    --language=python3  </span><br><span class="line">  </span><br><span class="line"> <span class="comment"># 关联地址 /home/ubuntu/.local/share/jupyter/kernels/.. </span></span><br><span class="line"> jupyter notebook --ip=0.0.0.0 --port=8001</span><br></pre></td></tr></table></figure>
<h1 id="Scanpy-–-Single-Cell-Analysis-in-Python"><a href="#Scanpy-–-Single-Cell-Analysis-in-Python" class="headerlink" title="Scanpy – Single-Cell Analysis in Python"></a>Scanpy – Single-Cell Analysis in Python</h1><p>Scanpy is a scalable toolkit for analyzing single-cell gene expression data built jointly with <a href="https://anndata.readthedocs.io/" target="_blank" rel="noopener">anndata</a>. It includes preprocessing, visualization, clustering, trajectory inference and differential expression testing. The Python-based implementation efficiently deals with datasets of more than one million cells.</p>
<p><a href="https://scanpy.readthedocs.io/en/stable/" target="_blank" rel="noopener">https://scanpy.readthedocs.io/en/stable/</a></p>
<p><a href="https://scanpy.readthedocs.io/en/stable/index.html" target="_blank" rel="noopener">https://scanpy.readthedocs.io/en/stable/index.html</a></p>
<figure class="highlight mipsasm"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">conda <span class="keyword">install </span>seaborn <span class="keyword">scikit-learn </span>statsmodels numba pytables</span><br><span class="line">conda <span class="keyword">install </span>-c conda-forge python-igraph leidenalg</span><br><span class="line">pip3 <span class="keyword">install </span><span class="keyword">scanpy</span></span><br></pre></td></tr></table></figure>
<hr>
<h2 id="SENIC-R例子学习"><a href="#SENIC-R例子学习" class="headerlink" title="SENIC R例子学习"></a><strong>SENIC R例子学习</strong></h2><h3 id="1-软件安装"><a href="#1-软件安装" class="headerlink" title="1. 软件安装"></a>1. 软件安装</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line"><span class="keyword">if</span> (!requireNamespace(<span class="string">"BiocManager"</span>, quietly = <span class="literal">TRUE</span>)) install.packages(<span class="string">"BiocManager"</span>)</span><br><span class="line">BiocManager::version()</span><br><span class="line"><span class="comment"># If your bioconductor version is previous to 3.9, see the section bellow</span></span><br><span class="line"></span><br><span class="line"><span class="comment">## Required</span></span><br><span class="line">BiocManager::install(c(<span class="string">"AUCell"</span>, <span class="string">"RcisTarget"</span>))</span><br><span class="line">BiocManager::install(c(<span class="string">"GENIE3"</span>)) <span class="comment"># Optional. Can be replaced by GRNBoost</span></span><br><span class="line"></span><br><span class="line"><span class="comment">## Optional (but highly recommended):</span></span><br><span class="line"><span class="comment"># To score the network on cells (i.e. run AUCell):</span></span><br><span class="line">BiocManager::install(c(<span class="string">"zoo"</span>, <span class="string">"mixtools"</span>, <span class="string">"rbokeh"</span>))</span><br><span class="line"><span class="comment"># For various visualizations and perform t-SNEs:</span></span><br><span class="line">BiocManager::install(c(<span class="string">"DT"</span>, <span class="string">"NMF"</span>, <span class="string">"pheatmap"</span>, <span class="string">"R2HTML"</span>, <span class="string">"Rtsne"</span>))</span><br><span class="line"><span class="comment"># To support paralell execution (not available in Windows):</span></span><br><span class="line">BiocManager::install(c(<span class="string">"doMC"</span>, <span class="string">"doRNG"</span>))</span><br><span class="line"><span class="comment"># To export/visualize in http://scope.aertslab.org</span></span><br><span class="line"><span class="keyword">if</span> (!requireNamespace(<span class="string">"devtools"</span>, quietly = <span class="literal">TRUE</span>)) install.packages(<span class="string">"devtools"</span>)</span><br><span class="line">devtools::install_github(<span class="string">"aertslab/SCopeLoomR"</span>, build_vignettes = <span class="literal">TRUE</span>)</span><br><span class="line"></span><br><span class="line"><span class="keyword">if</span> (!requireNamespace(<span class="string">"devtools"</span>, quietly = <span class="literal">TRUE</span>)) install.packages(<span class="string">"devtools"</span>)</span><br><span class="line">devtools::install_github(<span class="string">"aertslab/SCENIC"</span>) </span><br><span class="line">packageVersion(<span class="string">"SCENIC"</span>)</span><br></pre></td></tr></table></figure>
<blockquote>
<p>一定要注意的一个问题，运行时候一定要加载 foreach 包，否者会有莫名其妙的报错。</p>
</blockquote>
<h3 id="2-分析"><a href="#2-分析" class="headerlink" title="2.分析"></a>2.分析</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">library</span>(foreach)<span class="comment">#注意一定要加载这个包，不然会报错！</span></span><br><span class="line"><span class="keyword">library</span>(SCENIC)</span><br><span class="line">loomPath &lt;- system.file(package=<span class="string">"SCENIC"</span>, <span class="string">"examples/mouseBrain_toy.loom"</span>)</span><br><span class="line"><span class="keyword">library</span>(SCopeLoomR)</span><br><span class="line">loom &lt;- open_loom(loomPath, mode=<span class="string">"r"</span>)</span><br><span class="line">exprMat &lt;- get_dgem(loom)</span><br><span class="line">cellInfo &lt;- get_cellAnnotation(loom)</span><br><span class="line">close_loom(loom)</span><br><span class="line"></span><br><span class="line">cellInfo &lt;- data.frame(cellInfo)</span><br><span class="line">colnames(cellInfo)[which(colnames(cellInfo)==<span class="string">"Class"</span>)] &lt;- <span class="string">"CellType"</span></span><br><span class="line">cbind(table(cellInfo$CellType))</span><br><span class="line">saveRDS(cellInfo, file=<span class="string">"int/cellInfo.Rds"</span>)</span><br><span class="line"></span><br><span class="line">colVars &lt;- list(CellType=c(<span class="string">"microglia"</span>=<span class="string">"forestgreen"</span>,</span><br><span class="line">                           <span class="string">"endothelial-mural"</span>=<span class="string">"darkorange"</span>,</span><br><span class="line">                           <span class="string">"astrocytes_ependymal"</span>=<span class="string">"magenta4"</span>,</span><br><span class="line">                           <span class="string">"oligodendrocytes"</span>=<span class="string">"hotpink"</span>,</span><br><span class="line">                           <span class="string">"interneurons"</span>=<span class="string">"red3"</span>,</span><br><span class="line">                           <span class="string">"pyramidal CA1"</span>=<span class="string">"skyblue"</span>,</span><br><span class="line">                           <span class="string">"pyramidal SS"</span>=<span class="string">"darkblue"</span>))</span><br><span class="line">colVars$CellType &lt;- colVars$CellType[intersect(names(colVars$CellType), cellInfo$CellType)]</span><br><span class="line">saveRDS(colVars, file=<span class="string">"int/colVars.Rds"</span>)</span><br></pre></td></tr></table></figure>
<blockquote>
<p>关于scenicOptions 设置和操作问题</p>
</blockquote>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#默认的设置对大多数分析过程都有效。在所有分析中比较特殊的参数是物种这个选项，其中mgi表示小鼠，hgnc表示人，dmel表示果蝇。同时其中的dir应该是RcisTarget数据库保存的位置。</span></span><br><span class="line">org=<span class="string">"mgi"</span> <span class="comment"># or hgnc, or dmel</span></span><br><span class="line">dbDir=<span class="string">"cisTarget_databases"</span> <span class="comment"># RcisTarget databases location</span></span><br><span class="line">myDatasetTitle=<span class="string">"SCENIC example on Mouse brain"</span> <span class="comment"># choose a name for your analysis</span></span><br><span class="line">data(defaultDbNames) <span class="comment">##可以查看默认使用的数据库，也可以人为指定</span></span><br><span class="line">dbs &lt;- defaultDbNames[[org]]</span><br><span class="line">dbDir &lt;- <span class="string">"/data/CUHKSZ/SCENIC/cisTarget_databases"</span></span><br><span class="line">scenicOptions &lt;- initializeScenic(org=org, dbDir=dbDir, dbs=dbs, datasetTitle=myDatasetTitle, nCores=<span class="number">10</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># Modify if needed</span></span><br><span class="line">scenicOptions@inputDatasetInfo$cellInfo &lt;- <span class="string">"int/cellInfo.Rds"</span></span><br><span class="line">scenicOptions@inputDatasetInfo$colVars &lt;- <span class="string">"int/colVars.Rds"</span></span><br><span class="line"><span class="comment"># Databases:</span></span><br><span class="line"><span class="comment"># scenicOptions@settings$dbs &lt;- c("mm9-5kb-mc8nr"="mm9-tss-centered-5kb-10species.mc8nr.feather")</span></span><br><span class="line"><span class="comment"># scenicOptions@settings$db_mcVersion &lt;- "v8"</span></span><br><span class="line"></span><br><span class="line">scenicOptions@settings$verbose &lt;- <span class="literal">TRUE</span></span><br><span class="line">scenicOptions@settings$seed &lt;- <span class="number">123</span></span><br><span class="line">scenicOptions@settings$nCores &lt;- <span class="number">1</span></span><br><span class="line">scenicOptions@settings$dbs &lt;- scenicOptions@settings$dbs[<span class="string">"10kb"</span>] <span class="comment"># For toy run</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># Save to use at a later time...</span></span><br><span class="line">saveRDS(scenicOptions, file=<span class="string">"int/scenicOptions.Rds"</span>)</span><br></pre></td></tr></table></figure>
<blockquote>
<p>开始执行SCENIC</p>
</blockquote>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># (Adjust minimum values according to your dataset)</span></span><br><span class="line">genesKept &lt;- geneFiltering(exprMat, scenicOptions=scenicOptions,</span><br><span class="line">                           minCountsPerGene=<span class="number">3</span>*<span class="number">.01</span>*ncol(exprMat),</span><br><span class="line">                           minSamples=ncol(exprMat)*<span class="number">.01</span>)</span><br><span class="line"></span><br><span class="line">interestingGenes &lt;- c(<span class="string">"Sox9"</span>, <span class="string">"Sox10"</span>, <span class="string">"Dlx5"</span>)</span><br><span class="line"><span class="comment"># any missing?</span></span><br><span class="line">interestingGenes[which(interestingGenes %<span class="keyword">in</span>% genesKept)]</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">exprMat_filtered &lt;- exprMat[genesKept, ]</span><br><span class="line">runCorrelation(exprMat_filtered, scenicOptions)</span><br><span class="line"><span class="comment"># scenicOptions &lt;- readRDS("int/scenicOptions.Rds")</span></span><br><span class="line"><span class="comment"># Optional: add log (if it is not logged/normalized already)</span></span><br><span class="line">exprMat_filtered_log &lt;- log2(exprMat_filtered+<span class="number">1</span>) </span><br><span class="line">runGenie3(exprMat_filtered_log, scenicOptions)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line">scenicOptions@settings$verbose &lt;- <span class="literal">TRUE</span></span><br><span class="line">scenicOptions@settings$seed &lt;- <span class="number">123</span></span><br><span class="line">scenicOptions@settings$dbs</span><br><span class="line"><span class="comment">##                                        500bp                                           10kb </span></span><br><span class="line"><span class="comment">##"mm9-500bp-upstream-7species.mc9nr.feather" "mm9-tss-centered-10kb-7species.mc9nr.feather" </span></span><br><span class="line">scenicOptions@settings$dbs &lt;- scenicOptions@settings$dbs[<span class="string">"10kb"</span>] <span class="comment"># 只选择一个，节省时间</span></span><br><span class="line">runSCENIC_1_coexNetwork2modules(scenicOptions)</span><br><span class="line">runSCENIC_2_createRegulons(scenicOptions, coexMethod=c(<span class="string">"top5perTarget"</span>)) <span class="comment">#只选择一种方法，节省时间</span></span><br><span class="line"></span><br><span class="line">logMat &lt;- log2(exprMat+<span class="number">1</span>)</span><br><span class="line"><span class="comment">## 如果报错，把ncores 设为1可能能解决</span></span><br><span class="line">scenicOptions@settings$nCores &lt;- <span class="number">1</span></span><br><span class="line">runSCENIC_3_scoreCells(scenicOptions, logMat)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment"># Binarize activity?</span></span><br><span class="line"><span class="comment"># aucellApp &lt;- plotTsne_AUCellApp(scenicOptions, logMat)</span></span><br><span class="line"><span class="comment"># savedSelections &lt;- shiny::runApp(aucellApp)</span></span><br><span class="line"><span class="comment"># newThresholds &lt;- savedSelections$thresholds</span></span><br><span class="line"><span class="comment"># scenicOptions@fileNames$int["aucell_thresholds",1] &lt;- "int/newThresholds.Rds"</span></span><br><span class="line"><span class="comment"># saveRDS(newThresholds, file=getIntName(scenicOptions, "aucell_thresholds"))</span></span><br><span class="line"><span class="comment"># saveRDS(scenicOptions, file="int/scenicOptions.Rds")</span></span><br><span class="line">runSCENIC_4_aucell_binarize(scenicOptions)</span><br><span class="line"></span><br><span class="line"><span class="comment">### Exploring output </span></span><br><span class="line"><span class="comment"># Check files in folder 'output'</span></span><br><span class="line"><span class="comment"># .loom file @ http://scope.aertslab.org</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># output/Step2_MotifEnrichment_preview.html in detail/subset:</span></span><br><span class="line">motifEnrichment_selfMotifs_wGenes &lt;- loadInt(scenicOptions, <span class="string">"motifEnrichment_selfMotifs_wGenes"</span>)</span><br><span class="line">tableSubset &lt;- motifEnrichment_selfMotifs_wGenes[highlightedTFs==<span class="string">"Sox8"</span>]</span><br><span class="line">viewMotifs(tableSubset) </span><br><span class="line"></span><br><span class="line"><span class="comment"># output/Step2_regulonTargetsInfo.tsv in detail: </span></span><br><span class="line">regulonTargetsInfo &lt;- loadInt(scenicOptions, <span class="string">"regulonTargetsInfo"</span>)</span><br><span class="line">tableSubset_1 &lt;- regulonTargetsInfo[TF==<span class="string">"Stat6"</span> &amp; highConfAnnot==<span class="literal">TRUE</span>]</span><br><span class="line">viewMotifs(tableSubset_1)</span><br></pre></td></tr></table></figure>
<h2 id="生成文件列表"><a href="#生成文件列表" class="headerlink" title="生成文件列表"></a>生成文件列表</h2><figure class="highlight objectivec"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><span class="line">                                  fileName                                       </span><br><span class="line">genesKept                         <span class="string">"int/1.1_genesKept.Rds"</span>   <span class="meta">#通过了筛选的基因列表                      </span></span><br><span class="line">corrMat                           <span class="string">"int/1.2_corrMat.Rds"</span>          <span class="meta"># 基因相关性分析的结果矩阵                </span></span><br><span class="line">genie3wm                          <span class="string">"int/1.3_GENIE3_weightMatrix.Rds"</span>              </span><br><span class="line">genie3ll                          <span class="string">"int/1.4_GENIE3_linkList.Rds"</span>                  </span><br><span class="line">genie3weighPlot                   <span class="string">"int/1.5_weightPlot"</span>                           </span><br><span class="line">tfModules_asDF                    <span class="string">"int/1.6_tfModules_asDF.Rds"</span>                   </span><br><span class="line">tfModules_forEnrichment           <span class="string">"int/2.1_tfModules_forMotifEnrichmet.Rds"</span>      </span><br><span class="line">motifs_AUC                        <span class="string">"int/2.2_motifs_AUC.Rds"</span>                       </span><br><span class="line">motifEnrichment_full              <span class="string">"int/2.3_motifEnrichment.Rds"</span>                  </span><br><span class="line">motifEnrichment_selfMotifs_wGenes <span class="string">"int/2.4_motifEnrichment_selfMotifs_wGenes.Rds"</span></span><br><span class="line">regulonTargetsInfo                <span class="string">"int/2.5_regulonTargetsInfo.Rds"</span>  <span class="meta">#所有转录因子及其靶基因的信息，包含motif信息             </span></span><br><span class="line">regulons                          <span class="string">"int/2.6_regulons_asGeneSet.Rds"</span>     <span class="meta">#regulon及其调控的基因          </span></span><br><span class="line">regulons_incidMat                 <span class="string">"int/2.6_regulons_asIncidMat.Rds"</span>              </span><br><span class="line">aucell_regulons                   <span class="string">"int/3.1_regulons_forAUCell.Rds"</span>       <span class="meta">#与regulons中的信息一样        </span></span><br><span class="line">aucell_genesStatsPlot             <span class="string">"int/3.2_aucellGenesStats"</span>                     </span><br><span class="line">aucell_rankings                   <span class="string">"int/3.3_aucellRankings.Rds"</span>                   </span><br><span class="line">aucell_regulonAUC                 <span class="string">"int/3.4_regulonAUC.Rds"</span>    <span class="meta">#储存每个细胞的每个regulons的AUC数值，可以使用getAUC(loadInt(scenicOptions,<span class="meta-string">"aucell_regulonAUC"</span>)) 提取             </span></span><br><span class="line">aucell_thresholds                 <span class="string">"int/3.5_AUCellThresholds.Rds"</span>     <span class="meta">#储存用于进行二进制化的阈值，以及通过阈值的细胞名称            </span></span><br><span class="line">aucell_thresholdsTxt              <span class="string">"int/3.5_AUCellThresholds_Info.tsv"</span>            </span><br><span class="line">aucell_binary_full                <span class="string">"int/4.1_binaryRegulonActivity.Rds"</span>  <span class="meta">#储存 二进制化的矩阵         </span></span><br><span class="line">aucell_binary_nonDupl             <span class="string">"int/4.2_binaryRegulonActivity_nonDupl.Rds"</span>    </span><br><span class="line">aucell_regulonSelection           <span class="string">"int/4.3_regulonSelections.Rds"</span>                </span><br><span class="line">aucell_binaryRegulonOrder         <span class="string">"int/4.4_binaryRegulonOrder.Rds"</span></span><br></pre></td></tr></table></figure>
          
          <hr>
          <ul class="pager no-print">
              
              <li class="previous">
                  <a href="/hexo/2020/09/12/2020-09-12_jupyterNotebook/" data-toggle="tooltip" data-placement="left" title="Jupyter Notebook">&larr; Previous Post</a>
              </li>
              
              
              <li class="next">
                  <a href="/hexo/2020/08/03/2020-08-03_kinase/" data-toggle="tooltip" data-placement="top" title="激酶kinase 和 细胞外基质ECM, extracellular matrix ">Next Post&rarr;</a>
              </li>
              
          </ul>
          

      </div>
      
  <div class="hidden-xs col-sm-3 toc-col">
    <div class="toc-wrap">
        Table of Contents
        
          <ol class="toc"><li class="toc-item toc-level-2"><a class="toc-link" href="#SCENIC-single-cell-RNA-seq数据推断基因调控网络和细胞功能聚类"><span class="toc-number">1.</span> <span class="toc-text">SCENIC | single-cell RNA-seq数据推断基因调控网络和细胞功能聚类</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#SENIC-工作流程"><span class="toc-number">2.</span> <span class="toc-text">SENIC 工作流程</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#初始化SCENIC设置"><span class="toc-number">2.1.</span> <span class="toc-text">初始化SCENIC设置</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#不同数据格式的读入"><span class="toc-number">2.2.</span> <span class="toc-text">不同数据格式的读入</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#1-评分数据库cisTarget-databases"><span class="toc-number">3.</span> <span class="toc-text">1. 评分数据库cisTarget databases</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#2-SCENIC-docker运行指令"><span class="toc-number">4.</span> <span class="toc-text">2. SCENIC docker运行指令</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#docker-关联-jupyter-notebook（成功过一次，后来不成功）"><span class="toc-number">5.</span> <span class="toc-text">docker 关联 jupyter notebook（成功过一次，后来不成功）</span></a></li></ol><li class="toc-item toc-level-1"><a class="toc-link" href="#Scanpy-–-Single-Cell-Analysis-in-Python"><span class="toc-number"></span> <span class="toc-text">Scanpy – Single-Cell Analysis in Python</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#SENIC-R例子学习"><span class="toc-number">1.</span> <span class="toc-text">SENIC R例子学习</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#1-软件安装"><span class="toc-number">1.1.</span> <span class="toc-text">1. 软件安装</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#2-分析"><span class="toc-number">1.2.</span> <span class="toc-text">2.分析</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#生成文件列表"><span class="toc-number">2.</span> <span class="toc-text">生成文件列表</span></a></li></ol>
        
    </li></div>
  </div>


    </div>
</article>

<!-- Footer -->
<!-- footer.ejs -->
<footer class="no-print">

    <div class="text-center">

        <ul class="list-inline">
            
            
            

            

            

            

            

            

        </ul>
        <div class="text-muted copyright">
            &copy;
            
            2017 - 2020
            
            -
            xt
            <br>
            
            
            
            | 
            <p>
                <span id="busuanzi_container_site_pv"><span id="busuanzi_value_site_pv"></span> <b>PV</b></span> -
                <span id="busuanzi_container_site_uv"><span id="busuanzi_value_site_uv"></span> <b>UV</b></span> -
                174.8k <b>Words</b>
        </p></div>

        <div class="search-container">
            <form class="site-search-form">
                <span class="glyphicon glyphicon-search"></span>
                <input type="text" id="local-search-input" class="st-search-input" placeholder="Search...">
            </form>
            <div id="local-search-result" class="local-search-result-cls"></div>
        </div>
        <script type="text/javascript" id="local.search.active">
            var inputArea = document.querySelector("#local-search-input");
            inputArea.onclick = function () {
                var root = "/hexo/";
                getSearchFile(root);
                this.onclick = null
            }
            inputArea.onkeydown = function () {
                if (event.keyCode == 13) return false
            }
        </script>

    </div>
</footer>

<!-- Custom Theme JavaScript -->
<script src="/hexo/js/main.js"></script>

<!-- async load function -->
<script>
    function async(u, c) {
        var d = document, t = 'script',
            o = d.createElement(t),
            s = d.getElementsByTagName(t)[0];
        o.src = u;
        if (c) {
            o.addEventListener('load', function (e) {
                c(null, e);
            }, false);
        }
        s.parentNode.insertBefore(o, s);
    }
</script>

<!--fastClick.js -->
<script>
    async("//cdn.bootcss.com/fastclick/1.0.6/fastclick.min.js", function () {
        var $nav = document.querySelector("nav");
        if ($nav) FastClick.attach($nav);
    })
</script><!-- hexo-inject:begin --><!-- hexo-inject:end -->



</body>

</html>
