<!DOCTYPE html>
<html lang="en">

<!-- layout.ejs-->
<head><meta name="generator" content="Hexo 3.8.0">
    <!-- hexo-inject:begin --><!-- hexo-inject:end --><meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="Adelaide">
    <meta name="author" content="xt">
    <meta name="keyword" content="">
    <link rel="canonical" href="https://taoshengxu.github.io/hexo/hexo/2017/11/03/2017-11-03_1000software/">
    <link rel="shortcut icon" href="/hexo/img/favicon.png">
    <link rel="alternate" type="application/atom+xml" title="IIM" href="/atom.xml">
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/animate.css/3.5.2/animate.min.css">
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
    <link rel="stylesheet" href="https://ajax.googleapis.com/ajax/libs/jqueryui/1.12.1/themes/smoothness/jquery-ui.css">
    <script src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.12.1/jquery-ui.min.js"></script>
    <script src="/hexo/js/search.js"></script>

    <title>
        
        生物信息学常见1000个软件的安装代码｜undefined
        
    </title>

    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">

    <link rel="stylesheet" href="/hexo/css/main.css">

    
      <link href="//fonts.googleapis.com/css?family=Source+Code+Pro" rel="stylesheet" type="text/css">
      <link rel="stylesheet" href="/hexo/css/highlight.css">
    

    

    <script async src="//dn-lbstatics.qbox.me/busuanzi/2.3/busuanzi.pure.mini.js"></script>


    
      <meta name="google-site-verification" content="xxx">
    

    

    


    
  


    
<script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
                (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
            m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
    ga('create', 'UA-xxxxxx-xx', 'auto');
    ga('send', 'pageview');
</script>



<script>
    var _baId = 'xxx';
    // Originial
    var _hmt = _hmt || [];
    (function() {
        var hm = document.createElement("script");
        hm.src = "//hm.baidu.com/hm.js?" + _baId;
        var s = document.getElementsByTagName("script")[0];
        s.parentNode.insertBefore(hm, s);
    })();
</script>



    <script async defer src="https://buttons.github.io/buttons.js"></script>

<link rel="stylesheet" href="/hexo/css/prism-solarizedlight.css" type="text/css">
<link rel="stylesheet" href="/hexo/css/prism-line-numbers.css" type="text/css"><!-- hexo-inject:begin --><!-- hexo-inject:end --></head>

<style>
    header.intro-header {
        background-image: url('/hexo/img/northernlights-sisimiut-lake.jpg')
    }
</style>
<!-- hack iOS CSS :active style -->
<body ontouchstart="" class="animated fadeIn">
<!-- hexo-inject:begin --><!-- hexo-inject:end --><header>
  <nav class="navbar navbar-default header-navbar" id="nav-top" data-ispost="true" data-istags="false" data-ishome="false">
    <div class="container-fluid">
      <div class="navbar-header page-scroll">
        <button type="button" class="navbar-toggle" data-toggle="collapse" aria-expanded="false" data-target="#website_navbar">
          <span class="sr-only">Toggle navigation</span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <span class="navbar-brand animated pulse">
          <a class="brand-logo" href="/hexo/">
                <img src="/hexo/img/banner.jpg?h=350&amp;auto=compress&amp;cs=tinysrgb">
          </a>
        </span>
      </div>

      <div class="collapse navbar-collapse" id="website_navbar">
          <ul class="nav navbar-nav navbar-right">
              
                <li>
                  <a href="/hexo/">home</a>
                </li>
              
                <li>
                  <a href="/hexo/archives/">archives</a>
                </li>
              
                <li>
                  <a href="/hexo/categories/">categories</a>
                </li>
              
                <li>
                  <a href="/hexo/tags/">tags</a>
                </li>
              
                <li>
                  <a href="/hexo/columns/">columns</a>
                </li>
              
          </ul>
      </div>
  </div></nav>


  
    <style>
       .intro-header {
          background-image: url('/hexo/img/northernlights-sisimiut-lake.jpg');
      }
    </style>

    <div class="intro-header">
        <div class="container">
            <div class="row">
                <div class="col-lg-8 col-lg-offset-2 col-md-10 col-md-offset-1 text-center">
                    <div class="site-heading">
                        <h1>生物信息学常见1000个软件的安装代码</h1>
                        
                        

                        
                          <span class="meta">
                               <span class="meta-item">Author: xt</span>
                               <span class="meta-item">Date: Nov 3, 2017</span>
                               
                          </span>
                          <div class="tags text-center">
                              Categories: 
                          </div>
                          <div class="tags text-center">
                              Tags: 
                          </div>
                        
                    </div>
                </div>
            </div>
        </div>
    </div>
  
</header>


<!-- Main Content -->
<!-- post.ejs -->
<article>
    <div class="container">
      <div class="col-lg-8 col-lg-offset-1 col-sm-9">
          
          <span class="post-count">6,683 words in total, 41 minutes required.</span>
          <hr>
          
          <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/hint.css/2.4.1/hint.min.css"><p>引自Jiangming Zeng<a href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247484870&amp;idx=1&amp;sn=d336ed1951b5cff14c591201084622fd&amp;chksm=9b48457dac3fcc6bb25bdb2a0e744013a3c11d0b7b8cc8b9f274560260618c07bdf438611752&amp;mpshare=1&amp;scene=1&amp;srcid=08174RqFPmfWUyO1PO1Xh9Uz#rd" target="_blank" rel="noopener">原文</a></p>
<h1 id="分析软件"><a href="#分析软件" class="headerlink" title="分析软件"></a>分析软件</h1><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">## annovar and GATK </span></span><br><span class="line"><span class="comment">## Download and install sratoolkit</span></span><br><span class="line"><span class="comment">## http://www.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=software</span></span><br><span class="line"><span class="comment">## http://www.ncbi.nlm.nih.gov/books/NBK158900/</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir sratoolkit &amp;&amp;  cd sratoolkit</span><br><span class="line">wget http://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/<span class="number">2.6</span><span class="number">.3</span>/sratoolkit.2.6.3-centos_linux64.tar.gz</span><br><span class="line"><span class="comment">##</span></span><br><span class="line"><span class="comment">##  Length: 63453761 (61M) [application/x-gzip]</span></span><br><span class="line"><span class="comment">##  Saving to: "sratoolkit.2.6.3-centos_linux64.tar.gz"</span></span><br><span class="line">tar zxvf sratoolkit.2.6.3-centos_linux64.tar.gz</span><br><span class="line">~/biosoft/sratoolkit/sratoolkit.2.6.3-centos_linux64/bin/fastdump -h</span><br><span class="line">mkdir -p  ~/biosoft/myBin</span><br><span class="line">echo <span class="string">'export PATH=/home/jianmingzeng/biosoft/myBin/bin:$PATH'</span> &gt;&gt;~/.bashrc </span><br><span class="line"><span class="keyword">source</span> ~/.bashrc</span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir cmake &amp;&amp;  cd cmake</span><br><span class="line">wget http://cmake.org/files/v3.3/cmake-<span class="number">3.3</span><span class="number">.2</span>.tar.gz</span><br><span class="line">tar xvfz cmake-<span class="number">3.3</span><span class="number">.2</span>.tar.gz</span><br><span class="line">cd cmake-<span class="number">3.3</span><span class="number">.2</span> </span><br><span class="line">./configure --prefix=/home/jianmingzeng/biosoft/myBin</span><br><span class="line">make</span><br><span class="line">make install</span><br></pre></td></tr></table></figure>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br><span class="line">127</span><br><span class="line">128</span><br><span class="line">129</span><br><span class="line">130</span><br><span class="line">131</span><br><span class="line">132</span><br><span class="line">133</span><br><span class="line">134</span><br><span class="line">135</span><br><span class="line">136</span><br><span class="line">137</span><br><span class="line">138</span><br><span class="line">139</span><br><span class="line">140</span><br><span class="line">141</span><br><span class="line">142</span><br><span class="line">143</span><br><span class="line">144</span><br><span class="line">145</span><br><span class="line">146</span><br><span class="line">147</span><br><span class="line">148</span><br><span class="line">149</span><br><span class="line">150</span><br><span class="line">151</span><br><span class="line">152</span><br><span class="line">153</span><br><span class="line">154</span><br><span class="line">155</span><br><span class="line">156</span><br><span class="line">157</span><br><span class="line">158</span><br><span class="line">159</span><br><span class="line">160</span><br><span class="line">161</span><br><span class="line">162</span><br><span class="line">163</span><br><span class="line">164</span><br><span class="line">165</span><br><span class="line">166</span><br><span class="line">167</span><br><span class="line">168</span><br><span class="line">169</span><br><span class="line">170</span><br><span class="line">171</span><br><span class="line">172</span><br><span class="line">173</span><br><span class="line">174</span><br><span class="line">175</span><br><span class="line">176</span><br><span class="line">177</span><br><span class="line">178</span><br><span class="line">179</span><br><span class="line">180</span><br><span class="line">181</span><br><span class="line">182</span><br><span class="line">183</span><br><span class="line">184</span><br><span class="line">185</span><br><span class="line">186</span><br><span class="line">187</span><br><span class="line">188</span><br><span class="line">189</span><br><span class="line">190</span><br><span class="line">191</span><br><span class="line">192</span><br><span class="line">193</span><br><span class="line">194</span><br><span class="line">195</span><br><span class="line">196</span><br><span class="line">197</span><br><span class="line">198</span><br><span class="line">199</span><br><span class="line">200</span><br><span class="line">201</span><br><span class="line">202</span><br><span class="line">203</span><br><span class="line">204</span><br><span class="line">205</span><br><span class="line">206</span><br><span class="line">207</span><br><span class="line">208</span><br><span class="line">209</span><br><span class="line">210</span><br><span class="line">211</span><br><span class="line">212</span><br><span class="line">213</span><br><span class="line">214</span><br><span class="line">215</span><br><span class="line">216</span><br><span class="line">217</span><br><span class="line">218</span><br><span class="line">219</span><br><span class="line">220</span><br><span class="line">221</span><br><span class="line">222</span><br><span class="line">223</span><br><span class="line">224</span><br><span class="line">225</span><br><span class="line">226</span><br><span class="line">227</span><br><span class="line">228</span><br><span class="line">229</span><br><span class="line">230</span><br><span class="line">231</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">## Download and install samtools</span></span><br><span class="line"><span class="comment">## http://samtools.sourceforge.net/</span></span><br><span class="line"><span class="comment">## http://www.htslib.org/doc/samtools.html</span></span><br><span class="line"></span><br><span class="line"><span class="comment">##存放高通量测序比对结果的标准格式</span></span><br><span class="line"><span class="comment">##功能： Reading/writing/editing/indexing/viewing SAM/BAM/CRAM format</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir samtools &amp;&amp;  cd samtools</span><br><span class="line">wget https://github.com/samtools/samtools/releases/download/<span class="number">1.3</span><span class="number">.1</span>/samtools-<span class="number">1.3</span><span class="number">.1</span>.tar.bz2 </span><br><span class="line">tar xvfj samtools-<span class="number">1.3</span><span class="number">.1</span>.tar.bz2 </span><br><span class="line">cd samtools-<span class="number">1.3</span><span class="number">.1</span> </span><br><span class="line">./configure --prefix=/home/jianmingzeng/biosoft/myBin</span><br><span class="line">make</span><br><span class="line">make install</span><br><span class="line">~/biosoft/myBin/bin/samtools --help</span><br><span class="line">~/biosoft/myBin/bin/plot-bamstats --help</span><br><span class="line">cd htslib-<span class="number">1.3</span><span class="number">.1</span></span><br><span class="line">./configure --prefix=/home/jianmingzeng/biosoft/myBin</span><br><span class="line">make</span><br><span class="line">make install</span><br><span class="line">~/biosoft/myBin/bin/tabix</span><br><span class="line"><span class="comment"># ## Download and install tabix </span></span><br><span class="line"><span class="comment"># cd ~/biosoft</span></span><br><span class="line"><span class="comment"># mkdir tabix &amp;&amp;  cd tabix</span></span><br><span class="line"><span class="comment"># # http://genometoolbox.blogspot.com/2013/11/installing-tabix-on-unix.html</span></span><br><span class="line"><span class="comment"># tar xvfj tabix-0.2.6.tar.bz2 </span></span><br><span class="line"><span class="comment"># cd tabix-0.2.6</span></span><br><span class="line"><span class="comment"># make</span></span><br><span class="line"><span class="comment"># cd ~/biosoft</span></span><br><span class="line"><span class="comment"># ## http://samtools.github.io/bcftools/</span></span><br><span class="line"><span class="comment"># mkdir htslib &amp;&amp;  cd htslib  </span></span><br><span class="line"><span class="comment"># git clone git://github.com/samtools/htslib.git </span></span><br><span class="line"><span class="comment"># cd htslib</span></span><br><span class="line"><span class="comment"># make </span></span><br><span class="line"><span class="comment">## Download and install bcftools</span></span><br><span class="line"><span class="comment">## http://www.htslib.org/download/</span></span><br><span class="line"><span class="comment">## http://www.htslib.org/doc/bcftools-1.0.html</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir bcftools &amp;&amp;  cd bcftools</span><br><span class="line">wget https://github.com/samtools/bcftools/releases/download/<span class="number">1.3</span><span class="number">.1</span>/bcftools-<span class="number">1.3</span><span class="number">.1</span>.tar.bz2</span><br><span class="line">tar xvfj bcftools-<span class="number">1.3</span><span class="number">.1</span>.tar.bz2</span><br><span class="line">cd bcftools-<span class="number">1.3</span><span class="number">.1</span> </span><br><span class="line">make</span><br><span class="line">cp bcftools /home/jianmingzeng/biosoft/myBin</span><br><span class="line">~/biosoft/myBin/bin/bcftools --help</span><br><span class="line"><span class="comment">## Download and install vcftools</span></span><br><span class="line"><span class="comment">## https://vcftools.github.io/index.html </span></span><br><span class="line"><span class="comment">## http://vcftools.sourceforge.net/specs.html</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir vcftools &amp;&amp;  cd vcftools</span><br><span class="line"><span class="comment"># wget https://codeload.github.com/vcftools/vcftools/legacy.zip/master -O  vcftools-vcftools-v0.1.14-24-gac1bfd5.zip </span></span><br><span class="line"><span class="comment"># unzip vcftools-vcftools-v0.1.14-24-gac1bfd5.zip </span></span><br><span class="line"><span class="comment"># mv vcftools-vcftools-ac1bfd5 vcftools-v0.1.14-24</span></span><br><span class="line"><span class="comment"># cd vcftools-v0.1.14-24</span></span><br><span class="line"><span class="comment"># export PERL5LIB=/home/jianmingzeng/biosoft/vcftools/vcftools-v0.1.14-24/src/perl/</span></span><br><span class="line"><span class="comment"># ./autogen.sh </span></span><br><span class="line"><span class="comment"># ./configure     --prefix=/home/jianmingzeng/biosoft/myBin</span></span><br><span class="line"><span class="comment"># make </span></span><br><span class="line"><span class="comment"># make install </span></span><br><span class="line"><span class="comment"># ~/biosoft/myBin/bin/vcftools --help</span></span><br><span class="line">wget https://sourceforge.net/projects/vcftools/files/vcftools_0.1.13.tar.gz </span><br><span class="line">tar zxvf vcftools_0.1.13.tar.gz</span><br><span class="line">cd  vcftools_0.1.13</span><br><span class="line">make</span><br><span class="line"><span class="comment">## Download and install ANNOVAR  </span></span><br><span class="line">cd ~/biosoft</span><br><span class="line"><span class="comment"># The latest version of ANNOVAR (2016Feb01) can be downloaded here (registration required)</span></span><br><span class="line"><span class="comment"># http://www.openbioinformatics.org/annovar/annovar_download_form.php </span></span><br><span class="line">mkdir ANNOVAR  &amp;&amp;  cd ANNOVAR  </span><br><span class="line"><span class="comment">## Download and install samstat</span></span><br><span class="line"><span class="comment">## http://samstat.sourceforge.net/</span></span><br><span class="line"><span class="comment">## http://www.htslib.org/doc/samtools.html</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir samstat &amp;&amp;  cd samstat</span><br><span class="line">wget http://liquidtelecom.dl.sourceforge.net/project/samstat/samstat-<span class="number">1.5</span>.tar.gz</span><br><span class="line">tar zxvf  samstat-<span class="number">1.5</span>.tar.gz </span><br><span class="line">cd samstat-<span class="number">1.5</span> </span><br><span class="line">./configure --prefix=/home/jianmingzeng/biosoft/myBin</span><br><span class="line">make</span><br><span class="line">make install</span><br><span class="line">~/biosoft/myBin/bin/samstat --help</span><br><span class="line"><span class="comment">## Download and install picardtools</span></span><br><span class="line"><span class="comment">## https://sourceforge.net/projects/picard/</span></span><br><span class="line"><span class="comment">## https://github.com/broadinstitute/picard</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir picardtools &amp;&amp;  cd picardtools</span><br><span class="line">wget http://ncu.dl.sourceforge.net/project/picard/picard-tools/<span class="number">1.119</span>/picard-tools-<span class="number">1.119</span>.zip</span><br><span class="line">unzip picard-tools-<span class="number">1.119</span>.zip </span><br><span class="line"><span class="comment">## Download and install freebayes</span></span><br><span class="line"><span class="comment">## https://github.com/ekg/freebayes</span></span><br><span class="line"><span class="comment">## http://clavius.bc.edu/~erik/CSHL-advanced-sequencing/freebayes-tutorial.html</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir freebayes &amp;&amp;  cd freebayes</span><br><span class="line"><span class="comment">## wget -O freebayes-master.zip  https://codeload.github.com/ekg/freebayes/zip/master</span></span><br><span class="line"><span class="comment">## unzip freebayes-master.zip</span></span><br><span class="line">wget http://clavius.bc.edu/~erik/freebayes/freebayes-5d5b8ac0.tar.gz</span><br><span class="line">tar xzvf freebayes-5d5b8ac0.tar.gz</span><br><span class="line">cd freebayes</span><br><span class="line">make</span><br><span class="line"> ~/biosoft/freebayes/freebayes/bin/freebayes</span><br><span class="line">cd ~/biosoft</span><br><span class="line"><span class="comment">## https://sourceforge.net/projects/varscan/files/</span></span><br><span class="line"><span class="comment">## http://varscan.sourceforge.net/index.html</span></span><br><span class="line">mkdir VarScan  &amp;&amp;  cd VarScan  </span><br><span class="line">wget https://sourceforge.net/projects/varscan/files/VarScan.v2.3.9.jar </span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir SnpEff &amp;&amp;  cd SnpEff</span><br><span class="line"><span class="comment">##  http://snpeff.sourceforge.net/</span></span><br><span class="line"><span class="comment">##  http://snpeff.sourceforge.net/SnpSift.html </span></span><br><span class="line"><span class="comment">##  http://snpeff.sourceforge.net/SnpEff_manual.html</span></span><br><span class="line">wget http://sourceforge.net/projects/snpeff/files/snpEff_latest_core.zip </span><br><span class="line"><span class="comment">## java -jar snpEff.jar download GRCh37.75</span></span><br><span class="line"><span class="comment">## java -Xmx4G -jar snpEff.jar -i vcf -o vcf GRCh37.75 example.vcf &gt; example_snpeff.vcf</span></span><br><span class="line">unzip snpEff_latest_core.zip</span><br><span class="line"><span class="comment">## https://github.com/najoshi/sickle</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir sickle &amp;&amp; cd sickle</span><br><span class="line">wget https://codeload.github.com/najoshi/sickle/zip/master -O sickle.zip</span><br><span class="line">unzip sickle.zip</span><br><span class="line">cd sickle-master</span><br><span class="line">make</span><br><span class="line">~/biosoft/sickle/sickle-master/sickle -h</span><br><span class="line">cd ~/biosoft</span><br><span class="line"><span class="comment">## http://www.usadellab.org/cms/?page=trimmomatic</span></span><br><span class="line"><span class="comment">## http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/TrimmomaticManual_V0.32.pdf</span></span><br><span class="line">mkdir Trimmomatic &amp;&amp; cd Trimmomatic</span><br><span class="line">wget http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/Trimmomatic-<span class="number">0.36</span>.zip </span><br><span class="line">unzip Trimmomatic-<span class="number">0.36</span>.zip </span><br><span class="line">java -jar ~/biosoft/Trimmomatic/Trimmomatic-<span class="number">0.36</span>/trimmomatic-<span class="number">0.36</span>.jar -h</span><br><span class="line"><span class="comment">## Download and install bedtools</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir bedtools &amp;&amp;  cd bedtools</span><br><span class="line">wget https://github.com/arq5x/bedtools2/releases/download/v2.25.0/bedtools-<span class="number">2.25</span><span class="number">.0</span>.tar.gz</span><br><span class="line"><span class="comment">## Length: 19581105 (19M) [application/octet-stream] </span></span><br><span class="line">tar -zxvf bedtools-<span class="number">2.25</span><span class="number">.0</span>.tar.gz</span><br><span class="line">cd bedtools2</span><br><span class="line">make</span><br><span class="line"><span class="comment">#~/biosoft/bedtools/bedtools2/bin/</span></span><br><span class="line"><span class="comment">## Download and install PeakRanger</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir PeakRanger &amp;&amp;  cd PeakRanger</span><br><span class="line">wget https://sourceforge.net/projects/ranger/files/PeakRanger-<span class="number">1.18</span>-Linux-x86_64.zip </span><br><span class="line"><span class="comment">## Length: 1517587 (1.4M) [application/octet-stream]</span></span><br><span class="line">unzip PeakRanger-<span class="number">1.18</span>-Linux-x86_64.zip</span><br><span class="line">~/biosoft/PeakRanger/bin/peakranger -h</span><br><span class="line"><span class="comment">## Download and install bowtie</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir bowtie &amp;&amp;  cd bowtie</span><br><span class="line">wget https://sourceforge.net/projects/bowtie-bio/files/bowtie2/<span class="number">2.2</span><span class="number">.9</span>/bowtie2-<span class="number">2.2</span><span class="number">.9</span>-linux-x86_64.zip </span><br><span class="line"><span class="comment">#Length: 27073243 (26M) [application/octet-stream]</span></span><br><span class="line"><span class="comment">#Saving to: "download"   ## I made a mistake here for downloading the bowtie2 </span></span><br><span class="line">mv download  bowtie2-<span class="number">2.2</span><span class="number">.9</span>-linux-x86_64.zip</span><br><span class="line">unzip bowtie2-<span class="number">2.2</span><span class="number">.9</span>-linux-x86_64.zip</span><br><span class="line"><span class="comment">## Download and install BWA</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir bwa &amp;&amp;  cd bwa</span><br><span class="line"><span class="comment">#http://sourceforge.net/projects/bio-bwa/files/</span></span><br><span class="line">wget https://sourceforge.net/projects/bio-bwa/files/bwa-<span class="number">0.7</span><span class="number">.15</span>.tar.bz2 </span><br><span class="line">tar xvfj bwa-<span class="number">0.7</span><span class="number">.15</span>.tar.bz2 <span class="comment"># x extracts, v is verbose (details of what it is doing), f skips prompting for each individual file, and j tells it to unzip .bz2 files</span></span><br><span class="line">cd bwa-<span class="number">0.7</span><span class="number">.15</span></span><br><span class="line">make</span><br><span class="line"><span class="comment">#export PATH=$PATH:/path/to/bwa-0.7.15 # Add bwa to your PATH by editing ~/.bashrc file (or .bash_profile or .profile file)</span></span><br><span class="line"><span class="comment"># /path/to/ is an placeholder. Replace with real path to BWA on your machine</span></span><br><span class="line"><span class="comment">#source ~/.bashrc</span></span><br><span class="line"><span class="comment">## Download and install macs2  </span></span><br><span class="line"><span class="comment">## // https://pypi.python.org/pypi/MACS2/</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir macs2 &amp;&amp;  cd macs2</span><br><span class="line"><span class="comment">## just stick to PyPI release: https://pypi.python.org/pypi/MACS2</span></span><br><span class="line">wget https://pypi.python.org/packages/9f/<span class="number">99</span>/a8ac96b357f6b0a6f559fe0f5a81bcae12b98579551620ce07c5183aee2c/MACS2-<span class="number">2.1</span><span class="number">.1</span><span class="number">.20160309</span>.tar.gz -O MACS2-<span class="number">2.1</span><span class="number">.1</span>.tar.gz </span><br><span class="line">tar zxvf  MACS2-<span class="number">2.1</span><span class="number">.1</span>.tar.gz </span><br><span class="line">cd  MACS2-<span class="number">2.1</span><span class="number">.1</span><span class="number">.20160309</span>/</span><br><span class="line">python setup.py install --user </span><br><span class="line"><span class="comment">## https://docs.python.org/2/install/</span></span><br><span class="line">~/.local/bin/macs2 --help</span><br><span class="line"><span class="comment">#wget https://codeload.github.com/taoliu/MACS/zip/master -O MACS-master.zip</span></span><br><span class="line"><span class="comment">#unzip MACS-master.zip</span></span><br><span class="line"><span class="comment">#cd MACS-master </span></span><br><span class="line"><span class="comment">## So you can't just pull github snapshot then run setup.py to install MACS2. Instead</span></span><br><span class="line"><span class="comment"># ImageMagick</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir ImageMagick &amp;&amp;  cd ImageMagick</span><br><span class="line"><span class="comment">## http://www.imagemagick.org/ </span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir weblogo &amp;&amp;  cd weblogo</span><br><span class="line"><span class="comment">## http://weblogo.berkeley.edu/</span></span><br><span class="line">wget http://weblogo.berkeley.edu/release/weblogo.2.8.2.tar.gz</span><br><span class="line">tar zxvf weblogo.2.8.2.tar.gz</span><br><span class="line">cd weblogo</span><br><span class="line">export PATH=$PATH:/home/jianmingzeng/biosoft/weblogo/weblogo</span><br><span class="line"><span class="keyword">source</span> ~/.bashrc</span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir Ghostscript &amp;&amp;  cd Ghostscript</span><br><span class="line"><span class="comment"># http://www.ghostscript.com/download/gsdnld.html</span></span><br><span class="line"><span class="comment"># http://www.ghostscript.com/doc/9.20/Readme.htm</span></span><br><span class="line">wget https://github.com/ArtifexSoftware/ghostpdl-downloads/releases/download/gs920/ghostscript-<span class="number">9.20</span>-linux-x86_64.tgz </span><br><span class="line">tar zxvf ghostscript-<span class="number">9.20</span>-linux-x86_64.tgz</span><br><span class="line">cp ghostscript-<span class="number">9.20</span>-linux-x86_64/gs-<span class="number">920</span>-linux_x86_64  ~/biosoft/myBin/bin/gs</span><br><span class="line"><span class="comment">## make sure the "gs" program is executable </span></span><br><span class="line"><span class="comment">## Download and install homer (Hypergeometric Optimization of Motif EnRichment)</span></span><br><span class="line"><span class="comment">## // http://homer.salk.edu/homer/</span></span><br><span class="line"><span class="comment">## // http://blog.qiubio.com:8080/archives/3024 </span></span><br><span class="line"><span class="comment">## The commands gs, seqlogo, blat, and samtools should now work from the command line</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir homer &amp;&amp;  cd homer</span><br><span class="line">wget http://homer.salk.edu/homer/configureHomer.pl </span><br><span class="line">perl configureHomer.pl -install</span><br><span class="line">perl configureHomer.pl -install hg19</span><br><span class="line"><span class="comment">## Download and install SWEMBL</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir SWEMBL &amp;&amp;  cd SWEMBL</span><br><span class="line"><span class="comment">#### readme: http://www.ebi.ac.uk/~swilder/SWEMBL/beginners.html</span></span><br><span class="line">wget http://www.ebi.ac.uk/~swilder/SWEMBL/SWEMBL.3.3.1.tar.bz2 </span><br><span class="line">tar xvfj SWEMBL.3.3.1.tar.bz2</span><br><span class="line">cd SWEMBL.3.3.1/</span><br><span class="line">make</span><br><span class="line"><span class="comment">## error </span></span><br><span class="line"><span class="comment">## Download and install SISSRs</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir SISSRs &amp;&amp;  cd SISSRs</span><br><span class="line"><span class="comment">#### readme: https://dir.nhlbi.nih.gov/papers/lmi/epigenomes/sissrs/SISSRs-Manual.pdf</span></span><br><span class="line">wget http://dir.nhlbi.nih.gov/papers/lmi/epigenomes/sissrs/sissrs_v1.4.tar.gz</span><br><span class="line">tar xzvf sissrs_v1.4.tar.gz</span><br><span class="line">~/biosoft/SISSRs/sissrs.pl</span><br><span class="line"><span class="comment">## Download and install SISSRs</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir QuEST &amp;&amp;  cd QuEST</span><br><span class="line"><span class="comment">#### http://mendel.stanford.edu/SidowLab/downloads/quest/</span></span><br><span class="line">wget http://mendel.stanford.edu/SidowLab/downloads/quest/QuEST_2.4.tar.gz</span><br><span class="line">tar xzvf QuEST_2.4.tar.gz</span><br><span class="line">cd QuEST_2.4</span><br></pre></td></tr></table></figure>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">## Download and install fastqc</span></span><br><span class="line"><span class="comment">##可视化展示二代测序数据质量</span></span><br><span class="line"><span class="comment">##http://www.bioinformatics.babraham.ac.uk/projects/fastqc/</span></span><br><span class="line"></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir fastqc &amp;&amp;  cd fastqc</span><br><span class="line">wget http://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.5.zip</span><br><span class="line">unzip fastqc_v0.11.5.zip</span><br></pre></td></tr></table></figure>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br><span class="line">127</span><br><span class="line">128</span><br><span class="line">129</span><br><span class="line">130</span><br><span class="line">131</span><br><span class="line">132</span><br><span class="line">133</span><br><span class="line">134</span><br><span class="line">135</span><br><span class="line">136</span><br><span class="line">137</span><br><span class="line">138</span><br><span class="line">139</span><br><span class="line">140</span><br><span class="line">141</span><br><span class="line">142</span><br><span class="line">143</span><br><span class="line">144</span><br><span class="line">145</span><br><span class="line">146</span><br><span class="line">147</span><br><span class="line">148</span><br><span class="line">149</span><br><span class="line">150</span><br><span class="line">151</span><br><span class="line">152</span><br><span class="line">153</span><br><span class="line">154</span><br><span class="line">155</span><br><span class="line">156</span><br><span class="line">157</span><br><span class="line">158</span><br><span class="line">159</span><br><span class="line">160</span><br><span class="line">161</span><br><span class="line">162</span><br><span class="line">163</span><br><span class="line">164</span><br><span class="line">165</span><br><span class="line">166</span><br><span class="line">167</span><br><span class="line">168</span><br><span class="line">169</span><br><span class="line">170</span><br><span class="line">171</span><br><span class="line">172</span><br><span class="line">173</span><br><span class="line">174</span><br><span class="line">175</span><br><span class="line">176</span><br><span class="line">177</span><br><span class="line">178</span><br><span class="line">179</span><br><span class="line">180</span><br><span class="line">181</span><br><span class="line">182</span><br><span class="line">183</span><br><span class="line">184</span><br><span class="line">185</span><br><span class="line">186</span><br><span class="line">187</span><br><span class="line">188</span><br><span class="line">189</span><br><span class="line">190</span><br><span class="line">191</span><br><span class="line">192</span><br><span class="line">193</span><br><span class="line">194</span><br><span class="line">195</span><br><span class="line">196</span><br><span class="line">197</span><br><span class="line">198</span><br><span class="line">199</span><br><span class="line">200</span><br><span class="line">201</span><br><span class="line">202</span><br><span class="line">203</span><br><span class="line">204</span><br><span class="line">205</span><br><span class="line">206</span><br><span class="line">207</span><br><span class="line">208</span><br><span class="line">209</span><br><span class="line">210</span><br><span class="line">211</span><br><span class="line">212</span><br><span class="line">213</span><br><span class="line">214</span><br><span class="line">215</span><br><span class="line">216</span><br><span class="line">217</span><br><span class="line">218</span><br><span class="line">219</span><br><span class="line">220</span><br><span class="line">221</span><br><span class="line">222</span><br><span class="line">223</span><br><span class="line">224</span><br><span class="line">225</span><br><span class="line">226</span><br><span class="line">227</span><br><span class="line">228</span><br><span class="line">229</span><br><span class="line">230</span><br><span class="line">231</span><br><span class="line">232</span><br><span class="line">233</span><br><span class="line">234</span><br><span class="line">235</span><br><span class="line">236</span><br><span class="line">237</span><br><span class="line">238</span><br><span class="line">239</span><br><span class="line">240</span><br><span class="line">241</span><br><span class="line">242</span><br><span class="line">243</span><br><span class="line">244</span><br><span class="line">245</span><br><span class="line">246</span><br><span class="line">247</span><br><span class="line">248</span><br><span class="line">249</span><br><span class="line">250</span><br><span class="line">251</span><br><span class="line">252</span><br><span class="line">253</span><br><span class="line">254</span><br><span class="line">255</span><br><span class="line">256</span><br><span class="line">257</span><br><span class="line">258</span><br><span class="line">259</span><br><span class="line">260</span><br><span class="line">261</span><br><span class="line">262</span><br><span class="line">263</span><br><span class="line">264</span><br><span class="line">265</span><br><span class="line">266</span><br><span class="line">267</span><br><span class="line">268</span><br><span class="line">269</span><br><span class="line">270</span><br><span class="line">271</span><br><span class="line">272</span><br><span class="line">273</span><br><span class="line">274</span><br><span class="line">275</span><br><span class="line">276</span><br><span class="line">277</span><br><span class="line">278</span><br><span class="line">279</span><br><span class="line">280</span><br><span class="line">281</span><br><span class="line">282</span><br><span class="line">283</span><br><span class="line">284</span><br><span class="line">285</span><br><span class="line">286</span><br><span class="line">287</span><br><span class="line">288</span><br><span class="line">289</span><br><span class="line">290</span><br><span class="line">291</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">## Download and install CEAS    </span></span><br><span class="line"><span class="comment">## http://liulab.dfci.harvard.edu/CEAS/download.html</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir CEAS  &amp;&amp;  cd CEAS </span><br><span class="line">wget  http://liulab.dfci.harvard.edu/CEAS/src/CEAS-Package-<span class="number">1.0</span><span class="number">.2</span>.tar.gz</span><br><span class="line">tar zxvf CEAS-Package-<span class="number">1.0</span><span class="number">.2</span>.tar.gz</span><br><span class="line">cd  CEAS-Package-<span class="number">1.0</span><span class="number">.2</span></span><br><span class="line">python setup.py install --user </span><br><span class="line"><span class="comment">## http://liulab.dfci.harvard.edu/CEAS/usermanual.html</span></span><br><span class="line"> ~/.local/bin/ceas --help  </span><br><span class="line">mkdir annotation  &amp;&amp;  cd annotation  </span><br><span class="line">wget http://liulab.dfci.harvard.edu/CEAS/src/hg19.refGene.gz ; gunzip hg19.refGene.gz </span><br><span class="line"><span class="comment"># http://hgdownload.cse.ucsc.edu/goldenPath/hg19/database/refGene.txt.gz ##  gunzip refGene.txt.gz ; mv refGene.txt  hg19refGene.txt</span></span><br><span class="line"><span class="comment">## Download and install CEAS    </span></span><br><span class="line"><span class="comment">## http://liulab.dfci.harvard.edu/CEAS/download.html</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir crossmap  &amp;&amp;  cd crossmap </span><br><span class="line">pip install CrossMap --user</span><br><span class="line"><span class="comment">## http://crossmap.sourceforge.net/#use-pip-to-install-crossmap</span></span><br><span class="line">mkdir chain_files  &amp;&amp;  cd chain_files  </span><br><span class="line">wget http://hgdownload.soe.ucsc.edu/goldenPath/mm10/liftOver/mm10ToMm9.over.chain.gz</span><br><span class="line">wget http://hgdownload.soe.ucsc.edu/goldenPath/mm9/liftOver/mm9ToMm10.over.chain.gz </span><br><span class="line">wget http://hgdownload.soe.ucsc.edu/goldenPath/hg38/liftOver/hg38ToHg19.over.chain.gz </span><br><span class="line">wget http://hgdownload.soe.ucsc.edu/goldenPath/hg38/liftOver/hg38ToHg19.over.chain.gz </span><br><span class="line"><span class="comment"># http://hgdownload.cse.ucsc.edu/goldenPath/hg19/database/refGene.txt.gz ##  gunzip refGene.txt.gz ; mv refGene.txt  hg19refGene.txt</span></span><br><span class="line"><span class="comment"># Usage: CrossMap.py bed ~/biosoft/crossmap/chain_files/mm9ToMm10.over.chain.gz  test.mm9.bed3</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line"><span class="comment"># http://www.broadinstitute.org/cancer/cga/rnaseqc_run</span></span><br><span class="line"><span class="comment"># http://www.broadinstitute.org/cancer/cga/rnaseqc_download</span></span><br><span class="line">mkdir RNA-SeQC  &amp;&amp;  cd RNA-SeQC </span><br><span class="line"><span class="comment">#### readme: http://www.broadinstitute.org/cancer/cga/sites/default/files/data/tools/rnaseqc/RNA-SeQC_Help_v1.1.2.pdf</span></span><br><span class="line">wget http://www.broadinstitute.org/cancer/cga/tools/rnaseqc/RNA-SeQC_v1.1.8.jar </span><br><span class="line"><span class="comment">#TopHat+Cufflinks+ pipeline</span></span><br><span class="line"><span class="comment">## Download and install TopHat </span></span><br><span class="line"><span class="comment"># https://ccb.jhu.edu/software/tophat/index.shtml</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir TopHat  &amp;&amp;  cd TopHat </span><br><span class="line"><span class="comment">#### readme: https://ccb.jhu.edu/software/tophat/manual.shtml</span></span><br><span class="line">wget https://ccb.jhu.edu/software/tophat/downloads/tophat-<span class="number">2.1</span><span class="number">.1</span>.Linux_x86_64.tar.gz</span><br><span class="line">tar xzvf tophat-<span class="number">2.1</span><span class="number">.1</span>.Linux_x86_64.tar.gz </span><br><span class="line">ln -s tophat-<span class="number">2.1</span><span class="number">.1</span>.Linux_x86_64 current </span><br><span class="line"><span class="comment"># ~/biosoft/TopHat/current/tophat2</span></span><br><span class="line"><span class="comment">## Download and install Cufflinks </span></span><br><span class="line"><span class="comment">#  http://cole-trapnell-lab.github.io/cufflinks/</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir Cufflinks  &amp;&amp;  cd Cufflinks </span><br><span class="line"><span class="comment">#### readme: http://cole-trapnell-lab.github.io/cufflinks/manual/</span></span><br><span class="line"><span class="comment">#### install:http://cole-trapnell-lab.github.io/cufflinks/install/</span></span><br><span class="line">wget http://cole-trapnell-lab.github.io/cufflinks/assets/downloads/cufflinks-<span class="number">2.2</span><span class="number">.1</span>.Linux_x86_64.tar.gz</span><br><span class="line">tar xzvf cufflinks-<span class="number">2.2</span><span class="number">.1</span>.Linux_x86_64.tar.gz </span><br><span class="line">ln -s cufflinks-<span class="number">2.2</span><span class="number">.1</span>.Linux_x86_64 current</span><br><span class="line">~/biosoft/Cufflinks/current/cufflinks</span><br><span class="line"></span><br><span class="line"><span class="comment">#HISAT-Stringtie2-Ballgown pipeline</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">## Download and install HISAT </span></span><br><span class="line"><span class="comment">##https://ccb.jhu.edu/software/hisat2/index.shtml</span></span><br><span class="line"><span class="comment">##功能： 将测序结果比对到参考基因组上</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir HISAT  &amp;&amp;  cd HISAT </span><br><span class="line"><span class="comment">#### readme: https://ccb.jhu.edu/software/hisat2/manual.shtml</span></span><br><span class="line">wget ftp://ftp.ccb.jhu.edu/pub/infphilo/hisat2/downloads/hisat2-<span class="number">2.0</span><span class="number">.4</span>-Linux_x86_64.zip</span><br><span class="line">unzip hisat2-<span class="number">2.0</span><span class="number">.4</span>-Linux_x86_64.zip</span><br><span class="line">ln -s hisat2-<span class="number">2.0</span><span class="number">.4</span>  current </span><br><span class="line"><span class="comment">## ~/biosoft/HISAT/current/hisat2-build</span></span><br><span class="line"><span class="comment">## ~/biosoft/HISAT/current/hisat2  </span></span><br><span class="line"><span class="comment">## Download and install StringTie</span></span><br><span class="line"><span class="comment">## https://ccb.jhu.edu/software/stringtie/  ## https://ccb.jhu.edu/software/stringtie/index.shtml?t=manual</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir StringTie &amp;&amp;  cd StringTie </span><br><span class="line">wget http://ccb.jhu.edu/software/stringtie/dl/stringtie-<span class="number">1.2</span><span class="number">.3</span>.Linux_x86_64.tar.gz </span><br><span class="line">tar zxvf  stringtie-<span class="number">1.2</span><span class="number">.3</span>.Linux_x86_64.tar.gz</span><br><span class="line">ln -s stringtie-<span class="number">1.2</span><span class="number">.3</span>.Linux_x86_64 current</span><br><span class="line"><span class="comment"># ~/biosoft/StringTie/current/stringtie</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir RSEM &amp;&amp;  cd RSEM </span><br><span class="line">wget https://codeload.github.com/deweylab/RSEM/tar.gz/v1.2.31</span><br><span class="line">mv v1.2.31  RSEM.v1.2.31.tar.gz </span><br><span class="line">tar zxvf RSEM.v1.2.31.tar.gz  </span><br><span class="line"><span class="comment">## Download and install HTSeq  </span></span><br><span class="line"><span class="comment">## http://www-huber.embl.de/users/anders/HTSeq/doc/overview.html</span></span><br><span class="line"><span class="comment">## https://pypi.python.org/pypi/HTSeq</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir HTSeq &amp;&amp;  cd HTSeq</span><br><span class="line">wget  https://pypi.python.org/packages/<span class="number">72</span>/0f/566afae6c149762af301a19686cd5fd1876deb2b48d09546dbd5caebbb78/HTSeq-<span class="number">0.6</span><span class="number">.1</span>.tar.gz </span><br><span class="line">tar zxvf HTSeq-<span class="number">0.6</span><span class="number">.1</span>.tar.gz</span><br><span class="line">cd HTSeq-<span class="number">0.6</span><span class="number">.1</span></span><br><span class="line">python setup.py install --user </span><br><span class="line">~/.local/bin/htseq-count  --help</span><br><span class="line"><span class="comment">## ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_mouse/release_M1/</span></span><br><span class="line"><span class="comment">## http://hgdownload-test.cse.ucsc.edu/goldenPath/mm10/liftOver/</span></span><br><span class="line"><span class="comment">## GRCm38/mm10 (Dec, 2011) </span></span><br><span class="line"><span class="comment">## ls *bam |while read id;do ( ~/.local/bin/htseq-count  -f bam  $id   genecode/mm9/gencode.vM1.annotation.gtf.gz  1&gt;$&#123;id%%.*&#125;.gene.counts ) ;done </span></span><br><span class="line"><span class="comment">## ls *bam |while read id;do ( ~/.local/bin/htseq-count  -f bam -i exon_id  $id   genecode/mm9/gencode.vM1.annotation.gtf.gz  1&gt;$&#123;id%%.*&#125;.exon.counts ) ;done</span></span><br><span class="line"><span class="comment">## Download and install kallisto</span></span><br><span class="line"><span class="comment">## https://pachterlab.github.io/kallisto/starting</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir kallisto &amp;&amp;  cd kallisto </span><br><span class="line">wget https://github.com/pachterlab/kallisto/releases/download/v0.43.0/kallisto_linux-v0.43.0.tar.gz</span><br><span class="line"><span class="comment">#tar zxvf  </span></span><br><span class="line"><span class="comment">## Download and install Sailfish</span></span><br><span class="line"><span class="comment">## http://www.cs.cmu.edu/~ckingsf/software/sailfish/  ## </span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir Sailfish &amp;&amp;  cd Sailfish </span><br><span class="line">wget   https://github.com/kingsfordgroup/sailfish/releases/download/v0.9.2/SailfishBeta-<span class="number">0.9</span>.2_DebianSqueeze.tar.gz </span><br><span class="line"><span class="comment">#tar zxvf  </span></span><br><span class="line"><span class="comment">## Download and install salmon</span></span><br><span class="line"><span class="comment">## http://salmon.readthedocs.io/en/latest/salmon.html ## </span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir salmon &amp;&amp;  cd salmon </span><br><span class="line"><span class="comment">## https://github.com/COMBINE-lab/salmon</span></span><br><span class="line"><span class="comment">#tar zxvf  </span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir GDC  &amp;&amp;  cd GDC  </span><br><span class="line"><span class="comment"># https://gdc.cancer.gov/access-data/gdc-data-transfer-tool</span></span><br><span class="line"><span class="comment"># http://gdc-docs.nci.nih.gov/Data_Transfer_Tool/Users_Guide/Getting_Started/</span></span><br><span class="line">wget https://gdc.nci.nih.gov/files/public/file/gdc-client_v1.2.0_Ubuntu14.04_x64.zip </span><br><span class="line">unzip gdc-client_v1.2.0_Ubuntu14.04_x64.zip</span><br><span class="line">cd ~/biosoft/myBin/bin</span><br><span class="line"><span class="comment">## http://hgdownload.cse.ucsc.edu/admin/exe/</span></span><br><span class="line">wget http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/bedToBigBed</span><br><span class="line">wget http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/bedSort</span><br><span class="line">wget http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/bedGraphToBigWig</span><br><span class="line">wget http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/fetchChromSizes</span><br><span class="line">wget http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/wigToBigWig</span><br><span class="line">wget http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/liftOver</span><br><span class="line">wget http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/bigWigToBedGraph</span><br><span class="line">wget http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/bigBedToBed</span><br><span class="line">wget http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/blat/blat</span><br><span class="line">wget http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/blat/gfClient</span><br><span class="line">wget http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/blat/gfServer</span><br><span class="line"><span class="comment">## Download and install variationtoolkit</span></span><br><span class="line"><span class="comment">## https://code.google.com/archive/p/variationtoolkit/downloads </span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir variationtoolkit &amp;&amp;  cd variationtoolkit  </span><br><span class="line">wget https://storage.googleapis.com/google-code-archive-downloads/v2/code.google.com/variationtoolkit/archive.tar.gz</span><br><span class="line">tar zxvf archive.tar.gz </span><br><span class="line">cd variationtoolkit</span><br><span class="line">make</span><br><span class="line"><span class="comment">## Download and install transvar</span></span><br><span class="line"><span class="comment"># http://bioinformatics.mdanderson.org/main/Transvar</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line"><span class="comment"># https://bitbucket.org/wanding/transvar</span></span><br><span class="line">mkdir transvar &amp;&amp;  cd transvar  </span><br><span class="line">wget https://bitbucket.org/wanding/transvar/get/v2.1.19.20160221.zip </span><br><span class="line">unzip v2.1.19.20160221.zip </span><br><span class="line">cd wanding-transvar-5dd8a7366999 </span><br><span class="line">python setup.py install --user </span><br><span class="line">cd ~/biosoft</span><br><span class="line"><span class="comment"># http://kobas.cbi.pku.edu.cn/download.php </span></span><br><span class="line">mkdir kobas &amp;&amp;  cd kobas  </span><br><span class="line"><span class="comment"># wget http://kobas.cbi.pku.edu.cn/download_file.php?type=seq_pep&amp;filename=hsa.pep.fasta.gz </span></span><br><span class="line"><span class="comment"># wget http://kobas.cbi.pku.edu.cn/download_file.php?type=sqlite3&amp;filename=hsa.db.gz </span></span><br><span class="line">wget http://kobas.cbi.pku.edu.cn/kobas-<span class="number">2.1</span><span class="number">.1</span>/kobas-<span class="number">2.1</span><span class="number">.1</span>.tar.gz </span><br><span class="line">tar zxvf kobas-<span class="number">2.1</span><span class="number">.1</span>.tar.gz </span><br><span class="line">cd kobas-<span class="number">2.1</span>.1_<span class="number">20160822</span></span><br><span class="line"><span class="comment"># * Download the KOBAS organism data package (organism.db.gz) from KOBAS Backend databases download website</span></span><br><span class="line"><span class="comment"># * Download the KOBAS specific species data package from KOBAS Backend databases download website (for example, hsa.db.gz)</span></span><br><span class="line"><span class="comment"># * Download the specific sequence file from KOBAS sequence files download website (for example, hsa.pep.fasta.gz)</span></span><br><span class="line"><span class="comment"># * `gunzip organism.db.gz`</span></span><br><span class="line"><span class="comment"># * `gunzip hsa.db.gz`</span></span><br><span class="line"><span class="comment"># * Move all databases into $&#123;kobas_home&#125;/sqlite3/ (for example, organism.db, hsa.db)</span></span><br><span class="line"><span class="comment"># * `gunzip hsa.pep.fasta.gz`</span></span><br><span class="line"><span class="comment"># * Move the fasta sequence file into $&#123;kobas_home&#125;/seq_pep/</span></span><br><span class="line"><span class="comment"># * `makeblastdb -in hsa.pep.fasta -dbtype prot`</span></span><br><span class="line">pip install RPy2 --user </span><br><span class="line">pip install Numpy --user </span><br><span class="line">pip install Pandas --user </span><br><span class="line">pip install BioPython --user </span><br><span class="line">pip install matplotlib --user </span><br><span class="line">pip install PySQLite --user </span><br><span class="line"><span class="keyword">source</span>(<span class="string">"http://bioconductor.org/biocLite.R"</span>)</span><br><span class="line">biocLite(<span class="string">"qvalue"</span>)</span><br><span class="line"><span class="comment">## Download and install bamtools</span></span><br><span class="line"><span class="comment">## https://github.com/pezmaster31/bamtools/wiki/Building-and-installing</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir bamtools &amp;&amp;  cd bamtools  </span><br><span class="line">git clone git://github.com/pezmaster31/bamtools.git </span><br><span class="line">cd bamtools</span><br><span class="line">cmake --version  <span class="comment">## BamTools requires CMake (version &gt;= 2.6.4).</span></span><br><span class="line">mkdir build &amp;&amp;  cd build </span><br><span class="line">cmake ../ </span><br><span class="line">make</span><br><span class="line">~/biosoft/bamtools/bamtools/bin/bamtools</span><br><span class="line"><span class="comment">## Download and install BAMStats</span></span><br><span class="line"><span class="comment">## http://bamstats.sourceforge.net/</span></span><br><span class="line"><span class="comment">## https://sourceforge.net/projects/bamstats/files/</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir BAMStats &amp;&amp;  cd BAMStats  </span><br><span class="line">wget https://nchc.dl.sourceforge.net/project/bamstats/BAMStats-<span class="number">1.25</span>.zip </span><br><span class="line">unzip BAMStats-<span class="number">1.25</span>.zip</span><br><span class="line"><span class="comment">#java -jar  ~/biosoft/BAMStats/BAMStats-1.25/BAMStats-1.25.jar  --help</span></span><br><span class="line"><span class="comment">## Download and install Qualimap </span></span><br><span class="line"><span class="comment">## http://qualimap.bioinfo.cipf.es/</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir Qualimap &amp;&amp;  cd Qualimap  </span><br><span class="line">wget https://bitbucket.org/kokonech/qualimap/downloads/qualimap_v2.2.1.zip </span><br><span class="line"><span class="comment">## readme  http://qualimap.bioinfo.cipf.es/doc_html/index.html</span></span><br><span class="line"><span class="comment">## example results :http://kokonech.github.io/qualimap/HG00096.chr20_bamqc/qualimapReport.html </span></span><br><span class="line">unzip qualimap_v2.2.1.zip </span><br><span class="line">~/biosoft/bamtools/bamtools/bin/bamtools</span><br><span class="line">~/biosoft/Qualimap/qualimap_v2.2.1/qualimap --help <span class="comment">## --java-mem-size=4G</span></span><br><span class="line"><span class="comment">## modify ~/.bashrc by adding PATH=$PATH:~/.local/bin/</span></span><br><span class="line"><span class="comment">## Download and install deepTools</span></span><br><span class="line"><span class="comment">## https://github.com/fidelram/deepTools</span></span><br><span class="line"><span class="comment">## http://deeptools.readthedocs.io/en/latest/content/example_usage.html</span></span><br><span class="line">pip install pyBigWig --user </span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir deepTools &amp;&amp;  cd deepTools  </span><br><span class="line">git clone https://github.com/fidelram/deepTools <span class="comment">## 130M,</span></span><br><span class="line">cd deepTools</span><br><span class="line">python setup.py install --user</span><br><span class="line"><span class="comment">## 17 tools in ~/.local/bin/</span></span><br><span class="line">~/.local/bin/deeptools</span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir ngsplot &amp;&amp;  cd ngsplot  </span><br><span class="line"><span class="comment">## download by yourself :https://drive.google.com/drive/folders/0B1PVLadG_dCKN1liNFY0MVM1Ulk  </span></span><br><span class="line">tar -zxvf ngsplot-<span class="number">2.61</span>.tar.gz</span><br><span class="line">tar zxvf ngsplot.eg.bam.tar.gz</span><br><span class="line">echo <span class="string">'export PATH=/home/jianmingzeng/biosoft/ngsplot/ngsplot/bin:$PATH'</span> &gt;&gt;~/.bashrc  </span><br><span class="line">echo <span class="string">'export NGSPLOT=/home/jianmingzeng/biosoft/ngsplot/ngsplot'</span> &gt;&gt;~/.bashrc </span><br><span class="line"><span class="keyword">source</span> ~/.bashrc</span><br><span class="line">install.packages(<span class="string">"doMC"</span>, dep=<span class="literal">T</span>)</span><br><span class="line">install.packages(<span class="string">"caTools"</span>, dep=<span class="literal">T</span>)</span><br><span class="line">install.packages(<span class="string">"utils"</span>, dep=<span class="literal">T</span>)</span><br><span class="line"><span class="keyword">source</span>(<span class="string">"http://bioconductor.org/biocLite.R"</span>)</span><br><span class="line">biocLite( <span class="string">"BSgenome"</span> )</span><br><span class="line">biocLite( <span class="string">"Rsamtools"</span> )</span><br><span class="line">biocLite( <span class="string">"ShortRead"</span> )</span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir breakdancer &amp;&amp;  cd breakdancer  </span><br><span class="line"><span class="comment"># http://breakdancer.sourceforge.net/</span></span><br><span class="line"><span class="comment"># you need to install 2 perl module by yourself : http://breakdancer.sourceforge.net/moreperl.html</span></span><br><span class="line">wget https://sourceforge.net/projects/breakdancer/files/breakdancer-<span class="number">1.1</span>.2_2013_03_<span class="number">08.</span>zip </span><br><span class="line">unzip breakdancer-<span class="number">1.1</span>.2_2013_03_<span class="number">08.</span>zip </span><br><span class="line">cd breakdancer-<span class="number">1.1</span><span class="number">.2</span>/cpp</span><br><span class="line">make  <span class="comment">##something wrong !</span></span><br><span class="line"><span class="comment">## usage: http://breakdancer.sourceforge.net/pipeline.html</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line"><span class="comment"># http://boevalab.com/FREEC/</span></span><br><span class="line">mkdir Control-FREEC &amp;&amp; cd Control-FREEC</span><br><span class="line"><span class="comment"># https://github.com/BoevaLab/FREEC/releases</span></span><br><span class="line">wget https://github.com/BoevaLab/FREEC/archive/v10.3.zip </span><br><span class="line">unzip v10.3.zip </span><br><span class="line"><span class="comment"># https://www.ncbi.nlm.nih.gov/pubmed/22155870</span></span><br><span class="line"><span class="comment"># http://boevalab.com/FREEC/tutorial.html</span></span><br><span class="line"><span class="comment"># http://samtools.sourceforge.net/pileup.shtml</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line"><span class="comment"># https://github.com/dellytools/delly</span></span><br><span class="line">mkdir delly &amp;&amp; cd delly </span><br><span class="line"><span class="comment"># git clone --recursive https://github.com/dellytools/delly.git</span></span><br><span class="line"><span class="comment"># cd delly </span></span><br><span class="line"><span class="comment"># make all</span></span><br><span class="line"><span class="comment"># make PARALLEL=1 -B src/delly</span></span><br><span class="line">wget https://github.com/dellytools/delly/releases/download/v0.7.6/delly_v0.7.6_linux_x86_64bit </span><br><span class="line">chmod <span class="number">777</span> delly_v0.7.6_linux_x86_64bit </span><br><span class="line">~/biosoft/delly/delly_v0.7.6_linux_x86_64bit  --help </span><br><span class="line"><span class="comment">## delly call -t DEL -g hg19.fa -o s1.bcf -x hg19.excl sample1.bam</span></span><br><span class="line"><span class="comment">## ./delly/src/bcftools/bcftools view delly.bcf &gt; delly.vcf</span></span><br><span class="line"><span class="comment">## The SV type can be DEL, DUP, INV, TRA, or INS for deletions, tandem duplications, inversions, translocations and small insertions, respectively.</span></span><br><span class="line"><span class="comment">## In addition, you can filter input reads more stringently using -q 20 and -s 15.</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line"><span class="comment"># https://www.cog-genomics.org/plink2/data#merge_list</span></span><br><span class="line">mkdir PLINK &amp;&amp; cd PLINK </span><br><span class="line">wget https://www.cog-genomics.org/static/bin/plink170113/plink_linux_x86_64.zip </span><br><span class="line">unzip plink_linux_x86_64.zip</span><br><span class="line">~/biosoft/PLINK/plink</span><br><span class="line"><span class="comment">## Download and install Scalpel</span></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir Scalpel &amp;&amp;  cd Scalpel</span><br><span class="line">wget https://downloads.sourceforge.net/project/scalpel/scalpel-<span class="number">0.5</span><span class="number">.3</span>.tar.gz  </span><br><span class="line">tar zxvf scalpel-<span class="number">0.5</span><span class="number">.3</span>.tar.gz</span><br><span class="line">cd scalpel-<span class="number">0.5</span><span class="number">.3</span></span><br><span class="line">make</span><br><span class="line">~/biosoft/Scalpel/scalpel-<span class="number">0.5</span><span class="number">.3</span>/scalpel-discovery  --help</span><br><span class="line">~/biosoft/Scalpel/scalpel-<span class="number">0.5</span><span class="number">.3</span>/scalpel-export  --help</span><br><span class="line">cd ~/biosoft</span><br><span class="line"><span class="comment"># https://www.cog-genomics.org/plink2/data#merge_list</span></span><br><span class="line">mkdir firehose &amp;&amp; cd firehose </span><br><span class="line">wget http://gdac.broadinstitute.org/runs/code/firehose_get_latest.zip</span><br><span class="line">unzip firehose_get_latest.zip </span><br><span class="line">~/biosoft/firehose/firehose_get</span><br><span class="line">~/biosoft/firehose/firehose_get -tasks clinical analyses latest brca </span><br><span class="line">cd ~/biosoft</span><br><span class="line"><span class="comment"># https://www.cog-genomics.org/plink2/data#merge_list</span></span><br><span class="line">mkdir fastpop &amp;&amp; cd fastpop </span><br><span class="line">wget https://sourceforge.net/projects/fastpop/files/FastPop.tar.gz </span><br><span class="line">wget https://jaist.dl.sourceforge.net/project/fastpop/FastPop_Instruction.pdf</span><br><span class="line">tar zxvf FastPop.tar.gz  </span><br><span class="line">pip install cnvkit --user</span><br></pre></td></tr></table></figure>
<h1 id="转录组-de-novo分析流程的软件全代码"><a href="#转录组-de-novo分析流程的软件全代码" class="headerlink" title="转录组 de novo分析流程的软件全代码"></a>转录组 de novo分析流程的软件全代码</h1><p>Trinotate/Trinity/TransDecoder/sqlite/NCBI BLAST+/HMMER/PFAM signalP v4 /tmhmm v2 /RNAMMER 有些软件需要教育邮箱注册才行哦~~</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br></pre></td><td class="code"><pre><span class="line">## Trinotate/Trinity/TransDecoder/sqlite/NCBI BLAST+/HMMER/PFAM </span><br><span class="line">## signalP v4 /tmhmm v2 /RNAMMER</span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir hmmer &amp;&amp;  cd hmmer</span><br><span class="line">wget http://eddylab.org/software/hmmer/2.3/hmmer-2.3.tar.gz </span><br><span class="line">tar zxvf hmmer-2.3.tar.gz</span><br><span class="line">cd hmmer-2.3</span><br><span class="line">./configure --prefix=/home/jmzeng/my-bin</span><br><span class="line">#./configure --prefix=/home/jianmingzeng/biosoft/myBin</span><br><span class="line">make</span><br><span class="line">make install</span><br><span class="line">#for file in hmmalign hmmbuild hmmcalibrate hmmconvert hmmemit hmmfetch hmmindex hmmpfam hmmsearch ; do\</span><br><span class="line">#      cp src/$file /home/jmzeng/my-bin/bin/;\</span><br><span class="line">#   done</span><br><span class="line">#for file in hmmer hmmalign hmmbuild hmmcalibrate hmmconvert hmmemit hmmfetch hmmindex hmmpfam hmmsearch; do\</span><br><span class="line">#      cp documentation/man/$file.man /home/jmzeng/my-bin/man/man1/$file.1;\</span><br><span class="line">#   done</span><br><span class="line">cp /home/jmzeng/my-bin/bin/hmmsearch /home/jmzeng/my-bin/bin/hmmsearch2</span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir CBS &amp;&amp;  cd CBS</span><br><span class="line">#   signalP v4 (free academic download) http://www.cbs.dtu.dk/cgi-bin/nph-sw_request?signalp</span><br><span class="line">#   tmhmm v2 (free academic download) http://www.cbs.dtu.dk/cgi-bin/nph-sw_request?tmhmm</span><br><span class="line">#   RNAMMER (free academic download) http://www.cbs.dtu.dk/cgi-bin/sw_request?rnammer</span><br><span class="line">mkdir signalp-4.1</span><br><span class="line">mkdir rnammer-1.2</span><br><span class="line">## be sure to untar it in a new directory</span><br><span class="line">## it&apos;s a perl script, we need to modify it according to readme http://trinotate.github.io/#SoftwareRequired</span><br><span class="line">## vi ~/biosoft/CBS/signalp-4.1/signalp</span><br><span class="line">tar zxvf signalp-4.1e.Linux.tar.gz </span><br><span class="line">tar zxvf rnammer-1.2.src.tar.Z </span><br><span class="line">tar zxvf tmhmm-2.0c.Linux.tar.gz </span><br><span class="line">## it&apos;s a perl script, we need to modify it according to readme http://trinotate.github.io/#SoftwareRequired</span><br><span class="line">## vi ~/biosoft/CBS/tmhmm-2.0c/bin/tmhmm </span><br><span class="line">## vi ~/biosoft/CBS/tmhmm-2.0c/bin/tmhmmformat.pl</span><br><span class="line">which perl  ## /usr/bin/perl</span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir blastPlus &amp;&amp;  cd blastPlus</span><br><span class="line">#   ftp://ftp.ncbi.nlm.nih.gov/blast/executables/LATEST</span><br><span class="line">wget ftp://ftp.ncbi.nlm.nih.gov/blast/executables/LATEST/ncbi-blast-2.5.0+-x64-linux.tar.gz</span><br><span class="line">blastBinFolder=~/biosoft/blastPlus/ncbi-blast-2.5.0+/bin</span><br><span class="line">$blastBinFolder/makeblastdb -help</span><br><span class="line">#   http://www.cbs.dtu.dk/services/doc/signalp-4.1.readme</span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir TransDecoder &amp;&amp;  cd TransDecoder</span><br><span class="line">#   https://transdecoder.github.io/</span><br><span class="line"># https://github.com/TransDecoder/TransDecoder/releases</span><br><span class="line">wget https://github.com/TransDecoder/TransDecoder/archive/v3.0.0.tar.gz  -O TransDecoder.v3.0.0.tar.gz </span><br><span class="line">tar zxvf TransDecoder.v3.0.0.tar.gz </span><br><span class="line">cd TransDecoder-3.0.0 </span><br><span class="line">make</span><br><span class="line">~/biosoft/TransDecoder/TransDecoder-3.0.0/TransDecoder.LongOrfs -h</span><br><span class="line">~/biosoft/TransDecoder/TransDecoder-3.0.0/TransDecoder.Predict -h</span><br><span class="line">## sqlite3  --help</span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir Trinotate &amp;&amp;  cd Trinotate</span><br><span class="line">#   http://trinotate.github.io/</span><br><span class="line">#   https://github.com/Trinotate/Trinotate/releases</span><br><span class="line">wget https://github.com/Trinotate/Trinotate/archive/v3.0.1.tar.gz  -O Trinotate.v3.0.1.tar.gz </span><br><span class="line">tar zxvf Trinotate.v3.0.1.tar.gz</span><br><span class="line">~/biosoft/Trinotate/Trinotate-3.0.1/Trinotate -h</span><br><span class="line">wget https://data.broadinstitute.org/Trinity/Trinotate_v3_RESOURCES/Pfam-A.hmm.gz</span><br><span class="line">wget https://data.broadinstitute.org/Trinity/Trinotate_v3_RESOURCES/uniprot_sprot.pep.gz</span><br><span class="line">wget https://data.broadinstitute.org/Trinity/Trinotate_v3_RESOURCES/Trinotate_v3.sqlite.gz  -O Trinotate.sqlite.gz</span><br><span class="line">gunzip Trinotate.sqlite.gz</span><br><span class="line">gunzip uniprot_sprot.pep.gz</span><br><span class="line">makeblastdb -in uniprot_sprot.pep -dbtype prot</span><br><span class="line">gunzip Pfam-A.hmm.gz</span><br><span class="line">hmmpress Pfam-A.hmm</span><br></pre></td></tr></table></figure>
<h1 id="基因组，gtf，bed，注释"><a href="#基因组，gtf，bed，注释" class="headerlink" title="基因组，gtf，bed，注释"></a>基因组，gtf，bed，注释</h1><p>前面既然把我多年累积的软件安装代码共享了，就顺便把我最近做直播我的基因组的一些数据下载代码共享吧</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br><span class="line">127</span><br><span class="line">128</span><br><span class="line">129</span><br><span class="line">130</span><br><span class="line">131</span><br><span class="line">132</span><br><span class="line">133</span><br><span class="line">134</span><br><span class="line">135</span><br><span class="line">136</span><br><span class="line">137</span><br><span class="line">138</span><br><span class="line">139</span><br><span class="line">140</span><br><span class="line">141</span><br><span class="line">142</span><br><span class="line">143</span><br><span class="line">144</span><br><span class="line">145</span><br><span class="line">146</span><br><span class="line">147</span><br><span class="line">148</span><br><span class="line">149</span><br><span class="line">150</span><br><span class="line">151</span><br><span class="line">152</span><br><span class="line">153</span><br><span class="line">154</span><br><span class="line">155</span><br><span class="line">156</span><br><span class="line">157</span><br><span class="line">158</span><br><span class="line">159</span><br><span class="line">160</span><br><span class="line">161</span><br><span class="line">162</span><br><span class="line">163</span><br><span class="line">164</span><br><span class="line">165</span><br><span class="line">166</span><br><span class="line">167</span><br><span class="line">168</span><br><span class="line">169</span><br><span class="line">170</span><br><span class="line">171</span><br><span class="line">172</span><br><span class="line">173</span><br><span class="line">174</span><br><span class="line">175</span><br><span class="line">176</span><br><span class="line">177</span><br><span class="line">178</span><br><span class="line">179</span><br><span class="line">180</span><br><span class="line">181</span><br><span class="line">182</span><br><span class="line">183</span><br><span class="line">184</span><br><span class="line">185</span><br><span class="line">186</span><br><span class="line">187</span><br><span class="line">188</span><br><span class="line">189</span><br><span class="line">190</span><br><span class="line">191</span><br><span class="line">192</span><br><span class="line">193</span><br><span class="line">194</span><br><span class="line">195</span><br><span class="line">196</span><br><span class="line">197</span><br><span class="line">198</span><br></pre></td><td class="code"><pre><span class="line">cd ~/reference</span><br><span class="line">mkdir -p genome/hg19  &amp;&amp; cd genome/hg19 </span><br><span class="line">nohup wget http://hgdownload.cse.ucsc.edu/goldenPath/hg19/bigZips/chromFa.tar.gz &amp;</span><br><span class="line">tar zvfx chromFa.tar.gz</span><br><span class="line">cat *.fa &gt; hg19.fa</span><br><span class="line">rm chr*.fa</span><br><span class="line"> </span><br><span class="line"> </span><br><span class="line">cd ~/reference</span><br><span class="line">mkdir -p genome/hg38  &amp;&amp; cd genome/hg38 </span><br><span class="line">nohup wget http://hgdownload.cse.ucsc.edu/goldenPath/hg38/bigZips/hg38.fa.gz  &amp;</span><br><span class="line"> </span><br><span class="line">cd ~/reference</span><br><span class="line">mkdir -p  genome/mm10  &amp;&amp; cd genome/mm10 </span><br><span class="line">nohup wget http://hgdownload.cse.ucsc.edu/goldenPath/mm10/bigZips/chromFa.tar.gz  &amp;</span><br><span class="line">tar zvfx chromFa.tar.gz</span><br><span class="line">cat *.fa &gt; mm10.fa</span><br><span class="line">rm chr*.fa</span><br><span class="line"> </span><br><span class="line"> </span><br><span class="line">cd ~/biosoft/RNA-SeQC</span><br><span class="line">wget http://www.broadinstitute.org/cancer/cga/sites/default/files/data/tools/rnaseqc/ThousandReads.bam</span><br><span class="line">wget http://www.broadinstitute.org/cancer/cga/sites/default/files/data/tools/rnaseqc/gencode.v7.annotation_goodContig.gtf.gz</span><br><span class="line">wget http://www.broadinstitute.org/cancer/cga/sites/default/files/data/tools/rnaseqc/Homo_sapiens_assembly19.fasta.gz</span><br><span class="line">wget http://www.broadinstitute.org/cancer/cga/sites/default/files/data/tools/rnaseqc/Homo_sapiens_assembly19.other.tar.gz</span><br><span class="line">wget http://www.broadinstitute.org/cancer/cga/sites/default/files/data/tools/rnaseqc/gencode.v7.gc.txt</span><br><span class="line">wget http://www.broadinstitute.org/cancer/cga/sites/default/files/data/tools/rnaseqc/rRNA.tar.gz</span><br><span class="line"> </span><br><span class="line">cd ~/reference</span><br><span class="line">mkdir -p index/bowtie &amp;&amp; cd index/bowtie </span><br><span class="line">nohup time ~/biosoft/bowtie/bowtie2-2.2.9/bowtie2-build  ~/reference/genome/hg19/hg19.fa  ~/reference/index/bowtie/hg19 1&gt;hg19.bowtie_index.log 2&gt;&amp;1 &amp;</span><br><span class="line">nohup time ~/biosoft/bowtie/bowtie2-2.2.9/bowtie2-build  ~/reference/genome/hg38/hg38.fa  ~/reference/index/bowtie/hg38 1&gt;hg38.bowtie_index.log 2&gt;&amp;1 &amp;</span><br><span class="line">nohup time ~/biosoft/bowtie/bowtie2-2.2.9/bowtie2-build  ~/reference/genome/mm10/mm10.fa  ~/reference/index/bowtie/mm10 1&gt;mm10.bowtie_index.log 2&gt;&amp;1 &amp;</span><br><span class="line">  </span><br><span class="line">cd ~/reference</span><br><span class="line">mkdir -p index/bwa &amp;&amp; cd index/bwa </span><br><span class="line">nohup time ~/biosoft/bwa/bwa-0.7.15/bwa index   -a bwtsw   -p ~/reference/index/bwa/hg19  ~/reference/genome/hg19/hg19.fa 1&gt;hg19.bwa_index.log 2&gt;&amp;1   &amp;</span><br><span class="line">nohup time ~/biosoft/bwa/bwa-0.7.15/bwa index   -a bwtsw   -p ~/reference/index/bwa/hg38  ~/reference/genome/hg38/hg38.fa 1&gt;hg38.bwa_index.log 2&gt;&amp;1   &amp;</span><br><span class="line">nohup time ~/biosoft/bwa/bwa-0.7.15/bwa index   -a bwtsw   -p ~/reference/index/bwa/mm10  ~/reference/genome/mm10/mm10.fa 1&gt;mm10.bwa_index.log 2&gt;&amp;1   &amp;</span><br><span class="line">  </span><br><span class="line">cd ~/reference</span><br><span class="line">mkdir -p index/hisat &amp;&amp; cd index/hisat </span><br><span class="line">nohup wget ftp://ftp.ccb.jhu.edu/pub/infphilo/hisat2/data/hg19.tar.gz  &amp;</span><br><span class="line">nohup wget ftp://ftp.ccb.jhu.edu/pub/infphilo/hisat2/data/hg38.tar.gz  &amp;</span><br><span class="line">nohup wget ftp://ftp.ccb.jhu.edu/pub/infphilo/hisat2/data/grcm38.tar.gz &amp;</span><br><span class="line">nohup wget ftp://ftp.ccb.jhu.edu/pub/infphilo/hisat2/data/mm10.tar.gz  &amp;</span><br><span class="line">tar zxvf hg19.tar.gz</span><br><span class="line">tar zxvf grcm38.tar.gz</span><br><span class="line">tar zxvf hg38.tar.gz</span><br><span class="line">tar zxvf mm10.tar.gz </span><br><span class="line">  </span><br><span class="line">  </span><br><span class="line">mkdir -p ~/annotation/variation/human/ExAC</span><br><span class="line">cd ~/annotation/variation/human/ExAC</span><br><span class="line">## http://exac.broadinstitute.org/</span><br><span class="line">## ftp://ftp.broadinstitute.org/pub/ExAC_release/current</span><br><span class="line">wget ftp://ftp.broadinstitute.org/pub/ExAC_release/current/ExAC.r0.3.1.sites.vep.vcf.gz.tbi </span><br><span class="line">nohup wget ftp://ftp.broadinstitute.org/pub/ExAC_release/current/ExAC.r0.3.1.sites.vep.vcf.gz &amp;</span><br><span class="line">wget ftp://ftp.broadinstitute.org/pub/ExAC_release/current/cnv/exac-final-cnv.gene.scores071316 </span><br><span class="line">wget ftp://ftp.broadinstitute.org/pub/ExAC_release/current/cnv/exac-final.autosome-1pct-sq60-qc-prot-coding.cnv.bed</span><br><span class="line"> </span><br><span class="line"> </span><br><span class="line">mkdir -p ~/annotation/variation/human/dbSNP</span><br><span class="line">cd ~/annotation/variation/human/dbSNP</span><br><span class="line">## https://www.ncbi.nlm.nih.gov/projects/SNP/</span><br><span class="line">## ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b147_GRCh38p2/</span><br><span class="line">## ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b147_GRCh37p13/</span><br><span class="line">nohup wget ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b147_GRCh37p13/VCF/All_20160601.vcf.gz &amp;</span><br><span class="line">wget ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b147_GRCh37p13/VCF/All_20160601.vcf.gz.tbi </span><br><span class="line"> </span><br><span class="line"> </span><br><span class="line">mkdir -p ~/annotation/variation/human/1000genomes</span><br><span class="line">cd ~/annotation/variation/human/1000genomes </span><br><span class="line">## ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/ </span><br><span class="line">nohup wget  -c -r -nd -np -k -L -p  ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502 &amp;</span><br><span class="line"> </span><br><span class="line">mkdir -p ~/annotation/variation/human/cosmic</span><br><span class="line">cd ~/annotation/variation/human/cosmic</span><br><span class="line">## we need to register before we can download this file. </span><br><span class="line"> </span><br><span class="line">mkdir -p ~/annotation/variation/human/ESP6500</span><br><span class="line">cd ~/annotation/variation/human/ESP6500</span><br><span class="line"># http://evs.gs.washington.edu/EVS/</span><br><span class="line">nohup wget http://evs.gs.washington.edu/evs_bulk_data/ESP6500SI-V2-SSA137.GRCh38-liftover.snps_indels.vcf.tar.gz &amp; </span><br><span class="line"> </span><br><span class="line">mkdir -p ~/annotation/variation/human/UK10K</span><br><span class="line">cd ~/annotation/variation/human/UK10K</span><br><span class="line"># http://www.uk10k.org/</span><br><span class="line">nohup wget ftp://ngs.sanger.ac.uk/production/uk10k/UK10K_COHORT/REL-2012-06-02/UK10K_COHORT.20160215.sites.vcf.gz &amp; </span><br><span class="line"> </span><br><span class="line">mkdir -p ~/annotation/variation/human/gonl</span><br><span class="line">cd ~/annotation/variation/human/gonl</span><br><span class="line">## http://www.nlgenome.nl/search/</span><br><span class="line">## https://molgenis26.target.rug.nl/downloads/gonl_public/variants/release5/</span><br><span class="line">nohup wget  -c -r -nd -np -k -L -p  https://molgenis26.target.rug.nl/downloads/gonl_public/variants/release5  &amp;</span><br><span class="line"> </span><br><span class="line">mkdir -p ~/annotation/variation/human/omin</span><br><span class="line">cd ~/annotation/variation/human/omin</span><br><span class="line"> </span><br><span class="line">mkdir -p ~/annotation/variation/human/GWAS</span><br><span class="line">cd ~/annotation/variation/human/GWAS</span><br><span class="line"> </span><br><span class="line">mkdir -p ~/annotation/variation/human/hapmap</span><br><span class="line">cd ~/annotation/variation/human/hapmap</span><br><span class="line"># ftp://ftp.ncbi.nlm.nih.gov/hapmap/</span><br><span class="line">wget ftp://ftp.ncbi.nlm.nih.gov/hapmap/phase_3/relationships_w_pops_051208.txt </span><br><span class="line">nohup wget -c -r -np -k -L -p  -nd -A.gz ftp://ftp.ncbi.nlm.nih.gov/hapmap/phase_3/hapmap3_reformatted &amp;</span><br><span class="line"># ftp://ftp.hgsc.bcm.tmc.edu/pub/data/HapMap3-ENCODE/ENCODE3/ENCODE3v1/</span><br><span class="line">wget ftp://ftp.hgsc.bcm.tmc.edu/pub/data/HapMap3-ENCODE/ENCODE3/ENCODE3v1/bcm-encode3-QC.txt </span><br><span class="line">wget ftp://ftp.hgsc.bcm.tmc.edu/pub/data/HapMap3-ENCODE/ENCODE3/ENCODE3v1/bcm-encode3-submission.txt.gz</span><br><span class="line"> </span><br><span class="line"> </span><br><span class="line"> </span><br><span class="line"> </span><br><span class="line">## 1 million single nucleotide polymorphisms (SNPs) for DNA samples from each of the three ethnic groups in Singapore – Chinese, Malays and Indians.</span><br><span class="line">## The Affymetrix Genome-Wide Human SNP Array 6.0   &amp;&amp; The Illumina Human1M single BeadChip </span><br><span class="line">## http://www.statgen.nus.edu.sg/~SGVP/</span><br><span class="line">## http://www.statgen.nus.edu.sg/~SGVP/singhap/files-website/samples-information.txt</span><br><span class="line"># http://www.statgen.nus.edu.sg/~SGVP/singhap/files-website/genotypes/2009-01-30/QC/</span><br><span class="line"> </span><br><span class="line">## Singapore Sequencing Malay Project (SSMP) </span><br><span class="line">mkdir -p ~/annotation/variation/human/SSMP</span><br><span class="line">cd ~/annotation/variation/human/SSMP</span><br><span class="line">## http://www.statgen.nus.edu.sg/~SSMP/</span><br><span class="line">## http://www.statgen.nus.edu.sg/~SSMP/download/vcf/2012_05 </span><br><span class="line"> </span><br><span class="line"> </span><br><span class="line">## Singapore Sequencing Indian Project (SSIP) </span><br><span class="line">mkdir -p ~/annotation/variation/human/SSIP</span><br><span class="line">cd ~/annotation/variation/human/SSIP</span><br><span class="line"># http://www.statgen.nus.edu.sg/~SSIP/</span><br><span class="line">## http://www.statgen.nus.edu.sg/~SSIP/download/vcf/dataFreeze_Feb2013</span><br><span class="line"> </span><br><span class="line"> </span><br><span class="line"> </span><br><span class="line">wget ftp://ftp.ensembl.org/pub/release-75/gtf/homo_sapiens/Homo_sapiens.GRCh37.75.gtf.gz </span><br><span class="line">wget ftp://ftp.ensembl.org/pub/release-86/gtf/homo_sapiens/Homo_sapiens.GRCh38.86.chr.gtf.gz </span><br><span class="line"> </span><br><span class="line">mkdir -p ~/reference/gtf/gencode</span><br><span class="line">cd  ~/reference/gtf/gencode</span><br><span class="line">## https://www.gencodegenes.org/releases/current.html</span><br><span class="line">wget ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_25/gencode.v25.2wayconspseudos.gtf.gz</span><br><span class="line">wget ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_25/gencode.v25.long_noncoding_RNAs.gtf.gz </span><br><span class="line">wget ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_25/gencode.v25.polyAs.gtf.gz </span><br><span class="line">wget ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_25/gencode.v25.annotation.gtf.gz </span><br><span class="line">## https://www.gencodegenes.org/releases/25lift37.html </span><br><span class="line">wget ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_25/GRCh37_mapping/gencode.v25lift37.annotation.gtf.gz </span><br><span class="line">wget ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_25/GRCh37_mapping/gencode.v25lift37.metadata.HGNC.gz </span><br><span class="line">wget ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_25/GRCh37_mapping/gencode.v25lift37.metadata.EntrezGene.gz </span><br><span class="line">wget ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_25/GRCh37_mapping/gencode.v25lift37.metadata.RefSeq.gz </span><br><span class="line"> </span><br><span class="line"> </span><br><span class="line">mkdir -p ~/reference/gtf/ensembl/homo_sapiens_86</span><br><span class="line">cd  ~/reference/gtf/ensembl/homo_sapiens_86</span><br><span class="line">## http://asia.ensembl.org/info/data/ftp/index.html</span><br><span class="line"> </span><br><span class="line"> </span><br><span class="line"> </span><br><span class="line">cd ~/reference</span><br><span class="line">mkdir -p  genome/human_g1k_v37  &amp;&amp; cd genome/human_g1k_v37</span><br><span class="line"># http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/ </span><br><span class="line">nohup wget http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/human_g1k_v37.fasta.gz  &amp;</span><br><span class="line">gunzip human_g1k_v37.fasta.gz</span><br><span class="line">wget http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/human_g1k_v37.fasta.fai</span><br><span class="line">wget http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/README.human_g1k_v37.fasta.txt</span><br><span class="line">java -jar ~/biosoft/picardtools/picard-tools-1.119/CreateSequenceDictionary.jar R=human_g1k_v37.fasta O=human_g1k_v37.dict</span><br><span class="line"> </span><br><span class="line">## ftp://ftp.broadinstitute.org/bundle/b37/</span><br><span class="line">mkdir -p ~/annotation/GATK</span><br><span class="line">cd ~/annotation/variation/GATK</span><br><span class="line">wget ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/b37/1000G_phase1.snps.high_confidence.b37.vcf.gz </span><br><span class="line">wget ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/b37/dbsnp_138.b37.vcf.gz</span><br><span class="line">wget ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/b37/human_g1k_v37.fasta.gz </span><br><span class="line">wget ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/b37/NA12878.HiSeq.WGS.bwa.cleaned.raw.subset.b37.sites.vcf.gz</span><br><span class="line">wget ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/b37/Mills_and_1000G_gold_standard.indels.b37.vcf.gz </span><br><span class="line">wget ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/b37/hapmap_3.3.b37.vcf.gz</span><br><span class="line">wget ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/b37/1000G_phase1.indels.b37.vcf.gz </span><br><span class="line">wget ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/b37/1000G_phase1.indels.b37.vcf.idx.gz</span><br><span class="line">gunzip 1000G_phase1.indels.b37.vcf.idx.gz</span><br><span class="line">gunzip 1000G_phase1.indels.b37.vcf.gz</span><br><span class="line">  </span><br><span class="line">  </span><br><span class="line">mkdir -p  ~/institute/ENSEMBL/gtf</span><br><span class="line">cd  ~/institute/ENSEMBL/gtf</span><br><span class="line">wget ftp://ftp.ensembl.org/pub/release-87/gtf/homo_sapiens/Homo_sapiens.GRCh38.87.chr.gtf.gz </span><br><span class="line">wget ftp://ftp.ensembl.org/pub/release-87/gtf/mus_musculus/Mus_musculus.GRCm38.87.chr.gtf.gz</span><br><span class="line">wget ftp://ftp.ensembl.org/pub/release-87/gtf/danio_rerio/Danio_rerio.GRCz10.87.chr.gtf.gz</span><br><span class="line"> </span><br><span class="line"> </span><br><span class="line"> </span><br><span class="line"> </span><br><span class="line"> </span><br><span class="line">cd ~/institute/TCGA/firehose</span><br><span class="line">## https://gdac.broadinstitute.org/</span><br><span class="line">wget http://gdac.broadinstitute.org/runs/stddata__2016_01_28/data/ACC/20160128/gdac.broadinstitute.org_ACC.Merge_snp__genome_wide_snp_6__broad_mit_edu__Level_3__segmented_scna_minus_germline_cnv_hg19__seg.Level_3.2016012800.0.0.tar.gz  -O ACC.gistic.seg.tar.gz</span><br><span class="line">wget http://gdac.broadinstitute.org/runs/stddata__2016_01_28/data/ACC/20160128/gdac.broadinstitute.org_ACC.Merge_snp__genome_wide_snp_6__broad_mit_edu__Level_3__segmented_scna_hg19__seg.Level_3.2016012800.0.0.tar.gz  -O ACC.raw.seg.tar.gz </span><br><span class="line">wget http://gdac.broadinstitute.org/runs/stddata__2016_01_28/data/ACC/20160128/gdac.broadinstitute.org_ACC.Mutation_Packager_Calls.Level_3.2016012800.0.0.tar.gz -O ACC.maf.tar.gz</span><br><span class="line">wget http://gdac.broadinstitute.org/runs/stddata__2016_01_28/data/ACC/20160128/gdac.broadinstitute.org_ACC.Mutation_Packager_Oncotated_Calls.Level_3.2016012800.0.0.tar.gz -O ACC.maf.anno.tar.gz</span><br></pre></td></tr></table></figure>

          
          <hr>
          <ul class="pager no-print">
              
              <li class="previous">
                  <a href="/hexo/2017/11/03/2017-11-03_Bioconda_software/" data-toggle="tooltip" data-placement="left" title="使用Bioconda管理Linux系统中的生物信息软件">&larr; Previous Post</a>
              </li>
              
              
              <li class="next">
                  <a href="/hexo/2017/10/15/2017-10-15_Concordanceindex/" data-toggle="tooltip" data-placement="top" title="一致性指数Concordance index(CI, C-index)">Next Post&rarr;</a>
              </li>
              
          </ul>
          
  <br>

  


      </div>
      
  <div class="hidden-xs col-sm-3 toc-col">
    <div class="toc-wrap">
        Table of Contents
        
          <ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#分析软件"><span class="toc-number">1.</span> <span class="toc-text">分析软件</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#转录组-de-novo分析流程的软件全代码"><span class="toc-number">2.</span> <span class="toc-text">转录组 de novo分析流程的软件全代码</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#基因组，gtf，bed，注释"><span class="toc-number">3.</span> <span class="toc-text">基因组，gtf，bed，注释</span></a></li></ol>
        
    </div>
  </div>


    </div>
</article>

<!-- Footer -->
<!-- footer.ejs -->
<footer class="no-print">

    <div class="text-center">

        <ul class="list-inline">
            
            
            

            

            

            

            

            

        </ul>
        <div class="text-muted copyright">
            &copy;
            
            2017 - 2019
            
            -
            xt
            <br>
            
            
            
            | 
            <p>
                <span id="busuanzi_container_site_pv"><span id="busuanzi_value_site_pv"></span> <b>PV</b></span> -
                <span id="busuanzi_container_site_uv"><span id="busuanzi_value_site_uv"></span> <b>UV</b></span> -
                112.4k <b>Words</b>
        </p></div>

        <div class="search-container">
            <form class="site-search-form">
                <span class="glyphicon glyphicon-search"></span>
                <input type="text" id="local-search-input" class="st-search-input" placeholder="Search...">
            </form>
            <div id="local-search-result" class="local-search-result-cls"></div>
        </div>
        <script type="text/javascript" id="local.search.active">
            var inputArea = document.querySelector("#local-search-input");
            inputArea.onclick = function () {
                var root = "/hexo/";
                getSearchFile(root);
                this.onclick = null
            }
            inputArea.onkeydown = function () {
                if (event.keyCode == 13) return false
            }
        </script>

    </div>
</footer>

<!-- Custom Theme JavaScript -->
<script src="/hexo/js/main.js"></script>

<!-- async load function -->
<script>
    function async(u, c) {
        var d = document, t = 'script',
            o = d.createElement(t),
            s = d.getElementsByTagName(t)[0];
        o.src = u;
        if (c) {
            o.addEventListener('load', function (e) {
                c(null, e);
            }, false);
        }
        s.parentNode.insertBefore(o, s);
    }
</script>

<!--fastClick.js -->
<script>
    async("//cdn.bootcss.com/fastclick/1.0.6/fastclick.min.js", function () {
        var $nav = document.querySelector("nav");
        if ($nav) FastClick.attach($nav);
    })
</script><!-- hexo-inject:begin --><!-- hexo-inject:end -->



</body>

</html>
