<!DOCTYPE html>
<html lang="en">

<!-- layout.ejs-->
<head><meta name="generator" content="Hexo 3.8.0">
    <!-- hexo-inject:begin --><!-- hexo-inject:end --><meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="Adelaide">
    <meta name="author" content="xt">
    <meta name="keyword" content="">
    <link rel="canonical" href="https://taoshengxu.github.io/hexo/hexo/2020/05/21/2020-05-21_singlecell_analysis/">
    <link rel="shortcut icon" href="/hexo/img/favicon.png">
    <link rel="alternate" type="application/atom+xml" title="IIM" href="/atom.xml">
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/animate.css/3.5.2/animate.min.css">
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
    <link rel="stylesheet" href="https://ajax.googleapis.com/ajax/libs/jqueryui/1.12.1/themes/smoothness/jquery-ui.css">
    <script src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.12.1/jquery-ui.min.js"></script>
    <script src="/hexo/js/search.js"></script>

    <title>
        
        Single cell RNA 数据处理｜undefined
        
    </title>

    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">

    <link rel="stylesheet" href="/hexo/css/main.css">

    
      <link href="//fonts.googleapis.com/css?family=Source+Code+Pro" rel="stylesheet" type="text/css">
      <link rel="stylesheet" href="/hexo/css/highlight.css">
    

    

    <script async src="//dn-lbstatics.qbox.me/busuanzi/2.3/busuanzi.pure.mini.js"></script>


    

    

    


    
  


    




    <script async defer src="https://buttons.github.io/buttons.js"></script>

<link rel="stylesheet" href="/hexo/css/prism-solarizedlight.css" type="text/css">
<link rel="stylesheet" href="/hexo/css/prism-line-numbers.css" type="text/css"><!-- hexo-inject:begin --><!-- hexo-inject:end --></head>

<style>
    header.intro-header {
        background-image: url('/hexo/img/northernlights-sisimiut-lake.jpg')
    }
</style>
<!-- hack iOS CSS :active style -->
<body ontouchstart="" class="animated fadeIn">
<!-- hexo-inject:begin --><!-- hexo-inject:end --><header>
  <nav class="navbar navbar-default header-navbar" id="nav-top" data-ispost="true" data-istags="false" data-ishome="false">
    <div class="container-fluid">
      <div class="navbar-header page-scroll">
        <button type="button" class="navbar-toggle" data-toggle="collapse" aria-expanded="false" data-target="#website_navbar">
          <span class="sr-only">Toggle navigation</span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <span class="navbar-brand animated pulse">
          <a class="brand-logo" href="/hexo/">
                <img src="/hexo/img/banner.jpg?h=350&amp;auto=compress&amp;cs=tinysrgb">
          </a>
        </span>
      </div>

      <div class="collapse navbar-collapse" id="website_navbar">
          <ul class="nav navbar-nav navbar-right">
              
                <li>
                  <a href="/hexo/">home</a>
                </li>
              
                <li>
                  <a href="/hexo/archives/">archives</a>
                </li>
              
                <li>
                  <a href="/hexo/categories/">categories</a>
                </li>
              
                <li>
                  <a href="/hexo/tags/">tags</a>
                </li>
              
                <li>
                  <a href="/hexo/columns/">columns</a>
                </li>
              
          </ul>
      </div>
  </div></nav>


  
    <style>
       .intro-header {
          background-image: url('/hexo/post_cover_images/moon.jpg');
      }
    </style>

    <div class="intro-header">
        <div class="container">
            <div class="row">
                <div class="col-lg-8 col-lg-offset-2 col-md-10 col-md-offset-1 text-center">
                    <div class="site-heading">
                        <h1>Single cell RNA 数据处理</h1>
                        
                        

                        
                          <span class="meta">
                               <span class="meta-item">Author: XT</span>
                               <span class="meta-item">Date: May 21, 2020</span>
                               
                          </span>
                          <div class="tags text-center">
                              Categories: 
                              <a class="tag" href="/hexo/categories/#Bioinformatics" title="Bioinformatics">Bioinformatics</a>
                              
                          </div>
                          <div class="tags text-center">
                              Tags: 
                          </div>
                        
                    </div>
                </div>
            </div>
        </div>
    </div>
  
</header>


<!-- Main Content -->
<!-- post.ejs -->
<article>
    <div class="container">
      <div class="col-lg-8 col-lg-offset-1 col-sm-9">
          
          <span class="post-count">4,799 words in total, 21 minutes required.</span>
          <hr>
          
          <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/hint.css/2.4.1/hint.min.css"><h2 id="学习视频"><a href="#学习视频" class="headerlink" title="学习视频"></a>学习视频</h2><p><a href="https://www.bilibili.com/video/BV1dt411Y7nn" target="_blank" rel="noopener">https://www.bilibili.com/video/BV1dt411Y7nn</a></p>
<p><a href="https://github.com/jmzeng1314/scRNA_smart_seq2" target="_blank" rel="noopener">https://github.com/jmzeng1314/scRNA_smart_seq2</a></p>
<h2 id="scatter-包学习"><a href="#scatter-包学习" class="headerlink" title="scatter 包学习"></a>scatter 包学习</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># s4 对象操作</span></span><br><span class="line">assay(fluidigm)  &lt;- assays(fluidigm)$rsem_counts</span><br><span class="line">pheno_data &lt;- as.data.frame(colData(fluidigm))</span><br><span class="line"></span><br><span class="line">str(sce) 展示sce对象所有的信息</span><br><span class="line">str(sce,max.level=<span class="number">2</span>)</span><br><span class="line">exprs(sce) </span><br><span class="line">genesNames= rownames(rowData(sce))</span><br><span class="line">assays(sce)$counts 查看原始counts值</span><br></pre></td></tr></table></figure>
<h2 id="seurat-包学习"><a href="#seurat-包学习" class="headerlink" title="seurat 包学习"></a>seurat 包学习</h2><p><a href="https://satijalab.org/seurat/v3.1/pbmc3k_tutorial.html" target="_blank" rel="noopener">https://satijalab.org/seurat/v3.1/pbmc3k_tutorial.html</a></p>
<p><a href="https://www.jianshu.com/p/be12d25d2593" target="_blank" rel="noopener">https://www.jianshu.com/p/be12d25d2593</a></p>
<p><a href="https://www.plob.org/article/11101.html" target="_blank" rel="noopener">https://www.plob.org/article/11101.html</a></p>
<h4 id="seurat对象思维导图"><a href="#seurat对象思维导图" class="headerlink" title="seurat对象思维导图"></a>seurat对象思维导图</h4><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">library</span>(mindr)</span><br><span class="line">out &lt;- capture.output(str(du145_egf))</span><br><span class="line"></span><br><span class="line">aa=gsub(<span class="string">"\\.\\.\\$"</span>,<span class="string">"#"</span>,out)</span><br><span class="line">aa=gsub(<span class="string">"\\.\\.@"</span>,<span class="string">"# "</span>,gsub(<span class="string">"\\.\\. "</span>,<span class="string">"#"</span>,aa))</span><br><span class="line">out2&lt;- paste(aa, collapse=<span class="string">"\n"</span>)</span><br><span class="line">mm(out2,type =<span class="string">"text"</span>,root= <span class="string">"Seurat"</span>)</span><br></pre></td></tr></table></figure>
<h2 id="Monocle包学习"><a href="#Monocle包学习" class="headerlink" title="Monocle包学习"></a>Monocle包学习</h2><p><a href="http://cole-trapnell-lab.github.io/monocle-release/" target="_blank" rel="noopener">http://cole-trapnell-lab.github.io/monocle-release/</a> [官网]</p>
<p><a href="https://cole-trapnell-lab.github.io/monocle3/" target="_blank" rel="noopener">https://cole-trapnell-lab.github.io/monocle3/</a></p>
<p><a href="https://www.jianshu.com/p/8f57ac63f399" target="_blank" rel="noopener">https://www.jianshu.com/p/8f57ac63f399</a></p>
<p><a href="https://www.jianshu.com/p/66c387e1de3d" target="_blank" rel="noopener">https://www.jianshu.com/p/66c387e1de3d</a></p>
<blockquote>
<p>monocle 2 使用对象 CellDataSet类,继承ExpressionSet总结如下：</p>
</blockquote>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">## 取表达矩阵</span></span><br><span class="line">exprs(cds)</span><br><span class="line"><span class="comment">## 取注释信息</span></span><br><span class="line">pData(cds)</span><br><span class="line">fData(cds)</span><br><span class="line">colnames 取矩阵的列名</span><br></pre></td></tr></table></figure>
<blockquote>
<p>monocle 3 使用对象 cell_data_set,继承SingleCellExperiment  总结如下：</p>
<p>@meta.data一般会存放表型相关的信息如cluster、sample的来源、group等；<br>@assay中的@counts存放单细胞测序的raw data，data储存了归一化后的数据，而scale.data存储了标准化之后的数据（一般是z-score）。课程里使用的是归一化后的（实际上我认为需要使用最原始的，也就是counts里的数据，理由见下面的分析。但是由于为了跟课程一致，我这里用了和课程一样的归一化后的数据），需要注意的是，不要用scale.data，也就是不要用标准化之后的data来构建monocle对象，<br><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">## 直接输出 cell_data_set 对象的信息,这样打印可以能看到assay里面的对象是counts，所以后面可以用counts取出表达矩阵</span></span><br><span class="line">cds</span><br><span class="line"><span class="comment">## 查看slots信息</span></span><br><span class="line">str(cds)</span><br><span class="line"><span class="comment">#对象可以通过@取slot的内容</span></span><br><span class="line">cds@colData</span><br><span class="line"><span class="comment">## 取表达矩阵</span></span><br><span class="line"> SingleCellExperiment::counts(cds) </span><br><span class="line"><span class="comment">## 取注释信息</span></span><br><span class="line">colData(cds)</span><br><span class="line">rowData(cds)</span><br><span class="line">cds@colData</span><br></pre></td></tr></table></figure></p>
</blockquote>
<h4 id="Monocle相关函数功能"><a href="#Monocle相关函数功能" class="headerlink" title="Monocle相关函数功能"></a>Monocle相关函数功能</h4><p> <code>preprocess_cds</code> </p>
<p>first normalizes the data by log and size factor to address depth differences, or by size factor only. Next,  calculates a lower dimensional space that will be used as the input for further dimensionality reduction like tSNE and UMAP. </p>
<p>首先用log归一化数据，然后计算PCA, num_dim参数取PCA的维度，计算PCA的目的是为了以后在PCA基础上继续 tSNE或者UMAP降维。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">cds &lt;- preprocess_cds(cds, num_dim = <span class="number">50</span>)</span><br></pre></td></tr></table></figure>
<p><code>align_cds</code></p>
<p>重要的参数是alignment_group， residual_model_formula_str， 其中alignment_group是去除批次效应， 而residual_model_formula_str一般是背景噪声，这里是用lmFit 函数 对表达数据 关于背景参数的回归，然后表达数据减去这个数据。具体原理为什么这样，还是不清楚。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">cds &lt;- align_cds(cds, alignment_group = <span class="string">"batch"</span>, residual_model_formula_str = <span class="string">"~ bg.300.loading + bg.400.loading + bg.500.1.loading + bg.500.2.loading + bg.r17.loading + bg.b01.loading + bg.b02.loading"</span>)</span><br></pre></td></tr></table></figure>
<p><code>cluster_cells</code></p>
<p>对cells做划分， 得到的结果是”cluster”和“partition“。 其中”cluster” 部分是无监督聚类（Louvain/Leiden communities）结果，<strong>然后针对”cluster” 得到 “partition“（一个聚合的更大的类）</strong>represent superclusters of the Louvain/Leiden communities that are found using a kNN pruning method。通过 clusters(), partitions() 函数可以取到对应的分类结果.结果也可以通过 cds$cluster</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">cds &lt;- cluster_cells(cds)</span><br><span class="line">clusters(cds)</span><br><span class="line">partitions(cds) </span><br><span class="line">plot_cells(cds, color_cells_by = <span class="string">"cluster"</span>)</span><br><span class="line">plot_cells(cds, color_cells_by = <span class="string">"partition"</span>)</span><br></pre></td></tr></table></figure>
<p><code>learn_graph</code></p>
<p>学习进化轨迹函数</p>
<figure class="highlight routeros"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">cds &lt;- learn_graph(cds)</span><br><span class="line">plot_cells(cds,</span><br><span class="line">           color_cells_by = <span class="string">"cell.type"</span>,</span><br><span class="line">           <span class="attribute">label_groups_by_cluster</span>=<span class="literal">FALSE</span>,</span><br><span class="line">           <span class="attribute">label_leaves</span>=<span class="literal">FALSE</span>,</span><br><span class="line">           <span class="attribute">label_branch_points</span>=<span class="literal">FALSE</span>)</span><br></pre></td></tr></table></figure>
<p><code>aggregate_gene_expression</code></p>
<p>聚合表达 在这里是求 每个基因模块 每个 分组中的平均值。</p>
<h2 id="变异系数-Coefficient-of-variation-（CV）"><a href="#变异系数-Coefficient-of-variation-（CV）" class="headerlink" title="变异系数 Coefficient of variation （CV）"></a>变异系数 Coefficient of variation （CV）</h2><p>CV=<strong>标准差与均值的比值</strong></p>
<p>变异系数又称离散系数或<strong>相对偏差</strong>(我们肯定都听过标准偏差，也就是sd值，它描述了数据值偏离算术平均值的程度)，这个相对偏差描述的是标准偏差与平均值之比，即：<code>cv=sd/mean*100%</code></p>
<ul>
<li>如果两组数据的观测值在一个数量级，那么可以用标准差来比较它们的离散度</li>
<li>如果两个数据差别太大（比如一群老鼠和一群大象的重量），然后就要用<code>CV</code>或<code>CV^2</code>，这样可以去除不同量纲均值差异的影响</li>
</ul>
<h2 id="SCENIC-single-cell-RNA-seq数据推断基因调控网络和细胞功能聚类"><a href="#SCENIC-single-cell-RNA-seq数据推断基因调控网络和细胞功能聚类" class="headerlink" title="SCENIC | single-cell RNA-seq数据推断基因调控网络和细胞功能聚类"></a>SCENIC | single-cell RNA-seq数据推断基因调控网络和细胞功能聚类</h2><p><a href="https://github.com/aertslab/SCENIC" target="_blank" rel="noopener">https://github.com/aertslab/SCENIC</a></p>
<p><a href="https://cloud.tencent.com/developer/article/1606747" target="_blank" rel="noopener">https://cloud.tencent.com/developer/article/1606747</a> 这篇是论文大致翻译，写入了理解，good.</p>
<p>SCENIC是一种同时重建基因调控网络并从单细胞RNA-seq数据中鉴定stable cell states的工具。基于共表达和DNA模基序 （motif）分析推断基因调控网络 ，然后在每个细胞中分析网络活性以鉴定细胞状态。</p>
<p>workflow 包含3个主要步骤：</p>
<ul>
<li><p>1.用GENIE3（随机森林) 或GRNBoost (Gradient Boosting) 推断转录因子与候选靶基因之间的共表达模块。每个模块包含一个转录因子及其靶基因，纯粹基于共表达。</p>
</li>
<li><p>2.使用RcisTarget分析每个共表达模块中的基因，以鉴定enriched motifs；仅保留TF motif富集的模块和targets，<strong>每个TF及其潜在的直接targets gene被称作一个<code>调节子（regulon）</code></strong></p>
</li>
<li><p>3.使用AUCell评估每个细胞中每个regulon的活性，AUCell分数用于生成Regulon活性矩阵，通过为每个regulon设置AUC阈值，可以将该矩阵进行二值化（0|1，on|off），这将确定Regulon在哪些细胞中处于“打开”状态。</p>
</li>
</ul>
<blockquote>
<p><strong>使用RcisTarget是SCENIC不同于大多共表达算法的重要区别</strong>。</p>
</blockquote>
<p>由于GENIE3模块仅基于共表达，因此结果可能包含许多误报和间接target，为了鉴定推断的直接结合的靶标基因，使用RcisTarget对每个共表达模块进行顺式调控基序（motif）分析。仅保留具有正确基因上游调节子且显着富集TF motif的模块，并对它们进行修剪以除去缺乏基序支持的间接靶标，这些处理后的模块才称为regulon。</p>
<blockquote>
<p><strong>AUCell算法来对每个细胞中的每个regulon的活性进行评分。</strong></p>
</blockquote>
<p>对于一个给定的regulon，通过比较所有细胞间的AUCell打分值，我们可以识别哪些细胞具有更显著高的regulon活性。通过卡阈值得到的二元活性矩阵使矩阵维数减少（可理解为只有 0|1，on|off），对于下游分析很有用。例如，<strong>基于regulon二元活性矩阵的聚类，可以根据某个调控子网络（regulon）的活性来识别细胞群类型和细胞状态。</strong>由于regulon是整体评分的，而不是使用单个基因的表达，因此这种方法对于个别基因的dropouts很有效。</p>
<blockquote>
<p>SCENIC使用建议</p>
</blockquote>
<p>随着单细胞数据集大小的增加，作者建议两种补充方法来扩展网络推断：</p>
<p>1.取采样数据集的子集来推断出GRN，并在AUCell评分步骤中包括所有细胞</p>
<p>2.使用更高效的机器学习和大数据处理解决方案，作者应用GRNBoost 梯度增强来代替random-forest regression，这种实现方式大大减少了推断GRN所需的时间，并将为在非常大的数据集上进行网络推断铺平道路。</p>
<p>SCENIC是一种普遍适用的分析scRNA-seq数据的方法，利用TF和顺式调控序列来指导细胞状态的发现。文章的结果表明，GRNs是确定细胞状态的可靠方法，并且scRNA-seq数据非常适合跟踪基因调控过程，在基因调控过程中特定组合的TFs驱动细胞特异性的转录组。</p>
<figure class="highlight"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line">python3 -m remote_ikernel manage --add \</span><br><span class="line">    --kernel_cmd="source /opt/miniconda3/etc/profile.d/conda.sh &amp;&amp; conda activate scenic_protocol &amp;&amp; /data/home/xutaosheng5/.conda/envs/scenic_protocol/bin/ipython3 kernel -f &#123;connection_file&#125;" \</span><br><span class="line">    --name="pyscenic-0918-conda" \</span><br><span class="line">    --interface=ssh \</span><br><span class="line">    --host=hostname \</span><br><span class="line">    --workdir="~/" \</span><br><span class="line">    --language=python3</span><br><span class="line">    </span><br><span class="line">    </span><br><span class="line">    </span><br><span class="line">    </span><br><span class="line">    </span><br><span class="line">    python3 -m remote_ikernel manage --add \</span><br><span class="line">    --kernel_cmd="source /path/to/anaconda3/etc/profile.d/conda.sh &amp;&amp; conda activate scenic_protocol &amp;&amp; /path/to/envs/scenic_protocol/bin/ipython3 kernel -f &#123;connection_file&#125;" \</span><br><span class="line">    --name="pyscenic-0918-conda" \</span><br><span class="line">    --interface=ssh \</span><br><span class="line">    --host=hostname \</span><br><span class="line">    --workdir="~/" \</span><br><span class="line">    --language=python3</span><br><span class="line">    </span><br><span class="line">    </span><br><span class="line">    /data/home/xutaosheng5/.local/share/jupyter/kernels/scenic_protocol</span><br></pre></td></tr></table></figure>
<h2 id="单细胞测序质控相关footnote"><a href="#单细胞测序质控相关footnote" class="headerlink" title="单细胞测序质控相关footnote"></a>单细胞测序质控相关<sup><a href="#fn_footnote" id="reffn_footnote">footnote</a></sup></h2><p><a href="https://www.plob.org/article/20855.html" target="_blank" rel="noopener">https://www.plob.org/article/20855.html</a></p>
<h4 id="1-ERCC-spike-in-人为插入外源基因"><a href="#1-ERCC-spike-in-人为插入外源基因" class="headerlink" title="1. ERCC spike-in 人为插入外源基因"></a>1. <strong>ERCC spike-in 人为插入外源基因</strong></h4><p>  spike-in是已知浓度的外源RNA分子。在单细胞裂解液中加入spike-in后，再进行反转录。最广泛使用的spike-in是由External RNA Control Consortium （ERCC）提供的。目前使用的赛默飞公司提供的ERCC是包括<strong>92个不同长度和GC含量的细菌RNA序列</strong>，因此它和哺乳动物转录组不同，主要体现在转录本长度、核苷酸成分、polyA长度、没有内含子、没有二级结构。polyA尾大约15nt（一般保守的内源mRNA的polyA尾有250nt）。用它是为了更好地估计和消除<a href="https://www.plob.org/tag/单细胞测序" target="_blank" rel="noopener">单细胞测序</a>文库的系统误差(除此以外，还有一种UMI在10X中常用)。ERCC应该在样本解离后、建库前完成添加。除了ERCC，还其他的spike-in信息**，也就是Spike-In RNA Variants (SIRV)。</p>
<blockquote>
<p>挑选ERCC，我们这里用的正则表达匹配：<code>^ERCC</code> ，但是要注意，人类基因组中也会有以<code>ERCC</code>开头的基因集，比如来自GeneCard的结果。如ERCC1,ERCC2,..这样在挑选spike-in的时候，就可能会出错，因此在发布表达矩阵的时候，最好使用标准命名（Entrez、Ensembl），最好不要直接用Symbol命令</p>
<p><strong>比对到spike-in转录本的reads比例：分母是每个细胞文库的大小，分子就是细胞其中spike-in的数量。如果这个值很高，间接说明内源基因占比很低，细胞质量不好。**</strong></p>
</blockquote>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># grepl返回逻辑值</span></span><br><span class="line">isSpike(sce, <span class="string">"ERCC"</span>) &lt;- grepl(<span class="string">"^ERCC"</span>, rownames(sce))</span><br><span class="line">summary(isSpike(sce, <span class="string">"ERCC"</span>))</span><br><span class="line"><span class="comment">##    Mode   FALSE    TRUE </span></span><br><span class="line"><span class="comment">## logical   46611      92</span></span><br><span class="line"></span><br><span class="line">is.sirv &lt;- grepl(<span class="string">"^SIRV"</span>, rownames(sce))</span><br><span class="line">summary(is.sirv)</span><br><span class="line"><span class="comment">##    Mode   FALSE    TRUE </span></span><br><span class="line"><span class="comment">## logical   46696       7</span></span><br><span class="line">sce &lt;- sce[!is.sirv,]</span><br></pre></td></tr></table></figure>
<h4 id="2-线粒体基因"><a href="#2-线粒体基因" class="headerlink" title="2. 线粒体基因"></a>2. 线粒体基因</h4><p>如果不使用spike-in的信息时，可以用线粒体基因来替代。<strong>高比例的线粒体基因同样也代表了细胞质量低</strong>（Islam et al. 2014; Ilicic et al. 2016）。原因如下：pre-mRNA在细胞核内完成一系列复杂加工后，成熟的mRNA被转运到细胞质中进行翻译。如果一个<strong>细胞破裂，大量的细胞质mRNA会外流，但是线粒体的体积比mRNA分子要大得多</strong>，因此最后剩下的线粒体mRNA也更多，最后经过反转录、扩增、比对到线粒体基因组的reads数也更多。</p>
<blockquote>
<p>所以线粒体基因在每个单细胞中的比例，可以作为细胞质量好坏的控制条件。一般取5%。超过5%的单细胞被认为是低质量的细胞，应该被去除。</p>
</blockquote>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># https://zhuanlan.zhihu.com/p/111753425</span></span><br><span class="line"><span class="comment"># 以gencode数据库的gtf文件为标准，在人类的gencode.v32.annotation.gtf 文件里面，可以查找到37个</span></span><br><span class="line">gene_id <span class="string">"ENSG00000210049.1"</span>; gene_name <span class="string">"MT-TF"</span>; hgnc_id <span class="string">"HGNC:7481"</span></span><br><span class="line">gene_id <span class="string">"ENSG00000211459.2"</span>; gene_name <span class="string">"MT-RNR1"</span>; hgnc_id <span class="string">"HGNC:7470"</span></span><br><span class="line">gene_id <span class="string">"ENSG00000210077.1"</span>; gene_name <span class="string">"MT-TV"</span>; hgnc_id <span class="string">"HGNC:7500"</span></span><br><span class="line">gene_id <span class="string">"ENSG00000210082.2"</span>; gene_name <span class="string">"MT-RNR2"</span>; hgnc_id <span class="string">"HGNC:7471"</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 所以你会看到seurat包的函数：PercentageFeatureSet 可以用来计算线粒体基因含量。</span></span><br><span class="line">sce &lt;- CreateSeuratObject(Read10X(<span class="string">'../scRNA/filtered_feature_bc_matrix/'</span>), <span class="string">"sce"</span>)</span><br><span class="line">  head(sce@meta.data)  </span><br><span class="line">  GetAssayData(sce,<span class="string">'counts'</span>)</span><br><span class="line">  sce[[<span class="string">"percent.mt"</span>]] &lt;- PercentageFeatureSet(sce, pattern = <span class="string">"^MT-"</span>)</span><br><span class="line">  VlnPlot(sce, features = c(<span class="string">"nFeature_RNA"</span>, <span class="string">"nCount_RNA"</span>, <span class="string">"percent.mt"</span>), ncol = <span class="number">3</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># 上面的方法是修改 sce[["percent.mt"]] ，下面我们演示 AddMetaData 函数，同样是可以增加线粒体基因含量信息到我们的seurat对象。</span></span><br><span class="line">mt.genes &lt;- rownames(sce)[grep(<span class="string">"^MT-"</span>,rownames(sce))]</span><br><span class="line">C&lt;-GetAssayData(object = sce, slot = <span class="string">"counts"</span>)</span><br><span class="line">percent.mito &lt;- Matrix::colSums(C[mt.genes,])/Matrix::colSums(C)*<span class="number">100</span></span><br><span class="line">sce &lt;- AddMetaData(sce, percent.mito, col.name = <span class="string">"percent.mito"</span>)</span><br><span class="line">sce[[<span class="string">"percent.mito"</span>]]</span><br><span class="line"></span><br><span class="line"><span class="comment">#也可以是添加核糖体基因含量，同样的你需要知道核糖体基因的名字规则：</span></span><br><span class="line">rb.genes &lt;- rownames(sce)[grep(<span class="string">"^RP[SL]"</span>,rownames(sce))]</span><br><span class="line">percent.ribo &lt;- Matrix::colSums(C[rb.genes,])/Matrix::colSums(C)*<span class="number">100</span></span><br><span class="line">sce &lt;- AddMetaData(sce, percent.ribo, col.name = <span class="string">"percent.ribo"</span>)</span><br></pre></td></tr></table></figure>
<h2 id="全网第一的单细胞转录组实战"><a href="#全网第一的单细胞转录组实战" class="headerlink" title="全网第一的单细胞转录组实战"></a>全网第一的单细胞转录组实战</h2><h4 id="1-设置工作环境"><a href="#1-设置工作环境" class="headerlink" title="1. 设置工作环境"></a>1. 设置工作环境</h4><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">rm(list = ls())  <span class="comment">## 魔幻操作，一键清空~</span></span><br><span class="line">Sys.setenv(R_MAX_NUM_DLLS=<span class="number">999</span>) <span class="comment">##Sys.setenv修改环境设置，R的namespace是有上限的，如果导入包时超过这个上次就会报错,R_MAX_NUM_DLLS可以修改这个上限</span></span><br><span class="line">options(stringsAsFactors = <span class="literal">F</span>) <span class="comment">##options:允许用户对工作空间进行全局设置，stringsAsFactors防止R自动把字符串string的列辨认成factor</span></span><br><span class="line">options()$repos  <span class="comment">## 查看使用install.packages安装时的默认镜像</span></span><br><span class="line">options()$BioC_mirror <span class="comment">##查看使用bioconductor的默认镜像</span></span><br><span class="line">chooseBioCmirror()</span><br><span class="line"><span class="comment"># options(BioC_mirror="http://mirrors.tuna.tsinghua.edu.cn/bioconductor/")</span></span><br><span class="line"><span class="comment"># options(BioC_mirror="https://mirrors.ustc.edu.cn/bioc/") ##指定镜像，这个是中国科技大学镜像</span></span><br><span class="line">options(<span class="string">"repos"</span> = c(CRAN=<span class="string">"https://mirrors.tuna.tsinghua.edu.cn/CRAN/"</span>)) <span class="comment">##指定install.packages安装镜像，这个是清华镜像</span></span><br><span class="line">options()$repos </span><br><span class="line">options()$BioC_mirror</span><br></pre></td></tr></table></figure>
<h4 id="2-CPM-count-per-million，每百万碱基中每个转录本的count值"><a href="#2-CPM-count-per-million，每百万碱基中每个转录本的count值" class="headerlink" title="2. CPM (count-per-million，每百万碱基中每个转录本的count值)"></a>2. CPM (count-per-million，每百万碱基中每个转录本的count值)</h4><p>CPM只对基因的read count相对每个样本的 总reads数做了数量的均一化，去除文库大小差异。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">dat=log2(edgeR::cpm(dat)+<span class="number">1</span>)</span><br><span class="line"><span class="comment"># dat 行(基因), 列（样本），计算等同于下式</span></span><br><span class="line">log2(dat[<span class="number">1</span>,<span class="number">3</span>]*<span class="number">1000000</span>/sum(dat[,<span class="number">3</span>])+<span class="number">1</span>)</span><br></pre></td></tr></table></figure>
<h4 id="3-PCA做图"><a href="#3-PCA做图" class="headerlink" title="3. PCA做图"></a>3. PCA做图</h4><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">## dat 行是样本，列是特征</span></span><br><span class="line"><span class="keyword">library</span>(<span class="string">"FactoMineR"</span>)</span><br><span class="line"><span class="keyword">library</span>(<span class="string">"factoextra"</span>) </span><br><span class="line"><span class="comment"># group=c(1,1,2,1,3....) 样本分组信息</span></span><br><span class="line">dat.pca &lt;- PCA(dat, graph = <span class="literal">FALSE</span>) <span class="comment">#'-'表示“非”</span></span><br><span class="line">fviz_pca_ind(dat.pca,repel =<span class="literal">T</span>,</span><br><span class="line">               geom.ind = <span class="string">"point"</span>, <span class="comment"># show points only (nbut not "text")只显示点不显示文本</span></span><br><span class="line">               col.ind = group, <span class="comment"># color by groups 颜色组</span></span><br><span class="line">               <span class="comment"># palette = c("#00AFBB", "#E7B800"),</span></span><br><span class="line">               addEllipses = <span class="literal">TRUE</span>, <span class="comment"># Concentration ellipses 集中成椭圆</span></span><br><span class="line">               legend.title = <span class="string">"Groups"</span></span><br><span class="line">  )</span><br></pre></td></tr></table></figure>
<p><img src="https://taoshengxu.github.io/DocumentGit/img/PCA_circle.png" alt=""></p>
<h4 id="4-数据的变异系数"><a href="#4-数据的变异系数" class="headerlink" title="4.数据的变异系数"></a>4.数据的变异系数</h4><p> 在单细胞测序中起始RNA含量越低，技术重复样本的基因表达相关性也越低，因此生成标准化的基因表达谱之后，非常重要的一步是评估技术噪音（technical variability）。</p>
<p>比较常见的一种方法是计算基因表达值的变异系数的平方（CV^2），标准差与平均数的比值称为变异系数，记为C.V(Coefficient of Variance)。 </p>
<p>变异系数又称“标准差率”，是衡量资料中各观测值变异程度的另一个统计量。当进行两个或多个资料变异程度的比较时，如果度量单位与平均数相同，可以直接利用标准差来比较。平均绝对误差（Mean Absolute Deviation），又叫平均绝对离差。它是是所有单个观测值与算术平均值的偏差的绝对值的平均。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br></pre></td><td class="code"><pre><span class="line">mean_per_gene &lt;- apply(exprSet, <span class="number">1</span>, mean, na.rm = <span class="literal">TRUE</span>) <span class="comment">#对表达矩阵每行求均值</span></span><br><span class="line">sd_per_gene &lt;- apply(exprSet, <span class="number">1</span>, sd, na.rm = <span class="literal">TRUE</span>) <span class="comment">#对表达矩阵每行求标准差</span></span><br><span class="line">mad_per_gene &lt;-   apply(exprSet, <span class="number">1</span>, mad, na.rm = <span class="literal">TRUE</span>) <span class="comment">#对表达矩阵每行求绝对中位差</span></span><br><span class="line"><span class="comment">## 同样的apply函数，多次出现，请务必学透它！！！</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 构造一个数据框来存放结果。</span></span><br><span class="line">cv_per_gene &lt;- data.frame(mean = mean_per_gene,</span><br><span class="line">  sd = sd_per_gene,</span><br><span class="line">  mad=mad_per_gene,</span><br><span class="line">  cv = sd_per_gene/mean_per_gene)</span><br><span class="line">rownames(cv_per_gene) &lt;- rownames(exprSet)</span><br><span class="line">head(cv_per_gene)</span><br><span class="line"><span class="comment"># pairs(cv_per_gene)</span></span><br><span class="line">with(cv_per_gene,plot(log10(mean),log10(cv)))</span><br><span class="line">with(cv_per_gene,plot(log10(mean),log10(cv^<span class="number">2</span>)))</span><br><span class="line">cv_per_gene$log10cv2=log10(cv_per_gene$cv^<span class="number">2</span>)</span><br><span class="line">cv_per_gene$log10mean=log10(cv_per_gene$mean)</span><br></pre></td></tr></table></figure>
<h4 id="4-散点图"><a href="#4-散点图" class="headerlink" title="4.散点图"></a>4.散点图</h4><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br></pre></td><td class="code"><pre><span class="line">ggscatter(cv_per_gene, x = <span class="string">'log10mean'</span>, y = <span class="string">'log10cv2'</span>,</span><br><span class="line">          color = <span class="string">"black"</span>, shape = <span class="number">16</span>, size = <span class="number">1</span>, <span class="comment"># Points color, shape and size</span></span><br><span class="line">          xlab = <span class="string">'log10(mean)RPKM'</span>, ylab = <span class="string">"log10(cv^2)"</span>,</span><br><span class="line">          add = <span class="string">"loess"</span>, <span class="comment">#添加拟合曲线</span></span><br><span class="line">          add.params = list(color = <span class="string">"red"</span>,fill = <span class="string">"lightgray"</span>),</span><br><span class="line">          cor.coeff.args = list(method = <span class="string">"spearman"</span>), </span><br><span class="line">          label.x = <span class="number">3</span>,label.sep = <span class="string">"\n"</span>,</span><br><span class="line">          dot.size = <span class="number">2</span>,</span><br><span class="line">          ylim=c(-<span class="number">0.5</span>, <span class="number">3</span>),</span><br><span class="line">          xlim=c(<span class="number">0</span>,<span class="number">4</span>))</span><br><span class="line"></span><br><span class="line"><span class="keyword">attach</span>(draw)</span><br><span class="line"><span class="keyword">library</span>(scatterplot3d)</span><br><span class="line">scatterplot3d(G1, S, G2M, angle=<span class="number">20</span>,</span><br><span class="line">              color = rainbow(<span class="number">3</span>)[as.numeric(as.factor(assigned$phases))],</span><br><span class="line">              grid=<span class="literal">TRUE</span>, box=<span class="literal">FALSE</span>)</span><br><span class="line"><span class="keyword">detach</span>(draw)</span><br></pre></td></tr></table></figure>
<p><img src="https://taoshengxu.github.io/DocumentGit/img/Scatterplot.png" alt=""></p>
<h4 id="5-小提琴图"><a href="#5-小提琴图" class="headerlink" title="5. 小提琴图"></a>5. 小提琴图</h4><p>【小提琴图】是【箱线图】与【核密度图】的结合，【箱线图】展示了分位数的位置，【小提琴图】则展示了任意位置的密度，通过【小提琴图】可以知道哪些位置的密度较高。在上图中，白点是中位数，黑色盒型的范围是下四分位点到上四分位点，细黑线表示须。外部形状即为核密度估计（在概率论中用来估计未知的密度函数，属于非参数检验方法之一）。</p>
<p><img src="https://taoshengxu.github.io/DocumentGit/img/violin_1.png" alt=""></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">install.packages(<span class="string">"vioplot"</span>)</span><br><span class="line"><span class="comment">#载入vioplot包</span></span><br><span class="line"><span class="keyword">library</span>(<span class="string">"vioplot"</span>)</span><br><span class="line"><span class="comment">#定义需要绘制的变量数据</span></span><br><span class="line">x1 &lt;- mtcars$mpg[mtcars$cyl==<span class="number">4</span>]</span><br><span class="line">x2 &lt;- mtcars$mpg[mtcars$cyl==<span class="number">6</span>]</span><br><span class="line">x3 &lt;- mtcars$mpg[mtcars$cyl==<span class="number">8</span>]</span><br><span class="line"><span class="comment">#绘制小提琴图</span></span><br><span class="line">vioplot(x1,x2,x3,names=c(<span class="string">"4缸"</span>, <span class="string">"6缸"</span>, <span class="string">"8缸"</span>), col=<span class="string">"gold"</span>)</span><br><span class="line">title(<span class="string">"小提琴图：不同缸数对应每加仑汽油行驶里程数分布"</span>)</span><br><span class="line"></span><br><span class="line">数据化分析解读：</span><br><span class="line"><span class="number">1.</span>通常汽缸数量少的汽车每加仑汽油可以行驶更多里程数；</span><br><span class="line"><span class="number">2.4</span>缸汽车的油耗分布比较分散，<span class="number">6</span>缸汽车的油耗分布相对集中，<span class="number">8</span>缸汽车的油耗分布很不均匀（中间大两头小）；</span><br><span class="line"><span class="number">3.4</span>缸和<span class="number">8</span>缸汽车的油耗都存在比较明显的离散值（上侧的须或下侧的须较长）。</span><br></pre></td></tr></table></figure>
<p><img src="https://taoshengxu.github.io/DocumentGit/img/violin.png" alt=""></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 绘图必须强推 ggpubr  </span></span><br><span class="line"><span class="keyword">library</span>(ggpubr)</span><br><span class="line">ggviolin(df, x = <span class="string">"all"</span>, y = <span class="string">"n_g"</span>, fill = <span class="string">"all"</span>, </span><br><span class="line">         add = <span class="string">"boxplot"</span>, add.params = list(fill = <span class="string">"white"</span>)) </span><br><span class="line">ggviolin(df, x = <span class="string">"plate"</span>, y = <span class="string">"n_g"</span>, fill = <span class="string">"plate"</span>,</span><br><span class="line">         <span class="comment">#palette = c("#00AFBB", "#E7B800", "#FC4E07"),</span></span><br><span class="line">         add = <span class="string">"boxplot"</span>, add.params = list(fill = <span class="string">"white"</span>)) </span><br><span class="line">ggviolin(df, x = <span class="string">"g"</span>, y = <span class="string">"n_g"</span>, fill = <span class="string">"g"</span>, </span><br><span class="line">         add = <span class="string">"boxplot"</span>, add.params = list(fill = <span class="string">"white"</span>))  + stat_compare_means()</span><br></pre></td></tr></table></figure>
<h4 id="6-细胞周期"><a href="#6-细胞周期" class="headerlink" title="6. 细胞周期"></a>6. 细胞周期</h4><p>Gene sets reflecting five phases of the <strong>HeLa cell cycle (G1/S, S, G2/M, M and M/G1)</strong> were referred from Whitfield et al. (2002). 其中一篇提到了scran包的cyclone函数进行单细胞转录组的细胞周期状态推断。</p>
<p><a href="https://mp.weixin.qq.com/s/nFSa5hXuKHrGu_othopbWQ" target="_blank" rel="noopener">https://mp.weixin.qq.com/s/nFSa5hXuKHrGu_othopbWQ</a> </p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line">set.seed(<span class="number">6473</span>)</span><br><span class="line"><span class="keyword">library</span>(scRNAseq)</span><br><span class="line"><span class="keyword">library</span>(RColorBrewer)</span><br><span class="line"><span class="keyword">library</span>(scran)</span><br><span class="line">all.counts=assays(fluidigm)$rsem_counts</span><br><span class="line">dim(all.counts)</span><br><span class="line">head(rownames(all.counts))</span><br><span class="line">sce &lt;- SingleCellExperiment(list(counts=all.counts)) </span><br><span class="line"><span class="keyword">library</span>(org.Hs.eg.db)</span><br><span class="line">ensembl &lt;- mapIds(org.Hs.eg.db, keys=rownames(sce), keytype=<span class="string">"SYMBOL"</span>, column=<span class="string">"ENSEMBL"</span>)</span><br><span class="line">mm.pairs &lt;- readRDS(system.file(<span class="string">"exdata"</span>, <span class="string">"human_cycle_markers.rds"</span>, package=<span class="string">"scran"</span>))</span><br><span class="line">assigned &lt;- cyclone(sce, pairs=mm.pairs, gene.names=ensembl)</span><br><span class="line">head(assigned$scores)</span><br><span class="line">table(assigned$phases)</span><br><span class="line">plot(assigned$score$G1, assigned$score$G2M, </span><br><span class="line">     xlab=<span class="string">"G1 score"</span>, ylab=<span class="string">"G2/M score"</span>, pch=<span class="number">16</span>)</span><br></pre></td></tr></table></figure>
<h4 id="7-从基因数据库中获取基因长度"><a href="#7-从基因数据库中获取基因长度" class="headerlink" title="7,从基因数据库中获取基因长度"></a>7,从基因数据库中获取基因长度</h4><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">library</span>(<span class="string">"TxDb.Mmusculus.UCSC.mm10.knownGene"</span>)</span><br><span class="line">txdb &lt;- TxDb.Mmusculus.UCSC.mm10.knownGene</span><br><span class="line">?TxDb</span><br><span class="line"><span class="comment">## 下面是定义基因长度为 非冗余exon长度之和</span></span><br><span class="line"><span class="keyword">if</span>(<span class="literal">F</span>)&#123;</span><br><span class="line">  exon_txdb=exons(txdb)</span><br><span class="line">  genes_txdb=genes(txdb)</span><br><span class="line">  genes_txdb</span><br><span class="line">  ?GRanges</span><br><span class="line">  o = findOverlaps(exon_txdb,genes_txdb)</span><br><span class="line">  o</span><br><span class="line">  <span class="comment">## exon - 1 : chr1 4807893-4807982</span></span><br><span class="line">  <span class="comment">## 1        6523</span></span><br><span class="line">  <span class="comment">#  genes_txdb[6523]  # chr1 4807893-4846735 , 18777</span></span><br><span class="line">  t1=exon_txdb[queryHits(o)]</span><br><span class="line">  t2=genes_txdb[subjectHits(o)]</span><br><span class="line">  t1=as.data.frame(t1)</span><br><span class="line">  t1$geneid=mcols(t2)[,<span class="number">1</span>]</span><br><span class="line">  <span class="comment"># 如果觉得速度不够，就参考R语言实现并行计算</span></span><br><span class="line">  <span class="comment"># http://www.bio-info-trainee.com/956.html</span></span><br><span class="line">  <span class="comment">#lapply : 遍历列表向量内的每个元素，并且使用指定函数来对其元素进行处理。返回列表向量。</span></span><br><span class="line">  <span class="comment">#函数split()可以按照分组因子，把向量，矩阵和数据框进行适当的分组;</span></span><br><span class="line">  <span class="comment">#它的返回值是一个列表，代表分组变量每个水平的观测。</span></span><br><span class="line">  g_l = lapply(split(t1,t1$geneid),<span class="keyword">function</span>(x)&#123;</span><br><span class="line">    <span class="comment"># x=split(t1,t1$geneid)[[1]]</span></span><br><span class="line">    head(x)</span><br><span class="line">    tmp=apply(x,<span class="number">1</span>,<span class="keyword">function</span>(y)&#123;</span><br><span class="line">      y[<span class="number">2</span>]:y[<span class="number">3</span>]</span><br><span class="line">    &#125;)</span><br><span class="line">    length(unique(unlist(tmp)))</span><br><span class="line">    <span class="comment"># sum(x[,4])</span></span><br><span class="line">  &#125;)</span><br><span class="line">  head(g_l)</span><br><span class="line">  g_l=data.frame(gene_id=names(g_l),length=as.numeric(g_l))</span><br><span class="line">  </span><br><span class="line">  save(g_l,file = <span class="string">'step7-g_l.Rdata'</span>)</span><br><span class="line">&#125;</span><br><span class="line">head(g_l)</span><br><span class="line"><span class="keyword">library</span>(org.Mm.eg.db)</span><br><span class="line">s2g=toTable(org.Mm.egSYMBOL)</span><br><span class="line">head(s2g)</span><br><span class="line">g_l=merge(g_l,s2g,by=<span class="string">'gene_id'</span>) <span class="comment">#把g_l,s2g两个数据框以'gene_id'为连接进行拼接</span></span><br></pre></td></tr></table></figure>
<h4 id="8-RPKM计算"><a href="#8-RPKM计算" class="headerlink" title="8.RPKM计算"></a>8.RPKM计算</h4><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">rpkm &lt;- t(do.call( rbind,lapply(<span class="number">1</span>:length(total_count),<span class="keyword">function</span>(i)&#123;<span class="number">10</span>^<span class="number">9</span>*exprSet[,i]/lengths/total_count[i]&#125;) ))</span><br></pre></td></tr></table></figure>

          
          <hr>
          <ul class="pager no-print">
              
              <li class="previous">
                  <a href="/hexo/2020/05/22/2020-05-22_scRNA_sequcencing/" data-toggle="tooltip" data-placement="left" title="single cell测序技术">&larr; Previous Post</a>
              </li>
              
              
              <li class="next">
                  <a href="/hexo/2020/05/17/2020-05-17_Breast_cellline/" data-toggle="tooltip" data-placement="top" title="乳腺癌细胞系">Next Post&rarr;</a>
              </li>
              
          </ul>
          

      </div>
      
  <div class="hidden-xs col-sm-3 toc-col">
    <div class="toc-wrap">
        Table of Contents
        
          <ol class="toc"><li class="toc-item toc-level-2"><a class="toc-link" href="#学习视频"><span class="toc-number">1.</span> <span class="toc-text">学习视频</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#scatter-包学习"><span class="toc-number">2.</span> <span class="toc-text">scatter 包学习</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#seurat-包学习"><span class="toc-number">3.</span> <span class="toc-text">seurat 包学习</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#seurat对象思维导图"><span class="toc-number">3.0.1.</span> <span class="toc-text">seurat对象思维导图</span></a></li></ol></li></ol><li class="toc-item toc-level-2"><a class="toc-link" href="#Monocle包学习"><span class="toc-number">4.</span> <span class="toc-text">Monocle包学习</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#Monocle相关函数功能"><span class="toc-number">4.0.1.</span> <span class="toc-text">Monocle相关函数功能</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#变异系数-Coefficient-of-variation-（CV）"><span class="toc-number">5.</span> <span class="toc-text">变异系数 Coefficient of variation （CV）</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#SCENIC-single-cell-RNA-seq数据推断基因调控网络和细胞功能聚类"><span class="toc-number">6.</span> <span class="toc-text">SCENIC | single-cell RNA-seq数据推断基因调控网络和细胞功能聚类</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#单细胞测序质控相关footnote"><span class="toc-number">7.</span> <span class="toc-text">单细胞测序质控相关footnote</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#1-ERCC-spike-in-人为插入外源基因"><span class="toc-number">7.0.1.</span> <span class="toc-text">1. ERCC spike-in 人为插入外源基因</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#2-线粒体基因"><span class="toc-number">7.0.2.</span> <span class="toc-text">2. 线粒体基因</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#全网第一的单细胞转录组实战"><span class="toc-number">8.</span> <span class="toc-text">全网第一的单细胞转录组实战</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#1-设置工作环境"><span class="toc-number">8.0.1.</span> <span class="toc-text">1. 设置工作环境</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#2-CPM-count-per-million，每百万碱基中每个转录本的count值"><span class="toc-number">8.0.2.</span> <span class="toc-text">2. CPM (count-per-million，每百万碱基中每个转录本的count值)</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#3-PCA做图"><span class="toc-number">8.0.3.</span> <span class="toc-text">3. PCA做图</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#4-数据的变异系数"><span class="toc-number">8.0.4.</span> <span class="toc-text">4.数据的变异系数</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#4-散点图"><span class="toc-number">8.0.5.</span> <span class="toc-text">4.散点图</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#5-小提琴图"><span class="toc-number">8.0.6.</span> <span class="toc-text">5. 小提琴图</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#6-细胞周期"><span class="toc-number">8.0.7.</span> <span class="toc-text">6. 细胞周期</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#7-从基因数据库中获取基因长度"><span class="toc-number">8.0.8.</span> <span class="toc-text">7,从基因数据库中获取基因长度</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#8-RPKM计算"><span class="toc-number">8.0.9.</span> <span class="toc-text">8.RPKM计算</span></a></li></ol></li>
        
    </div>
  </div>


    </div>
</article>

<!-- Footer -->
<!-- footer.ejs -->
<footer class="no-print">

    <div class="text-center">

        <ul class="list-inline">
            
            
            

            

            

            

            

            

        </ul>
        <div class="text-muted copyright">
            &copy;
            
            2017 - 2020
            
            -
            xt
            <br>
            
            
            
            | 
            <p>
                <span id="busuanzi_container_site_pv"><span id="busuanzi_value_site_pv"></span> <b>PV</b></span> -
                <span id="busuanzi_container_site_uv"><span id="busuanzi_value_site_uv"></span> <b>UV</b></span> -
                170.1k <b>Words</b>
        </p></div>

        <div class="search-container">
            <form class="site-search-form">
                <span class="glyphicon glyphicon-search"></span>
                <input type="text" id="local-search-input" class="st-search-input" placeholder="Search...">
            </form>
            <div id="local-search-result" class="local-search-result-cls"></div>
        </div>
        <script type="text/javascript" id="local.search.active">
            var inputArea = document.querySelector("#local-search-input");
            inputArea.onclick = function () {
                var root = "/hexo/";
                getSearchFile(root);
                this.onclick = null
            }
            inputArea.onkeydown = function () {
                if (event.keyCode == 13) return false
            }
        </script>

    </div>
</footer>

<!-- Custom Theme JavaScript -->
<script src="/hexo/js/main.js"></script>

<!-- async load function -->
<script>
    function async(u, c) {
        var d = document, t = 'script',
            o = d.createElement(t),
            s = d.getElementsByTagName(t)[0];
        o.src = u;
        if (c) {
            o.addEventListener('load', function (e) {
                c(null, e);
            }, false);
        }
        s.parentNode.insertBefore(o, s);
    }
</script>

<!--fastClick.js -->
<script>
    async("//cdn.bootcss.com/fastclick/1.0.6/fastclick.min.js", function () {
        var $nav = document.querySelector("nav");
        if ($nav) FastClick.attach($nav);
    })
</script><!-- hexo-inject:begin --><!-- hexo-inject:end -->



</body>

</html>
