<!DOCTYPE html>
<html lang="en">

<!-- layout.ejs-->
<head><meta name="generator" content="Hexo 3.8.0">
    <!-- hexo-inject:begin --><!-- hexo-inject:end --><meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="Adelaide">
    <meta name="author" content="xt">
    <meta name="keyword" content="">
    <link rel="canonical" href="https://taoshengxu.github.io/hexo/hexo/2017/11/10/2017-11-10_gtf_gene_position/">
    <link rel="shortcut icon" href="/hexo/img/favicon.png">
    <link rel="alternate" type="application/atom+xml" title="IIM" href="/atom.xml">
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/animate.css/3.5.2/animate.min.css">
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
    <link rel="stylesheet" href="https://ajax.googleapis.com/ajax/libs/jqueryui/1.12.1/themes/smoothness/jquery-ui.css">
    <script src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.12.1/jquery-ui.min.js"></script>
    <script src="/hexo/js/search.js"></script>

    <title>
        
        根据gtf格式的基因注释文件得到人所有基因的染色体坐标｜undefined
        
    </title>

    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">

    <link rel="stylesheet" href="/hexo/css/main.css">

    
      <link href="//fonts.googleapis.com/css?family=Source+Code+Pro" rel="stylesheet" type="text/css">
      <link rel="stylesheet" href="/hexo/css/highlight.css">
    

    

    <script async src="//dn-lbstatics.qbox.me/busuanzi/2.3/busuanzi.pure.mini.js"></script>


    

    

    


    
  


    




    <script async defer src="https://buttons.github.io/buttons.js"></script>

<link rel="stylesheet" href="/hexo/css/prism-solarizedlight.css" type="text/css">
<link rel="stylesheet" href="/hexo/css/prism-line-numbers.css" type="text/css"><!-- hexo-inject:begin --><!-- hexo-inject:end --></head>

<style>
    header.intro-header {
        background-image: url('/hexo/img/northernlights-sisimiut-lake.jpg')
    }
</style>
<!-- hack iOS CSS :active style -->
<body ontouchstart="" class="animated fadeIn">
<!-- hexo-inject:begin --><!-- hexo-inject:end --><header>
  <nav class="navbar navbar-default header-navbar" id="nav-top" data-ispost="true" data-istags="false" data-ishome="false">
    <div class="container-fluid">
      <div class="navbar-header page-scroll">
        <button type="button" class="navbar-toggle" data-toggle="collapse" aria-expanded="false" data-target="#website_navbar">
          <span class="sr-only">Toggle navigation</span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <span class="navbar-brand animated pulse">
          <a class="brand-logo" href="/hexo/">
                <img src="/hexo/img/banner.jpg?h=350&amp;auto=compress&amp;cs=tinysrgb">
          </a>
        </span>
      </div>

      <div class="collapse navbar-collapse" id="website_navbar">
          <ul class="nav navbar-nav navbar-right">
              
                <li>
                  <a href="/hexo/">home</a>
                </li>
              
                <li>
                  <a href="/hexo/archives/">archives</a>
                </li>
              
                <li>
                  <a href="/hexo/categories/">categories</a>
                </li>
              
                <li>
                  <a href="/hexo/tags/">tags</a>
                </li>
              
                <li>
                  <a href="/hexo/columns/">columns</a>
                </li>
              
          </ul>
      </div>
  </div></nav>


  
    <style>
       .intro-header {
          background-image: url('/hexo/img/northernlights-sisimiut-lake.jpg');
      }
    </style>

    <div class="intro-header">
        <div class="container">
            <div class="row">
                <div class="col-lg-8 col-lg-offset-2 col-md-10 col-md-offset-1 text-center">
                    <div class="site-heading">
                        <h1>根据gtf格式的基因注释文件得到人所有基因的染色体坐标</h1>
                        
                        

                        
                          <span class="meta">
                               <span class="meta-item">Author: xt</span>
                               <span class="meta-item">Date: Nov 10, 2017</span>
                               
                          </span>
                          <div class="tags text-center">
                              Categories: 
                          </div>
                          <div class="tags text-center">
                              Tags: 
                          </div>
                        
                    </div>
                </div>
            </div>
        </div>
    </div>
  
</header>


<!-- Main Content -->
<!-- post.ejs -->
<article>
    <div class="container">
      <div class="col-lg-8 col-lg-offset-1 col-sm-9">
          
          <span class="post-count">1,061 words in total, 5 minutes required.</span>
          <hr>
          
          <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/hint.css/2.4.1/hint.min.css"><p><strong>原文</strong> ：<a href="http://www.biotrainee.com/thread-472-1-1.html" target="_blank" rel="noopener">http://www.biotrainee.com/thread-472-1-1.html</a></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br></pre></td><td class="code"><pre><span class="line">mkdir -p ~/reference/gtf/gencode</span><br><span class="line"><span class="built_in">cd</span>  ~/reference/gtf/gencode</span><br><span class="line"><span class="comment">## https://www.gencodegenes.org/releases/current.html</span></span><br><span class="line">wget ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_25/gencode.v25.2wayconspseudos.gtf.gz</span><br><span class="line">wget ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_25/gencode.v25.long_noncoding_RNAs.gtf.gz </span><br><span class="line">wget ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_25/gencode.v25.polyAs.gtf.gz </span><br><span class="line">wget ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_25/gencode.v25.annotation.gtf.gz </span><br><span class="line"><span class="comment">## https://www.gencodegenes.org/releases/25lift37.html </span></span><br><span class="line">wget ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_25/GRCh37_mapping/gencode.v25lift37.annotation.gtf.gz </span><br><span class="line">wget ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_25/GRCh37_mapping/gencode.v25lift37.metadata.HGNC.gz </span><br><span class="line">wget ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_25/GRCh37_mapping/gencode.v25lift37.metadata.EntrezGene.gz </span><br><span class="line">wget ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_25/GRCh37_mapping/gencode.v25lift37.metadata.RefSeq.gz </span><br><span class="line"></span><br><span class="line">zcat  gencode.v25.long_noncoding_RNAs.gtf.gz |perl -alne <span class="string">'&#123;next unless $F[2] eq "gene" ;/gene_name \"(.*?)\";/; print "$F[0]\t$F[3]\t$F[4]\t$1" &#125;'</span> &gt;lncRNA.hg38.position</span><br><span class="line">zcat  gencode.v25.2wayconspseudos.gtf.gz     |perl -alne <span class="string">'&#123;next unless $F[2] eq "transcript" ;/gene_name \"(.*?)\";/; print "$F[0]\t$F[3]\t$F[4]\t$1" &#125;'</span> &gt;pseudos.hg38.position</span><br><span class="line">zcat  gencode.v25.annotation.gtf.gz| grep   protein_coding |perl -alne <span class="string">'&#123;next unless $F[2] eq "gene" ;/gene_name \"(.*?)\";/; print "$F[0]\t$F[3]\t$F[4]\t$1" &#125;'</span> &gt;protein_coding.hg38.position</span><br><span class="line">zcat  gencode.v25.annotation.gtf.gz|perl -alne <span class="string">'&#123;next unless $F[2] eq "gene" ;/gene_name \"(.*?)\";/; print "$F[0]\t$F[3]\t$F[4]\t$1" &#125;'</span> &gt;allGene.hg38.position</span><br><span class="line"></span><br><span class="line">zcat  gencode.v25lift37.annotation.gtf.gz | grep   protein_coding |perl -alne <span class="string">'&#123;next unless $F[2] eq "gene" ;/gene_name \"(.*?)\";/; print "$F[0]\t$F[3]\t$F[4]\t$1" &#125;'</span> &gt;protein_coding.hg19.position</span><br><span class="line">zcat  gencode.v25lift37.annotation.gtf.gz | perl -alne <span class="string">'&#123;next unless $F[2] eq "gene" ;/gene_name \"(.*?)\";/; print "$F[0]\t$F[3]\t$F[4]\t$1" &#125;'</span> &gt;allGene.hg19.position</span><br></pre></td></tr></table></figure>
<h2 id="在R语言中读取GTF文件的最好方法"><a href="#在R语言中读取GTF文件的最好方法" class="headerlink" title="在R语言中读取GTF文件的最好方法"></a>在R语言中读取GTF文件的最好方法</h2><p>下载的TCGA数据是没有注释的，需要从ensemble上面下载GTF文件，现在需要把GTF文件读入R</p>
<h4 id="第一种方法rtracklayer-import"><a href="#第一种方法rtracklayer-import" class="headerlink" title="第一种方法rtracklayer::import"></a>第一种方法rtracklayer::import</h4><figure class="highlight kotlin"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">source(<span class="string">"https://bioconductor.org/biocLite.R"</span>)</span><br><span class="line">biocLite(<span class="string">"rtracklayer"</span>)</span><br><span class="line">biocLite(<span class="string">"SummarizedExperiment"</span>)</span><br><span class="line">gtf1 &lt;- rtracklayer::<span class="keyword">import</span>(<span class="string">'Homo_sapiens.GRCh38.90.chr.gtf'</span>)</span><br><span class="line">gtf_df &lt;- <span class="keyword">as</span>.<span class="keyword">data</span>.frame(gtf1)</span><br></pre></td></tr></table></figure>
<p>最终读入27个变量，2612129个观测，测试一下显示的不错</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">test</span> &lt;- gtf_df[1:5,]</span><br><span class="line">View(<span class="built_in">test</span>)</span><br></pre></td></tr></table></figure>
<p>取出我需要的gene_id,gene_biotype,gene_name，成为新的数据框</p>
<figure class="highlight cpp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">geneid_df &lt;- dplyr::select(gtf_df,c(gene_name,gene_id,gene_biotype))</span><br></pre></td></tr></table></figure>
<h4 id="第二种方法read-table"><a href="#第二种方法read-table" class="headerlink" title="第二种方法read.table"></a>第二种方法read.table</h4><figure class="highlight php"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">gtf2 &lt;- read.table(<span class="string">'Homo_sapiens.GRCh38.90.chr.gtf'</span>, header = <span class="keyword">FALSE</span>, sep = <span class="string">'\t'</span>)</span><br></pre></td></tr></table></figure>
<p>最后发现，速度奇慢无比,取消参考了Y叔的帖子增加参数，读入成功，需要1分钟,读入9个变量<br> <a href="https://link.jianshu.com?t=https://mp.weixin.qq.com/s?__biz=MzI5NjUyNzkxMg==&amp;mid=2247483986&amp;idx=1&amp;sn=43214f77337bc660a733bbc80ab1124b&amp;chksm=ec43b315db343a03cd51ee6cb96e1646975d08e9ea7a980bed236472b77bc6166e601a977dab&amp;mpshare=1&amp;scene=1&amp;srcid=1116dPlFaunZobi46nhFTAM5#rd" target="_blank" rel="noopener">根据GTF画基因的多个转录本结构</a></p>
<figure class="highlight php"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">gtf2 &lt;- read.table(<span class="string">'Homo_sapiens.GRCh38.90.chr.gtf'</span>, stringsAsFactors = F, header = <span class="keyword">FALSE</span>, sep = <span class="string">'\t'</span>,comment = <span class="string">"#"</span>)</span><br><span class="line">test2  &lt;- gtf2[<span class="number">1</span>:<span class="number">5</span>,]</span><br></pre></td></tr></table></figure>
<h4 id="第三种方法readr包里面的read-table2"><a href="#第三种方法readr包里面的read-table2" class="headerlink" title="第三种方法readr包里面的read_table2"></a>第三种方法readr包里面的read_table2</h4><p>速度很快，有进度条，读入了18个变量，但是最gene_biotype显示不对</p>
<figure class="highlight kotlin"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">gtf3 &lt;- readr::read_table2(<span class="string">"Homo_sapiens.GRCh38.90.chr.gtf"</span>,comment = <span class="string">"#"</span>)</span><br><span class="line">gtf_df3 &lt;- <span class="keyword">as</span>.<span class="keyword">data</span>.frame(gtf3) #转成<span class="keyword">data</span>.frame</span><br><span class="line">test3 &lt;- gtf_df3[<span class="number">1</span>:<span class="number">5</span>,]</span><br><span class="line">View(test3)</span><br></pre></td></tr></table></figure>
<h4 id="第四种方法read-table，fread"><a href="#第四种方法read-table，fread" class="headerlink" title="第四种方法read.table，fread"></a>第四种方法read.table，fread</h4><p>读入数据跟read.table一样，</p>
<figure class="highlight css"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="selector-tag">library</span>(<span class="selector-tag">data</span><span class="selector-class">.table</span>)</span><br><span class="line"><span class="selector-tag">genes</span> &lt;<span class="selector-tag">-</span> <span class="selector-tag">fread</span>("<span class="selector-tag">Homo_sapiens</span><span class="selector-class">.GRCh38</span><span class="selector-class">.90</span><span class="selector-class">.chr</span><span class="selector-class">.gtf</span>")</span><br><span class="line"><span class="selector-tag">test4</span>&lt;<span class="selector-tag">-</span> <span class="selector-tag">genes</span><span class="selector-attr">[1:5,]</span></span><br><span class="line"><span class="selector-tag">View</span>(<span class="selector-tag">test4</span>)</span><br></pre></td></tr></table></figure>
<p><strong>上一步用时12秒，速度极快,只是读入后没有名字，需要自己添加</strong><br> 读入9个变量,跟read.table一样</p>
<figure class="highlight swift"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">setnames(genes, names(genes), <span class="built_in">c</span>(<span class="string">"chr"</span>,<span class="string">"source"</span>,<span class="string">"type"</span>,<span class="string">"start"</span>,<span class="string">"end"</span>,<span class="string">"score"</span>,<span class="string">"strand"</span>,<span class="string">"phase"</span>,<span class="string">"attributes"</span>) )</span><br></pre></td></tr></table></figure>
<p>可选步骤，type那一项有转录本和gene，选取gene，其他有很多转录本</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">genes &lt;- genes[<span class="built_in">type</span> == <span class="string">"gene"</span>]</span><br></pre></td></tr></table></figure>
<p>我要的信息粗存在attributes中，构建函数提取</p>
<figure class="highlight kotlin"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">extract_attributes &lt;- function(gtf_attributes, att_of_interest)&#123;</span><br><span class="line">  att &lt;- strsplit(gtf_attributes,<span class="string">"; "</span>)</span><br><span class="line">  att &lt;- gsub(<span class="string">"\""</span>,<span class="string">""</span>,unlist(att))</span><br><span class="line">  <span class="keyword">if</span>(!<span class="keyword">is</span>.<span class="literal">null</span>(unlist(strsplit(att[grep(att_of_interest, att)], <span class="string">" "</span>))))&#123;</span><br><span class="line">    <span class="keyword">return</span>( unlist(strsplit(att[grep(att_of_interest, att)], <span class="string">" "</span>))[<span class="number">2</span>])</span><br><span class="line">  &#125;<span class="keyword">else</span>&#123;</span><br><span class="line">    <span class="keyword">return</span>(NA)&#125;</span><br><span class="line">&#125;</span><br></pre></td></tr></table></figure>
<p>举例子解释,加入我们需要gene_id</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">gtf_attributes &lt;- genes[1,9]</span><br><span class="line">att &lt;- strsplit(gtf_attributes,<span class="string">"; "</span>)</span><br></pre></td></tr></table></figure>
<p>这一步提示non-character argument，后面我查询才知道, 对于第九列的第一行的的获取两种方式返回的格式不一样</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">genes[1,9] <span class="comment">#data.frame</span></span><br><span class="line">genes<span class="variable">$attributes</span>[1] <span class="comment">#character</span></span><br></pre></td></tr></table></figure>
<p>而strsplit的对象是character,重来</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">gtf_attributes &lt;- genes<span class="variable">$attributes</span>[1]</span><br><span class="line">att &lt;- strsplit(gtf_attributes,<span class="string">"; "</span>)</span><br><span class="line">att &lt;- gsub(<span class="string">"\""</span>,<span class="string">""</span>,unlist(att))</span><br></pre></td></tr></table></figure>
<p>grep获取位置，获取到内容后分隔,得到的第二个元素是我们需要的</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">unlist(strsplit(att[grep(<span class="string">"gene_id"</span>, att)], <span class="string">" "</span>))[2]</span><br></pre></td></tr></table></figure>
<p>利用函数获得基因列表</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">genes<span class="variable">$gene_id</span> &lt;- unlist(lapply(genes<span class="variable">$attributes</span>, extract_attributes, <span class="string">"gene_id"</span>))</span><br></pre></td></tr></table></figure>
<p>如果我们选择使用第一种方法rtracklayer::import，这些事情都是不需要做的！！！</p>
<h4 id="总结：rtracklayer-import最简单，fread最快，read-table可选，read-table没戏！"><a href="#总结：rtracklayer-import最简单，fread最快，read-table可选，read-table没戏！" class="headerlink" title="总结：rtracklayer::import最简单，fread最快，read.table可选，read_table没戏！"></a>总结：rtracklayer::import最简单，fread最快，read.table可选，read_table没戏！</h4><p>作者：果子学生信</p>
<p>链接：<a href="https://www.jianshu.com/p/7f96adec2b0d" target="_blank" rel="noopener">https://www.jianshu.com/p/7f96adec2b0d</a></p>
<p>来源：简书</p>
<p>著作权归作者所有。商业转载请联系作者获得授权，非商业转载请注明出处。</p>

          
          <hr>
          <ul class="pager no-print">
              
              <li class="previous">
                  <a href="/hexo/2017/11/10/2017-11-10_genomic_live_streaming/" data-toggle="tooltip" data-placement="left" title="Jimmy 直播基因组学习笔记">&larr; Previous Post</a>
              </li>
              
              
              <li class="next">
                  <a href="/hexo/2017/11/09/2017-11-09_wes/" data-toggle="tooltip" data-placement="top" title="外显子测序数据分析学习笔记">Next Post&rarr;</a>
              </li>
              
          </ul>
          
  <br>

  


      </div>
      
  <div class="hidden-xs col-sm-3 toc-col">
    <div class="toc-wrap">
        Table of Contents
        
          <ol class="toc"><li class="toc-item toc-level-2"><a class="toc-link" href="#在R语言中读取GTF文件的最好方法"><span class="toc-number">1.</span> <span class="toc-text">在R语言中读取GTF文件的最好方法</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#第一种方法rtracklayer-import"><span class="toc-number">1.0.1.</span> <span class="toc-text">第一种方法rtracklayer::import</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#第二种方法read-table"><span class="toc-number">1.0.2.</span> <span class="toc-text">第二种方法read.table</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#第三种方法readr包里面的read-table2"><span class="toc-number">1.0.3.</span> <span class="toc-text">第三种方法readr包里面的read_table2</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#第四种方法read-table，fread"><span class="toc-number">1.0.4.</span> <span class="toc-text">第四种方法read.table，fread</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#总结：rtracklayer-import最简单，fread最快，read-table可选，read-table没戏！"><span class="toc-number">1.0.5.</span> <span class="toc-text">总结：rtracklayer::import最简单，fread最快，read.table可选，read_table没戏！</span></a></li></ol></li></ol>
        
    </div>
  </div>


    </div>
</article>

<!-- Footer -->
<!-- footer.ejs -->
<footer class="no-print">

    <div class="text-center">

        <ul class="list-inline">
            
            
            

            

            

            

            

            

        </ul>
        <div class="text-muted copyright">
            &copy;
            
            2017 - 2019
            
            -
            xt
            <br>
            
            
            
            | 
            <p>
                <span id="busuanzi_container_site_pv"><span id="busuanzi_value_site_pv"></span> <b>PV</b></span> -
                <span id="busuanzi_container_site_uv"><span id="busuanzi_value_site_uv"></span> <b>UV</b></span> -
                121.9k <b>Words</b>
        </p></div>

        <div class="search-container">
            <form class="site-search-form">
                <span class="glyphicon glyphicon-search"></span>
                <input type="text" id="local-search-input" class="st-search-input" placeholder="Search...">
            </form>
            <div id="local-search-result" class="local-search-result-cls"></div>
        </div>
        <script type="text/javascript" id="local.search.active">
            var inputArea = document.querySelector("#local-search-input");
            inputArea.onclick = function () {
                var root = "/hexo/";
                getSearchFile(root);
                this.onclick = null
            }
            inputArea.onkeydown = function () {
                if (event.keyCode == 13) return false
            }
        </script>

    </div>
</footer>

<!-- Custom Theme JavaScript -->
<script src="/hexo/js/main.js"></script>

<!-- async load function -->
<script>
    function async(u, c) {
        var d = document, t = 'script',
            o = d.createElement(t),
            s = d.getElementsByTagName(t)[0];
        o.src = u;
        if (c) {
            o.addEventListener('load', function (e) {
                c(null, e);
            }, false);
        }
        s.parentNode.insertBefore(o, s);
    }
</script>

<!--fastClick.js -->
<script>
    async("//cdn.bootcss.com/fastclick/1.0.6/fastclick.min.js", function () {
        var $nav = document.querySelector("nav");
        if ($nav) FastClick.attach($nav);
    })
</script><!-- hexo-inject:begin --><!-- hexo-inject:end -->



</body>

</html>
