<!DOCTYPE html>
<html lang="en">

<!-- layout.ejs-->
<head><meta name="generator" content="Hexo 3.8.0">
    <!-- hexo-inject:begin --><!-- hexo-inject:end --><meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="Adelaide">
    <meta name="author" content="xt">
    <meta name="keyword" content="">
    <link rel="canonical" href="https://taoshengxu.github.io/hexo/hexo/2017/11/04/2017-11-04_geo/">
    <link rel="shortcut icon" href="/hexo/img/favicon.png">
    <link rel="alternate" type="application/atom+xml" title="IIM" href="/atom.xml">
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/animate.css/3.5.2/animate.min.css">
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
    <link rel="stylesheet" href="https://ajax.googleapis.com/ajax/libs/jqueryui/1.12.1/themes/smoothness/jquery-ui.css">
    <script src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.12.1/jquery-ui.min.js"></script>
    <script src="/hexo/js/search.js"></script>

    <title>
        
        GEO数据｜undefined
        
    </title>

    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">

    <link rel="stylesheet" href="/hexo/css/main.css">

    
      <link href="//fonts.googleapis.com/css?family=Source+Code+Pro" rel="stylesheet" type="text/css">
      <link rel="stylesheet" href="/hexo/css/highlight.css">
    

    

    <script async src="//dn-lbstatics.qbox.me/busuanzi/2.3/busuanzi.pure.mini.js"></script>


    

    

    


    
  


    




    <script async defer src="https://buttons.github.io/buttons.js"></script>

<link rel="stylesheet" href="/hexo/css/prism-solarizedlight.css" type="text/css">
<link rel="stylesheet" href="/hexo/css/prism-line-numbers.css" type="text/css"><!-- hexo-inject:begin --><!-- hexo-inject:end --></head>

<style>
    header.intro-header {
        background-image: url('/hexo/img/northernlights-sisimiut-lake.jpg')
    }
</style>
<!-- hack iOS CSS :active style -->
<body ontouchstart="" class="animated fadeIn">
<!-- hexo-inject:begin --><!-- hexo-inject:end --><header>
  <nav class="navbar navbar-default header-navbar" id="nav-top" data-ispost="true" data-istags="false" data-ishome="false">
    <div class="container-fluid">
      <div class="navbar-header page-scroll">
        <button type="button" class="navbar-toggle" data-toggle="collapse" aria-expanded="false" data-target="#website_navbar">
          <span class="sr-only">Toggle navigation</span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <span class="navbar-brand animated pulse">
          <a class="brand-logo" href="/hexo/">
                <img src="/hexo/img/banner.jpg?h=350&amp;auto=compress&amp;cs=tinysrgb">
          </a>
        </span>
      </div>

      <div class="collapse navbar-collapse" id="website_navbar">
          <ul class="nav navbar-nav navbar-right">
              
                <li>
                  <a href="/hexo/">home</a>
                </li>
              
                <li>
                  <a href="/hexo/archives/">archives</a>
                </li>
              
                <li>
                  <a href="/hexo/categories/">categories</a>
                </li>
              
                <li>
                  <a href="/hexo/tags/">tags</a>
                </li>
              
                <li>
                  <a href="/hexo/columns/">columns</a>
                </li>
              
          </ul>
      </div>
  </div></nav>


  
    <style>
       .intro-header {
          background-image: url('/hexo/img/northernlights-sisimiut-lake.jpg');
      }
    </style>

    <div class="intro-header">
        <div class="container">
            <div class="row">
                <div class="col-lg-8 col-lg-offset-2 col-md-10 col-md-offset-1 text-center">
                    <div class="site-heading">
                        <h1>GEO数据</h1>
                        
                        

                        
                          <span class="meta">
                               <span class="meta-item">Author: xt</span>
                               <span class="meta-item">Date: Nov 4, 2017</span>
                               
                          </span>
                          <div class="tags text-center">
                              Categories: 
                          </div>
                          <div class="tags text-center">
                              Tags: 
                          </div>
                        
                    </div>
                </div>
            </div>
        </div>
    </div>
  
</header>


<!-- Main Content -->
<!-- post.ejs -->
<article>
    <div class="container">
      <div class="col-lg-8 col-lg-offset-1 col-sm-9">
          
          <span class="post-count">984 words in total, 5 minutes required.</span>
          <hr>
          
          <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/hint.css/2.4.1/hint.min.css"><h3 id="GEO-基础"><a href="#GEO-基础" class="headerlink" title="GEO 基础"></a>GEO 基础</h3><ul>
<li>GEO Platform (GPL) 芯片平台</li>
<li>GEO Sample (GSM) 样本ID号</li>
<li>GEO Series (GSE) study的ID号</li>
<li>GEO Dataset (GDS) 数据集的ID号 ## 用法</li>
</ul>
<h3 id="数据搜索"><a href="#数据搜索" class="headerlink" title="数据搜索"></a>数据搜索</h3><ul>
<li><strong>方法一</strong></li>
</ul>
<ol>
<li>在<a href="https://www.ncbi.nlm.nih.gov/" target="_blank" rel="noopener">https://www.ncbi.nlm.nih.gov/</a> 中搜索 GSE81916 选择 BioProject查询 Accession：PRJNA323422; GEO: GSE81916</li>
<li><a href="https://trace.ncbi.nlm.nih.gov/Traces/study/?acc=PRJNA323422" target="_blank" rel="noopener">https://trace.ncbi.nlm.nih.gov/Traces/study/?acc=PRJNA323422</a> 可以查询数据具体信息</li>
</ol>
<ul>
<li><strong>方法二</strong></li>
</ul>
<ol>
<li>在 Gene Expression Omnibus (<a href="https://www.ncbi.nlm.nih.gov/geo/)搜素GSE81916进入" target="_blank" rel="noopener">https://www.ncbi.nlm.nih.gov/geo/)搜素GSE81916进入</a></li>
<li><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE81916" target="_blank" rel="noopener">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE81916</a> 数据地址</li>
</ol>
<h3 id="数据下载"><a href="#数据下载" class="headerlink" title="数据下载"></a>数据下载</h3><h4 id="原始GSE数据，以GSE114727为例"><a href="#原始GSE数据，以GSE114727为例" class="headerlink" title="原始GSE数据，以GSE114727为例"></a>原始GSE数据，以GSE114727为例</h4><p>ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE114nnn/GSE114727/suppl/GSE114727_RAW.tar<br>用迅雷下载即可</p>
<hr>
<p><strong>ftp地址</strong><br>ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByStudy/sra/SRP/SRP075/SRP075747</p>
<p>可以分为以下几个部分</p>
<ul>
<li>所有SRA数据的共同部分： ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant</li>
<li>reads表示存放reads数据，在FTP可以看到另一个选项是analysis，表示分析结果</li>
<li>ByStudy表示根据Study进行分类，其他还可以根据实验ByExp,根据Run,ByRun.</li>
<li>sra/SRP/SRP075/SRP075747: 后面部分都是为了便于检索。</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#/bin/bash</span></span><br><span class="line"><span class="comment"># @author: xt</span></span><br><span class="line"><span class="comment"># @date: 2017-11-04</span></span><br><span class="line"></span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> ` seq 56 62`;</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">axel ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByStudy/sra/SRP/SRP075/SRP075747/SRR35899<span class="variable">$&#123;i&#125;</span>/SRR35899<span class="variable">$&#123;i&#125;</span>.sra</span><br><span class="line"><span class="comment">#wget ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByStudy/sra/SRP/SRP075/SRP075747/SRR35899$&#123;i&#125;/SRR35899$&#123;i&#125;.sra</span></span><br><span class="line"><span class="comment">#echo $i </span></span><br><span class="line"><span class="keyword">done</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># https://www.ncbi.nlm.nih.gov/gquery/?term=GSE81916</span></span><br><span class="line"><span class="comment"># esearch -db sra -query PRJNA299273  | efetch -format runinfo &gt; runinfo.txt # 这个命令是把所有的结果放到一个文件里，也可以通过 https://trace.ncbi.nlm.nih.gov/Traces/study/?acc=PRJNA323422下载SRR的编号</span></span><br><span class="line"><span class="comment"># cat runinfo.txt | cut -f 1 -d ',' | grep SRR &gt; sra.ids</span></span><br><span class="line"><span class="comment"># ~/biosoft/sratoolkit.2.8.2-1-centos_linux64/bin/prefetch --option-file sra.ids # 数据存在/home/shenmy/ncbi/public/sra这个文件下面，找了半天</span></span><br><span class="line">mkdir /mnt/d/rna_seq/data  &amp;&amp; <span class="built_in">cd</span> /mnt/d/rna_seq/data</span><br><span class="line">perl -lne <span class="string">'$id=substr($_,0,6);print "axel ftp://ftp-trace.ncbi.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/$id/$_/$_.sra"'</span> SRR_Acc_List.txt &gt;sra_down.sh</span><br><span class="line">bash sra_down.sh</span><br><span class="line"><span class="comment"># 改成用axel下是因为prefetch下载总是不成功</span></span><br><span class="line">ls *.sra|<span class="keyword">while</span> <span class="built_in">read</span> id;<span class="keyword">do</span>(/mnt/d/Software/Biosoft/sratoolkit/sratoolkit.2.8.2-1-ubuntu64/bin/fastq-dump --split-3 <span class="variable">$id</span>);<span class="keyword">done</span></span><br><span class="line">rm *.sra</span><br><span class="line">chmod u-w *</span><br></pre></td></tr></table></figure>
<p>======================================</p>
<p>2019.12.25 重新学习B站</p>
<p><a href="https://www.bilibili.com/video/av26731585" target="_blank" rel="noopener">https://www.bilibili.com/video/av26731585</a><br><a href="https://github.com/jmzeng1314/GEO" target="_blank" rel="noopener">https://github.com/jmzeng1314/GEO</a></p>
<p>===================================================</p>
<h3 id="GEO中国镜像-2019-12-26"><a href="#GEO中国镜像-2019-12-26" class="headerlink" title="GEO中国镜像 2019.12.26"></a>GEO中国镜像 2019.12.26</h3><p><a href="https://github.com/jmzeng1314/GEOmirror" target="_blank" rel="noopener">https://github.com/jmzeng1314/GEOmirror</a></p>
<p><a href="https://taoshengxu.github.io/DocumentGit/html/GEO数据库中国区镜像横空出世.html">GEO数据库中国区镜像横空出世</a></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br></pre></td><td class="code"><pre><span class="line">BiocManager::install(<span class="string">"GEOquery"</span>)</span><br><span class="line"><span class="keyword">library</span>(devtools)</span><br><span class="line">install_github(<span class="string">"jmzeng1314/GEOmirror"</span>)</span><br><span class="line"><span class="keyword">library</span>(GEOmirror)</span><br><span class="line"><span class="keyword">library</span>(GEOquery)</span><br><span class="line"><span class="keyword">library</span>(GEOmirror)</span><br><span class="line"><span class="comment">#eSet=geoChina('GSE1009') </span></span><br><span class="line"><span class="comment">#eSet=geoChina('GSE27533') </span></span><br><span class="line">eSet=geoChina(<span class="string">'GSE95166'</span>) </span><br><span class="line"></span><br><span class="line"><span class="comment">## download GSE95166 data</span></span><br><span class="line"><span class="comment"># https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE95166</span></span><br><span class="line"><span class="comment">#eSet=getGEO('GSE95166', destdir=".", AnnotGPL = F, getGPL = F)[[1]]</span></span><br><span class="line">eSet</span><br><span class="line">eSet=eSet[[<span class="number">1</span>]]</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">probes_expr &lt;- exprs(eSet);dim(probes_expr)</span><br><span class="line">head(probes_expr[,<span class="number">1</span>:<span class="number">4</span>])</span><br><span class="line">boxplot(probes_expr,las=<span class="number">2</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment">## pheno info</span></span><br><span class="line">phenoDat &lt;- pData(eSet)</span><br><span class="line">head(phenoDat[,<span class="number">1</span>:<span class="number">4</span>])</span><br><span class="line"><span class="comment"># https://www.ncbi.nlm.nih.gov/pubmed/31430288</span></span><br><span class="line"></span><br><span class="line">groupList=factor(c(rep(<span class="string">'npc'</span>,<span class="number">4</span>),rep(<span class="string">'normal'</span>,<span class="number">4</span>)))</span><br><span class="line">table(groupList)</span><br><span class="line">eSet@annotation</span><br><span class="line"><span class="comment"># GPL15314    Arraystar Human LncRNA microarray V2.0 (Agilent_033010 Probe Name version)</span></span><br><span class="line"></span><br><span class="line"><span class="comment">### PCA+limma</span></span><br><span class="line">genes_expr=probes_expr</span><br><span class="line"><span class="keyword">library</span>(<span class="string">"FactoMineR"</span>)</span><br><span class="line"><span class="keyword">library</span>(<span class="string">"factoextra"</span>)</span><br><span class="line">dat.pca &lt;- PCA(t(genes_expr) , graph = <span class="literal">FALSE</span>)</span><br><span class="line">dat.pca</span><br><span class="line">fviz_pca_ind(dat.pca,</span><br><span class="line">             geom.ind = <span class="string">"point"</span>,</span><br><span class="line">             col.ind = groupList,</span><br><span class="line">             addEllipses = <span class="literal">TRUE</span>,</span><br><span class="line">             legend.title = <span class="string">"Groups"</span></span><br><span class="line">)</span><br><span class="line"><span class="keyword">library</span>(limma)</span><br><span class="line">design=model.matrix(~factor(groupList))</span><br><span class="line">design</span><br><span class="line">fit=lmFit(genes_expr,design)</span><br><span class="line">fit=eBayes(fit)</span><br><span class="line">DEG=topTable(fit,coef=<span class="number">2</span>,n=<span class="literal">Inf</span>)</span><br><span class="line">head(DEG)</span><br><span class="line"><span class="comment"># We observed that 2107 lncRNAs were upregulated</span></span><br><span class="line"><span class="comment"># while 2090 lncRNAs were downregulated by more than 2-fold,</span></span><br><span class="line"><span class="comment"># NKILA among these downregulated lncRNAs (Fig 1A, GSE95166).</span></span><br><span class="line"></span><br><span class="line"><span class="comment">## for volcano plot</span></span><br><span class="line">df=DEG</span><br><span class="line"><span class="keyword">attach</span>(df)</span><br><span class="line">df$v= -log10(P.Value)</span><br><span class="line">df$g=ifelse(df$P.Value&gt;<span class="number">0.05</span>,<span class="string">'stable'</span>,</span><br><span class="line">            ifelse( df$logFC &gt;<span class="number">1</span>,<span class="string">'up'</span>,</span><br><span class="line">                    ifelse( df$logFC &lt; -<span class="number">1</span>,<span class="string">'down'</span>,<span class="string">'stable'</span>) )</span><br><span class="line">)</span><br><span class="line">table(df$g)</span><br><span class="line">df$name=rownames(df)</span><br><span class="line">head(df)</span><br><span class="line"><span class="keyword">library</span>(ggpubr)</span><br><span class="line">ggpubr::ggscatter(df, x = <span class="string">"logFC"</span>, y = <span class="string">"v"</span>, color = <span class="string">"g"</span>,size = <span class="number">0.5</span>,</span><br><span class="line">          label = <span class="string">"name"</span>, repel = <span class="literal">T</span>,</span><br><span class="line">          label.select =head(rownames(df)),</span><br><span class="line">          palette = c(<span class="string">"#00AFBB"</span>, <span class="string">"#E7B800"</span>, <span class="string">"#FC4E07"</span>) )</span><br><span class="line"><span class="keyword">detach</span>(df)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">x=DEG$logFC</span><br><span class="line">names(x)=rownames(DEG)</span><br><span class="line">cg=c(names(head(sort(x),<span class="number">100</span>)),</span><br><span class="line">     names(tail(sort(x),<span class="number">100</span>)))</span><br><span class="line">cg</span><br><span class="line"><span class="keyword">library</span>(pheatmap)</span><br><span class="line">n=t(scale(t(genes_expr[cg,])))</span><br><span class="line">n[n&gt;<span class="number">2</span>]=<span class="number">2</span></span><br><span class="line">n[n&lt; -<span class="number">2</span>]= -<span class="number">2</span></span><br><span class="line">n[<span class="number">1</span>:<span class="number">4</span>,<span class="number">1</span>:<span class="number">4</span>]</span><br><span class="line">ac=data.frame(groupList=groupList)</span><br><span class="line">rownames(ac)=colnames(n)  </span><br><span class="line">pheatmap(n,show_colnames =<span class="literal">F</span>,show_rownames = <span class="literal">F</span>,</span><br><span class="line">         annotation_col=ac)</span><br></pre></td></tr></table></figure>

          
          <hr>
          <ul class="pager no-print">
              
              <li class="previous">
                  <a href="/hexo/2017/11/05/2017-11-05_nat123/" data-toggle="tooltip" data-placement="left" title="nat123">&larr; Previous Post</a>
              </li>
              
              
              <li class="next">
                  <a href="/hexo/2017/11/03/2017-11-03_1000software/" data-toggle="tooltip" data-placement="top" title="生物信息学常见1000个软件的安装代码">Next Post&rarr;</a>
              </li>
              
          </ul>
          
  <br>

  


      </div>
      
  <div class="hidden-xs col-sm-3 toc-col">
    <div class="toc-wrap">
        Table of Contents
        
          <ol class="toc"><li class="toc-item toc-level-3"><a class="toc-link" href="#GEO-基础"><span class="toc-number">1.</span> <span class="toc-text">GEO 基础</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#数据搜索"><span class="toc-number">2.</span> <span class="toc-text">数据搜索</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#数据下载"><span class="toc-number">3.</span> <span class="toc-text">数据下载</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#原始GSE数据，以GSE114727为例"><span class="toc-number">3.1.</span> <span class="toc-text">原始GSE数据，以GSE114727为例</span></a></li></ol></li><li class="toc-item toc-level-3"><a class="toc-link" href="#GEO中国镜像-2019-12-26"><span class="toc-number">4.</span> <span class="toc-text">GEO中国镜像 2019.12.26</span></a></li></ol>
        
    </div>
  </div>


    </div>
</article>

<!-- Footer -->
<!-- footer.ejs -->
<footer class="no-print">

    <div class="text-center">

        <ul class="list-inline">
            
            
            

            

            

            

            

            

        </ul>
        <div class="text-muted copyright">
            &copy;
            
            2017 - 2019
            
            -
            xt
            <br>
            
            
            
            | 
            <p>
                <span id="busuanzi_container_site_pv"><span id="busuanzi_value_site_pv"></span> <b>PV</b></span> -
                <span id="busuanzi_container_site_uv"><span id="busuanzi_value_site_uv"></span> <b>UV</b></span> -
                123.4k <b>Words</b>
        </p></div>

        <div class="search-container">
            <form class="site-search-form">
                <span class="glyphicon glyphicon-search"></span>
                <input type="text" id="local-search-input" class="st-search-input" placeholder="Search...">
            </form>
            <div id="local-search-result" class="local-search-result-cls"></div>
        </div>
        <script type="text/javascript" id="local.search.active">
            var inputArea = document.querySelector("#local-search-input");
            inputArea.onclick = function () {
                var root = "/hexo/";
                getSearchFile(root);
                this.onclick = null
            }
            inputArea.onkeydown = function () {
                if (event.keyCode == 13) return false
            }
        </script>

    </div>
</footer>

<!-- Custom Theme JavaScript -->
<script src="/hexo/js/main.js"></script>

<!-- async load function -->
<script>
    function async(u, c) {
        var d = document, t = 'script',
            o = d.createElement(t),
            s = d.getElementsByTagName(t)[0];
        o.src = u;
        if (c) {
            o.addEventListener('load', function (e) {
                c(null, e);
            }, false);
        }
        s.parentNode.insertBefore(o, s);
    }
</script>

<!--fastClick.js -->
<script>
    async("//cdn.bootcss.com/fastclick/1.0.6/fastclick.min.js", function () {
        var $nav = document.querySelector("nav");
        if ($nav) FastClick.attach($nav);
    })
</script><!-- hexo-inject:begin --><!-- hexo-inject:end -->



</body>

</html>
