<!DOCTYPE html>
<html lang="en">

<!-- layout.ejs-->
<head><meta name="generator" content="Hexo 3.8.0">
    <!-- hexo-inject:begin --><!-- hexo-inject:end --><meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="Adelaide">
    <meta name="author" content="xt">
    <meta name="keyword" content="">
    <link rel="canonical" href="https://taoshengxu.github.io/hexo/hexo/2020/05/21/2020-05-20_singlecell_package/">
    <link rel="shortcut icon" href="/hexo/img/favicon.png">
    <link rel="alternate" type="application/atom+xml" title="IIM" href="/atom.xml">
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/animate.css/3.5.2/animate.min.css">
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
    <link rel="stylesheet" href="https://ajax.googleapis.com/ajax/libs/jqueryui/1.12.1/themes/smoothness/jquery-ui.css">
    <script src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.12.1/jquery-ui.min.js"></script>
    <script src="/hexo/js/search.js"></script>

    <title>
        
        Singlecell-scRNASeq 分析包｜undefined
        
    </title>

    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">

    <link rel="stylesheet" href="/hexo/css/main.css">

    
      <link href="//fonts.googleapis.com/css?family=Source+Code+Pro" rel="stylesheet" type="text/css">
      <link rel="stylesheet" href="/hexo/css/highlight.css">
    

    

    <script async src="//dn-lbstatics.qbox.me/busuanzi/2.3/busuanzi.pure.mini.js"></script>


    

    

    


    
  


    




    <script async defer src="https://buttons.github.io/buttons.js"></script>

<link rel="stylesheet" href="/hexo/css/prism-solarizedlight.css" type="text/css">
<link rel="stylesheet" href="/hexo/css/prism-line-numbers.css" type="text/css"><!-- hexo-inject:begin --><!-- hexo-inject:end --></head>

<style>
    header.intro-header {
        background-image: url('/hexo/img/northernlights-sisimiut-lake.jpg')
    }
</style>
<!-- hack iOS CSS :active style -->
<body ontouchstart="" class="animated fadeIn">
<!-- hexo-inject:begin --><!-- hexo-inject:end --><header>
  <nav class="navbar navbar-default header-navbar" id="nav-top" data-ispost="true" data-istags="false" data-ishome="false">
    <div class="container-fluid">
      <div class="navbar-header page-scroll">
        <button type="button" class="navbar-toggle" data-toggle="collapse" aria-expanded="false" data-target="#website_navbar">
          <span class="sr-only">Toggle navigation</span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <span class="navbar-brand animated pulse">
          <a class="brand-logo" href="/hexo/">
                <img src="/hexo/img/banner.jpg?h=350&amp;auto=compress&amp;cs=tinysrgb">
          </a>
        </span>
      </div>

      <div class="collapse navbar-collapse" id="website_navbar">
          <ul class="nav navbar-nav navbar-right">
              
                <li>
                  <a href="/hexo/">home</a>
                </li>
              
                <li>
                  <a href="/hexo/archives/">archives</a>
                </li>
              
                <li>
                  <a href="/hexo/categories/">categories</a>
                </li>
              
                <li>
                  <a href="/hexo/tags/">tags</a>
                </li>
              
                <li>
                  <a href="/hexo/columns/">columns</a>
                </li>
              
          </ul>
      </div>
  </div></nav>


  
    <style>
       .intro-header {
          background-image: url('/hexo/post_cover_images/moon.jpg');
      }
    </style>

    <div class="intro-header">
        <div class="container">
            <div class="row">
                <div class="col-lg-8 col-lg-offset-2 col-md-10 col-md-offset-1 text-center">
                    <div class="site-heading">
                        <h1>Singlecell-scRNASeq 分析包</h1>
                        
                        

                        
                          <span class="meta">
                               <span class="meta-item">Author: XT</span>
                               <span class="meta-item">Date: May 21, 2020</span>
                               
                          </span>
                          <div class="tags text-center">
                              Categories: 
                              <a class="tag" href="/hexo/categories/#Bioinformatics" title="Bioinformatics">Bioinformatics</a>
                              
                          </div>
                          <div class="tags text-center">
                              Tags: 
                          </div>
                        
                    </div>
                </div>
            </div>
        </div>
    </div>
  
</header>


<!-- Main Content -->
<!-- post.ejs -->
<article>
    <div class="container">
      <div class="col-lg-8 col-lg-offset-1 col-sm-9">
          
          <span class="post-count">1,117 words in total, 5 minutes required.</span>
          <hr>
          
          <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/hint.css/2.4.1/hint.min.css"><h1 id="1-Seurat-和-Monocle-对象slots思维导图"><a href="#1-Seurat-和-Monocle-对象slots思维导图" class="headerlink" title="1. Seurat 和 Monocle 对象slots思维导图"></a>1. Seurat 和 Monocle 对象slots思维导图</h1><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">library</span>(mindr)</span><br><span class="line">out &lt;- capture.output(str(du145_egf))</span><br><span class="line"></span><br><span class="line">aa=gsub(<span class="string">"\\.\\.\\$"</span>,<span class="string">"#"</span>,out)</span><br><span class="line">aa=gsub(<span class="string">"\\.\\.@"</span>,<span class="string">"# "</span>,gsub(<span class="string">"\\.\\. "</span>,<span class="string">"#"</span>,aa))</span><br><span class="line">out2&lt;- paste(aa, collapse=<span class="string">"\n"</span>)</span><br><span class="line">mm(out2,type =<span class="string">"text"</span>,root= <span class="string">"Seurat"</span>)</span><br></pre></td></tr></table></figure>
<h1 id="2-scatter-包学习"><a href="#2-scatter-包学习" class="headerlink" title="2. scatter 包学习"></a>2. scatter 包学习</h1><blockquote>
<p>这是一个单细胞数据预处理包</p>
</blockquote>
<p><a href="https://www.jianshu.com/p/4c20a7866ce6" target="_blank" rel="noopener">https://www.jianshu.com/p/4c20a7866ce6</a></p>
<p>scater包可以做下面几件事：<br>（1）QC<br>（2）从read数据里进行转录本定量（根据伪时间排序）<br>（3）数据格式标准化<br>（4）可视化<br>（5）Seamless integration into the Bioconductor universe<br>（6）标准化数据</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># s4 对象操作</span></span><br><span class="line">assay(fluidigm)  &lt;- assays(fluidigm)$rsem_counts</span><br><span class="line">pheno_data &lt;- as.data.frame(colData(fluidigm))</span><br><span class="line"></span><br><span class="line">str(sce) 展示sce对象所有的信息</span><br><span class="line">str(sce,max.level=<span class="number">2</span>)</span><br><span class="line">exprs(sce) </span><br><span class="line">genesNames= rownames(rowData(sce))</span><br><span class="line">assays(sce)$counts 查看原始counts值</span><br></pre></td></tr></table></figure>
<h1 id="3-Seurat-包学习"><a href="#3-Seurat-包学习" class="headerlink" title="3. Seurat 包学习"></a>3. Seurat 包学习</h1><p><a href="https://satijalab.org/seurat/v3.1/pbmc3k_tutorial.html" target="_blank" rel="noopener">https://satijalab.org/seurat/v3.1/pbmc3k_tutorial.html</a></p>
<p><a href="https://www.jianshu.com/p/be12d25d2593" target="_blank" rel="noopener">https://www.jianshu.com/p/be12d25d2593</a></p>
<p><a href="https://www.plob.org/article/11101.html" target="_blank" rel="noopener">https://www.plob.org/article/11101.html</a></p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h1 id="4-Monocle包学习"><a href="#4-Monocle包学习" class="headerlink" title="4. Monocle包学习"></a>4. Monocle包学习</h1><p><a href="http://cole-trapnell-lab.github.io/monocle-release/" target="_blank" rel="noopener">http://cole-trapnell-lab.github.io/monocle-release/</a> [官网]</p>
<p><a href="https://cole-trapnell-lab.github.io/monocle3/" target="_blank" rel="noopener">https://cole-trapnell-lab.github.io/monocle3/</a></p>
<p><a href="https://www.jianshu.com/p/8f57ac63f399" target="_blank" rel="noopener">https://www.jianshu.com/p/8f57ac63f399</a></p>
<p><a href="https://www.jianshu.com/p/66c387e1de3d" target="_blank" rel="noopener">https://www.jianshu.com/p/66c387e1de3d</a></p>
<h2 id="4-1-monocle-2"><a href="#4-1-monocle-2" class="headerlink" title="4.1 monocle 2"></a>4.1 monocle 2</h2><p>使用对象 CellDataSet类,继承ExpressionSet总结如下：</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">## 取表达矩阵</span></span><br><span class="line">exprs(cds)</span><br><span class="line"><span class="comment">## 取注释信息</span></span><br><span class="line">pData(cds)</span><br><span class="line">fData(cds)</span><br><span class="line">colnames 取矩阵的列名</span><br></pre></td></tr></table></figure>
<h2 id="4-2-monocle-3"><a href="#4-2-monocle-3" class="headerlink" title="4.2 monocle 3"></a>4.2 monocle 3</h2><p>使用对象 cell_data_set,继承SingleCellExperiment  总结如下：</p>
<blockquote>
<p>@meta.data一般会存放表型相关的信息如cluster、sample的来源、group等；<br>@assay中的@counts存放单细胞测序的raw data，data储存了归一化后的数据，而scale.data存储了标准化之后的数据（一般是z-score）。课程里使用的是归一化后的（实际上我认为需要使用最原始的，也就是counts里的数据，理由见下面的分析。但是由于为了跟课程一致，我这里用了和课程一样的归一化后的数据），需要注意的是，不要用scale.data，也就是不要用标准化之后的data来构建monocle对象，<br><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">## 直接输出 cell_data_set 对象的信息,这样打印可以能看到assay里面的对象是counts，所以后面可以用counts取出表达矩阵</span></span><br><span class="line">cds</span><br><span class="line"><span class="comment">## 查看slots信息</span></span><br><span class="line">str(cds)</span><br><span class="line"><span class="comment">#对象可以通过@取slot的内容</span></span><br><span class="line">cds@colData</span><br><span class="line"><span class="comment">## 取表达矩阵</span></span><br><span class="line">SingleCellExperiment::counts(cds) </span><br><span class="line">exprs(cds)</span><br><span class="line">counts(cds)</span><br><span class="line">logcounts(cds)</span><br><span class="line"></span><br><span class="line"><span class="comment">## 取注释信息</span></span><br><span class="line">colData(cds)</span><br><span class="line">rowData(cds)</span><br><span class="line">cds@colData</span><br><span class="line"><span class="comment">## 取轨迹的伪时间值</span></span><br><span class="line">pseudotime=cds@principal_graph_aux[[UMAP]]$pseudotime</span><br><span class="line"></span><br><span class="line"><span class="comment">## 取降维度之后的数据</span></span><br><span class="line">cc=reducedDims(cds_stimu)</span><br><span class="line"></span><br><span class="line"><span class="comment">## 把对象当成SingleCellExperiment  对象，调用其他包对数据进行归一化</span></span><br><span class="line">cds_stimu_1 &lt;- scran::computeSumFactors(cds_stimu)</span><br><span class="line"><span class="comment">#最常用的函数是logNormCounts()，结果储存在"logcounts"里。它是用每一个count除以自身的size factor，加一个 pseudo-count 和log-transforming。</span></span><br><span class="line">cds_stimu_1 &lt;- scater::logNormCounts(cds_stimu_1)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment"># 函数estimateSizeFactors 估计统计数据的大小因子</span></span><br><span class="line">cds = estimateSizeFactors( cds )</span><br><span class="line">sizeFactors( cds )</span><br><span class="line">如果统计数据的每列除以这列的大小因子， 这样统计值就变成同一规模， 使它们具有可比性。</span><br><span class="line">函数counts 可以做这个计算。</span><br><span class="line">head( counts( cds, normalized=<span class="literal">TRUE</span> ) )</span><br></pre></td></tr></table></figure></p>
</blockquote>
<h2 id="4-3-Monocle相关函数功能"><a href="#4-3-Monocle相关函数功能" class="headerlink" title="4.3 Monocle相关函数功能"></a>4.3 Monocle相关函数功能</h2><h4 id="preprocess-cds"><a href="#preprocess-cds" class="headerlink" title="preprocess_cds"></a><code>preprocess_cds</code></h4><blockquote>
<p>first normalizes the data by log and size factor to address depth differences, or by size factor only. Next,  calculates a lower dimensional space that will be used as the input for further dimensionality reduction like tSNE and UMAP. 首先用log归一化数据，然后计算PCA, num_dim参数取PCA的维度，计算PCA的目的是为了以后在PCA基础上继续 tSNE或者UMAP降维。</p>
</blockquote>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">cds &lt;- preprocess_cds(cds, num_dim = <span class="number">50</span>)</span><br></pre></td></tr></table></figure>
<h4 id="align-cds"><a href="#align-cds" class="headerlink" title="align_cds"></a><code>align_cds</code></h4><blockquote>
<p>重要的参数是alignment_group， residual_model_formula_str， 其中alignment_group是去除批次效应， 而residual_model_formula_str一般是背景噪声，这里是用lmFit 函数 对表达数据 关于背景参数的回归，然后表达数据减去这个数据。具体原理为什么这样，还是不清楚。</p>
</blockquote>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">cds &lt;- align_cds(cds, alignment_group = <span class="string">"batch"</span>, residual_model_formula_str = <span class="string">"~ bg.300.loading + bg.400.loading + bg.500.1.loading + bg.500.2.loading + bg.r17.loading + bg.b01.loading + bg.b02.loading"</span>)</span><br></pre></td></tr></table></figure>
<h4 id="cluster-cells"><a href="#cluster-cells" class="headerlink" title="cluster_cells"></a><code>cluster_cells</code></h4><blockquote>
<p>对cells做划分， 得到的结果是”cluster”和“partition“。 其中”cluster” 部分是无监督聚类（Louvain/Leiden communities）结果，<strong>然后针对”cluster” 得到 “partition“（一个聚合的更大的类）</strong>represent superclusters of the Louvain/Leiden communities that are found using a kNN pruning method。通过 clusters(), partitions() 函数可以取到对应的分类结果.结果也可以通过 cds$cluster</p>
</blockquote>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">cds &lt;- cluster_cells(cds)</span><br><span class="line">clusters(cds)</span><br><span class="line">partitions(cds) </span><br><span class="line">plot_cells(cds, color_cells_by = <span class="string">"cluster"</span>)</span><br><span class="line">plot_cells(cds, color_cells_by = <span class="string">"partition"</span>)</span><br></pre></td></tr></table></figure>
<h4 id="learn-graph"><a href="#learn-graph" class="headerlink" title="learn_graph"></a><code>learn_graph</code></h4><blockquote>
<p>学习进化轨迹函数</p>
</blockquote>
<figure class="highlight routeros"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">cds &lt;- learn_graph(cds)</span><br><span class="line">plot_cells(cds,</span><br><span class="line">           color_cells_by = <span class="string">"cell.type"</span>,</span><br><span class="line">           <span class="attribute">label_groups_by_cluster</span>=<span class="literal">FALSE</span>,</span><br><span class="line">           <span class="attribute">label_leaves</span>=<span class="literal">FALSE</span>,</span><br><span class="line">           <span class="attribute">label_branch_points</span>=<span class="literal">FALSE</span>)</span><br></pre></td></tr></table></figure>
<h4 id="aggregate-gene-expression"><a href="#aggregate-gene-expression" class="headerlink" title="aggregate_gene_expression"></a><code>aggregate_gene_expression</code></h4><blockquote>
<p>聚合表达 在这里是求 每个基因模块 每个 分组中的平均值。</p>
</blockquote>

          
          <hr>
          <ul class="pager no-print">
              
              <li class="previous">
                  <a href="/hexo/2020/05/21/2020-05-21_singlecell_analysis/" data-toggle="tooltip" data-placement="left" title="Single cell RNA 数据处理">&larr; Previous Post</a>
              </li>
              
              
              <li class="next">
                  <a href="/hexo/2020/05/17/2020-05-17_Breast_cellline/" data-toggle="tooltip" data-placement="top" title="乳腺癌细胞系">Next Post&rarr;</a>
              </li>
              
          </ul>
          

      </div>
      
  <div class="hidden-xs col-sm-3 toc-col">
    <div class="toc-wrap">
        Table of Contents
        
          <ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#1-Seurat-和-Monocle-对象slots思维导图"><span class="toc-number">1.</span> <span class="toc-text">1. Seurat 和 Monocle 对象slots思维导图</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#2-scatter-包学习"><span class="toc-number">2.</span> <span class="toc-text">2. scatter 包学习</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#3-Seurat-包学习"><span class="toc-number">3.</span> <span class="toc-text">3. Seurat 包学习</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#4-Monocle包学习"><span class="toc-number">4.</span> <span class="toc-text">4. Monocle包学习</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#4-1-monocle-2"><span class="toc-number">4.1.</span> <span class="toc-text">4.1 monocle 2</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#4-2-monocle-3"><span class="toc-number">4.2.</span> <span class="toc-text">4.2 monocle 3</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#4-3-Monocle相关函数功能"><span class="toc-number">4.3.</span> <span class="toc-text">4.3 Monocle相关函数功能</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#preprocess-cds"><span class="toc-number">4.3.0.1.</span> <span class="toc-text">preprocess_cds</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#align-cds"><span class="toc-number">4.3.0.2.</span> <span class="toc-text">align_cds</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#cluster-cells"><span class="toc-number">4.3.0.3.</span> <span class="toc-text">cluster_cells</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#learn-graph"><span class="toc-number">4.3.0.4.</span> <span class="toc-text">learn_graph</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#aggregate-gene-expression"><span class="toc-number">4.3.0.5.</span> <span class="toc-text">aggregate_gene_expression</span></a></li></ol></li></ol></li></ol>
        
    </div>
  </div>


    </div>
</article>

<!-- Footer -->
<!-- footer.ejs -->
<footer class="no-print">

    <div class="text-center">

        <ul class="list-inline">
            
            
            

            

            

            

            

            

        </ul>
        <div class="text-muted copyright">
            &copy;
            
            2017 - 2020
            
            -
            xt
            <br>
            
            
            
            | 
            <p>
                <span id="busuanzi_container_site_pv"><span id="busuanzi_value_site_pv"></span> <b>PV</b></span> -
                <span id="busuanzi_container_site_uv"><span id="busuanzi_value_site_uv"></span> <b>UV</b></span> -
                174.8k <b>Words</b>
        </p></div>

        <div class="search-container">
            <form class="site-search-form">
                <span class="glyphicon glyphicon-search"></span>
                <input type="text" id="local-search-input" class="st-search-input" placeholder="Search...">
            </form>
            <div id="local-search-result" class="local-search-result-cls"></div>
        </div>
        <script type="text/javascript" id="local.search.active">
            var inputArea = document.querySelector("#local-search-input");
            inputArea.onclick = function () {
                var root = "/hexo/";
                getSearchFile(root);
                this.onclick = null
            }
            inputArea.onkeydown = function () {
                if (event.keyCode == 13) return false
            }
        </script>

    </div>
</footer>

<!-- Custom Theme JavaScript -->
<script src="/hexo/js/main.js"></script>

<!-- async load function -->
<script>
    function async(u, c) {
        var d = document, t = 'script',
            o = d.createElement(t),
            s = d.getElementsByTagName(t)[0];
        o.src = u;
        if (c) {
            o.addEventListener('load', function (e) {
                c(null, e);
            }, false);
        }
        s.parentNode.insertBefore(o, s);
    }
</script>

<!--fastClick.js -->
<script>
    async("//cdn.bootcss.com/fastclick/1.0.6/fastclick.min.js", function () {
        var $nav = document.querySelector("nav");
        if ($nav) FastClick.attach($nav);
    })
</script><!-- hexo-inject:begin --><!-- hexo-inject:end -->



</body>

</html>
