<!DOCTYPE html>
<html lang="en">

<!-- layout.ejs-->
<head><meta name="generator" content="Hexo 3.8.0">
    <!-- hexo-inject:begin --><!-- hexo-inject:end --><meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="Adelaide">
    <meta name="author" content="xt">
    <meta name="keyword" content="">
    <link rel="canonical" href="https://taoshengxu.github.io/hexo/hexo/2020/04/12/2020-03-28_RepeatMasker/">
    <link rel="shortcut icon" href="/hexo/img/favicon.png">
    <link rel="alternate" type="application/atom+xml" title="IIM" href="/atom.xml">
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/animate.css/3.5.2/animate.min.css">
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
    <link rel="stylesheet" href="https://ajax.googleapis.com/ajax/libs/jqueryui/1.12.1/themes/smoothness/jquery-ui.css">
    <script src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.12.1/jquery-ui.min.js"></script>
    <script src="/hexo/js/search.js"></script>

    <title>
        
        RepeatMasker｜undefined
        
    </title>

    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">

    <link rel="stylesheet" href="/hexo/css/main.css">

    
      <link href="//fonts.googleapis.com/css?family=Source+Code+Pro" rel="stylesheet" type="text/css">
      <link rel="stylesheet" href="/hexo/css/highlight.css">
    

    

    <script async src="//dn-lbstatics.qbox.me/busuanzi/2.3/busuanzi.pure.mini.js"></script>


    

    

    


    
  


    




    <script async defer src="https://buttons.github.io/buttons.js"></script>

<link rel="stylesheet" href="/hexo/css/prism-solarizedlight.css" type="text/css">
<link rel="stylesheet" href="/hexo/css/prism-line-numbers.css" type="text/css"><!-- hexo-inject:begin --><!-- hexo-inject:end --></head>

<style>
    header.intro-header {
        background-image: url('/hexo/img/northernlights-sisimiut-lake.jpg')
    }
</style>
<!-- hack iOS CSS :active style -->
<body ontouchstart="" class="animated fadeIn">
<!-- hexo-inject:begin --><!-- hexo-inject:end --><header>
  <nav class="navbar navbar-default header-navbar" id="nav-top" data-ispost="true" data-istags="false" data-ishome="false">
    <div class="container-fluid">
      <div class="navbar-header page-scroll">
        <button type="button" class="navbar-toggle" data-toggle="collapse" aria-expanded="false" data-target="#website_navbar">
          <span class="sr-only">Toggle navigation</span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <span class="navbar-brand animated pulse">
          <a class="brand-logo" href="/hexo/">
                <img src="/hexo/img/banner.jpg?h=350&amp;auto=compress&amp;cs=tinysrgb">
          </a>
        </span>
      </div>

      <div class="collapse navbar-collapse" id="website_navbar">
          <ul class="nav navbar-nav navbar-right">
              
                <li>
                  <a href="/hexo/">home</a>
                </li>
              
                <li>
                  <a href="/hexo/archives/">archives</a>
                </li>
              
                <li>
                  <a href="/hexo/categories/">categories</a>
                </li>
              
                <li>
                  <a href="/hexo/tags/">tags</a>
                </li>
              
                <li>
                  <a href="/hexo/columns/">columns</a>
                </li>
              
          </ul>
      </div>
  </div></nav>


  
    <style>
       .intro-header {
          background-image: url('/hexo/post_cover_images/moon.jpg');
      }
    </style>

    <div class="intro-header">
        <div class="container">
            <div class="row">
                <div class="col-lg-8 col-lg-offset-2 col-md-10 col-md-offset-1 text-center">
                    <div class="site-heading">
                        <h1>RepeatMasker</h1>
                        
                        

                        
                          <span class="meta">
                               <span class="meta-item">Author: XT</span>
                               <span class="meta-item">Date: Apr 12, 2020</span>
                               
                          </span>
                          <div class="tags text-center">
                              Categories: 
                              <a class="tag" href="/hexo/categories/#Bioinformatics" title="Bioinformatics">Bioinformatics</a>
                              
                          </div>
                          <div class="tags text-center">
                              Tags: 
                          </div>
                        
                    </div>
                </div>
            </div>
        </div>
    </div>
  
</header>


<!-- Main Content -->
<!-- post.ejs -->
<article>
    <div class="container">
      <div class="col-lg-8 col-lg-offset-1 col-sm-9">
          
          <span class="post-count">1,827 words in total, 8 minutes required.</span>
          <hr>
          
          <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/hint.css/2.4.1/hint.min.css"><p>重复序列注释有两种常用策略，基于同源序列相似性和基于重复序列结构特征。</p>
<p>RepeatMasker是基于同源序列相似性注释序列的常用工具, </p>
<p>RepeatModeler可用来从头对基因组的重复序列家族进行建模注释，它的核心组件是RECON和RepatScout。</p>
<p><a href="https://blog.csdn.net/woodcorpse/article/details/73148464" target="_blank" rel="noopener"><strong>使用RepeatModeler和RepeatMasker注释基因组重复序列</strong></a></p>
<p>RepeatMasker是一款专门用于基因组重复序列识别注释，并分类统计的软件，几乎用于所有物种。是研究基因组、非编码RNA、转座子和着丝粒领等相关领域的必备软件。很多small RNA, lncRNA与Repeat区有密切关系。</p>
<p><strong>软件安装与配置</strong></p>
<p>本次安装环境为 Ubuntu 16.04.2 x64，所有相关软件和数据库全为文章发布时最新版。本文是以Root权限安装提供服务所有用户使用，没有权限的小伙伴只需将软件下载安装在自己的文件夹内，配置repeatmasker时设置所有相关软件的位置即可，不会设置环境变量的一律使用程序完整路径名运行RepeatMasker即可。</p>
<p><strong>1. RMBlast序列搜索引擎</strong></p>
<p><a href="http://www.repeatmasker.org/RMBlast.html" target="_blank" rel="noopener">http://www.repeatmasker.org/RMBlast.html</a></p>
<p>2.6.0 ver 2 2017-3-29</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">\<span class="comment"># 下载RMBlast源码包并编辑</span></span><br><span class="line"><span class="built_in">cd</span> ~/bin/</span><br><span class="line">wget ftp://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/2.6.0/ncbi-blast-2.6.0+-src.tar.gz</span><br><span class="line">wget http://www.repeatmasker.org/isb-2.6.0+-changes-vers2.patch.gz</span><br><span class="line">tar zxvf ncbi-blast-2.6.0+-src.tar.gz</span><br><span class="line">gunzip isb-2.6.0+-changes-vers2.patch.gz</span><br><span class="line"><span class="built_in">cd</span> ncbi-blast-2.6.0+-src</span><br><span class="line">patch -p1 &lt; ../isb-2.6.0+-changes-vers2.patch</span><br><span class="line"><span class="built_in">cd</span> c++</span><br><span class="line">./configure --with-mt --prefix=/usr/<span class="built_in">local</span>/rmblast --without-debug</span><br><span class="line">make</span><br><span class="line">\<span class="comment"># 安装程序及库至系统目录，有报误，但我们需要的rmblastn已经可以正常使用了</span></span><br><span class="line">sudo make install <span class="comment"># Makefile:40: recipe for target 'install-toolkit' failed</span></span><br><span class="line"><span class="comment"># 测试程序是否安装成功</span></span><br><span class="line">/usr/<span class="built_in">local</span>/rmblast/bin/rmblastn -h</span><br></pre></td></tr></table></figure>
<p><strong>2. TRF(Tandem Repeat Finder)搜寻串联重复序列</strong></p>
<p><a href="http://tandem.bu.edu/trf/trf.download.html" target="_blank" rel="noopener">http://tandem.bu.edu/trf/trf.download.html</a></p>
<p>4.09 2016-2-22</p>
<p>最新版4.09，本操作系统需要安装其中的legacy的64位版才能运行</p>
<p>cd ~/bin/</p>
<p>wget <a href="http://tandem.bu.edu/trf/downloads/trf409.legacylinux64" target="_blank" rel="noopener">http://tandem.bu.edu/trf/downloads/trf409.legacylinux64</a></p>
<p>chmod +x trf409.legacylinux64</p>
<p>sudo cp trf409.legacylinux64 /usr/local/bin/trf</p>
<p># 测试有帮助信息即可用</p>
<p>trf </p>
<p><strong>3. RepeatMasker程序</strong></p>
<p><a href="http://www.repeatmasker.org/RMDownload.html" target="_blank" rel="noopener">http://www.repeatmasker.org/RMDownload.html</a></p>
<p>4.0.7 2017-2-1</p>
<p># 下载267MB安装包，也比较慢，放在后台慢慢下</p>
<p>nohup wget -c <a href="http://www.repeatmasker.org/RepeatMasker-open-4-0-7.tar.gz" target="_blank" rel="noopener">http://www.repeatmasker.org/RepeatMasker-open-4-0-7.tar.gz</a> &amp;bg</p>
<p>tar xvzf RepeatMasker-open-4-0-7.tar.gz</p>
<p><strong>4. Repbase数据库</strong></p>
<p><a href="http://www.girinst.org/server/RepBase/index.php" target="_blank" rel="noopener">http://www.girinst.org/server/RepBase/index.php</a></p>
<p>需要注册才能下载，人工审批，可能要等两天。</p>
<p>RepBaseRepeatMaskerEdition-20170127.tar.gz (48.84 MB)</p>
<p>也可以在我的百度网盘下载，并上传服务器至RepeatMasker下载的相同目录。</p>
<p><a href="http://pan.baidu.com/s/1c2zSMKo" target="_blank" rel="noopener">http://pan.baidu.com/s/1c2zSMKo</a></p>
<p>mv RepBaseRepeatMaskerEdition-20170127.tar.gz RepeatMasker/cd RepeatMasker/tar xvzf RepBaseRepeatMaskerEdition-20170127.tar.gz</p>
<p><strong>5. 配置RepeatMasker依赖关系</strong></p>
<p><em># 默认perl, repatmasker, trf安装位置正确的就一路回车，搜索引挚选择2 RMBlast，输入RMBlast安装目录/usr/local/rmblast/bin，再选5 done完成；例如我刚才编绎最新版安装目录：/usr/local/rmblast/bin；如果新版安装失败的，下载的旧版预编辑 安装目录为：/usr/local/rmblast-2.2.28/bin</em>./configure <em>#添加至全局环境变量</em>sudo ln -s <code>pwd</code>/RepeatMasker /usr/local/bin/RepeatMasker </p>
<p><strong>软件使用实例</strong></p>
<p><strong>1. 以拟南芥, 短柄草基因组为例</strong></p>
<figure class="highlight hsp"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta"># 显示程序基本用法、参数和说明*</span></span><br><span class="line"><span class="symbol">*RepeatMasker</span>* </span><br><span class="line">*<span class="meta"># 显示程序详细帮助手册*</span></span><br><span class="line"><span class="symbol">*RepeatMasker</span> -help* </span><br><span class="line">*<span class="meta"># 拟南芥分析实例*</span></span><br><span class="line">*<span class="meta"># 进入我存放拟南芥基因组的目录*</span></span><br><span class="line"><span class="symbol">*cd</span> ~/ref/phytozome/Athaliana/TAIR10/assembly*</span><br><span class="line">*<span class="meta"># 建立结果输出目录*</span></span><br><span class="line"><span class="symbol">*mkdir</span> <span class="keyword">repeat</span>*</span><br><span class="line">*<span class="meta"># 运行程序：parallel是选择线程数; species是物种名，常见物种看帮助，没有的写小写拉丁属名或引号全名;  html和gff是输出html和gff格式结果，方便查看和下游分析; dir输出结果目录；基因组fa文件必须放在所有参数最后；用时<span class="number">8</span>min*</span></span><br><span class="line">time RepeatMasker -parallel <span class="number">30</span> -species arabidopsis -html -gff -dir <span class="keyword">repeat</span> Athaliana_167_TAIR9.fa</span><br><span class="line"></span><br><span class="line">time RepeatMasker -parallel <span class="number">30</span> -species all -html -gff -dir <span class="keyword">repeat</span> Athaliana_167_TAIR9.fa</span><br><span class="line"></span><br><span class="line">*<span class="meta"># 短柄草分析实例, <span class="number">274</span>MB基因组<span class="number">30</span>线程用时<span class="number">13</span>min*</span></span><br><span class="line"><span class="symbol">*cd</span> ~/ref/phytozome/Bdistachyon/v3<span class="number">.1</span>/assembly*</span><br><span class="line"><span class="symbol">*mkdir</span> <span class="keyword">repeat</span>*</span><br><span class="line"><span class="symbol">*time</span> RepeatMasker -parallel <span class="number">30</span> -species brachypodium -html -gff -dir <span class="keyword">repeat</span> Bdistachyon_314_v3<span class="number">.0</span>.fa</span><br></pre></td></tr></table></figure>
<p>运行开始会显示数据库的发布时间版本和物种特异数据信息，需 注释核对</p>
<p>( Complete Database: dc20170127-rb20170127 )</p>
<p>Building species libraries in: /mnt/bai/public/bin/RepeatMasker/Libraries/dc20170127-rb20170127/brachypodium</p>
<p>- 201 ancestral and ubiquitous sequence(s) for brachypodium</p>
<p>- 282 lineage specific sequence(s) for brachypodium</p>
<p><strong>2. 结果文件说明</strong></p>
<p>*代表你基因组的名字</p>
<figure class="highlight css"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">\1. *<span class="selector-class">.out</span><span class="selector-class">.gff</span>：重复序列基因组注释文件，与基因注释类似，最重要结果</span><br><span class="line">\# 结果预览</span><br><span class="line"><span class="selector-tag">Chr1</span>    <span class="selector-tag">RepeatMasker</span>    <span class="selector-tag">similarity</span>      1       107     13<span class="selector-class">.2</span>    <span class="selector-tag">-</span>       .       <span class="selector-tag">Target</span> "<span class="selector-tag">Motif</span><span class="selector-pseudo">:ATREP18"</span> 561 649  </span><br><span class="line"><span class="selector-tag">Chr1</span>    <span class="selector-tag">RepeatMasker</span>    <span class="selector-tag">similarity</span>      1066    1097    10<span class="selector-class">.0</span>    +       .       <span class="selector-tag">Target</span> "<span class="selector-tag">Motif</span><span class="selector-pseudo">:(C)n"</span> 1 32  </span><br><span class="line"><span class="selector-tag">Chr1</span>    <span class="selector-tag">RepeatMasker</span>    <span class="selector-tag">similarity</span>      1155    1187    17<span class="selector-class">.1</span>    +       .       <span class="selector-tag">Target</span> "<span class="selector-tag">Motif</span><span class="selector-pseudo">:(TTTCTT)n"</span> 1 33 </span><br><span class="line"></span><br><span class="line">1. *<span class="selector-class">.tbl</span>：重复序列注释结果报告信息汇总表格 <span class="selector-tag">overview</span></span><br><span class="line">2. *<span class="selector-class">.out</span><span class="selector-class">.html</span>: 网页版结果详细，同<span class="selector-tag">RepeatMasker</span>在线注释结果报告</span><br><span class="line">3. *<span class="selector-class">.masked</span>: 将注释为重复序列区的大项替换为<span class="selector-tag">N</span>的基因组</span><br><span class="line">4. *<span class="selector-class">.out</span>：<span class="selector-tag">RepeatMasker</span>默认输入结果格式，信息基本与<span class="selector-tag">gff</span>相关</span><br><span class="line">5. *<span class="selector-class">.cat</span><span class="selector-class">.gz</span>: 序列与重复序列比对的文件</span><br></pre></td></tr></table></figure>
<p><strong>软件安装使用常见问题</strong></p>
<p><strong>1. RMBlast安装问题</strong></p>
<ul>
<li>NCBI自从2013年2.2.28后再没有更新过rmblast <a href="https://blog.csdn.net/woodcorpse/article/details/ftp://ftp.ncbi.nlm.nih.gov/blast/executables/rmblast/LATEST" target="_blank" rel="noopener">ftp://ftp.ncbi.nlm.nih.gov/blast/executables/rmblast/LATEST</a> ，我尝试安装源码包在Ubuntu16.04是无法安装的；</li>
<li>在RepeatMasker的页面上有最近ncbi-blast-2.6.0+-src源代码和补定，按要求安装，即本文中的操作，make编辑成功，但make install有错误，不过关键程序rmblastn已经成功，可正常使用；</li>
<li>如果新版安装失败，可尝试安装2.2.28的预译版</li>
</ul>
<p>cd /usr/local</p>
<p>wget ftp://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/2.2.28/ncbi-blast-2.2.28+-x64-linux.tar.gz</p>
<p>wget ftp://ftp.ncbi.nlm.nih.gov/blast/executables/rmblast/2.2.28/ncbi-rmblastn-2.2.28-x64-linux.tar.gz</p>
<p>tar zxvf ncbi-blast-2.2.28+-x64-linux.tar.gz</p>
<p>tar zxvf ncbi-rmblastn-2.2.28-x64-linux.tar.gz</p>
<p>cp -R ncbi-rmblastn-2.2.28/* ncbi-blast-2.2.28+/</p>
<p>rm -rf ncbi-rmblastn-2.2.28</p>
<p>mv ncbi-blast-2.2.28+ rmblast-2.2.28</p>
<p>/usr/local/rmblast-2.2.28/bin/rmblastn -h</p>
<p>如果安装成功了2.2.28，则配置repeatmasker中rmblast位置则改为/usr/local/rmblast-2.2.28/bin/rmblastn</p>
<p><strong>2. trf运行报错</strong></p>
<p>是由于Linux legacy GLIBC的版本兼容性问题，故作者提供了两个版本，原文中的如果不可用，试试下面另一个版本</p>
<p>wget <a href="http://tandem.bu.edu/trf/downloads/trf409.linux64chmod" target="_blank" rel="noopener">http://tandem.bu.edu/trf/downloads/trf409.linux64chmod</a> +x trf409.linux64./trf409.linux64</p>
<p><strong>3. Rpeatmasker运行时找不到依赖程序</strong></p>
<p>是RepeatMasker文件中./configure步骤设置错了，再新再一次，仔细核对每个依赖程序的位置，即可正常运行。</p>
<p>前提是你先运行下相关依赖的程序是否可以运行！</p>
<p><strong>4. 没有结果目录及结果</strong></p>
<p>添加了-dir 指定输出目录，但没有结果</p>
<p>time RepeatMasker -parallel30-species arabidopsis -html-gff-dir repeat Bdistachyon_314_v3.0.fa  </p>
<p>你一定是忘记建立结果文件夹了，程序不会自己建目录，mkdir repeat是必须的。你有两个选择，要么提前建文件夹，要么直接不用-dir result参数，把结果全都输出至当前目录。</p>

          
          <hr>
          <ul class="pager no-print">
              
              <li class="previous">
                  <a href="/hexo/2020/05/17/2020-05-17_Breast_cellline/" data-toggle="tooltip" data-placement="left" title="乳腺癌细胞系">&larr; Previous Post</a>
              </li>
              
              
              <li class="next">
                  <a href="/hexo/2020/04/03/2020-04-03_NGS_server/" data-toggle="tooltip" data-placement="top" title="NGS server集群">Next Post&rarr;</a>
              </li>
              
          </ul>
          

      </div>
      
  <div class="hidden-xs col-sm-3 toc-col">
    <div class="toc-wrap">
        Table of Contents
        
          
        
    </div>
  </div>


    </div>
</article>

<!-- Footer -->
<!-- footer.ejs -->
<footer class="no-print">

    <div class="text-center">

        <ul class="list-inline">
            
            
            

            

            

            

            

            

        </ul>
        <div class="text-muted copyright">
            &copy;
            
            2017 - 2020
            
            -
            xt
            <br>
            
            
            
            | 
            <p>
                <span id="busuanzi_container_site_pv"><span id="busuanzi_value_site_pv"></span> <b>PV</b></span> -
                <span id="busuanzi_container_site_uv"><span id="busuanzi_value_site_uv"></span> <b>UV</b></span> -
                160.7k <b>Words</b>
        </p></div>

        <div class="search-container">
            <form class="site-search-form">
                <span class="glyphicon glyphicon-search"></span>
                <input type="text" id="local-search-input" class="st-search-input" placeholder="Search...">
            </form>
            <div id="local-search-result" class="local-search-result-cls"></div>
        </div>
        <script type="text/javascript" id="local.search.active">
            var inputArea = document.querySelector("#local-search-input");
            inputArea.onclick = function () {
                var root = "/hexo/";
                getSearchFile(root);
                this.onclick = null
            }
            inputArea.onkeydown = function () {
                if (event.keyCode == 13) return false
            }
        </script>

    </div>
</footer>

<!-- Custom Theme JavaScript -->
<script src="/hexo/js/main.js"></script>

<!-- async load function -->
<script>
    function async(u, c) {
        var d = document, t = 'script',
            o = d.createElement(t),
            s = d.getElementsByTagName(t)[0];
        o.src = u;
        if (c) {
            o.addEventListener('load', function (e) {
                c(null, e);
            }, false);
        }
        s.parentNode.insertBefore(o, s);
    }
</script>

<!--fastClick.js -->
<script>
    async("//cdn.bootcss.com/fastclick/1.0.6/fastclick.min.js", function () {
        var $nav = document.querySelector("nav");
        if ($nav) FastClick.attach($nav);
    })
</script><!-- hexo-inject:begin --><!-- hexo-inject:end -->



</body>

</html>
